---
title: "Cluster stability"
author: "almut"
date: "7 MÃ¤rz 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, message=FALSE, warning=FALSE}
library(clusterExperiment)
library(DESeq2)
library(DEFormats)

```




## Aim: Find an optimal cluster and its stability for CLL data 

load data
```{r datasets}
#gene expression
#load("/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")

#gene expression
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")
#load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV.RData")  #trimmed dataset with Tsig included by Junyan, variant annotations, and IGHVmod as generated by IGHV_clustering.Rmd

#exclude NAs
ddsCLL <- ddsCLL[,!is.na(colData(ddsCLL)[,"IGHV"])]
ddsCLL <- ddsCLL[,!is.na(colData(ddsCLL)[,"trisomy12"])]

```


normalize data
```{r norm data}

#Variance stabilization transformation of the raw data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:1000],]

ddsCLL_var <- ddsCLL[rownames(exprMat),]
RNAnorm_var <- RNAnorm[rownames(exprMat),]



#generate a summarizedExperiment data set from deseqDataset
seNorm <- as(RNAnorm_var, "SummarizedExperiment")
se <-as(ddsCLL_var, "SummarizedExperiment")

```

#RSEC
```{r RSEC}
#system.time(rsecCLL <-RSEC(se, isCount = TRUE,nPCADims=10,combineMinSize=3,random.seed=176201))
rsecCLL <-RSEC(seNorm, isCount = TRUE,nPCADims=10,combineMinSize=3,random.seed=176201)
rsecCLL
#save(rsecCLL, file = "/home/almut/Dokumente/git/Transcriptome_CLL/dataset/clusterExp_080318.RData")

#explore ClusterExperiment data set
head(primaryCluster(rsecCLL),20)
head(clusterLabels(rsecCLL))
head(getClusterManyParams(rsecCLL))

```
#Visualize RSEC results
```{r, fig.height=15, fig.width=25}
defaultMar<-par("mar")
plotCMar<-c(1.1,8.1,4.1,1.1)
par(mar=plotCMar)
plotClusters(rsecCLL,main="Clusters from RSEC", whichClusters="workflow", sampleData=c("trisomy12","IGHV", "Methylation", "TP53"), axisLine=-1)
plotBarplot(rsecCLL,main="Final Clustering")
plotBarplot(rsecCLL,whichClusters=c("mergeClusters" ,"combineMany"))
```

```{r other dimensions}

plotCoClustering(rsecCLL,whichClusters=c("mergeClusters","combineMany"))
plotDendrogram(rsecCLL,whichClusters=c("combineMany","mergeClusters"),plotType="colorblock",leafType="sample")
plotDimReduce(rsecCLL)
plotDimReduce(rsecCLL,whichDims=c(1:4))
```
#try diff parameter
```{r}
rsecCLL<-RSEC(rsecCLL,isCount=TRUE,combineProportion=0.6,mergeMethod="JC",mergeCutoff=0.05)
defaultMar<-par("mar")
plotCMar<-c(1.1,8.1,4.1,1.1)
par(mar=plotCMar)
plotClusters(rsecCLL,main="Clusters from RSEC", whichClusters=c("mergeClusters.1","combineMany.1","mergeClusters","combineMany"), sampleData=c("trisomy12","IGHV", "Methylation", "TP53"), axisLine=-1)


```



