---
title: "Individual_cluster"
author: "Almut Luetge"
date: "September 9, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Aim: examine the individual cluster to find variance source:
Get a detailed view on the single variants
1.) How do sample cluster using all genes (add traits to sample distance matrix)
2.) What do we see unsupervised? Obvious? 
3.) What do we need to consider when analysing data? IGHV? c1c2-groups?


##Data preprocessing
load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
#library(VennDiagram)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(pheatmap)
library(corrplot)
library(ggplot2)
library(gtable)
library(grid)
library(dendextend)
library(ComplexHeatmap)
library(gridExtra)
library(factoextra)
library(Rtsne)
library(RColorBrewer)
library(circlize)
library(ggpubr)
library("fpc")
library("cluster")
 library(wesanderson)
library(pvclust)
```


load data
```{r datasets}
#gene expression
#load("/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")

#gene expression
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")
#load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV.RData")  #trimmed dataset with Tsig included by Junyan, variant annotations, and IGHVmod as generated by IGHV_clustering.Rmd

#exclude NAs
ddsCLL <- ddsCLL[,!is.na(colData(ddsCLL)[,"IGHV"])]
ddsCLL <- ddsCLL[,!is.na(colData(ddsCLL)[,"trisomy12"])]

```


normalize data
```{r norm data}

#Variance stabilization transformation of the raw data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)


```


1.) Cluster based on sample to sample dist including annotation of variants

```{r s-to-s heatmap, fig.height= 15, fig.width= 20}
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:1000],]



sampleDists <- dist(t(exprMat)) #, method = "minkowski")
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- colData(RNAnorm)$PatID

dend = as.dendrogram(hclust(sampleDists, method = "ward.D2"))
#dend_pv <- pvclust(exprMat, method.hclust = "ward.D2", nboot = 100)

#colors
#colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
#colors = colorRamp2(c(110,90,70,0), c("#ffffd9","#7fcdbb" ,"#1d91c0","#081d58"))
colors = colorRamp2(c(120,100,80,0), c("#fee8c8","#fdbb84", "#d7301f","#b30000")) #,"#7f0000"))
#colors= colorRamp2(seq4min(sampleDistMat), max(sampleDistMat), length = 3), c("blue", "#EEEEEE", "red"), space = "LAB")
#colors = ggplot2::scale_fill_gradient2(low = "blue", high = "red", midpoint = 120, limits = c(0, 150))


annocol <- colorRampPalette( rev(brewer.pal(12,"Paired")) )(17)
annocol <- get_palette("jco", 10)
#annocol <- c("#155F83", "#8A9045", "#800000", "#B1746F", "#497467", "#FFB547", "#5B8FA8", "#800000", "#54503B", "#350E20")
annocolor <- list(IGHV = c("M" = annocol[1], "U" = annocol[2]) , trisomy12 = c( "1" = annocol[8], "0" = annocol[4]), TP53= c( "1" = annocol[5], "0" = annocol[6]), del13q14 = c( "1" = annocol[7], "0" = annocol[8]), del11q22.3 = c( "1" = annocol[9], "0" = annocol[10]), Methylation = c("IP" = annocol[5], "LP" = annocol[6], "HP" =  annocol[7]))



#ha_row = rowAnnotation(df = data.frame(colData(RNAnorm)[, c("IGHV", "trisomy12", "TP53","del13q14", "del11q22.3", "Methylation")]), col = annocolor, annotation_width = unit(c(rep(1.8, 6)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 70), labels_gp = gpar(fontsize = 50),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm")))

#ha_row = rowAnnotation(df = data.frame(colData(RNAnorm)[, c("IGHV", "trisomy12", "Methylation")]), col = annocolor, annotation_width = unit(c(rep(1.8, 3)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 70), labels_gp = gpar(fontsize = 50),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm")))


ha_top = HeatmapAnnotation(df = data.frame(colData(RNAnorm)[, c("IGHV", "trisomy12", "Methylation")]), col = annocolor, annotation_width = unit(c(rep(1.8, 3)), "cm"), show_legend = FALSE, annotation_legend_param = list(title_gp = gpar(fontsize = 70), labels_gp = gpar(fontsize = 55),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(2, "cm")))

anno_legend_list = lapply(ha_top@anno_list[c("IGHV", "trisomy12", "Methylation")], function(anno) color_mapping_legend(anno@color_mapping, plot = FALSE,title_gp = gpar(fontsize = 45, fontface = "bold"),
    grid_height = unit(1.5, "cm"),
    grid_width = unit(0.5, "cm"),
    labels_gp = gpar(fontsize = 35)))


h1 <- Heatmap(sampleDistMat, col = colors , heatmap_legend_param = list(title = "Sample distance", title_gp = gpar(fontsize = 45), grid_height = unit(1.5, "cm"), grid_width = unit(0.5, "cm"), gap = unit(5, "cm"), labels_gp = gpar(fontsize = 55)), column_dend_height = unit(6, "cm"), show_row_dend = FALSE, show_column_names = FALSE , show_row_names = FALSE,   cluster_columns = dend, cluster_rows  = dend, row_dend_reorder = FALSE, column_dend_reorder = FALSE, top_annotation = ha_top, top_annotation_height = unit(2.5, "cm"), show_heatmap_legend = FALSE)

heatmap_legend = color_mapping_legend(h1@matrix_color_mapping, plot = FALSE, title = "Sample distance",
    title_gp = gpar(fontsize = 45, fontface = "bold"),
    grid_height = unit(1.5, "cm"),
    grid_width = unit(0.5, "cm"),
    labels_gp = gpar(fontsize = 35))

#pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/cluster1000exprgenes.pdf", width=20, height=15)
#draw(ha_row + h1)
draw(h1, heatmap_legend_list = c(list(heatmap_legend), anno_legend_list))
#dev.off()




```


##Data overview using tsne
```{r tsne, fig.height=7, fig.width=7}


#Calculate t-sne (t-distributed Stochastic Neighbor Embedding)
distSamp <- sampleDistMat
tsneRes <- Rtsne(distSamp, perplexity = 10, theta = 0, max_iter = 5000, is_distance = T, dims =2)
tsneRes <-tsneRes$Y
rownames(tsneRes) <- labels(distSamp)[[1]]
colnames(tsneRes) <- c("x", "y")
tsneRes <- data.frame(tsneRes)

cond <-c("del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")

tsne_feat <- sapply(c(cond, "Methylation", "IGHV", "trisomy12"), function(condition){
  tsneRes[,condition] <-colData(ddsCLL)[which(colData(ddsCLL)$PatID %in% rownames(tsneRes)), condition]
})


tsne_feat <- as.data.frame(tsne_feat)
tsneRes <- cbind(tsneRes, tsne_feat)

#plot using ggplot2
plist <- lapply(c("Methylation", "IGHV", "trisomy12"), function(feature){
  ggplot(tsneRes, aes_string(x = "x", y = "y", color = feature)) + geom_point(size = 7) + theme_classic() + theme(panel.border = element_rect(color = 'black', size = 1, linetype = 'solid', fill = NA), axis.ticks = element_line(color = "black", size = 1), text = element_text(size = 20)) + xlab("") + ylab("") + ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 40)) +
  theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 30), legend.title = element_text(face = "bold", size = 40))
})


#grid.arrange(grobs = plist, ncol =2)

#save as ggpubr ggscatter plot for thesis
#lapply(c(cond,"Methylation", "IGHV", "trisomy12"), function(feature){
lapply("Methylation", function(feature){ 
  p <- ggscatter(tsneRes, x = "x", y = "y", color = feature, palette =c(annocol[7], annocol[5], annocol[6]),#c("#00AFBB", "#E7B800", "#FC4E07"),
  ylab = "", xlab = "", legend = "right", main = feature, font.legend = c(23, "plain", "black"), font.tickslab = c(28, "plain", "black"), font.main = 33, font.submain = 28, font.caption = 28) + coord_fixed()
  #ggsave(file= paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/tsnes/tsne_", feature, ".svg"), plot=p, width=7, height=7)
  p
})




```

#hierachical clustering tree
```{r tree, fig.height= 10, fig.width= 25}
# Compute hierarchical clustering
#labels <- paste0(colData(RNAnorm)$IGHV, "_", colData(RNAnorm)$Methylation, "_", colData(RNAnorm)$trisomy12)

res.hc <- sampleDists %>% 
  hclust(method = "ward.D2")

##verify the cluster tree
# Compute cophentic distance
res.coph <- cophenetic(res.hc)
# Correlation between cophenetic distance and
# the original distance
cor(sampleDists, res.coph)

res.hc$labels <- paste0(colData(RNAnorm)$Methylation, "_", colData(RNAnorm)$trisomy12)    # Compute hierachical clustering


##visualize with dendextend
#dend <- as.dendrogram(res.hc)
mycols <- c("#2E9FDF", "#00AFBB", "#E7B800", "#FC4E07")
mycols <-c( "#969696", "#525252", "#969696" , "#000000")
dend <-  as.dendrogram(res.hc) %>%
   set("branches_lwd", 2) %>% # Branches line width
   set("branches_k_color", mycols, k = 4)


#colormatrix
colMat <- matrix(nrow = 160, ncol = 3)
colMat[,1] <- ifelse(colData(RNAnorm)$IGHV == "M", annocol[1], annocol[2])
colMat[,2] <- ifelse(colData(RNAnorm)$trisomy12 == "1", annocol[8], annocol[4])
colMat[,3] <- ifelse(colData(RNAnorm)$Methylation == "HP", annocol[7], ifelse(colData(RNAnorm)$Methylation == "IP", annocol[5], annocol[6]))
colMat[which(is.na(colMat[,3])),3] <- "grey"

colnames(colMat) <- c("IGHV", "trisomy12", "Methylation")

pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/dendogramm.pdf", width=25, height=10)
#plot(dend)
colored_bars(colMat, dend, rowLabels = c("IGHV", "trisomy12", "Methylation"), y_scale = 60, y_shift = 5, cex.rowLabels = 1.8)
#dev.off()

# Visualize using factoextra
# Cut in 4 groups and color by groups
#fviz_dend(res.hc, k = 4, # Cut in four groups
#          cex = 1, # label size
#          k_colors = c("#00AFBB", "#E7B800", "#FC4E07", "#2E9FDF"),
#          color_labels_by_k = T, # color labels by groups
#          rect = TRUE # Add rectangle around groups
#          )


```

#cluster stability
```{r cluster stability}

clusterResampling = function(x, ngenes = 1000, k = 4, B = 250,
  prob = 0.67) {
  mat = exprs(x)
  ce = cl_ensemble(list = lapply(seq_len(B), function(b) {
  selSamps = sample(ncol(mat), size = round(prob * ncol(mat)),
  replace = FALSE)
  submat = mat[, selSamps, drop = FALSE]
  sel = order(rowVars(submat), decreasing = TRUE)[seq_len(ngenes)]
  submat = submat[sel,, drop = FALSE]
  pamres = pam(t(submat), k = k)
  pred = cl_predict(pamres, t(mat[sel, ]), "memberships")
  as.cl_partition(pred)
  }))
  cons = cl_consensus(ce)
  ag = sapply(ce, cl_agreement, y = cons)
  list(agreements = ag, consensus = cons)
}






```







#compare dendogramm methods 
```{r different dendogramms}
# Create multiple dendrograms by chaining
dend1 <- sampleDists %>% hclust("complete") %>% as.dendrogram
dend2 <- sampleDists %>% hclust("ward.D2") %>% as.dendrogram
dend3 <- sampleDists %>% hclust("average") %>% as.dendrogram
dend4 <- sampleDists %>% hclust("centroid") %>% as.dendrogram
# Compute correlation matrix
dend_list <- dendlist("Complete" = dend1, "Ward.D2" = dend2,
                      "Average" = dend3, "Centroid" = dend4)
cors <- cor.dendlist(dend_list)
# Print correlation matrix
round(cors, 2)

corrplot(cors, "pie", "lower")


```


#PCA using factoextra
```{r pca factoextra}
#define the number of genes to use
exprs <- assay(RNAnorm)
colnames(exprs) <- colData(RNAnorm)$PatID
#top 5000 most variant genes
exprs <- exprs[order(sds, decreasing = T)[1:500],]

metaTab <- colData(RNAnorm)
rownames(metaTab) <- metaTab$PatID
#PCA
pcaRes <- prcomp(t(exprs), scale =T)

#visualize using factoextra
#screeplot
fviz_eig(pcaRes, addlabels = TRUE, ylim = c(0, 50))

#plot
fviz_pca_ind(pcaRes, 
            #ind
             geom.ind = "point", # show points only (nbut not "text")
             col.ind = metaTab$IGHV, # color by groups
             palette = "jco",
             addEllipses = TRUE, # Concentration ellipses
             legend.title = "Groups", axes = c(1,2)
             )





```


```{r pca ggscatter, fig.width=7, fig.height=7}

#Plot PCA
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:1000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))


#plot PCA and color samples based on annotations

  
p <- ggscatter(pcaTab, x = "PC1", y = "PC2", color = "IGHV", palette = "jco",
  ylab = sprintf("PC2 (%2.1f%%)",varExp[2]), xlab = sprintf("PC1 (%2.1f%%)",varExp[1]), legend = "right", main = "PCA IGHV status", font.legend = c(23, "plain", "black"), font.tickslab = c(23, "plain", "black"), font.main = 33, font.submain = 28, font.caption = 28, font.x = 28, font.y= 28) + coord_fixed()

#ggsave(file= "/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/pca_IGHV.svg", plot=p, width=7, height=7)
p

```


