---
title: "Correlation of genomic variants"
author: "Almut Luetge"
date: "September 11, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#How do the single variants correlate with each other?
Some of the genomic variants correlate or show tendencies towards being mutually exclusive. Junyan found this for trisomy12 and del13q14, but almost not or week for del11q22.3. The corrplot from script: gene signature shows the opposite. Why? 

Check corrplot analysis


load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(dplyr)
library(beeswarm)
library(Rtsne)
library(ggplot2)
library(sva)
library(gridExtra)
library(robustbase)
library(ggrepel)
library(polycor)
library(corrplot)
library(reshape2)

```


load data
```{r data}
#gene expression
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_trimmed_tsig_modIGHV.RData")
     #2017-08-16_ddsrna_trimmed.RData")


```

#select interesting feature, based on the occurence in the trimmed dds-object 2017-08-16_ddsrna_trimmed.RData

```{r prepare metadata}
##Prepare dds dataset --> add patmeta data and remove non CLL patients
variants <- c("trisomy12", "del13q14", "RNAprep", "batch", "Tsig", "Methylation", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12", "c1c2", "IGHV")

patMat <-colData(ddsCLL)[,c("PatID", variants)]
patMat <- as.data.frame(patMat)
patMat <- lapply(patMat, function(x) as.factor(x))
patMat <- as.data.frame(patMat)
rownames(patMat) <- colData(ddsCLL)$PatID
```

# 1. Test co-occurence using Fisher exact test (following Junyan)

```{r fishers exact}
tri12 <- patMat[, c("PatID","trisomy12")]
rownames(tri12) <- patMat$PatID
tri12 <- tri12[which(!is.na(tri12$trisomy12)),]
geneBack <- patMat[rownames(tri12), -which(colnames(patMat) %in% c("PatID", "Methylation_Cluster", "trisomy12"))]



coRes <- apply(geneBack,2,function(x) {
  nonNA <- !is.na(x)
  res <- chisq.test(tri12$trisomy12[nonNA], x[nonNA], correct=FALSE)
  data.frame(p=res$p.value, stat = res$statistic[[1]])
}) %>% do.call(rbind,.)

coRes$p.adj <- p.adjust(coRes$p)
coRes[coRes$p <= 0.05,,drop=FALSE]



```

del11q22.3 is not significant exclusive any more, but del13q14. But I get different p-values, probably due to slightly different selection and different input data. I don't differentiate between monoallelic and biallelic... 


perform chisqared on all variants
```{r corMat chi, fig.height=10, fig.width= 10}
variants_all <- as.vector(colnames(patMat[-which(colnames(patMat) %in% c("PatID", "Methylation", "Tsig", "RNAprep", "c1c2", "batch"))]))

#Chisquare p.adj matrix
my_chisquare <- function(variant_in){
  variant1 <- patMat[, c("PatID",variant_in)]
  rownames(variant1) <- patMat$PatID
  variant1 <- variant1[which(!is.na(variant1[,variant_in])),]
  geneBack <- patMat[rownames(variant1), variants_all]

  coRes <- apply(geneBack,2,function(x) {
  nonNA <- !is.na(x)
  res <- chisq.test(variant1[,variant_in][nonNA], x[nonNA], correct=FALSE)
  data.frame(p=res$p.value, stat = res$statistic[[1]], residu = ifelse(res$residuals[1] > 0, 1, -1))
  }) %>% do.call(rbind,.)
coRes$p.adj <- p.adjust(coRes$p)
resultDat <- as.data.frame(-log10(coRes$p.adj) * coRes$residu)
#resultDat <- as.data.frame(coRes$p.adj)
rownames(resultDat) <- rownames(coRes)
colnames(resultDat) <- variant_in
resultDat[variant_in,] <- 0
resultDat
}

resChi <-lapply(variants_all,my_chisquare) %>% do.call(cbind,.)
resChiMat <- as.matrix(resChi)

col <- colorRampPalette(rev(c("#BB4444", "#EE9988", "#FFFFFF", "#77AADD", "#4477AA")))(50)
col3 <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
#corplot from ci-square test
#corrplot.mixed(resChiMat, is.corr = FALSE, number.cex= 1.7, tl.cex=2, tl.pos = "lt", cl.cex=2, diag = "n", col = col3, pch.col = "black")



#svg(filename="~/git/figures_thesis/corplot_chiquare.svg", width=10, height=10)
corrplot(resChiMat, is.corr = FALSE, method="color", col = col,
 type = "upper", order = "hclust", number.cex = 1.2,
         addCoef.col = "black", # Add coefficient of correlation
         tl.col = "black", tl.srt = 90, # Text label color and rotation
         # hide correlation coefficient on the principal diagonal
         tl.cex=1.4,
         cl.lim = c(-4, 12),
         cl.cex = 1.0,
         diag = FALSE)
#dev.off()

```





#2.Correlation analysis using the polychor package

Corplots
```{r corplots, fig.height=15, fig.width= 15}
#Generate matrix with correlation coefficients

corMat <- hetcor(patMat[,-1], use = "pairwise.complete.obs")


corrplot.mixed(corMat$correlations, number.cex= 1.7, tl.cex=2, tl.pos = "lt", cl.cex=2, diag = "n")

cor.mtest <- function(mat, conf.level = 0.95){
  mat <- as.matrix(mat)
    n <- ncol(mat)
    p.mat <- lowCI.mat <- uppCI.mat <- matrix(NA, n, n)
    diag(p.mat) <- 0
    diag(lowCI.mat) <- diag(uppCI.mat) <- 1
    for(i in 1:(n-1)){
        for(j in (i+1):n){
            tmp <- cor.test(mat[,i], mat[,j], method = "pearson", conf.level = conf.level)
            p.mat[i,j] <- p.mat[j,i] <- tmp$p.value
            lowCI.mat[i,j] <- lowCI.mat[j,i] <- tmp$conf.int[1]
            uppCI.mat[i,j] <- uppCI.mat[j,i] <- tmp$conf.int[2]
        }
    }
    return(list(p.mat, lowCI.mat, uppCI.mat))
}

res1 <- cor.mtest(corMat$correlations,0.95)
res2 <- cor.mtest(corMat$correlations,0.99)

corrplot.mixed(corMat$correlations, p.mat = res1[[1]], sig.level=0.05, insig = "blank", number.cex= 1.7, tl.cex=2, tl.pos = "lt", cl.cex=2, diag = "n")

```

Results differ.. Is this caused by different test statistics? Do I use the wrong parameter? Is Pearson the wrong test statistic - not normally distributed data? 



## Check the occurence of variants sorted by trisomy12 status in the patMet object
```{r common variants, fig.height=10, fig.width=30}
#Interesting variants


patMat_new <- data.frame(colData(ddsCLL)[,-which(names(colData(ddsCLL)) %in% c("PatID", "Methylation", "HipoID", "Gender",  "diag", "IGVH", "Treatment","center", "RNAprep", "batch", "Tsig",  "c1c2", "sizeFactor"))])

patMat_new[,c(1:12)] <- apply(patMat_new[,c(1:12)], 2, function(x) as.numeric(as.character(x)))

###Change VAR if other than IGHV is used

#function to create summary:
myVariant <- function(variant){
  variant_counts <- patMat_new[,variant]

  overview <- as_data_frame(patMat_new) %>% dplyr::group_by_(variant) %>% dplyr::summarize_if(is.numeric, sum, na.rm =T) 
  Sum <- as_data_frame(patMat_new) %>% dplyr::summarize_if(is.numeric, sum, na.rm =T)
  VAR <- as_data_frame(variant_counts) %>% dplyr::summarize("0" = sum(value %in% "0"), "1"= sum(value %in% "1"), "NA" = sum(is.na(value)), Total = n())
  #VAR <- as_data_frame(variant_counts) %>% dplyr::summarize("U" = sum(value %in% "U"), "M"= sum(value %in% "M"), "NA" = sum(is.na(value)), Total = n())
  
  Sum[,variant] <- "Total"
  Sum <- Sum[,colnames(overview)]

  VAR <- t(VAR)

  sum_tab <- rbind(overview,Sum)
  sum_tab <- cbind(sum_tab, VAR)
  colnames(sum_tab) <- gsub("VAR", paste0(variant, "_total"), colnames(sum_tab))


  sum_melted <- melt(sum_tab)
  
  #Image as object to save as svg
  svg <- ggplot(sum_melted, aes_string(x = "variable", y = variant)) +  geom_tile(aes(fill = value)) + 
    geom_text(aes(label = round(value, 1)), size =15) + ylab(variant) +
    scale_fill_gradient(low = "white", high = "red") + theme(axis.title.x = element_text(face="bold", size=50), axis.text.x=element_text(size=40, angle = 90)) + theme(axis.title.y = element_text(face="bold", size=50), axis.text.y= element_text(size=30)) + theme(legend.key.size = unit(1, "cm"), legend.text = element_text(size = 30), legend.title = element_text(size = 40, face = "bold")) 
  
  #save as svg
  #ggsave(file="~/git/figures_thesis/datatable_overview_trisomy12.svg", plot=svg, width=30, height=10)
  
  #plot
  ggplot(sum_melted, aes_string(x = "variable", y = variant)) +  geom_tile(aes(fill = value)) + 
    geom_text(aes(label = round(value, 1)), size =15) + ylab(variant) +
    scale_fill_gradient(low = "white", high = "red") + theme(axis.title.x = element_text(face="bold", size=50), axis.text.x=element_text(size=40, angle = 90)) + theme(axis.title.y = element_text(face="bold", size=50), axis.text.y= element_text(size=30)) + theme(legend.key.size = unit(1, "cm"), legend.text = element_text(size = 30), legend.title = element_text(size = 40, face = "bold")) 
  
}

myVariant("del17p13")

```
Both parameter have a predominant occoruence in IGHV-U samples. This makes sense from the point of view that the mutational rate is much higher in IGHV-U status. It will help to integrate the IGHV-M/U status of the last batch (in progress). 


#Open questions:
What is the mathematical analysis behind the hetcor function? How does it differ from the chi-square function of Junyan? Which one is better to use? 


















