---
title: "tsne_diffGenes"
author: "almut"
date: "13 Mai 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Aim: How can we distinguish between variants with distinct and without distinct gene expression?
Idea show that samples can be seperated on the basis of their differentially expressed genes


libraries
```{r load libraries}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(piano)
library(ggsci)
library(RColorBrewer)
library(circlize)
library(Rtsne)
library(dendextend)
library(corrplot)
```

data
```{r data }
#dds data set. gene expression data + patmetadata
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)

#filter for sign. genes in Trisomy12 
exprMat <- assay(RNAnorm)


cond <-c("IGHV", "trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")

```

tsne plot function

```{r tsne}

tsne_plot <- function(variant){
  diff_genes <- read.csv(file=paste0("/home/almut/Dokumente/masterarbeit/results/noFilter/", variant, "_diffGenes.csv"))
  #filter
  diff_genes <- diff_genes[which(diff_genes$padj < 0.05 & abs(diff_genes$log2FoldChange) > 2 ),-1]

  #sampledists on basis of diff genes
  exprVar <- exprMat[diff_genes$ID,]
  colnames(exprVar) <- colData(ddsCLL)$PatID
  sampleDists <- dist(t(exprVar)) #, method = "minkowski")
  sampleDistMat <- as.matrix(sampleDists)
  rownames(sampleDistMat) <- colData(RNAnorm)$PatID
  
  #tsne
  tsneRes <- Rtsne(sampleDistMat, perplexity = 10, theta = 0, max_iter = 5000, is_distance = T, dims =2)
  tsneRes <-tsneRes$Y
  rownames(tsneRes) <- labels(sampleDistMat)[[1]]
  colnames(tsneRes) <- c("x", "y")
  tsneRes <- data.frame(tsneRes)
  tsneRes[,variant] <-colData(ddsCLL)[which(colData(ddsCLL)$PatID %in% rownames(tsneRes)), variant]
  
  #plot
    p <- ggscatter(tsneRes, x = "x", y = "y", color = variant, palette =c("#00AFBB", "#E7B800"),
  ylab = "", xlab = "", legend = "right", main = variant, font.legend = c(23, "plain", "black"), font.tickslab = c(28, "plain", "black"), font.main = 33, font.submain = 28, font.caption = 28) + coord_fixed()
  #ggsave(file= paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/tsnes/tsne_", variant, ".svg"), plot=p, width=7, height=7)
  p
}

lapply(cond, tsne_plot)


```

```{r pca, fig.width=10, fig.height=10}

pca_plot <- function(variant){
  diff_genes <- read.csv(file=paste0("/home/almut/Dokumente/masterarbeit/results/noFilter/", variant, "_diffGenes.csv"))
  #filter
  diff_genes <- diff_genes[which(diff_genes$padj < 0.1 & abs(diff_genes$log2FoldChange) > 2 ),-1]

  #sampledists on basis of diff genes
  exprVar <- exprMat[diff_genes$ID,]
  colnames(exprVar) <- colData(ddsCLL)$PatID
 #Calculate PCA
  pcaRes <- prcomp(t(exprVar), scale =T)
  varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
  pcaTab <- data.frame(pcaRes$x[,c(1:10)])
  names(varExp) <- colnames(pcaRes$x)

#add background information
  pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))


#plot PCA and color samples based on annotations

  
  p <- ggscatter(pcaTab, x = "PC3", y = "PC4", color = variant, palette = "jco",
  ylab = sprintf("PC4 (%2.1f%%)",varExp[4]), xlab = sprintf("PC3 (%2.1f%%)",varExp[3]), legend = "right", main =paste0("PCA", variant), font.legend = c(23, "plain", "black"), font.tickslab = c(23, "plain", "black"), font.main = 33, font.submain = 28, font.caption = 28, font.x = 28, font.y= 28) + coord_fixed()

#ggsave(file= "/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/pca_IGHV.svg", plot=p, width=7, height=7)
p

  variant1 <- colData(RNAnorm)[, c("PatID",variant)]
  rownames(variant1) <- colData(RNAnorm)$PatID
  variant1 <- variant1[which(!is.na(variant1[,variant])),]
  pcaTab1 <- pcaTab[which(pcaTab$PatID %in% rownames(variant1)), ]
  variant1 <- variant1[pcaTab1$PatID,]
  
  resT <- apply(pcaTab1[,c(1:10)], 2, function(x) {
  res <- t.test(x ~ variant1[, variant], var.equal=FALSE)
  data.frame(p = res$p.value, fc = res$estimate[[2]]/res$estimate[[1]])
}) %>% do.call(rbind,.)
  
  resT[,variant] <- p.adjust(resT$p) #-log10(p.adjust(resT$p))  #* varExp[c(1:10)]
  resT[,variant]
  
}

resTtest <- lapply(cond, pca_plot) %>% do.call(cbind,.)

colnames(resTtest) <- cond
rownames(resTtest) <- c(paste0("PC", c(1:10)))
resTMat <- as.matrix(resTtest)


col <- colorRampPalette(rev(c("#BB4444", "#EE9988", "#FFFFFF", "#77AADD", "#4477AA")))(50)

#pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/pca_variant_specific_genes.pdef", width=9, height=9)
corrplot(resTMat, is.corr = FALSE, method="color", col = col, number.cex = 1.2,
         addCoef.col = "black", # Add coefficient of correlation
         tl.col = "black", tl.srt = 90, # Text label color and rotation
         # hide correlation coefficient on the principal diagonal
         tl.cex=1.4,
         cl.cex = 1.1,
          cl.lim = c(0, 17),
         diag = TRUE)
#dev.off()


  
  

  




```

#hier clustering
```{r hier clust, fig.height= 10, fig.width= 25}

hier_plot <- function(variant){
  diff_genes <- read.csv(file=paste0("/home/almut/Dokumente/masterarbeit/results/noFilter/", variant, "_diffGenes.csv"))
  #filter
  diff_genes <- diff_genes[which(diff_genes$padj < 0.01 & abs(diff_genes$log2FoldChange) > 0 ),-1]

  #sampledists on basis of diff genes
  exprVar <- exprMat[diff_genes$ID,]
  colnames(exprVar) <- colData(ddsCLL)$PatID
  sampleDists <- dist(t(exprVar)) #, method = "minkowski")
  sampleDistMat <- as.matrix(sampleDists)
  rownames(sampleDistMat) <- colData(RNAnorm)$PatID
  
  res.hc <- sampleDists %>% 
  hclust(method = "ward.D2")

  ##verify the cluster tree
  # Compute cophentic distance
  res.coph <- cophenetic(res.hc)
  # Correlation between cophenetic distance and
  # the original distance
  cor(sampleDists, res.coph)

  res.hc$labels <- res.hc$labels <- paste(colData(RNAnorm)[,variant] ) #, "_", colData(RNAnorm)[,variant])    # Compute hierachical clustering
  #res.hc$labels <- factor(res.hc$labels, levels = levels(addNA(res.hc$labels)), labels = c(levels(res.hc$labels), 2), exclude = NULL)

  ##visualize with dendextend
  #dend <- as.dendrogram(res.hc)
  mycols <- c("#2E9FDF", "#00AFBB", "#E7B800", "#FC4E07")
  mycols <-c( "#969696", "#525252", "#969696" , "#000000")
  dend <-  as.dendrogram(res.hc) %>%
    set("branches_lwd", 2) %>% # Branches line width
    set("branches_k_color", mycols, k = 2)
  
  
  colMat <- matrix(nrow = ncol(RNAnorm), ncol = 1)
colMat[,1] <- ifelse(colData(RNAnorm)[,variant] == levels(colData(RNAnorm)[,variant])[1], "#00AFBB", ifelse(colData(RNAnorm)[,variant] == levels(colData(RNAnorm)[,variant])[2], "#E7B800", "grey"))

  plot(dend, main = variant)
  colored_bars(colMat, dend, rowLabels = variant, y_scale = 30, y_shift = 0.1, cex.rowLabels = 1.8)
}

lapply(cond, hier_plot)


```
