---
title: "Gene_signatures"
author: "Almut Luetge"
date: "September 6, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

06.09.2017
#Aim: Find gene signatures for mutational status of CLL patients

1.) Find differentially expressed genes
    a) Which mutations are we interested in? All mutations with at least 7 cases (3%)?
    b) What factors shall we account for? IGHV? c1c2? Are there any differences

2.) Perform gene set enrichment analysis
    a)Using Kegg annotation
    b)Using Hallmarks
    
3.) Summarizing gene signatures for interesting mutations
    a)Which mutations are considered interesting? How do we define them? By Deseq/GSEA results?
    b)How do we describe "signatures"? Literature?
    c) Is there any possibility to proof our results?


###1. Differential expression analysis
load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(reshape2)
library(genefilter)
library(Biobase)
library(SummarizedExperiment)
library(ggplot2)
library(gtable)
library(grid)
library(IHW)
library(corrplot)
library(polycor)
library(gridExtra)
library(factoextra)
library(ggpubr)
library(Rtsne)
library(RColorBrewer)

```


load data
```{r datasets}
#gene expression
load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV.RData")


#metaData
load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData")

#define plot directory
plotdir <-"/home/almut/Documents/CLL/results/plots/IGHV_Tsig_Tri12/"

```


#Differential expression analysis
Use multifactor design with T cell contamination (Tsig), Trisomy12 status and IGHV status as co-variable
```{r deseq}
###Deseq
ddsCLL <- estimateSizeFactors(ddsCLL)


#write a function to perform deseq for different genetic conditions 
diff <- function(cond){
  ddsCLL_new <- ddsCLL[,!is.na(colData(ddsCLL)[,cond])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"IGHV"])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"trisomy12"])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"Tsig"])]
  design(ddsCLL_new) <- as.formula(paste("~ trisomy12 + Tsig +", paste(cond)))
  rnaRaw <- DESeq(ddsCLL_new, betaPrior = FALSE)
  res <- results(rnaRaw)
  resOrdered <- res[order(res$pvalue),]
}

gene_conditions <- c("del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")
  



## Caution: Takes a lot of time - load data instead
res_list <- lapply(gene_conditions, diff)

names(res_list) <- gene_conditions
#save(res_list, file="/home/almut/Documents/CLL/data/desResIGHV_Tsig_Tri12_25102017.RData")
#tri12: load("/home/almut/Documents/CLL/data/desResTrisomy12_Tsig_IGHV_25102017.RData")
#IGHV: load("/home/almut/Documents/CLL/data/desResIGHV_Tsig_Tri12_25102017.RData")
#Other: load("/home/almut/Documents/CLL/data/desRes_Tsig_IGHV_tri12_25102017.RData")


load("/home/almut/Documents/CLL/data/desRes_Tsig_IGHV_tri12_25102017.RData")
```


```{r DE}
DEres <- res_list
names(DEres) <- gene_conditions
```

How can we visualize the results to find interesting ones? 
Questions to adress:
How many genes are diff expressed?

```{r create csv of diff genes}

pCut <- 1

difftab <- function(condition){
  dataTab <- data.frame(DEres[[condition]])
  dataTab$ID <- rownames(dataTab)
  #filter using pvalues
  dataTab <- filter(dataTab, padj <= pCut) %>%
    arrange(padj) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol) #%>%
    #filter(log2FoldChange > 0)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  #dataTab <- data.frame(row.names = dataTab$Symbol, stat = dataTab$stat)
  rownames(dataTab) <- dataTab$ID
  #write.csv(dataTab, file=paste0("/home/almut/Documents/CLL/results/deseq/IGHV_Tsig_Tri12/", condition, "_diffGenes.csv"))
  dataTab
}

cond <- gene_conditions

#Only run when you want to write result tables! Change path according to test! 
sigRes <- lapply(cond, difftab)
names(sigRes) <- cond


```

#MA-plots
```{r ma-plots}
#Check Ma plots

myMaPlot <-function(condition){
  DESeq2::plotMA(DEres[[condition]], ylim=c(-5,5), main= paste(condition))
}

lapply(cond, myMaPlot)

```
Conditions with low abundance seem to have a weird shape. Genes with a low norm mean tend to have a high negative log fold change. 
What does this mean? Sig. genes seem to be only affected to a small amount.. Better not to set the pCut higher?!


#Histogramm of pvalues

```{r hist}

myHist <- function(condition){
  #pdf(paste0(plotdir,"pvalues/", condition, ".pdf"), height = 20, width = 20)
  res <- DEres[[condition]]
  hist(res$pvalue, breaks=100, col="skyblue", border="slateblue", main=paste0("Histogramm of pvalues:", condition))
  #dev.off()
}

lapply(cond, myHist)

```

#Volcano plot
```{r volcano plots, fig.width=5, fig.height=5}
myVolc <-function(condition){
  #pdf(paste0(plotdir,"volcano/", condition, ".pdf"), height = 20, width = 20)
  res<- data.frame(DEres[[condition]])
  plotTab <- mutate(res, group = ifelse(pvalue <= 0.01, ifelse(log2FoldChange  >1,"up","down"), "Not significant"))
  
  ggplot(data=plotTab,aes(x=log2FoldChange,y=-log10(pvalue),col=group)) + geom_point(size=2) + 
  geom_hline(yintercept = -log10(0.01), linetype="dotted")+ theme_bw() + 
  scale_color_manual(values =  c("Not significant" = "grey80", "up" = "pink", "down" = "lightblue")) + 
  #geom_text_repel(data = subset(pRes.no13q, p <= 0.01), aes(label=name),size=4,col="red",force=20) + 
  ylab("-log10(P value)") + xlab("log2(Fold change)") + ggtitle(condition) 
  #dev.off()
}
  
  
 volc <- lapply(cond, myVolc)
grid.arrange(grobs = volc, ncol=1)
```




##Gene expression table
```{r hierachical clustering, fig.height= 50, fig.width=20}
#Variance stabilization transformation of the raw data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
exprMat <- assay(RNAnorm)

#function to plot heatmap

myHeatmap <- function(condition){
  #svg(paste0(plotdir,"hi_clustering/", condition, ".svg"), height = 80, width = 20)
  colAnno <- as.data.frame(colData(RNAnorm)[,c("IGHV","trisomy12", condition)])
  sig_genes <- sigRes[[condition]]$ID
  #scale and censor
  exprMat.new <- log2(exprMat[sig_genes,])
  exprMat.new <- t(scale(t(exprMat.new)))
  exprMat.new[exprMat.new > 4] <- 4
  exprMat.new[exprMat.new < -4] <- -4

  colors <- colorRampPalette( rev(brewer.pal(8,"RdYlBu")) )(255)
  
  pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, col = colors,
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol), fontsize_row = 10,
         clustering_method = "ward.D2", fontsize = 25, main = paste(condition))
  #dev.off()
}

#Better run for specific mutations only --> many plots = slow!
#heatmapPlots<-lapply(cond, myHeatmap)
#lapply("del13q14", myHeatmap)

```

##Data overview using tsne
```{r tsne, fig.height=30, fig.width=30}
#top5000 most variant genes
sds <- rowSds(exprMat)
exprCLL <- t(exprMat[order(sds, decreasing = T),])
rownames(exprCLL) <-ddsCLL$PatID

#Calculate t-sne (t-distributed Stochastic Neighbor Embedding)
distSamp <- dist(exprCLL[,c(1:5000)])
tsneRes <- Rtsne(distSamp, perplexity = 10, theta = 0, max_iter = 2000, is_distance = T, dims =2)
tsneRes <-tsneRes$Y
rownames(tsneRes) <- labels(distSamp)
colnames(tsneRes) <- c("x", "y")
tsneRes <- data.frame(tsneRes)



tsne_feat <- sapply(c(cond,"Methylation", "IGHV", "trisomy12"), function(condition){
  tsneRes[,condition] <-colData(ddsCLL)[which(colData(ddsCLL)$PatID %in% rownames(tsneRes)), condition]
})


tsne_feat <- as.data.frame(tsne_feat)
tsneRes <- cbind(tsneRes, tsne_feat)

#plot using ggplot2
plist <- lapply(c(cond, "Methylation", "IGHV", "trisomy12"), function(feature){
  ggplot(tsneRes, aes_string(x = "x", y = "y", color = feature)) + geom_point(size = 7) + theme_classic() + theme(panel.border = element_rect(color = 'black', size = 1, linetype = 'solid', fill = NA), axis.ticks = element_line(color = "black", size = 1), text = element_text(size = 20)) + xlab("") + ylab("") + ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 40)) +
  theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 30), legend.title = element_text(face = "bold", size = 40))
})


grid.arrange(grobs = plist, ncol =2)

#save as ggpubr ggscatter plot for thesis
#lapply(c(cond, "Methylation", "IGHV", "trisomy12"), function(feature){
 # p <- ggscatter(tsneRes, x = "x", y = "y", color = feature, palette = c("#00AFBB", "#E7B800", "#FC4E07"),
  #ylab = "", xlab = "", legend = "right", main = feature) + coord_fixed()
  #ggsave(file= paste0("~/git/figures_thesis/tsnes/tsne_", feature, ".svg"), plot=p, width=7, height=7)
#})




```



#Enrichment analysis
#Enrichment - Piano package GSEA.
```{r GSEA, fig.width = 20, fig.height= 15}

#load gene set collection
#Hallmark
gsc <- loadGSC("/home/almut/Documents/CLL/data/h.all.v6.0.symbols.gmt", type="gmt")
#Kegg
gsc_Kegg <- loadGSC("/home/almut/Documents/CLL/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
tft <- loadGSC("/home/almut/Documents/CLL/data/c3.tft.v6.0.symbols.gmt", type="gmt")
c5 <- loadGSC("/home/almut/Documents/CLL/data/c5.all.v6.0.symbols.gmt", type="gmt")
c6 <- loadGSC("/home/almut/Documents/CLL/data/c6.all.v6.0.symbols.gmt", type="gmt")
c7 <- loadGSC("/home/almut/Documents/CLL/data/c7.all.v6.0.symbols.gmt", type="gmt")
cgp <- loadGSC("/home/almut/Documents/CLL/data/c2.cgp.v6.0.symbols.gmt", type="gmt")


setWiseGSA <- function(cond, gmtFile, numGenes){
  geneNam <- sigRes[[cond]]$Symbol
  pVal <- sigRes[[cond]]$padj
  fchange <- sigRes[[cond]]$log2FoldChange
  stat <- sigRes[[cond]]$stat
  gsaTab <- data.frame(row.names = geneNam, stat = stat)
  res <- runGSA(geneLevelStats = gsaTab,
                      geneSetStat = "gsea",
                     adjMethod = "fdr", gsc=gmtFile,
                     signifMethod = 'geneSampling', nPerm = 1000,
                      gsSizeLim=c(numGenes, Inf))
  Res <- arrange(GSAsummaryTable(res),desc(`Stat (dist.dir)`))
  #rename results to plot
  resPlot <- Res[, c(1:7)]
  colnames(resPlot) <- c("pathway", "gene_number", "stat", "p_up", "p.adj_up", "p_dn", "p.adj_dn")
  #write.csv(resPlot, file = paste0("/home/almut/Documents/CLL/results/deseq/all_IGHV_Tsig_tri12/", cond, "_diffPathways_GSEAKegg.csv"))
  resPlot <- resPlot %>% mutate(group = ifelse(stat > 0, "up", "down")) %>% filter(ifelse(group %in% "up", p_up < 0.1, p_dn < 0.1))
   
    

  p <- ggbarplot(resPlot, x = "pathway", y = "stat",
          fill = "group",           # change fill color by mpg_level
          color = "white",            # Set bar border colors to white
          palette = "jco",            # jco journal color palett. see ?ggpar
          sort.val = "asc",          # Sort the value in ascending order
          sort.by.groups = FALSE,     # Don't sort inside each group
          x.text.angle = 90,          # Rotate vertically x axis texts
          ylab = "Enrichment stat",
          legend.title = "Pathway",
          rotate = TRUE,
          font.x = 30, font.y = 30, font.legend = 30, legend = "right", 
          title = paste0("GSEA in ", cond),
          ggtheme = theme_pubr()
          ) + font("xy.text", size = 20) + font("title", size = 30, face = "bold")
  
 #ggsave(file= paste0("~/git/figures_thesis/enrichment/gseaKegg_", cond, ".svg"), plot=p, width=20, height=18)
  p
  }

cond <- gene_conditions
lapply(cond, FUN = "setWiseGSA", gmtFile = gsc_Kegg, numGenes = 1)

```


```{r Page, fig.width = 20, fig.height= 20}

#GSA on cluster defined by kmeans clustering. See cluster in heatmap...
setWiseGSA <- function(cond, gmtFile, numGenes){
  geneNam <- sigRes[[cond]]$Symbol
  pVal <- sigRes[[cond]]$padj
  fchange <- sigRes[[cond]]$log2FoldChange
  gsaTab <- data.frame(row.names = geneNam, stat = fchange)
  res <- runGSA(geneLevelStats = gsaTab, directions = fchange,
                      geneSetStat = "page",
                     adjMethod = "fdr", gsc=gmtFile,
                     signifMethod = 'nullDist',
                      gsSizeLim=c(numGenes, Inf))
  Res <- arrange(GSAsummaryTable(res),desc(`Stat (dist.dir)`))
  #rename results to plot
  resPlot <- Res[, c(1:7)]
  colnames(resPlot) <- c("pathway", "gene_number", "stat", "p_up", "p.adj_up", "p_dn", "p.adj_dn")
  resPlot <- resPlot %>% mutate(group = ifelse(stat > 0, "up", "down")) %>% filter(ifelse(group %in% "up", p_up < 0.1, p_dn < 0.1))

  ggbarplot(resPlot, x = "pathway", y = "stat",
          fill = "group",           # change fill color by mpg_level
          color = "white",            # Set bar border colors to white
          palette = "jco",            # jco journal color palett. see ?ggpar
          sort.val = "asc",          # Sort the value in ascending order
          sort.by.groups = FALSE,     # Don't sort inside each group
          x.text.angle = 90,          # Rotate vertically x axis texts
          ylab = "Enrichment z-score",
          legend.title = "Pathway",
          rotate = TRUE,
          font.x = 30, font.y = 30, font.legend = 30,
          ggtheme = theme_minimal()
          ) + font("xy.text", size = 20)
  }

cond <- gene_conditions
lapply(cond, FUN = "setWiseGSA", gmtFile = gsc_Kegg, numGenes = 5)

```
