---
title: "Gene_signatures"
author: "Almut Luetge"
date: "September 6, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##06.09.2017
##Aim: Find gene signatures for mutational status of CLL patients

#1.) Find differentially expressed genes
    a) Which mutations are we interested in? All mutations with at least 7 cases (3%)?
    b) What factors shall we account for? IGHV? c1c2? Are there any differences

#2.) Perform gene set enrichment analysis
    a)Using Kegg annotation
    b)Using Hallmarks
    
#3.) Summarizing gene signatures for interesting mutations
    a)Which mutations are considered interesting? How do we define them? By Deseq/GSEA results?
    b)How do we describe "signatures"? Literature?
    c) Is there any possibility to proof our results?


###1. Differential expression analysis
load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(reshape2)
library(genefilter)
library(Biobase)
library(SummarizedExperiment)
library(ggplot2)
library(gtable)
library(grid)
library(IHW)
library(corrplot)
library(polycor)
library(gridExtra)
library(factoextra)
library(Rtsne)

gmts = list(H=system.file("extdata","h.all.v5.1.symbols.gmt", package="pace"),
            C6=system.file("extdata","c6.all.v5.1.symbols.gmt", package="pace"))
```


load data
```{r datasets}
#gene expression
#load("/home/almut/Documents/CLL/data/ddsrna_20170908.RData")  #trimmed dataset with Tsig included by Junyan, to intergrate T-cell contamination as confounding factor 


load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_variants.RData")


#metaData
load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData")

#define plot directory
plotdir <-"/home/almut/Documents/CLL/results/plots/MF4_7_IGHV/"

```


#Add metadata to colData (dds)
```{r prepare metadata}
##Prepare dds dataset --> add patmeta data and remove non CLL patients

#remove Y-chromosome
dds <- dds[!rowData(dds)$chromosome %in% c("Y"),]

#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
ddsCLL <- dds[,colData(dds)$diag %in% cll_vector]


#number genes and samples
dim(dds)
dim(ddsCLL)

# Remove rows/genes with too few counts
keep <-apply(counts(ddsCLL), 1, function(x) any(x >= 10))
ddsCLL <- ddsCLL[keep,]

#Add meta data
#generate Matrix with interesting features
patMat <- patMeta[which(patMeta$Patient.ID %in% colData(ddsCLL)$PatID),]
genomDat <- apply(patMat[,c(12:98)], 2, function(x) as.numeric(as.character(x)))
intGenom <- genomDat[, which(colSums(genomDat, na.rm=T) > 6)]

patMat <-patMat[, c("Patient.ID", "IGHV.status","Methylation_Cluster", colnames(intGenom), "c1c2")]
rownames(intGenom) <- patMat$Patient.ID
patMat <- as.data.frame(patMat)
patMat <- lapply(patMat, function(x) as.factor(x))
patMat <- as.data.frame(patMat)


rownames(patMat) <- patMat$Patient.ID

patMat <- patMat[colData(ddsCLL)$PatID,]

colData(ddsCLL) <- cbind(colData(ddsCLL), patMat[,-which(colnames(patMat) %in% c("Patient.ID", "trisomy12", "IGHV.status", "del13q14"))])


```

##Data overview

Summarize data of most common variants
For our analysis we only consider Variants with more then 3% prevalence as interesting
```{r common variants, fig.height=10, fig.width=30}
#Interesting variants
#write.csv(patMat, file=paste0("/home/almut/Documents/CLL/results/mostCommonVariants_dds16082017.csv"))

patMat_new <-intGenom[rownames(patMat),]
patMat_new <- cbind(patMat[,c("IGHV.status","Methylation_Cluster", "c1c2")], patMat_new)
IGHV_counts <- patMat_new$IGHV.status 

overview <- as_data_frame(patMat_new) %>% dplyr::group_by(IGHV.status) %>% dplyr::summarize_if(is.numeric, sum, na.rm =T) 
Sum <- as_data_frame(patMat_new) %>% dplyr::summarize_if(is.numeric, sum, na.rm =T)
IGHV <- as_data_frame(IGHV_counts) %>% dplyr::summarize(M = sum(value %in% "M"), U = sum(value %in% "U"), "NA" = sum(is.na(value)), Total = n())

Sum$IGHV.status <- "Total"
Sum <- Sum[,colnames(overview)]

IGHV <- t(IGHV)

sum_tab <- rbind(overview,Sum)
sum_tab <- cbind(sum_tab, IGHV)



sum_melted <- melt(sum_tab)


#plot
ggplot(sum_melted, aes(x = variable, y = factor(IGHV.status, levels = c("Total", "NA", "M", "U")))) +  geom_tile(aes(fill = value)) + 
    geom_text(aes(label = round(value, 1)), size =15) + ylab("IGHV.status") +
    scale_fill_gradient(low = "white", high = "red") + theme(axis.title.x = element_text(face="bold", size=50), axis.text.x=element_text(size=40, angle = 90)) + theme(axis.title.y = element_text(face="bold", size=50), axis.text.y= element_text(size=30))


```

```{r common variants, fig.height=10, fig.width=30}
#Interesting variants

patMat_new <-intGenom[rownames(patMat),]
patMat_new <- cbind(patMat[,c("IGHV.status","Methylation_Cluster", "c1c2")], patMat_new)

#function to create summary:
myVariant <- function(variant){
  variant_counts <- patMat_new[,variant]

  overview <- as_data_frame(patMat_new) %>% dplyr::group_by_(variant) %>% dplyr::summarize_if(is.numeric, sum, na.rm =T) 
  Sum <- as_data_frame(patMat_new) %>% dplyr::summarize_if(is.numeric, sum, na.rm =T)
  VAR <- as_data_frame(variant_counts) %>% dplyr::summarize("0" = sum(value %in% "0"), "1"= sum(value %in% "1"), "NA" = sum(is.na(value)), Total = n())

  Sum[,variant] <- "Total"
  Sum <- Sum[,colnames(overview)]

  VAR <- t(VAR)

  sum_tab <- rbind(overview,Sum)
  sum_tab <- cbind(sum_tab, VAR)



  sum_melted <- melt(sum_tab)


  #plot
  ggplot(sum_melted, aes_string(x = "variable", y = variant)) +  geom_tile(aes(fill = value)) + 
    geom_text(aes(label = round(value, 1)), size =15) + ylab(variant) +
    scale_fill_gradient(low = "white", high = "red") + theme(axis.title.x = element_text(face="bold", size=50), axis.text.x=element_text(size=40, angle = 90)) + theme(axis.title.y = element_text(face="bold", size=50), axis.text.y= element_text(size=30))
}

myVariant("trisomy12")

```




##Correlation of common variants
Corplots
```{r corplots, fig.height=15, fig.width= 15}
#Generate matrix with correlation coefficients

corMat <- hetcor(patMat[,-1], use = "pairwise.complete.obs")


corrplot.mixed(corMat$correlations, number.cex= 1.7, tl.cex=2, tl.pos = "lt", cl.cex=2, diag = "n")

cor.mtest <- function(mat, conf.level = 0.95){
  mat <- as.matrix(mat)
    n <- ncol(mat)
    p.mat <- lowCI.mat <- uppCI.mat <- matrix(NA, n, n)
    diag(p.mat) <- 0
    diag(lowCI.mat) <- diag(uppCI.mat) <- 1
    for(i in 1:(n-1)){
        for(j in (i+1):n){
            tmp <- cor.test(mat[,i], mat[,j], conf.level = conf.level)
            p.mat[i,j] <- p.mat[j,i] <- tmp$p.value
            lowCI.mat[i,j] <- lowCI.mat[j,i] <- tmp$conf.int[1]
            uppCI.mat[i,j] <- uppCI.mat[j,i] <- tmp$conf.int[2]
        }
    }
    return(list(p.mat, lowCI.mat, uppCI.mat))
}

res1 <- cor.mtest(corMat$correlations,0.95)
res2 <- cor.mtest(corMat$correlations,0.99)

corrplot.mixed(corMat$correlations, p.mat = res1[[1]], sig.level=0.1, insig = "blank", number.cex= 1.7, tl.cex=2, tl.pos = "lt", cl.cex=2, diag = "n")

```









#Do differential expression analysis
#Shall we add IGHV status as multifactor design to deseqmodel? c1c2groups? 
#Starting without multifactordesign because many of the mutations show so low frequencies, but should do another run with at least ighv for the more common ones: gene_conditions <- c("del11q22.3","trisomy12", "del13q14", "del17p13", "SF3B1", "TP53")
```{r deseq}
###Deseq
ddsCLL <- estimateSizeFactors(ddsCLL)


#write a function to perform deseq for different genetic conditions 
diff <- function(cond){
  ddsCLL_new <- ddsCLL[,!is.na(colData(ddsCLL)[,cond])]
  #ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"IGHVmeth"])]
  #ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"trisomy12"])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"Tsig"])]
  colData(ddsCLL_new)[,"IGHVmeth"] <-droplevels(colData(ddsCLL_new)[,"IGHVmeth"])
  design(ddsCLL_new) <- as.formula(paste("~ c1c2 + Tsig +", paste(cond)))
  rnaRaw <- DESeq(ddsCLL_new, betaPrior = FALSE)
  res <- results(rnaRaw)
  resOrdered <- res[order(res$pvalue),]
  #resSig <- subset(resOrdered, padj < 0.01)
  #rownames(resSig)
}

gene_conditions <- c("IGHVmeth", "trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")

##Takes a lot of time - load data instead
res_list <- lapply(gene_conditions, diff)

names(res_list) <- gene_conditions
save(res_list, file="/home/almut/Documents/CLL/data/desRes_MF2_7IGHV_06092017.RData")

#load("/home/almut/Documents/CLL/data/desRes_MF3_7_06092017.RData")
```


```{r DE}
DEres <- res_list
names(DEres) <-c("del13q14", "IGVH", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")
```

How can we visualize the results to find interesting ones? 
Questions to adress:
How many genes are diff expressed?

```{r create csv of diff genes}

pCut <- 0.05

difftab <- function(condition){
  dataTab <- data.frame(DEres[[condition]])
  dataTab$ID <- rownames(dataTab)
  #filter using pvalues
  dataTab <- filter(dataTab, padj <= pCut) %>%
    arrange(padj) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol) #%>%
    #filter(log2FoldChange > 0)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  #dataTab <- data.frame(row.names = dataTab$Symbol, stat = dataTab$stat)
  rownames(dataTab) <- dataTab$ID
 # write.csv(dataTab, file=paste0("/home/almut/Documents/CLL/results/deseq/MF4_7/", condition, "_diffGenes.csv"))
  dataTab
}

cond <- c("del13q14","IGHV", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")

#Only run when you want to write result tables! Change path according to test! 
sigRes <- lapply(cond, difftab)
names(sigRes) <- cond

```

#MA-plots
```{r ma-plots}
#Check Ma plots

myMaPlot <-function(condition){
  DESeq2::plotMA(DEres[[condition]], ylim=c(-5,5), main= paste(condition))
}

lapply(cond, myMaPlot)

```
Conditions with low abundance seem to have a weird shape. Genes with a low norm mean tend to have a high negative log fold change. 
What does this mean? Sig. genes seem to be only affected to a small amount.. Better not to set the pCut higher?!


#Histogramm of pvalues

```{r hist}

myHist <- function(condition){
  pdf(paste0(plotdir,"pvalues/", condition, ".pdf"), height = 20, width = 20)
  res <- DEres[[condition]]
  hist(res$pvalue, breaks=100, col="skyblue", border="slateblue", main=paste0("Histogramm of pvalues:", condition))
  dev.off()
}

lapply(cond, myHist)

```

#Volcano plot
```{r volcano plots, fig.width=10, fig.height=20}
myVolc <-function(condition){
  #pdf(paste0(plotdir,"volcano/", condition, ".pdf"), height = 20, width = 20)
  res<- data.frame(DEres[[condition]])
  plotTab <- mutate(res, group = ifelse(pvalue <= 0.01, ifelse(log2FoldChange  >1,"up","down"), "Not significant"))
  
  ggplot(data=plotTab,aes(x=log2FoldChange,y=-log10(pvalue),col=group)) + geom_point(size=2) + 
  geom_hline(yintercept = -log10(0.01), linetype="dotted")+ theme_bw() + 
  scale_color_manual(values =  c("Not significant" = "grey80", "up" = "pink", "down" = "lightblue")) + 
  #geom_text_repel(data = subset(pRes.no13q, p <= 0.01), aes(label=name),size=4,col="red",force=20) + 
  ylab("-log10(P value)") + xlab("log2(Fold change)") + ggtitle(condition) 
  #dev.off()
}
  
  
 volc <- lapply(cond, myVolc)
grid.arrange(grobs = volc, ncol=2)
```




##Gene expression table
```{r hierachical clustering, fig.height= 50, fig.width=20}
#Variance stabilization transformation of the raw data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
exprMat <- assay(RNAnorm)

#function to plot heatmap

myHeatmap <- function(condition){
  #pdf(paste0(plotdir,"hi_clustering/", condition, ".pdf"), height = 100, width = 40)
  colAnno <- as.data.frame(colData(RNAnorm)[,c("IGVH","trisomy12", condition)])
  sig_genes <- sigRes[[condition]]$ID
  #scale and censor
  exprMat.new <- log2(exprMat[sig_genes,])
  exprMat.new <- t(scale(t(exprMat.new)))
  exprMat.new[exprMat.new > 4] <- 4
  exprMat.new[exprMat.new < -4] <- -4

  colors <- colorRampPalette( rev(brewer.pal(8,"RdYlBu")) )(255)
  
  pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, col = colors,
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol), fontsize_row = 10,
         clustering_method = "ward.D2", fontsize = 25, main = paste(condition))
  #dev.off()
}

#Better run for specific mutations only --> many plots = slow!
#heatmapPlots<-lapply(cond, myHeatmap)
lapply("del13q14", myHeatmap)

```

##Data overview using tsne
```{r tsne, fig.height=100, fig.width=35}
#top5000 most variant genes
sds <- rowSds(exprMat)
exprCLL <- t(exprMat[order(sds, decreasing = T),])
rownames(exprCLL) <-patMat$Patient.ID

#Calculate t-sne (t-distributed Stochastic Neighbor Embedding)
distSamp <- dist(exprCLL[,c(1:5000)])
tsneRes <- Rtsne(distSamp, perplexity = 10, theta = 0, max_iter = 2000, is_distance = T, dims =2)
tsneRes <-tsneRes$Y
rownames(tsneRes) <- labels(distSamp)
colnames(tsneRes) <- c("x", "y")
tsneRes <- data.frame(tsneRes)



colnames(patMat) <- gsub("IGHV.status", "IGVH", colnames(patMat))


tsne_feat <- sapply(c(cond,"Methylation_Cluster"), function(condition){
  tsneRes[,condition] <-patMat[rownames(tsneRes), condition]
})


tsne_feat <- as.data.frame(tsne_feat)
tsneRes <- cbind(tsneRes, tsne_feat)

#plot using ggplot2
plist <- lapply(c(cond, "Methylation_Cluster"), function(feature){
  ggplot(tsneRes, aes_string(x = "x", y = "y", color = feature)) + geom_point(size = 7) + theme_classic() + theme(panel.border = element_rect(color = 'black', size = 1, linetype = 'solid', fill = NA), axis.ticks = element_line(color = "black", size = 1), text = element_text(size = 20)) + xlab("") + ylab("") + ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 40)) +
  theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 30), legend.title = element_text(face = "bold", size = 40))
})


grid.arrange(grobs = plist, ncol =2)



```





###2.) Enrichment analysis
The gene set enrichment analysis will be performed by using the MSigDB gene set collections C6 and Hallmark (http://software.broadinstitute.org/gsea/msigdb/ ). For each collection we will show the top five enriched gene sets and respective differentially expressed genes. Gene set enrichment analysis will be performed on the ranked gene lists using the Parametric Analysis of Gene Set Enrichment (PAGE).


### Functions for enrichment analysis and plots

Define cut-off.
```{r}
pCut = 0.05
```

Function to run GSEA or PAGE in R.
```{r gsea_function}
runGSEA <- function(inputTab, gmtFile, GSAmethod="gsea", nPerm=1000){
    inGMT <- loadGSC(gmtFile,type="gmt")
    rankTab <- inputTab[order(inputTab[,1],decreasing = TRUE),,drop=FALSE] #re-rank by score 
    if (GSAmethod == "gsea"){
        #readin geneset database
        #GSEA analysis
        res <- runGSA(geneLevelStats = rankTab,
                      geneSetStat = GSAmethod,
                      adjMethod = "fdr", gsc=inGMT,
                      signifMethod = 'geneSampling', nPerm = nPerm)
        GSAsummaryTable(res)
    } else if (GSAmethod == "page"){
        res <- runGSA(geneLevelStats = rankTab,
                      geneSetStat = GSAmethod,
                      adjMethod = "fdr", gsc=inGMT,
                      signifMethod = 'nullDist')
        GSAsummaryTable(res)
    }
}
```

Function which run the GSE for each response group.
```{r}
runGSE = function(gmt) {
  
  Res <- list()
  for (i in names(DEres)) {
    dataTab <- data.frame(DEres[[i]])
    dataTab$ID <- rownames(dataTab)
    #filter using pvalues
    dataTab <- filter(dataTab, pvalue <= pCut) %>%
                  arrange(pvalue) %>% 
                  mutate(Symbol = rowData(ddsCLL[ID,])$symbol)
    dataTab <- dataTab[!duplicated(dataTab$Symbol),]
    dataTab <- dataTab[!is.na(dataTab$Symbol),]
    statTab <- data.frame(row.names = dataTab$Symbol, stat = dataTab$stat)
    resTab <- runGSEA(inputTab=statTab, gmtFile=gmt, GSAmethod="page")
    Res[[i]] <- arrange(resTab,desc(`Stat (dist.dir)`))
  }
  Res
}
```

Function to get the list of genes enriched in a set.
```{r get_genes_function}
getGenes <- function(inputTab, gmtFile){
  geneList <- loadGSC(gmtFile,type="gmt")$gsc
  enrichedUp <- lapply(geneList, function(x) 
    intersect(rownames(inputTab[inputTab[,1] >0,,drop=FALSE]),x))
  enrichedDown <- lapply(geneList, function(x)
    intersect(rownames(inputTab[inputTab[,1] <0,,drop=FALSE]),x))
  return(list(up=enrichedUp, down=enrichedDown))
}
```

A function to plot the heat map of intersection of genes in different gene sets.
```{r intersection_heatmap}
plotSetHeatmap <- 
  function(geneTab, enrichTab, topN, gmtFile, tittle="", anno=TRUE) {
    if (nrow(enrichTab) < topN) topN <- nrow(enrichTab)
    enrichTab <- enrichTab[seq(1,topN),]

    geneList <- getGenes(geneTab,gmtFile)
    
    geneList$up <- geneList$up[enrichTab[,1]]
    geneList$down <- geneList$down[enrichTab[,1]]
    
    #form a table 
    allGenes <- unique(c(unlist(geneList$up),unlist(geneList$down)))
    allSets <- unique(c(names(geneList$up),names(geneList$down)))
    plotTable <- matrix(data=NA,ncol = length(allSets),
                        nrow = length(allGenes),
                        dimnames = list(allGenes,allSets))
    for (setName in names(geneList$up)) {
      plotTable[geneList$up[[setName]],setName] <- 1
    }
    for (setName in names(geneList$down)) {
      plotTable[geneList$down[[setName]],setName] <- -1
    }
    geneOrder <- rownames(plotTable[order(rowSums(plotTable, na.rm = TRUE),
                                          decreasing =FALSE),])
    plotTable <- melt(plotTable)
    colnames(plotTable) <- c("gene","set","value")
    plotTable$gene <- as.character(plotTable$gene)
    plotTable$value <- replace(plotTable$value,
                               plotTable$value %in% c(1), "Up")
    plotTable$value <- replace(plotTable$value,
                               plotTable$value %in%  c(-1), "Down")
    geneSymbol <- geneOrder
    
    if (anno) { #if add functional annotations in addition to gene names
      annoTab <- tibble(symbol = rowData(ddsCLL)$symbol, 
                        anno = sapply(rowData(ddsCLL)$description,
                                      function(x) unlist(strsplit(x,"[[]"))[1]))
      annoTab <- annoTab[!duplicated(annoTab$symbol),]
      annoTab$combine <- sprintf("%s (%s)",annoTab$symbol, annoTab$anno)
      plotTable$gene <- annoTab[match(plotTable$gene,annoTab$symbol),]$combine
      geneOrder <- annoTab[match(geneOrder,annoTab$symbol),]$combine
    }
    
    plotTable$gene <- factor(plotTable$gene, levels =geneOrder)
    plotTable$set <- factor(plotTable$set, levels = enrichTab[,1])
    
    g <- ggplot(plotTable, aes(x=set, y = gene)) +
      geom_tile(aes(fill=value), color = "black") +
      scale_fill_manual(values = c("Up"="red","Down"="blue")) +
      xlab("") + ylab("") + ggtitle(names(geneTab))+theme_classic() +
      theme(axis.text.x=element_text(size=8, angle = 60, hjust = 0),
            axis.text.y=element_text(size=8),
            axis.ticks = element_line(color="white"),
            axis.line = element_line(color="white"),
            legend.position = "none") +
      scale_x_discrete(position = "top") +
      scale_y_discrete(position = "right")
    
    # construct the gtable
    #wdths = c(0.05, 0.25*length(levels(plotTable$set)), 5)
    #hghts = c(2.8, 0.1*length(levels(plotTable$gene)), 0.05)
    
    wdths = c(3, 0.25*length(levels(plotTable$set)), 7)
    hghts = c(3, 0.1*length(levels(plotTable$gene)), 0.05)
    
    gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    ## make grobs
    ggr = ggplotGrob(g)
    ## fill in the gtable
    gt = gtable_add_grob(gt, gtable_filter(ggr, "panel"), 2, 2)
    gt = gtable_add_grob(gt, ggr$grobs[[5]], 1, 2) # top axis
    gt = gtable_add_grob(gt, ggr$grobs[[9]], 2, 3) # right axis
    
    return(list(list(plot=gt,
                     width=sum(wdths),
                     height=sum(hghts),
                     genes=geneSymbol)))
}    
```

Prepare stats per gene for plotting.
```{r}
statTab = setNames(lapply(c("trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12"), function(gr) {
  dataTab <- data.frame(DEres[[gr]])
  dataTab$ID <- rownames(dataTab)
  #filter using pvalues
  dataTab <- filter(dataTab, pvalue <= pCut) %>%
    arrange(pvalue) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol) #%>%
    #filter(log2FoldChange > 0)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  data.frame(row.names = dataTab$Symbol, stat = dataTab$stat)
  
}), nm=c("trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12"))
```

### Geneset enrichment based on Hallmark set (H)

Perform enrichment analysis using PAGE method.
```{r run_GSE_hallmark, warning=FALSE, message= FALSE}
hallmarkRes = runGSE(gmt=gmts[["H"]])
```

Plot heat map for enriched set.
```{r heatmapfunction}
grpl = sapply( c("trisomy12","del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12"), function(gr) {
  plotSetHeatmap(geneTab=statTab[[gr]],
                 enrichTab=hallmarkRes[[gr]],
                topN=7,gmtFile=gmts[["H"]], anno=TRUE, i)
})
```

Function to plot enriched pathways

```{r Enrichment plots, echo=TRUE, fig.path=plotdir, dev=c("png", "pdf"), fig.height=100, fig.width=50}
myPathways <- function(condition){
  pdf(paste0(plotdir, condition, ".pdf"), height = grpl[[condition]][["height"]], width = grpl[[condition]][["width"]])  
  grid.arrange(grpl[[condition]]$plot)
  dev.off()
}

lapply(cond, myPathways)

```











