---
title: "gene_expression_signature_all"
author: "Almut Luetge"
date: "November 8, 2017"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Aim: Define a gene varaiant specific gene expression signature


libraries
```{r load libraries}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(VennDiagram)
library(RColorBrewer)
library(circlize)
library(piano)
```

#data
```{r load data}
#dds data set. gene expression data + patmetadata
load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV_drugs.RData")

variant <- "BRAF"
int_genes <- c("BRAF", "RAB25", "RAB38", "MAPK13")
#int_genes <- c("EBF1", "SF3B1", "TERT", "TERK", "UQCC1")
#int_genes <- c("EBF1", "MED13L", "TERT", "TERK")
#int_genes <- c("NOTCH4", "MED13L", "TERT", "TERK")

#differentially expressed genes between groups with Tsig as confounding factor
#diff_genes <- read.csv(file=paste0("/home/almut/Documents/CLL/results/deseq/all_IGHV_Tsig_tri12/", variant, "_diffGenes.csv"))
diff_genes <- read.csv(file=paste0("/home/almut/Documents/CLL/results/deseq/multifactor_reduced/", variant, "_diffGenes.csv"))
rownames(diff_genes) <- diff_genes$X
diff_genes<- diff_genes[which(diff_genes$padj < 0.05),-1]
intr_genes <- diff_genes[which(diff_genes$padj < 0.01 & abs(diff_genes$log2FoldChange) > 4),-1]

#chromosome
#chromosome <- rowData(ddsCLL)[which(rownames(ddsCLL) %in% rownames(diff_genes)),]
#rownames(chromosome) <-rownames(ddsCLL)[which(rownames(ddsCLL) %in% rownames(diff_genes))]
#chromosome <- chromosome[rownames(diff_genes),]
#diff_genes$chromosome <- chromosome$chromosome
#intr_genes <- diff_genes[which(diff_genes$chromosome %in% "8" & abs(diff_genes$log2FoldChange) > 0.7),]


#Add diff pathway information
pathways <- read.csv(file=paste0("/home/almut/Documents/CLL/results/deseq/all_IGHV_Tsig_tri12/", variant, "_diffPathways_GSEAHallmark.csv"))
pathways <- pathways[,-1] 
pathways <- pathways %>% mutate(group = ifelse(stat > 0, "up", "down")) %>% filter(ifelse(group %in% "up", p_up < 0.1, p_dn < 0.1))


```


Add all relevant information into one matrix
```{r del17 gene expression}
#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)


#filter for sign. genes in del13
exprMat <- assay(RNAnorm)
exprVariant <- exprMat[rownames(diff_genes),]
colnames(exprVariant) <- colData(ddsCLL)$PatID
exprVariant.new <- log2(exprVariant)
exprVariant.new <- t(scale(t(exprVariant.new)))
exprVariant.new[exprVariant.new > 4] <- 4
exprVariant.new[exprVariant.new < -4] <- -4
rownames(exprVariant.new) <- rowData(RNAnorm[rownames(diff_genes),])$symbol


#Rowdata
#chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[rownames(diff_genes),])$chromosome)
rownames(chrom) <- rownames(exprVariant.new) 
colnames(chrom ) <- "chromosome"

#select for chromosome 8
chrom$chromosome <- ifelse(chrom$chromosome == 8, 8, "other")

#pathway Annotation
topGO <- sort.int(abs(pathways$stat), index.return = T, decreasing = T)$ix[1:5]
topPath <- pathways[topGO,]


#Annotate Genes in this pathways
gsc <- loadGSC("/home/almut/Documents/CLL/data/h.all.v6.0.symbols.gmt", type="gmt")
gsc_Kegg <- loadGSC("/home/almut/Documents/CLL/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
tft <- loadGSC("/home/almut/Documents/CLL/data/c3.tft.v6.0.symbols.gmt", type="gmt")
c5 <- loadGSC("/home/almut/Documents/CLL/data/c5.all.v6.0.symbols.gmt", type="gmt")
c6 <- loadGSC("/home/almut/Documents/CLL/data/c6.all.v6.0.symbols.gmt", type="gmt")
c7 <- loadGSC("/home/almut/Documents/CLL/data/c7.all.v6.0.symbols.gmt", type="gmt")
cgp <- loadGSC("/home/almut/Documents/CLL/data/c2.cgp.v6.0.symbols.gmt", type="gmt")
topGsc <- gsc$gsc[as.character(topPath$pathway)]

topPathGenes <- lapply(topGsc, function(x){
  genes <- ifelse(diff_genes$Symbol %in% unlist(x), 1 , 0)
})  %>% do.call(cbind,.)

topPathGenes <- apply(topPathGenes, 2, function(x) as.factor(as.character(x)))
rownames(topPathGenes) <- diff_genes$Symbol

#Notch1:intr_genes <- gsc_Kegg$gsc[["KEGG_NOTCH_SIGNALING_PATHWAY"]] 
#BRAF:
intr_genes <- gsc$gsc[["HALLMARK_G2M_CHECKPOINT"]]

```


#Plot heatmap using complex heatmaps to show expression signature
```{r complex heatmap del13, fig.width= 35, fig.height=45}

#colors
colors <- colorRampPalette(rev( brewer.pal(10,"RdBu")) )(20)
annocol <- colorRampPalette( brewer.pal(11,"Paired") )(11)
annocolor <- list(variant = c("0" = annocol[3], "1" = annocol[4])) #, del8p12 = c("0" = annocol[7], "1" = annocol[8])) #, drugGroup = c("MEK" = "#c9a600", "mTOR" = "#de0066", "BTK" = "#ff9730", "none" = "#0076fa"))
names(annocolor)[1] <- variant

rowcolors <-colorRampPalette(brewer.pal(5, "Set1"))(5)
rowcolors[6] <- "white"
#chromcol <- list(chromosome = c("1" ="#00d3b4", "2" ="#f443b9", "3" = "#6cde56", "4" = "#932db4", "5" ="#dcc608", "6"= "#0a54d9", "7" = "#fea114", "8"= "#0098ec","12" ="#fd4b43","10" = "#01e09f", "11" = "#ff6aab", "9" = "#a8d37e", "13" = "#a59eff","14" = "#be6f00", "15" = "#03d1f5", "16" = "#a02008", "17" = "#89c8ff", "18" = "#7a7600", "19" ="#005b8f", "20" = "#ffad61","21" = "#80386c", "22" = "#e3c27d", "X" ="#843e29", "MT" = "#ff938b"))

chromcol <- list(chromosome = c("8" =annocol[6], "other" = annocol[5]))  
                
annoRowcolor <- list()
  for(i in c(1:length(topGO)))(
  annoRowcolor[[i]] = c("0"= rowcolors[6], "1" = rowcolors[i]) 
  )
  
names(annoRowcolor) <- topPath$pathway


 
ha_row = rowAnnotation(df = topPathGenes, col = annoRowcolor, annotation_width = unit(c(rep(1.4, 5)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 30), labels_gp = gpar(fontsize = 22),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm"), gap = unit(3, "cm")))


ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(1.7, "cm"), annotation_legend_param = list(ncol = 2, title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 35),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm"), gap = unit(1,"cm")))

feature <- as.data.frame(colData(ddsCLL)[,c(variant)]) #, "del8p12")])#,"drugGroup")])
colnames(feature) <- c(variant) #,"del8p12") #, "drugGroup")

ha_col <- HeatmapAnnotation(feature, col = annocolor, annotation_height = unit(c(rep(1.7, 1)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 35),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm")))

h1 <- Heatmap(exprVariant.new ,                                                     #Cluster param
              km = 2,
              clustering_distance_columns = "euclidean",
              clustering_method_columns = "ward.D2",
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = paste0("Gene signature: ", variant),                      #Graphic parameter
              col = colors,
              column_title_gp = gpar(fontsize = 60, fontface = "bold"), 
              heatmap_legend_param = list(title = "expr", 
                                          title_gp = gpar(fontsize = 40), 
                                          grid_height = unit(1.9, "cm"), 
                                          grid_width = unit(1.9, "cm"), 
                                          gap = unit(2, "cm"), 
                                          labels_gp = gpar(fontsize = 30)), 
              column_dend_height = unit(2.9, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = FALSE, 
              top_annotation = ha_col)


#Annotate known genes from litertaure
subset <- as.data.frame(rowData(RNAnorm[rownames(diff_genes),]))
subset1 <- subset[which(subset$symbol %in% int_genes),]
#subset2 <- subset[which(subset$symbol %in% intr_genes$Symbol),]
subset2 <- subset[which(subset$symbol %in% intr_genes),]
subset <- rbind(subset1, subset2)
rownames(subset) <- subset$symbol
geneIDs <- which(rownames(exprVariant.new) %in% rownames(subset))
labels <- rownames(exprVariant.new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 35)), width = unit(9, "cm"))

#svg(filename=paste0("~/git/figures_thesis/gene_expr/gene_expr", variant, "_gsea_Hallmark.svg"), width=40, height=45)
#pdf(file=paste0("~/git/figures_thesis/gene_expr/gene_expr", variant, "_gsea_Hallmark.pdf"), width=35, height=40)
draw(h1  + ha_genes + ha_row )# +  ha_row + ha_chrom)
#dev.off()




```

#Plot chromosome distributions
Is there any connection to genes on chromosome 12 (more or less expressed?)
```{r plot chromosome dist, fig.width= 8, fig.height=8}

VariantGroups <- diff_genes %>%  mutate(group = ifelse(stat > 0, "up", "down"))
rownames(VariantGroups) <- rownames(diff_genes)


chromDist <- lapply(c("up", "down"), function(group){
  geneID <- rownames(VariantGroups)[which(VariantGroups$group %in% group)]
  chromo <- as.data.frame(rowData(RNAnorm[geneID,])$chromosome)
  rownames(chromo) <- geneID 
  colnames(chromo ) <- "chromosome"
  
  
  chrom <- chromo %>% dplyr::select(chromosome) %>% group_by(chromosome) %>% summarise(total = n())

  chrom$label <- paste0(chrom$chromosome, "(", chrom$total, ")")
  
  #svg(filename=paste0("~/git/figures_thesis/", variant, "/chromosom_dist_", group, ".svg"), width=6, height=6)
  pie(chrom$total, col=colorRampPalette(brewer.pal(11,'Spectral'))(25), border="white", labels=chrom$label, main = paste0("Chromosomal gene distribution in ", variant, ": ", group))
  #dev.off()
})

```


