---
title: "Multivariable model"
author: "Almut Luetge"
date: "December 4, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##06.09.2017
##Aim: Find gene signatures for mutational status of CLL patients in a multi variable model

#1.) Find differentially expressed genes
    - creating a multi variable model with all genetic variants as confounding factors

#2.) What are the changes to our previous results?
    
#3.) How can we visualize them? 

#4.) Is a multivariable model better?


###1. Differential expression analysis
load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(genefilter)
library(Biobase)
library(SummarizedExperiment)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)

```


load data
```{r datasets}
#gene expression
load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV.RData")


#metaData
load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData")

#define plot directory
#plotdir <-"/home/almut/Documents/CLL/results/plots/IGHV_Tsig_Tri12/"

```

```{r}
###Deseq
#remove NAs in all integrated variables (variants)
patnoNA <- as_tibble(colData(ddsCLL)) %>% filter(!is.na(IGHV) & !is.na(trisomy12) & !is.na(del13q14) & !is.na(del8p12) & !is.na(gain8q24) & !is.na(del11q22.3) & !is.na(del17p13) & !is.na(BRAF) & !is.na(NOTCH1) & !is.na(SF3B1) & !is.na(TP53) & !is.na(ATM) & !is.na(MED12) & !is.na(Tsig)) %>% select(PatID)
ddsCLL_new <- ddsCLL[,which(colData(ddsCLL)$PatID %in% patnoNA$PatID)]
ddsCLL_new <- estimateSizeFactors(ddsCLL_new)


#write a function to perform deseq for different genetic conditions 
diff <- function(cond){
  #colData(ddsCLL_new)[,"IGHV"] <-droplevels(colData(ddsCLL_new)[,"IGHV"])
  multiVars <- gene_conditions[-which(gene_conditions %in% cond)]
  #mult <-c("Tsig", "IGHV", "trisomy12")
  #nocond <-ifelse(cond %in% "IGHV", "trisomy12", "IGHV")
  #multi1 <- paste0("Tsig  + ", nocond)
  #multiVars2 <- ifelse(cond %in% mult, multi1 , "Tsig + IGHV + trisomy12")           
  design(ddsCLL_new) <- as.formula(paste("~ Tsig +", paste(multiVars, collapse = " + "), " + ", paste(cond)))
  rnaRaw <- DESeq(ddsCLL_new, betaPrior = FALSE, test="LRT", reduced=as.formula(paste("~  Tsig + ", paste(multiVars, collapse = " + "))))
  res <- results(rnaRaw)
  resOrdered <- res[order(res$pvalue),]
  #resSig <- subset(resOrdered, padj < 0.01)
  #rownames(resSig)
}

gene_conditions <- c("IGHV", "trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")



##Takes a lot of time - load data instead
res_list <- lapply(gene_conditions, diff)

names(res_list) <- gene_conditions
save(res_list, file="/home/almut/Documents/CLL/data/desRes_MultifactorReducedII.RData")

#load("/home/almut/Documents/CLL/data/desRes_Multifactor.RData")

```



```{r DE}
DEres <- res_list
names(DEres) <- gene_conditions
```

How can we visualize the results to find interesting ones? 
Questions to adress:
How many genes are diff expressed?

```{r create csv of diff genes}

pCut <- 0.01

difftab <- function(condition){
  dataTab <- data.frame(DEres[[condition]])
  dataTab$ID <- rownames(dataTab)
  #filter using pvalues
  dataTab <- filter(dataTab, padj <= pCut) %>%
    arrange(padj) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol) #%>%
    #filter(abs(log2FoldChange) > 2)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  #dataTab <- data.frame(row.names = dataTab$Symbol, stat = dataTab$stat)
  rownames(dataTab) <- dataTab$ID
  write.csv(dataTab, file=paste0("/home/almut/Documents/CLL/results/deseq/multifactor_reducedII/", condition, "_diffGenes.csv"))
  dataTab
}

#Only run when you want to write result tables! Change path according to test! 
sigRes_new <- lapply(gene_conditions, difftab)
names(sigRes_new) <- gene_conditions


```

#2.compare with previous results
load data
```{r old results}
load("/home/almut/Documents/CLL/data/sig_DiffGenes_250917.RData")
sigRes_old <- sigRes
```


```{r venn diagramm}
#create a venn diagramm with old and new diffGenes
myVenn <-function(variant){
  new <- sigRes_new[[variant]]
  old <- sigRes_old[[variant]]
  
  venntab <- unique(c(as.vector(new$Symbol), as.vector(old$Symbol)))
  vennTab_new <- as.data.frame(venntab) %>% mutate(new = ifelse(as.character(venntab) %in% new$Symbol, 1, 0)) %>% mutate(old = ifelse(as.character(venntab) %in% old$Symbol, 1, 0))
  
  new_only <- vennTab_new %>% filter(old == 0 & new == 1) %>% select(venntab)
  old_only <- vennTab_new %>% filter(old == 1 & new == 0) %>% select(venntab)
  
  vennTab_new <- vennTab_new[,-1]
  rownames(vennTab_new) <- venntab
  new_area <- vennTab_new %>% filter(new == 1)
  old_area <- vennTab_new %>% filter(old == 1)
  Both_area <- vennTab_new %>% filter(new == 1 & old == 1)

  venn_pair <- draw.pairwise.venn(nrow(new_area), nrow(old_area), nrow(Both_area), c("multi", "original"), fill = c("#00AFBB", "#E7B800"), cex = rep(2, 3), cat.cex = rep(2, 2),  ind = TRUE)

  svg(filename=paste0("~/git/figures_thesis/multifactor_reducedII/venn_", variant ,".svg"), width=8.5, height=8)
  grid.draw(venn_pair)
  #grid.arrange(grobs=g, top=paste(variant))
  dev.off()
  grid.newpage()
  }
lapply(gene_conditions, myVenn)



```

#Vergleiche Gene expression multivariate - original Model

```{r expression model}

my_compare <-function(variant){
  oriGene <- data.frame(sigRes_old[[variant]])
  multiGene <- data.frame(DEres[[variant]])
  multiGene <- multiGene[oriGene$ID,]
  pVal <- cbind(oriGene$padj,multiGene$padj)
  colnames(pVal) <-c("originalModel", "multiVariableModel")
  pVal <- as.data.frame(pVal)
  ggscatter(pVal,  x = "originalModel", y = "multiVariableModel")
   )
} 



```



