---
title: "trisomy12"
author: "Almut Luetge"
date: "November 3, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Aim: Define a Trisomy12 expression signature

###paper
- Which genes shall be annotated?
- Do we have a certain pathway/message we want to emphazise?
- How many genes? Pattern is much clearer without filtering for fold changes
- dendogramm and unsupervised clustering on samples? (emphasizes the connectivity, but looks less distinct)


libraries
```{r load libraries}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(piano)
library(ggsci)
library(RColorBrewer)
library(circlize)
```

#data
```{r load data}
#dds data set. gene expression data + patmetadata
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

#filter for parients without NA in trisomy12 
ddsCLL <- ddsCLL[, !is.na(colData(ddsCLL)["IGHV"]) & !is.na(colData(ddsCLL)["trisomy12"])]

#differentially expressed genes between IGHV groups with Tsig as confounding factor
tri12 <- read.csv(file="/home/almut/Dokumente/masterarbeit/results/noTsig/trisomy12_diffGenes.csv")
#multi <- read.csv(file="/home/almut/Dokumente/masterarbeit/results/trisomy12_diffGenes.csv")


rownames(tri12) <- tri12$X
tri12<- tri12[which(tri12$padj < 0.01 ),-1]
tri12<- tri12[which(abs(tri12$log2FoldChange) > 2 & tri12$baseMean > 500),]
tri12<- tri12[which(tri12$baseMean > 500),]

#NOT located on chromosome 12 or an IG gene
#chrom12 <- as_tibble(rowData(ddsCLL[rownames(tri12),])) %>%  mutate(geneID = rownames(tri12)) %>% filter(!grepl("IG",symbol))
#filter(chromosome != "12") %>%

int_genes <- tri12[which(tri12$padj < 0.001 & abs(tri12$log2FoldChange) > 5),]

#tri12 <- tri12[chrom12$geneID,]

#Add diff pathway information
#pathways <- read.csv(file="/home/almut/Dokumente/masterarbeit/results/deseq/Trisomy12_IGHV_Tsig/trisomy12_diffPathways_GSEAHallmark.csv")
#pathways <- pathways[,-1] 
#pathways <- pathways %>% mutate(group = ifelse(stat > 0, "up", "down")) %>% filter(ifelse(group %in% "up", p_up < 0.1, p_dn < 0.1))

mutStatus <- data.frame(colData(ddsCLL)) %>% arrange(trisomy12)
colnames(ddsCLL) <-colData(ddsCLL)$PatID
ddsCLL <- ddsCLL[, mutStatus$PatID]


```


Add all relevant information into one matrix
```{r tri12 gene expression}
#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)


#filter for sign. genes in Trisomy12 
exprMat <- assay(RNAnorm)
exprTri12 <- exprMat[rownames(tri12),]
colnames(exprTri12) <- colData(ddsCLL)$PatID
exprTri12.new <- log2(exprTri12)
exprTri12.new <- t(scale(t(exprTri12.new)))
exprTri12.new[exprTri12.new > 4] <- 4
exprTri12.new[exprTri12.new < -4] <- -4
rownames(exprTri12.new) <- rowData(RNAnorm[rownames(tri12),])$symbol

#Rowdata
#chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[rownames(tri12),])$chromosome)
rownames(chrom) <- rownames(tri12) 
colnames(chrom ) <- "chromosome"

#select for chromosome12
chrom$chromosome <- ifelse(chrom$chromosome == 12, 12, "other")

#pathway Annotation
#topGO <- sort.int(abs(pathways$stat), index.return = T, decreasing = T)$ix[1:5]
#topPath <- pathways[topGO,]


#Annotate Genes in this pathways
#gsc <- loadGSC("/home/almut/Dokumente/masterarbeit/data/h.all.v6.0.symbols.gmt", type="gmt")
#gsc_Kegg <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
#tft <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c3.tft.v6.0.symbols.gmt", type="gmt")
#c5 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c5.all.v6.0.symbols.gmt", type="gmt")
#c6 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c6.all.v6.0.symbols.gmt", type="gmt")
#c7 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c7.all.v6.0.symbols.gmt", type="gmt")
#cgp <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cgp.v6.0.symbols.gmt", type="gmt")
#topGsc <- gsc$gsc[as.character(topPath$pathway)]

#topPathGenes <- lapply(topGsc, function(x){
#  genes <- ifelse(tri12$Symbol %in% unlist(x), 1 , 0)
#})  %>% do.call(cbind,.)

#topPathGenes <- apply(topPathGenes, 2, function(x) as.factor(as.character(x)))
#rownames(topPathGenes) <- tri12$Symbol

```


#Plot heatmap using complex heatmaps to show expression signature
```{r complex heatmap, fig.width= 22, fig.height=25}

#colors
#colors <- colorRampPalette(rev( brewer.pal(20,"RdBu")) )(20)
colors = colorRamp2(c(-4,-2,0,2,4), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
annocol <- colorRampPalette( brewer.pal(11,"Paired") )(11)
annocol <- get_palette("jco", 10)
chromcolor <- pal_jama("default")(7)
annocolor <- list(trisomy12 = c("0" = annocol[4], "1" = annocol[8])) #, IGHV = c("M" = annocol[11], "U" = annocol[5])) 
rowcolors <-colorRampPalette(brewer.pal(5, "Set1"))(5)
rowcolors[6] <- "white"


chromcol <- list(chromosome = c("12" = chromcolor[1], "other" = chromcolor[5]))
                
#annoRowcolor <- list()
#  for(i in c(1:length(topGO)))(
#  annoRowcolor[[i]] = c("0"= rowcolors[6], "1" = rowcolors[i]) 
#  )
  
#names(annoRowcolor) <- topPath$pathway
#chromcol <- list(chromosome = c("1" ="#00d3b4", "2" ="#f443b9", "3" = "#6cde56", "4" = "#932db4", "5" ="#dcc608", "6"= "#0a54d9", "7" = "#fea114", "8"= "#0098ec","12" ="#fd4b43","10" = "#01e09f", "11" = "#ff6aab", "9" = "#a8d37e", "13" = "#a59eff","14" = "#be6f00", "15" = "#03d1f5", "16" = "#a02008", "17" = "#89c8ff", "18" = "#7a7600", "19" ="#005b8f", "20" = "#ffad61","21" = "#80386c", "22" = "#e3c27d", "X" ="#843e29", "MT" = "#ff938b"))

 
#ha_row = rowAnnotation(df = topPathGenes, col = annoRowcolor, annotation_width = unit(c(rep(1.9, 5)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 30), labels_gp = gpar(fontsize = 20),  grid_height = unit(1, "cm"), grid_width = unit(1.3, "cm"), gap = unit(3, "cm")))


ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(1.9, "cm"), annotation_legend_param = list(ncol = 1, title_gp = gpar(fontsize = 50), labels_gp = gpar(fontsize = 40),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm"), gap = unit(1.3,"cm")))

feature <- as.data.frame(colData(ddsCLL)[,c("trisomy12")]) #, "IGHV")])
colnames(feature) <- c("trisomy12")#, "IGHV")

ha_col <- HeatmapAnnotation(feature, col = annocolor, annotation_height = unit(c(rep(1.9, 1)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 50), labels_gp = gpar(fontsize = 45),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm")))


h1 <- Heatmap(exprTri12.new ,                                                     #Cluster param
              km = 2,
              gap = unit(0.5, "cm"),
              cluster_columns = F,
              #clustering_distance_columns = "euclidean",
              #clustering_method_columns = "ward.D2",
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = "Gene signature: Trisomy12",                      #Graphic parameter
              col = colors,
              column_title_gp = gpar(fontsize = 60, fontface = "bold"), 
              heatmap_legend_param = list(title = "expr", 
                                          title_gp = gpar(fontsize = 50), 
                                          grid_height = unit(1.9, "cm"), 
                                          grid_width = unit(1.9, "cm"), 
                                          gap = unit(2, "cm"), 
                                          labels_gp = gpar(fontsize = 45)), 
              column_dend_height = unit(2.5, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = TRUE, 
              row_names_gp = gpar(fontsize = 40),
              top_annotation = ha_col)

#Annotate known genes from litertaure
subset <- as.data.frame(rowData(RNAnorm[rownames(tri12),]))
subset1 <- subset[which(subset$symbol %in% c("ITGA4", "RAP1B", "ITGAM", "ITGB2", "ITGB1", "ITGB7")),]
subset2 <- subset[which(subset$symbol %in% int_genes$Symbol),]
subset <- rbind(subset1, subset2)
rownames(subset) <- subset$symbol
geneIDs <- which(rownames(exprTri12.new) %in% rownames(subset))
labels <- rownames(exprTri12.new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 30)), width = unit(8, "cm"))


#svg(filename="~/git/figures_thesis/gene_expr/gene_exprTrisomy12_gsea_Hallmark.svg", width=30, height=45)
#pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/gene_exprTrisomy12_noTsig.pdf", width=22, height=25)

draw(h1 + ha_chrom) #  + ha_genes) #+ ha_row ha_chrom +
#dev.off()

```

#gene counts for specific genes
```{r single gene counts}
#function to create stripchart plots for specific genes
gene_count <- function(gene_nam){
  geneEnsID <- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% gene_nam)]
  geneNum <- counts(ddsCLL)[geneEnsID,]
  mutPat <- as.data.frame(colData(ddsCLL)[, c("trisomy12")])
  colnames(mutPat) <- c("genotype")
  geneDat <- cbind(mutPat, geneNum)
  colnames(geneDat) <- c("genotype", "counts")
  
  p <- ggstripchart(geneDat, x = "genotype", y = "counts",
          color = "genotype",
          palette = "jco",
          add = "mean_sd",
          title = paste(gene_nam),
          font.x = 18, font.y = 18, font.legend = 16,
          ylab = "normalized counts") + font("xy.text", size = 15) + font("title", size = 20, face = "bold")
  ggsave(file=paste0("/home/almut/Dokumente/masterarbeit/workinprogress/tri12/genetic_interaction_", gene_nam, ".svg"), plot=p, width=6, height=5)
  p
}

geneList <- c("CHFR", "LIX1", "GIT2", "CORO2B", "ADD2", "SAGE1", "DENND3", "ADGRG7", "UACA", "TEAD1", "IRAK4", "RAPGEF5", "ITGB2-AS1", "SOCS3")#, "TRAK2", "EML6","SYBU", "SLC30A4", "RAB27B", "MAP2K6", "MAPK13", "PLIN4", "TRAK2", "BCRP3", "PAX9", "EBF4", "IRAK4", "TWF1", "TMEM117", "TRAF6", "TLR4", "TLR9", "BRG1")

lapply(geneList, gene_count)


```





#Plot chromosome distributions
Is there any connection to genes on chromosome 12 (more or less expressed?)
```{r plot chromosome dist, fig.width= 7, fig.height=7}

tri12Groups <- tri12 %>%  mutate(group = ifelse(stat > 0, "up", "down"))
rownames(tri12Groups) <- rownames(tri12)


chromDist <- lapply(c("up", "down"), function(group){
  geneID <- rownames(tri12Groups)[which(tri12Groups$group %in% group)]
  chromo <- as.data.frame(rowData(RNAnorm[geneID,])$chromosome)
  rownames(chromo) <- geneID 
  colnames(chromo ) <- "chromosome"
  
  
  chrom <- chromo %>% dplyr::select(chromosome) %>% group_by(chromosome) %>% summarise(total = n()) %>% mutate(chromosome1 = gsub("MT", nrow(chrom), chromosome)) %>%  mutate(chromosome1 =gsub("X", (nrow(chrom) - 1), chromosome1)) %>% arrange(as.numeric(chromosome1))
  
  #chrom$label <- paste0(chrom$chromosome, "(", chrom$total, ")")
  chrom$label <- paste0( "(", chrom$total, ")")
  
  #pdf(file=paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/chromosom_dist_", group, ".pdf"), width=7, height=7)
  pie(chrom$total, col=colorRampPalette(brewer.pal(11,'Spectral'))(25), border="white", labels=chrom$label, main = paste0("Chromosomal gene distribution in tri12: ", group), cex = 1.5) 
  legend("bottomright",legend=chrom$chromosome, bty="n", fill=colorRampPalette(brewer.pal(11,'Spectral'))(25), cex = 1)
  #dev.off()
})

```


#plot chromosome distribution as stripchart
```{r stripchart, fig.width=8, fig.height=5}

#specify colors/change order to get a better color for chromosome 12
palette <- pal_igv(palette = c("default"), alpha = 1)(24)
palette <- c(palette[c(5:24)], palette[c(1:4)])

#chrom dist
tri12_chrom <- as.tibble(tri12Groups) %>% mutate(Chromo=rowData(RNAnorm[as.character(ID),])$chromosome) %>% group_by(Chromo, group) %>% summarise(total = n()) %>% mutate(Chromosome= ifelse(Chromo == 12, "12", "other"))

#plot
p <- ggstripchart(tri12_chrom, x = "group", y = "total",
          color = "Chromosome",
          palette = c("#5B8FA8FF", "#FFB547FF"),          
          title = "Chromosomal distribution",
          ylab = "# upregulated genes",
          size = 6,
          jitter = 0.3,
          legend= "right",
          font.legend = c(18, "plain", "black"),
          font.main = 25,
          font.x =23,
          font.y = 23,
          font.tickslab=18
)
p
 
#save plot 
#ggsave(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/chromosom_dist_trisomy12_plain.svg", plot=p, width=8, height=5)  
  


```





