
%%%%%%%%%%%%%%%%%%%%%%%% Material and Methods %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Material and Methods}

\subsection{Data acquisition}

\subsubsection{RNA-sequencing}
RNA isolation and library preparation were performed as described \citet{Dietrich}. In short, total RNA was isolated from patient blood samples using the RNA RNeasy mini kit (Qiagen). For quantification a Qubit 2.0 Fluorometer was used, and RNA integrity was evaluated with an Agilent 2100 Bioanalyzer. Samples with RNA integrity number (RIN) <8 were excluded from the study. Library preparation was performed according to the Illumina TruSeq RNA sample preparation v2 protocol. Samples were paired-end sequenced at the DKFZ Genomics and Proteomics Core Facility. Two to three samples were multiplexed per lane on Illumina HiSeq 2000, Illumina HiSeq3000/4000 or Illumina HiSeqX machines.
 

Raw RNA-seq reads were demultiplexed and quality control was performed using FastQC version 0.11.5 \citep{Andrews}. Internal trimming with STAR version 2.5.2a \citep{Dobin2015} was used to remove adapters before mapping \citep{Dobin2015}. Mapping was performed using \texttt{STAR} version 2.5.2a \citep{Dobin2015} against the Ensembl human reference genome release 75 (Homo sapiens GRCh37.75) \citep{Flicek2014}. \texttt{STAR} was run in default mode with internal adapter trimming using the \texttt{clip3pAdapterSeq} option. Mapped reads were summarized into counts using \texttt{htseq-count} version 0.9.0 \citep{Anders2010} with default parameters and \texttt{union} mode. Thus, only fragments unambiguously overlapping with one gene were counted \citep{Anders2010}. The count data were then imported into R (version 3.4) for subsequent analysis. 

Among the RNA-seq data from 184 CLL samples presented in the study, 136 samples were included in the Primary Blood Cancer Cell Encyclopedia (PACE) \citep{Dietrich}.

\subsubsection{Somatic variants}
Mutations, including 66 gene mutations and 22 structural variants, of 184 CLL samples were retreaved as previously described in the Primary Blood Cancer Cell Encyclopedia (PACE) project \citep{Dietrich}. Statistical analyses were restricted to genetic variants with a prevalence  $\ge3.5\%$ ($7/184$ samples), i.e., to $7$ gene mutations and $5$ SVs. In addition, the somatic hypermutation status of the immunoglobulin heavy-chain variable region (IGHV) and CLL subtype classifications defined by global levels of CpG methylation \citep{Oakes2016} level  were also obtained from PACE.


Genomic variations, including 66 gene mutations and 22 structural variants, of the 184 CLL samples were extensively profiled in one of our previous studies and can be queried from the Primary Blood Cancer Cell Encyclopedia (PACE) project \citep{Dietrich}. Statistical analyses were restricted to those genetic variants with a prevalence $\ge3.5\%$ ($7/184$ samples), i.e., to $7$ gene mutations and $5$ SVs. In addition, the somatic hypermutation status of the immunoglobulin heavy-chain variable region (IGHV) and CLL subtype classifications defined by global levels of CpG methylation level \citep{Oakes2016} were also obtained from PACE.

\subsection{Statistical analysis}

\subsubsection{Statistical analysis}

\subsubsection{Exploratory data analysis}

All statistical analyses were performed using R version 3.4. Raw count data were transformed using variance stablizing transformation (VST) \citep{Love2014} before exploratory data analysis. Genes were ranked by overall variance, and the most variable genes were used in the subsequent analyses. PCAs were generated with the R function \texttt{prcomp} using the 1000 most variable genes. For hierarchical clustering of samples, expression data of the 1000 most variable genes were centred and scaled and distances were calculated by the R \texttt{dist} function with Euclidean distance. The \texttt{complexHeatmaps} package \citep{Gu2016} was further used to cluster and display gene expression using \texttt{ward.D2} as method \citep{Murtagh2014}. Two-dimensional $t$-distributed stochastic neighbour embedding ($t$-SNE) was performed using the \texttt{Rtsne} function of the \texttt{Rtsne} package \citep{R-Rtsne} on sample distances calculated based on the 150 most variable genes.

\subsubsection{Batch effect estimation}
Due to the large sample size, transcriptome data were generated over a period of four years and platforms were changed due to technological development during the period of sequencing, which lead to the changes in sequencing depth and read length (101, 125 and 150 nucleotides). Therefore, the batch effect due to platform differences were estimated. While the amount of sequences belonging to adapters differed depending on the read length, we could not find associations between the top 10 principle components and different batches after adapter trimming (data not shown).  


\subsubsection{Differential Expression Analysis}
Differentially expressed genes between samples with and without each of the genetic variants were identified using \texttt{DESeq2} version 1.16.1 \citep{Love2014}. Exploratory data analysis revealed IGHV hypermutation status and trisomy12 as dominating factors for gene expression. Both were used as co-variates in the GLMs of the remaining variants. To account for multiple testing, the method of Benjamini and Hochberg \citep{Benjamini} was applied to control FDR at $\alpha=0.1$.   

\subsubsection{Gene set enrichment analysis}
Gene set enrichment analysis was performed using the Bioconductor package \texttt{piano} version 1.18.0 \citep{Varemo} using adjusted p.values from \texttt{DESeq2} and Fisherâ€™s combined probability test as method \citep{Fisher1932}. \texttt{Hallmark} and \texttt{KEGG} gene set collections version 4.0 were downloaded from \texttt{MSigDB} \citep{Liberzon2015, Kanehisa2017}. Significance of gene sets was determined based on a background distribution of sampled genes with 1000 permutations. P-values were corrected for multiple testing using the method of Benjamini and Hochberg, with a FDR = 0.1~\citep{Benjamini}.


\subsubsection{Tumor epistasis model}
Genetic interactions were modeled with an epistasis model describing observed gene expression \textit{g} by a linear combination of the expression in the two interacting variants \textit{variantA} and \textit{variantB}.
\begin{equation}
 g \sim variantA + variantB + variantA:variantB
\end{equation}
The model was fit using \texttt{DESeq2} \citep{Love2014}. P-values were adjusted for multiple testing with FDR = 0.1~\citep{Benjamini}.


