---
title: "2017-16-08-dds_all_trimmed"
author: "Almut Luetge"
date: "August 15, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

16.08.2017
#Building a new DeSeq Data set from all RNAseq data. All samples have been adapter trimmed. 

```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(Rtsne)

```

#1. Get data
Metadata from former dds dataset
countdata from counttable

```{r data, message=FALSE, warning=FALSE}
###1.)Load available RNA data sets
##RNA dataset with clinical trait data
load("/home/almut/Dokumente/masterarbeit/data/2017-06-18_ddsrna_clipped.RData")         #dds-object containg previous data
colnames(dds) <- colData(dds)$HipoID

##Countdata
#dataset0
fileName <- paste("/home/almut/Dokumente/masterarbeit/data/seqCounts_trimmed_batch0.txt")
seqDat0 <- read.table(fileName,sep = ",",row.names = 1,header = TRUE)

#dataset1
fileName <- paste("/home/almut/Dokumente/masterarbeit/data/seqCounts_trimmed_batch1.txt")
seqDat1 <- read.table(fileName,sep = ",",row.names = 1,header = TRUE)

#dataset2
fileName <- paste("/home/almut/Dokumente/masterarbeit/data/seqCounts_trimmed_batch2.txt")
seqDat2 <- read.table(fileName,sep = ",",row.names = 1,header = TRUE)

seqDat <- cbind(seqDat0, seqDat1, seqDat2)
```



#2.) Processing metadata:

```{r  Metadata and annotation, warning=FALSE}
##New dataset
#Replace H005. in the beginning of the HipoIDs

colnames(seqDat) <- gsub("H005.", "", colnames(seqDat))
hipoID <- colnames(seqDat)

#get patient informations from dds object
metaTab <- colData(dds)

#order metadata and countdata
metaTab <- metaTab[which(metaTab$HipoID %in% colnames(seqDat)),]
countTable <- seqDat[,metaTab$HipoID]

# to later build dds-dataset (function will break otherwise)
rownames(metaTab) <- seq(nrow(metaTab))
colnames(countTable) <- seq(ncol(countTable))


```


#3.) Gene annotation

```{r Gene annotation,  message=FALSE, warning=FALSE}
#Get annotations for genes (using biomaRt)
## Caution do not load earlier as it mask function select
library(biomaRt)
library(genefilter)
library(geneplotter)
library(pheatmap)
library(RColorBrewer)
library(stats)

ensembl <- useMart("ensembl",dataset = "hsapiens_gene_ensembl")
anno <- getBM(values = rownames(countTable),mart = ensembl, attributes = c("ensembl_gene_id","hgnc_symbol","description","chromosome_name","entrezgene","gene_biotype"),filters = "ensembl_gene_id")
anno <- anno[!duplicated(anno$ensembl_gene_id),]
rownames(anno) <- anno[,1]
geneAnno <- anno[rownames(countTable),]
geneAnno$ensembl_gene_id <- NULL


```


#4.) Create Deseq dataset

```{r deseq}
#Generate a new DESeq object
stopifnot(ncol(countTable) == nrow(metaTab))
dds <- DESeqDataSetFromMatrix(countData = countTable,colData = metaTab, design = ~1)

mcols(dds)$symbol <- geneAnno$hgnc_symbol
mcols(dds)$description <- geneAnno$description
mcols(dds)$chromosome <- geneAnno$chromosome_name
mcols(dds)$entrezgene <- geneAnno$entrezgene
mcols(dds)$biotype <- geneAnno$gene_biotype

#save the object
#save(dds,file="/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")

```



#######################################################
#5.) Exploratory data analysis

```{r Preprocessing}
#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
ddsCLL <- dds[,colData(dds)$diag %in% cll_vector]

#number genes and samples
dim(dds)
dim(ddsCLL)
ngenes <-nrow(ddsCLL)
nsamples <-ncol(ddsCLL)

# Remove rows/genes with too few counts
keep <-apply(counts(ddsCLL), 1, function(x) any(x >= 10))
ddsCLL <- ddsCLL[keep,]

#Variance stabilization transformation of the raw data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)


```


Checking transformed data by plotting densities of counts for each sample and plotting the empirical cummulative distribution functions (ECDFs).
Successful normalization should lead to overlapped curves

```{r quality check, fig.width=15, fig.height=7, cache=TRUE}

par(mfrow = c(1,2))
multidensity (assay(RNAnorm), xlim = c(0,30), legend = F, xlab = "mean counts")
multiecdf(assay(RNAnorm), legend = F, xlab="mean counts")


```





#Hierachical Clustering on Sample distances.

```{r s-to-s-distances, fig.width=10, fig.height=10, cache=TRUE}
sampleDists <- dist(t(assay(RNAnorm)))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- paste0(colData(RNAnorm)$PatID, "_", colData(RNAnorm)$IGVH, "_", colData(RNAnorm)$batch)

colnames(sampleDistMat) <- NULL

#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"Blues")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")

```


#Principal Component Analysis
Try different number of genes - most variant can be related to gender (remove y/x-chromosomes first?!)

```{r pca, fig.height=22, fig.width=16}
#Plot PCA
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:5000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))
pcaTab$batch <- as.factor(pcaTab$batch)

#plot PCA and color samples based on annotations
featureList <- c("batch","IGVH","trisomy12","Gender","RNAprep", "del13q14")

pList <- lapply(featureList, function(feature) {
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 3) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) +
    theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) 
 })

grid.arrange(grobs = pList, ncol =2)

```

Hierachical clustering
```{r, hierachical clustering, fig.width=18, fig.height=40}
colAnno <- pcaTab[,c("batch","IGVH","trisomy12","Gender","RNAprep", "del13q14")]

#scale and censor
exprMat.new <- log2(exprMat[c(1:200),])
exprMat.new <- t(scale(t(exprMat.new)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, 
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol),
         clustering_method = "ward.D2", fontsize = 20, fontsize_row = 15)
```



#Check pseudogene expression in particular
Due to Adaptercontamination previous data, showed a strong batch effect associated with high pseudo gene mapping. The sequenced read length changed with batches, while the insert size distribution was the same. This led batch associated adapter contamination and pseudogene mapping. Check here whether this effect has been removed? 
```{r sub samples, fig.width=18, fig.heigh
```

#1. Get data
Metadata from former dds dataset
countdata from counttable

```{r data, message=FALSE, warning=FALSE}
###1.)Load available RNA data sets
##RNA dataset with clinical trait data
load("/home/almut/Documents/CLL/data/2017-06-18_ddsrna_clipped.RData")         #dds-object containg previous data
colnames(dds) <- colData(dds)$HipoID

##Countdata
#dataset0
fileName <- paste("/home/almut/Documents/CLL/data/seqCounts_trimmed_batch0.txt")
seqDat0 <- read.table(fileName,sep = ",",row.names = 1,header = TRUE)

#dataset1
fileName <- paste("/home/almut/Documents/CLL/data/seqCounts_trimmed_batch1.txt")
seqDat1 <- read.table(fileName,sep = ",",row.names = 1,header = TRUE)

#dataset2
fileName <- paste("/home/almut/Documents/CLL/data/seqCounts_trimmed_batch2.txt")
seqDat2 <- read.table(fileName,sep = ",",row.names = 1,header = TRUE)

seqDat <- cbind(seqDat0, seqDat1, seqDat2)
```



#2.) Processing metadata:

```{r  Metadata and annotation, warning=FALSE}
##New dataset
#Replace H005. in the beginning of the HipoIDs

colnames(seqDat) <- gsub("H005.", "", colnames(seqDat))
hipoID <- colnames(seqDat)

#get patient informations from dds object
metaTab <- colData(dds)

#order metadata and countdata
metaTab <- metaTab[which(metaTab$HipoID %in% colnames(seqDat)),]
countTable <- seqDat[,metaTab$HipoID]

# to later build dds-dataset (function will break otherwise)
rownames(metaTab) <- seq(nrow(metaTab))
colnames(countTable) <- seq(ncol(countTable))


```


#3.) Gene annotation

```{r Gene annotation,  message=FALSE, warning=FALSE}
#Get annotations for genes (using biomaRt)
## Caution do not load earlier as it mask function select
library(biomaRt)
library(genefilter)
library(geneplotter)
library(pheatmap)
library(RColorBrewer)
library(stats)

ensembl <- useMart("ensembl",dataset = "hsapiens_gene_ensembl")
anno <- getBM(values = rownames(countTable),mart = ensembl, attributes = c("ensembl_gene_id","hgnc_symbol","description","chromosome_name","entrezgene","gene_biotype"),filters = "ensembl_gene_id")
anno <- anno[!duplicated(anno$ensembl_gene_id),]
rownames(anno) <- anno[,1]
geneAnno <- anno[rownames(countTable),]
geneAnno$ensembl_gene_id <- NULL


```


#4.) Create Deseq dataset

```{r deseq}
#Generate a new DESeq object
stopifnot(ncol(countTable) == nrow(metaTab))
dds <- DESeqDataSetFromMatrix(countData = countTable,colData = metaTab, design = ~1)

mcols(dds)$symbol <- geneAnno$hgnc_symbol
mcols(dds)$description <- geneAnno$description
mcols(dds)$chromosome <- geneAnno$chromosome_name
mcols(dds)$entrezgene <- geneAnno$entrezgene
mcols(dds)$biotype <- geneAnno$gene_biotype

#save the object
#save(dds,file="/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")

```



#######################################################
#5.) Exploratory data analysis

```{r Preprocessing}
#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
ddsCLL <- dds[,colData(dds)$diag %in% cll_vector]

#number genes and samples
dim(dds)
dim(ddsCLL)
ngenes <-nrow(ddsCLL)
nsamples <-ncol(ddsCLL)

# Remove rows/genes with too few counts
keep <-apply(counts(ddsCLL), 1, function(x) any(x >= 10))
ddsCLL <- ddsCLL[keep,]

#Variance stabilization transformation of the raw data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)


```


Checking transformed data by plotting densities of counts for each sample and plotting the empirical cummulative distribution functions (ECDFs).
Successful normalization should lead to overlapped curves

```{r quality check, fig.width=15, fig.height=7, cache=TRUE}

par(mfrow = c(1,2))
multidensity (assay(RNAnorm), xlim = c(0,30), legend = F, xlab = "mean counts")
multiecdf(assay(RNAnorm), legend = F, xlab="mean counts")


```





#Hierachical Clustering on Sample distances.

```{r s-to-s-distances, fig.width=10, fig.height=10, cache=TRUE}
sampleDists <- dist(t(assay(RNAnorm)))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- paste0(colData(RNAnorm)$PatID, "_", colData(RNAnorm)$IGVH, "_", colData(RNAnorm)$batch)

colnames(sampleDistMat) <- NULL

#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"Blues")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")

```


#Principal Component Analysis
Try different number of genes - most variant can be related to gender (remove y/x-chromosomes first?!)

```{r pca, fig.height=22, fig.width=16}
#Plot PCA
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:5000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))
pcaTab$batch <- as.factor(pcaTab$batch)

#plot PCA and color samples based on annotations
featureList <- c("batch","IGVH","trisomy12","Gender","RNAprep", "del13q14")

pList <- lapply(featureList, function(feature) {
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 3) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) +
    theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) 
 })

grid.arrange(grobs = pList, ncol =2)

```

Hierachical clustering
```{r, hierachical clustering, fig.width=18, fig.height=40}
colAnno <- pcaTab[,c("batch","IGVH","trisomy12","Gender","RNAprep", "del13q14")]

#scale and censor
exprMat.new <- log2(exprMat[c(1:200),])
exprMat.new <- t(scale(t(exprMat.new)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, 
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol),
         clustering_method = "ward.D2", fontsize = 20, fontsize_row = 15)
```



#Check pseudogene expression in particular
Due to Adaptercontamination previous data, showed a strong batch effect associated with high pseudo gene mapping. The sequenced read length changed with batches, while the insert size distribution was the same. This led batch associated adapter contamination and pseudogene mapping. Check here whether this effect has been removed? 
```{r sub samples, fig.width=18, fig.height=40}

#pseudogenes
pseudogene <- rownames(RNAnorm[rowData(RNAnorm)$biotype %in% "processed_pseudogene"])


#heatmap with top100 genes before correction
exprMat <- assay(RNAnorm)
exprMat <- exprMat[pseudogene,]

sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:5000],]

#scale and censor
exprMat.new <- log2(exprMat[c(1:500),])
exprMat.new <- t(scale(t(exprMat.new)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4


colAnno <- pcaTab[,c("batch","IGVH","trisomy12","Gender","RNAprep", "del13q14")]

pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, 
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol, "_", rowData(RNAnorm[rownames(exprMat.new),])$biotype),
         clustering_method = "ward.D2", main = "pseudogenes", fontsize = 20, fontsize_row = 7)





```

There still is a batch-depended pattern in the pseudogene expression. The expression pattern is less dominant and pseudogene expression in batch 2 is at least 10 reduced after adapter trimming. I assume that there are still some adapter left after trimming and more conservative trimming could finally remove this. This is also in line with the star mapping parameter outcome (variance of unmapped reads etc. is still highest in batch 2, but much closer to the other than before mapping). This batch pattern in pseudogenes should not affect further analysis and can be disregarded for further analysis. It might is reasonable still to filter for coding genes for further analysis. (careful del13p14 is very depended on the expression of miR22? (microRNA) - so better just exclude pseudogenes?) 



