---
title: "Dataset overview"
author: "Almut Luetge"
date: "October 13, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#13.10.2017
#Data overview
Aim: Plot overview on the Pace dataset containing and their different variants

```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(RColorBrewer)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(matrixStats)
library(limma)
library(ggplot2)
library(ggpubr)
library(ComplexHeatmap)

```


load data
```{r data}

load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_trimmed_tsig_modIGHV.RData")
dim(ddsCLL)

```

#metadata
Complex heatmaps to present patient metadata

```{r complex heatmaps - data overview, fig.width=60, fig.height=40}

#data to plot
#metaDat <- data.frame(colData(ddsCLL)[,c(1,3, 6,13, 26,18:20,22,23,21,17,8,9,14:16)])
metaDat <- data.frame(colData(ddsCLL)[,c(1, 13, 26,18:20,22,23,21,17,8,9,14:16)])
#metaDat[which(metaDat$Treatment == "No"), "Treatment"] <-  0

metaTib <- as.tibble(metaDat) %>% arrange(IGHV, desc(BRAF, NOTCH1, SF3B1, ATM, MED12,TP53, del17p13, trisomy12, del13q14, del8p12, gain8q24, del11q22.3))
metaDat <- as.data.frame(metaTib)

#colors
colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
annocol <- c("#6389c0","#67b748","white","#6f4acb","#baa342","#ca4cc0","white","#4e6632","#ce4673","#5eaf98","#d35330","#603976")
#annocolor <- list(Gender = c("m" = annocol[1], "f" = annocol[2]) , IGHV = c("U" = annocol[3], "M" = annocol[4]), Treatment = c("1" = annocol[5], "0" = annocol[6]), trisomy12 = c( "0" = annocol[7], "1" = annocol[8]), del13q14 = c("0" = annocol[7], "1" = annocol[8]), del8p12 = c("0" = annocol[7], "1" = annocol[8]),  gain8q24 = c("0" = annocol[7], "1" = annocol[8]),  del11q22.3 = c("0" = annocol[7], "1" = annocol[8]), TP53= c( "0" = annocol[3], "1" = annocol[4]), del17p13 = c( "0" = annocol[7], "1" = annocol[8]), BRAF = c( "0" = annocol[3], "1" = annocol[4]), NOTCH1 = c( "0" = annocol[3], "1" = annocol[4]), SF3B1 = c( "0" = annocol[3], "1" = annocol[4]), ATM = c( "0" = annocol[3], "1" = annocol[4]), MED12 = c( "0" = annocol[3], "1" = annocol[4]), Methylation = c("HP"= annocol[9], "IP" = annocol[10], "LP" = annocol[11]))
                  
                  #drugGroup = c( "BTK" = annocol[9], "mTOR" = annocol[10], "none" = annocol[11], "MEK" = annocol [12]))

annocolor <- list(IGHV = c("U" = annocol[3], "M" = annocol[4]), trisomy12 = c( "0" = annocol[7], "1" = annocol[8]), del13q14 = c("0" = annocol[7], "1" = annocol[8]), del8p12 = c("0" = annocol[7], "1" = annocol[8]),  gain8q24 = c("0" = annocol[7], "1" = annocol[8]),  del11q22.3 = c("0" = annocol[7], "1" = annocol[8]), TP53= c( "0" = annocol[3], "1" = annocol[4]), del17p13 = c( "0" = annocol[7], "1" = annocol[8]), BRAF = c( "0" = annocol[3], "1" = annocol[4]), NOTCH1 = c( "0" = annocol[3], "1" = annocol[4]), SF3B1 = c( "0" = annocol[3], "1" = annocol[4]), ATM = c( "0" = annocol[3], "1" = annocol[4]), MED12 = c( "0" = annocol[3], "1" = annocol[4]), Methylation = c("HP"= annocol[9], "IP" = annocol[10], "LP" = annocol[11]))


#Column annotation
ha_col = HeatmapAnnotation(df = metaDat[,-1], col = annocolor, annotation_height = unit(c(rep(4, 15)),"cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 80), labels_gp = gpar(fontsize = 60),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(20, "mm")))


zero_row_mat = matrix(nrow = 0, ncol = nrow(metaDat))
ht = Heatmap(zero_row_mat, top_annotation = ha_col, column_title = "Overview PACE dataset - CLL patients", column_title_gp = gpar(fontsize = 80, fontface = "bold"), column_names_gp = gpar(fontsize = 40))

feature <- c( "Methylation", "IGHV", "BRAF", "NOTCH1", "SF3B1", "ATM", "MED12", "TP53","del17p13", "trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3")

svg(filename="~/git/figures_thesis/data_overview.svg", width=60, height=45)

draw(ht, padding = unit(c(15, 15, 15, 15), "cm"),  annotation_legend_side = "right")
lapply(feature,function(feat){
  decorate_annotation(feat, {grid.text(feat, unit(-2, "mm"), just = "right", gp=gpar(fontsize=60))})
})

dev.off()

```





