---
title: "EDA thesis"
author: "Almut Luetge"
date: "October 14, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Aim: Exploratory data analysis - structured for thesis

load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(ggdendro)
library(reshape2)
library(genefilter)
library(Biobase)
library(SummarizedExperiment)
library(ggplot2)
library(gtable)
library(grid)
library(ggpubr)
library(ComplexHeatmap)
library(gridExtra)
library(factoextra)
library(Rtsne)
library(RColorBrewer)
library(corrplot)
library("IHW")

```


load data
```{r datasets}
#gene expression
#load("/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")

#gene expression
load("/home/almut/Documents/CLL/data/ddsrna_20170908.RData")  #trimmed dataset with Tsig included by Junyan, to intergrate T-cell contamination as confounding factor 

#metaData
load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData")

#methylation data for the new methylation cluster associations
meth <- read_csv("/home/almut/Documents/CLL/data/MethylomeData092017.csv")
colnames(meth) <- c("sampleID", "LP-CLL", "IP-CLL", "HP-CLL", "NON-CLL MBN", "AML", "Impure", "Silhouette QC", "Epitype", "HighConfidenceEpitype")
meth$HighConfidenceEpitype <- gsub("-CLL", "", meth$HighConfidenceEpitype)
meth$HighConfidenceEpitype <- gsub("Impure", NA, meth$HighConfidenceEpitype)
meth$HighConfidenceEpitype <- gsub("unassignable", NA, meth$HighConfidenceEpitype)
```

#Data preprocessing

Add latest methylation Cluster information from Oakes
```{r methData}
#add new methylation cluster to patmeta object 
methConf <- meth[,c("sampleID", "HighConfidenceEpitype")]
rownames(methConf) <- methConf$sampleID
methConf <- as.data.frame(methConf)
methConf <- methConf[patMeta$Patient.ID,]
patMeta$Methylation <- methConf$HighConfidenceEpitype

#impute IGHV status from new methylation dataset
patMeta$IGHVmeth <- methConf$HighConfidenceEpitype
patMeta$IGHVmeth <- gsub("IP", "M", patMeta$IGHVmeth)
patMeta$IGHVmeth <- gsub("HP", "M", patMeta$IGHVmeth)
patMeta$IGHVmeth <- gsub("LP", "U", patMeta$IGHVmeth)

patMeta$IGHVmeth <- as.factor(patMeta$IGHVmeth)

```

#Add metadata to colData (dds)
```{r prepare metadata}
##Prepare dds dataset --> add patmeta data and remove non CLL patients
#remove y-chromosome
dds <- dds[!rowData(dds)$chromosome %in% c("Y"),]

#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
ddsCLL <- dds[,colData(dds)$diag %in% cll_vector]


#number genes and samples
dim(dds)


# Remove rows/genes with too few counts
keep <-apply(counts(ddsCLL), 1, function(x) any(x >= 10))
ddsCLL <- ddsCLL[keep,]

dim(ddsCLL)

#Add meta data
#generate Matrix with interesting features
patMat <- patMeta[which(patMeta$Patient.ID %in% colData(ddsCLL)$PatID),]
genomDat <- apply(patMat[,c(12:98)], 2, function(x) as.numeric(as.character(x)))
intGenom <- genomDat[, which(colSums(genomDat, na.rm=T) > 6)]

patMat <-patMat[, c("Patient.ID", "IGHV.status","Methylation", colnames(intGenom), "c1c2", "IGHVmeth", "drugGroup")]
rownames(intGenom) <- patMat$Patient.ID
patMat <- as.data.frame(patMat)
patMat <- lapply(patMat, function(x) as.factor(x))
patMat <- as.data.frame(patMat)


rownames(patMat) <- patMat$Patient.ID

patMat <- patMat[colData(ddsCLL)$PatID,]

colData(ddsCLL) <- cbind(colData(ddsCLL), patMat[,-which(colnames(patMat) %in% c("Patient.ID", "trisomy12", "IGHV.status", "del13q14"))])

#write new Rdata object - avoid filtering/preprocessing in future. Use this dataset for further analysis!


#save(ddsCLL, file = "/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_variants.RData")
```


Normalize data
```{r norm data}

#Variance stabilization transformation of the raw data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
```


#Cluster groups to add IGHv status to NA samples
6.Hierachical clustering
```{r s-to-s-distances, fig.width=20, fig.height=20, cache=TRUE}
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:300],]


sampleDists <- dist(t(exprMat))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- paste0(colData(RNAnorm)$Methylation, "_", colData(RNAnorm)$IGVH)

colnames(sampleDistMat) <- NULL

sampleDistDat <- t(exprMat)
rownames(sampleDistDat) <-colData(RNAnorm)$PatID


#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"Blues")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")


```
#Clustering 
Basically 3 Cluster: 
  - LP-U cluster --> NA can be set to U with high confidence
  - HP(IP)-M cluster --> NA can be set to M with high confindence
  - IP-M/U cluster --> NA can not be determined with high enough confidence. 
  (In line with prior observations this cluster is closer to the HP/M cluster than to LP/U cluster)



```{r hclust, fig.width = 40, fig.heigth = 90}
#hclust object
IGHV_cluster <- hclust(dist(sampleDistDat), method = "ward.D2", members = NULL)

## S3 method for class 'hclust'
plot(IGHV_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "IGHV_cluster_new",
     sub = NULL, xlab = "CLL_patients", ylab = "Height")


dendr    <- dendro_data(IGHV_cluster, type="rectangle") # convert for ggplot
clust    <- cutree(IGHV_cluster,k=3)                    # find 2 clusters
clust.df <- data.frame(label=names(clust), cluster=factor(clust))

# dendr[["labels"]] has the labels, merge with clust.df based on label column
dendr[["labels"]] <- merge(dendr[["labels"]], clust.df, by="label")

# plot the dendrogram; note use of color=cluster in geom_text(...)
ggplot() + geom_segment(data=segment(dendr), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr), aes(x, y, label=label, hjust=0, color=cluster), 
           size=8) +
        coord_flip() + scale_y_reverse(expand=c(0.9, 0)) + 
        ggtitle("Clustering by IGHVcluster") +
        theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_text(size = 20),
        axis.text.x=element_text(size = 20),
        axis.title = element_blank(),
        panel.background=element_rect(fill="white"),
        panel.grid=element_blank(),
        plot.title = element_text(size = 80)) 
        


## S3 method for class 'hclust'
plot(IGHV_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "Cluster by IGHV_cluster",
     sub = NULL, xlab = "CLL_patients", ylab = "Height")


```

Add ighv mod to ddsCLL dataset
```{r add ighvmod}
##Add group information to patients
colData(ddsCLL)$IGHVmod <- clust.df$cluster[match(colData(ddsCLL)$PatID,rownames(clust.df))] 

#replace Clusternumber with M,U,NA
colData(ddsCLL)$IGHVmod <- gsub("1", "U", colData(ddsCLL)$IGHVmod)
colData(ddsCLL)$IGHVmod <- gsub("2", "M", colData(ddsCLL)$IGHVmod)
colData(ddsCLL)$IGHVmod <- gsub("3", "NA", colData(ddsCLL)$IGHVmod)

colData(ddsCLL)$IGHVmod <- as.factor(colData(ddsCLL)$IGHVmod)

#Change only NA in original classification
colData(ddsCLL)$IGHV <- colData(ddsCLL)$IGVH
colData(ddsCLL)$IGHV[which(is.na(colData(ddsCLL)$IGHV))] <- colData(ddsCLL)$IGHVmod[which(is.na(colData(ddsCLL)$IGVH))]
colData(ddsCLL) <- colData(ddsCLL)[-which(names(colData(ddsCLL)) %in% c("IGHVmeth", "IGHVmod"))]

##Use this ddsdataset for further analysis. it contains Tsig, IGHV (with imputed IGHV status based on sample clustering and variants)
#save(ddsCLL, file = "/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV_drugs.RData")


colData(RNAnorm)$IGHV <- colData(ddsCLL)$IGHV 

```


Principal Component Analysis
Try different number of genes - most variant can be related to gender (remove y/x-chromosomes first?!)

```{r pca, fig.height=25, fig.width=18}
#Plot PCA
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:5000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))

#plot the eigenvalues of the principal components
fviz_eig(pcaRes)

#plot PCA and color samples based on annotations
featureList <- c("Methylation","IGHV","trisomy12","Gender","RNAprep", "del13q14")


pList <- lapply(featureList, function(feature) {
  
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 3) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title =element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) +
    theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) + coord_fixed() 
 })

grid.arrange(grobs = pList, ncol =2)


#plots for thesis as ggpubr plots in Adaptercontamination.RMD

```

#PCA and variants
#test coocourrence of each variant with the first principal components

select interesting feature, based on the occurence in the trimmed dds-object 2017-08-16_ddsrna_trimmed.RData

```{r prepare metadata}
##Prepare dds dataset --> add patmeta data and remove non CLL patients
variants <- c("trisomy12", "del13q14", "RNAprep", "batch", "Tsig", "Methylation", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12", "c1c2", "IGHV")

patMat <-colData(ddsCLL)[,c("PatID", variants)]
patMat <- as.data.frame(patMat)
patMat <- lapply(patMat, function(x) as.factor(x))
patMat <- as.data.frame(patMat)
rownames(patMat) <- colData(ddsCLL)$PatID
```


perform T-test on all variants
```{r corMat chi, fig.height=10, fig.width= 10}
#variants_all <- as.vector(colnames(patMat[-which(colnames(patMat) %in% c("PatID", "Methylation", "Tsig", "RNAprep", "c1c2", "batch"))]))

variants_all <- as.vector(colnames(patMat[-which(colnames(patMat) %in% c("PatID","Methylation", "Tsig", "batch", "c1c2"))])) #, "RNAprep", "c1c2", "batch"))]))

#varstab <- pcaTab[,variants_all] 
#IGHV <- ifelse(varstab$IGHV %in% "M", 1, 0)
#RNAprep <- ifelse(varstab$RNAprep %in% "Pellet", 1, 0)

#varstab$IGHV <- IGHV
#varstab$RNAprep <- RNAprep

#varsAll <- apply(varstab, 2, function(x){
 # p.est <- mean(10 * as.numeric(x), na.rm = T)
#variance <- (p.est*(1-p.est))/length(x)
#})

#overVars <- cbind(colnames(varstab), varsAll)

#Chisquare p.adj matrix
my_ttest <- function(variant_in){
  variant1 <- patMat[, c("PatID",variant_in)]
  rownames(variant1) <- patMat$PatID
  variant1 <- variant1[which(!is.na(variant1[,variant_in])),]
  pcaTab1 <- pcaTab[which(pcaTab$PatID %in% rownames(variant1)), ]
  variant1 <- variant1[pcaTab1$PatID,]

  
  
  #overVari <- overVars[variant_in, "varsAll"]
  
  resT <- apply(pcaTab1[,c(1:10)], 2, function(x) {
  res <- t.test(x ~ variant1[, variant_in], var.equal=FALSE)
  data.frame(p = res$p.value, fc = res$estimate[[2]]/res$estimate[[1]])
}) %>% do.call(rbind,.)

  #resT$overVar <- overVari
  #resT$variant <- c(rep(variant_in, 10))
  #use ihw
  #ihwRes <- ihw(p ~ overVar,  data = resT, alpha = 0.1)
  
  #resT[,variant_in] <- -log10(ihwRes$p.adjust)
resT[,variant_in] <- -log10(p.adjust(resT$p))
resT[,variant_in]
}


resTtest <-lapply(variants_all,my_ttest) %>% do.call(cbind,.)
#resTtest$overVar <- as.numeric(resTtest$overVar)
#use ihw
#ihwRes <- ihw(p ~ overVar,  data = resTtest, alpha = 0.1)



colnames(resTtest) <- variants_all
rownames(resTtest)<- colnames(pcaTab[,c(1:10)])
resTMat <- as.matrix(resTtest)


col <- colorRampPalette(rev(c("#BB4444", "#EE9988", "#FFFFFF", "#77AADD", "#4477AA")))(50)
col3 <- colorRampPalette( brewer.pal(2,"YlGnBu")) (255)
#col4 <- (c("#99a666", "#96568b" ))
#corplot from ci-square test
corrplot.mixed(resTMat, is.corr = FALSE, number.cex= 3, tl.cex=3.5, tl.pos = "lt", cl.cex=3, diag = "n", pch.col = "black")



#svg(filename="~/git/figures_thesis/corplot_pca_variants.svg", width=10, height=10)
corrplot(resTMat, is.corr = FALSE, method="color", col = col, number.cex = 1.2,
         addCoef.col = "black", # Add coefficient of correlation
         tl.col = "black", tl.srt = 90, # Text label color and rotation
         # hide correlation coefficient on the principal diagonal
         tl.cex=1.4,
         cl.cex = 1.1,
          cl.lim = c(0, 40),
         diag = TRUE)
#dev.off()

```
