---
title: "Diff_genes_summary_plot"
author: "almut"
date: "6 Mai 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Aim: Generate a summary plot giving an overview on the number of differntially expressed genes / number upreagulated / number downregulated genes and chromosomal distribution for CNVs.


libraries
```{r load libraries , warning=F, message=FALSE}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(piano)
library(ggsci)
library(RColorBrewer)
library(circlize)
library(reshape2)
library(ggsci)
```

data
```{r data set}
#dds data set. gene expression data + patmetadata
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

variants <- c("trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1","TP53", "ATM", "MED12","IGHV")

```

Summarize all information in a table
```{r summary table}
variant_stats <- function(variant, chr){
  name <- variant
  diff_genes <-read.csv(file=paste0("/home/almut/Dokumente/masterarbeit/results/noTsig/", variant, "_diffGenes.csv"))
  sig_genes <- as.tibble(diff_genes) %>% filter(padj < 0.1, abs(log2FoldChange) >= 2)
  up_genes <- sig_genes %>% filter(log2FoldChange > 0) %>% mutate(chrom = rowData(ddsCLL)$chromosome[which(rownames(ddsCLL) %in% X)])
  dn_genes <- sig_genes %>% filter(log2FoldChange < 0) %>% mutate(chrom = rowData(ddsCLL)$chromosome[which(rownames(ddsCLL) %in% X)])
  chr_up <- ifelse(!is.na(chr), length(which(up_genes$chrom %in% chr)), 0)
  chr_dn <- ifelse(!is.na(chr), length(which(dn_genes$chrom %in% chr)), 0)
  summary <- tibble(
    variant = name,
    n_genes = nrow(sig_genes),
    up_genes = nrow(up_genes),
    dn_genes = nrow(dn_genes),
    nr_up = chr_up,
    nr_dn = chr_dn,
    nr_chrom = chr_dn + chr_up,
    nr_other = n_genes - chr_dn - chr_up
  )
  }

chromo <- c("12", "13", "8", "8", "11", "17", NA, NA, NA, NA, NA, NA, NA)
input <- cbind(variants, chromo)

summary <- mapply(variant_stats, variants, chromo)
sum_dat <- data.frame(matrix(unlist(summary), nrow=13, byrow=T),stringsAsFactors=FALSE)
colnames(sum_dat) <- c("variant", "n_genes", "up_genes", "dn_genes", "nr_up", "nr_dn", "nr_chrom", "nr_other")

#sum_dat_long <- melt(sum_dat[,c("variant","up_genes", "dn_genes")], id.vars = c("variant")) #, variable_name = c("up_genes", "dn_genes"))

sum_dat_long <- melt(sum_dat[,c("variant","nr_other", "nr_chrom")], id.vars = c("variant")) #, variable_name = c("up_genes", "dn_genes"))

colnames(sum_dat_long) <- c("variant", "location", "genes")
sum_dat_long$dir_color <-paste0(sum_dat_long$variant, "_", sum_dat_long$location)

sum_dat_long$genes <- as.numeric(as.character(sum_dat_long$genes))

```


create a barplot
```{r barplot, fig.height= 5, fig.width=10}
col_pal1 <- pal_igv(palette = c("default"), alpha = 1)(13)
col_pal2 <- pal_igv(palette = c("default"), alpha = 0.6)(13)

color <- c(col_pal1, col_pal2)

ggbarplot(sum_dat_long, "variant", "genes",
  fill = "dir_color", palette = color, orientation = "horiz",
  label = FALSE)  #, position = position_dodge())


```

