---
title: "del13q14"
author: "Almut Luetge"
date: "November 7, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Aim: Define a del13q14 expression signature


libraries
```{r load libraries}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(VennDiagram)
library(RColorBrewer)
library(circlize)
library(piano)
```

#data
```{r load data}
#dds data set. gene expression data + patmetadata
load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV_drugs.RData")

#differentially expressed genes between del17p13 groups with Tsig as confounding factor
del13 <- read.csv(file="/home/almut/Documents/CLL/results/deseq/all_IGHV_Tsig_tri12/del13q14_diffGenes.csv")
#del13 <- read.csv(file="/home/almut/Documents/CLL/results/deseq/multifactor/del13q14_diffGenes.csv")
rownames(del13) <- del13$X
del13<- del13[,-1]
#del13<- del13[which(del13$padj < 0.01),-1]

int_genes <- del13[which(del13$padj < 0.01 & abs(del13$log2FoldChange) > 4),]

#Add diff pathway information
pathways <- read.csv(file="/home/almut/Documents/CLL/results/deseq/all_IGHV_Tsig_tri12/del13q14_diffPathways_GSEAKegg.csv")
pathways <- pathways[,-1] 
pathways <- pathways %>% mutate(group = ifelse(stat > 0, "up", "down")) %>% filter(ifelse(group %in% "up", p_up < 0.1, p_dn < 0.1))


```


#Add fish data of the percentage of del13q14 loss
```{r sequenzier daten}
#read in table with sequencing data 
fish <- read.csv( file = "/home/almut/Documents/CLL/data/SampleIDInfoRNA_sequencing_fish.csv")

fishID <- fish[which(duplicated(fish$Patient.ID)),] %>% filter(!Patient.ID %in% "") %>% select(Patient.ID) 
fish_new <- fish %>% filter(Patient.ID %in% fishID$Patient.ID) %>% mutate(year = gsub("[[:digit:]][[:digit:]][[:punct:]]", "", FISH))
fish_unique <- fish
patList <- as.character(unique(fishID$Patient.ID))

for(PatID in patList){
  latest <- max(fish_new[which(fish_new$Patient.ID %in% PatID),"year"])
    replacement <- fish_new[which(fish_new$Patient.ID %in% PatID & fish_new$year == latest),-51]
    num_replacement <- nrow(fish_unique[which(fish_unique$Patient.ID %in% PatID),])
    if(nrow(replacement) != num_replacement){
    rep_all <- rbind(replacement, replacement, replacement, replacement, replacement)
    rep_all <- rep_all[c(1:num_replacement),]
    fish_unique[which(fish_unique$Patient.ID %in% PatID),] <- rep_all
}}

fish_unique[49,] <- fish_unique[50,]
fish_unique <- fish_unique[which(!duplicated(fish_unique$Patient.ID)),] 

rownames(fish_unique) <- fish_unique[,1]
fish13 <- fish_unique[, c("X13q14","X13q14.")]
fish13 <- fish13[colData(ddsCLL)$PatID,]

colData(ddsCLL)$fish_del13q14 <- fish13[,2]

```





Add all relevant information into one matrix
```{r del17 gene expression}
#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)


#filter for sign. genes in del13
exprMat <- assay(RNAnorm)
exprdel13 <- exprMat[rownames(del13),]
colnames(exprdel13) <- colData(ddsCLL)$PatID
exprdel13.new <- log2(exprdel13)
exprdel13.new <- t(scale(t(exprdel13.new)))
exprdel13.new[exprdel13.new > 4] <- 4
exprdel13.new[exprdel13.new < -4] <- -4
rownames(exprdel13.new) <- rowData(RNAnorm[rownames(del13),])$symbol

#Rowdata
#chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[rownames(del13),])$chromosome)
rownames(chrom) <- rownames(exprdel13.new) 
colnames(chrom ) <- "chromosome"

#select for chromosome 17
chrom$chromosome <- ifelse(chrom$chromosome == 13, 13, "other")

#pathway Annotation
topGO <- sort.int(abs(pathways$stat), index.return = T, decreasing = T)$ix[1:5]
topPath <- pathways[topGO,]


#Annotate Genes in this pathways
gsc <- loadGSC("/home/almut/Documents/CLL/data/h.all.v6.0.symbols.gmt", type="gmt")
gsc_Kegg <- loadGSC("/home/almut/Documents/CLL/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
tft <- loadGSC("/home/almut/Documents/CLL/data/c3.tft.v6.0.symbols.gmt", type="gmt")
c5 <- loadGSC("/home/almut/Documents/CLL/data/c5.all.v6.0.symbols.gmt", type="gmt")
c6 <- loadGSC("/home/almut/Documents/CLL/data/c6.all.v6.0.symbols.gmt", type="gmt")
c7 <- loadGSC("/home/almut/Documents/CLL/data/c7.all.v6.0.symbols.gmt", type="gmt")
cgp <- loadGSC("/home/almut/Documents/CLL/data/c2.cgp.v6.0.symbols.gmt", type="gmt")
topGsc <- gsc_Kegg$gsc[as.character(topPath$pathway)]

topPathGenes <- lapply(topGsc, function(x){
  genes <- ifelse(del13$Symbol %in% unlist(x), 1 , 0)
})  %>% do.call(cbind,.)

topPathGenes <- apply(topPathGenes, 2, function(x) as.factor(as.character(x)))
rownames(topPathGenes) <- del13$Symbol

```


#Plot heatmap using complex heatmaps to show expression signature
```{r complex heatmap del13, fig.width= 35, fig.height=40}

#colors
colors <- colorRampPalette(rev( brewer.pal(10,"RdBu")) )(20)
annocol <- colorRampPalette( brewer.pal(11,"Paired") )(11)
annocolor <- list(del13q14 = c("0" = annocol[1], "1" = annocol[2]), IGHV = c("U" = annocol[3], "M" = annocol[4]), fish_del13q14 = colorRamp2(c(0, 100), c("white", "red")) , drugGroup = c("MEK" = "#4fe481", "mTOR" = "#de0066", "BTK" = "#ff9730", "none" = "#0076fa")) 
rowcolors <-colorRampPalette(brewer.pal(5, "Set1"))(5)
rowcolors[6] <- "white"
#chromcol <- list(chromosome = c("1" ="#00d3b4", "2" ="#f443b9", "3" = "#6cde56", "4" = "#932db4", "5" ="#dcc608", "6"= "#0a54d9", "7" = "#fea114", "8"= "#0098ec","12" ="#fd4b43","10" = "#01e09f", "11" = "#ff6aab", "9" = "#a8d37e", "13" = "#a59eff","14" = "#be6f00", "15" = "#03d1f5", "16" = "#a02008", "17" = "#89c8ff", "18" = "#7a7600", "19" ="#005b8f", "20" = "#ffad61","21" = "#80386c", "22" = "#e3c27d", "X" ="#843e29", "MT" = "#ff938b"))

chromcol <- list(chromosome = c("13" =annocol[8], "other" = annocol[7])) 

                
annoRowcolor <- list()
  for(i in c(1:length(topGO)))(
  annoRowcolor[[i]] = c("0"= rowcolors[6], "1" = rowcolors[i]) 
  )
  
names(annoRowcolor) <- topPath$pathway


 
ha_row = rowAnnotation(df = topPathGenes, col = annoRowcolor, annotation_width = unit(c(rep(1, 5)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15),  grid_height = unit(1, "cm"), grid_width = unit(1, "cm"), gap = unit(3, "cm")))


ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(1.9, "cm"), annotation_legend_param = list(ncol = 1, title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 35),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm"), gap = unit(2.1,"cm")))

feature <- as.data.frame(colData(ddsCLL)[,c("IGHV" ,"del13q14", "fish_del13q14", "drugGroup")])
colnames(feature) <- c("IGHV" ,"del13q14", "fish_del13q14", "drugGroup")

ha_col <- HeatmapAnnotation(feature, col = annocolor, annotation_height = unit(c(rep(1.9, 4)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 35),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm")))


h1 <- Heatmap(exprdel13.new ,                                                     #Cluster param
              km = 2,
              clustering_distance_columns = "euclidean",
              clustering_method_columns = "ward.D2",
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = "Gene signature: del13q14",                      #Graphic parameter
              col = colors,
              column_title_gp = gpar(fontsize = 50, fontface = "bold"), 
              heatmap_legend_param = list(title = "expr", 
                                          title_gp = gpar(fontsize = 40), 
                                          grid_height = unit(1.9, "cm"), 
                                          grid_width = unit(1.9, "cm"), 
                                          gap = unit(2, "cm"), 
                                          labels_gp = gpar(fontsize = 35)), 
              column_dend_height = unit(3.5, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = FALSE, 
              top_annotation = ha_col)


#Annotate genes on chrom 13
subset <- as.data.frame(rowData(RNAnorm[rownames(del13),]))
subset1 <- subset[which(subset$symbol %in% c("DLEU1", "DLEU2", "BCL2", "MIR-15A", "MIR-16A", "KPNA3", "CCLD6", "CCLD7", "DLEU7", "KCNRG", "DLEU5", "RFP2", "PHF11", "ARL11", "RCBTB1")),]
subset2 <- subset[which(subset$symbol %in% int_genes$Symbol),]
#subset <- subset1
subset <- rbind(subset1, subset2)
rownames(subset) <- subset$symbol
geneIDs <- which(rownames(exprdel13.new) %in% rownames(subset))
labels <- rownames(exprdel13.new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 35)), width = unit(8, "cm"))



#svg(filename="~/git/figures_thesis/gene_expr/gene_exprDel13q14_gsea_Kegg.svg", width=35, height=45)
#pdf(file="~/git/figures_thesis/gene_expr/gene_exprDel13q14_gsea_Kegg.pdf", width=39, height=45)
draw( h1+ ha_chrom + ha_genes) # + ha_row)
#dev.off()




```

#Plot chromosome distributions
Is there any connection to genes on chromosome 12 (more or less expressed?)
```{r plot chromosome dist, fig.width= 8, fig.height=8}

del13Groups <- del13 %>%  mutate(group = ifelse(stat > 0, "up", "down"))
rownames(del13Groups) <- rownames(del13)


chromDist <- lapply(c("up", "down"), function(group){
  geneID <- rownames(del13Groups)[which(del13Groups$group %in% group)]
  chromo <- as.data.frame(rowData(RNAnorm[geneID,])$chromosome)
  rownames(chromo) <- geneID 
  colnames(chromo ) <- "chromosome"
  
  
  chrom <- chromo %>% dplyr::select(chromosome) %>% group_by(chromosome) %>% summarise(total = n())

  chrom$label <- paste0(chrom$chromosome, "(", chrom$total, ")")
  
  #svg(filename=paste0("~/git/figures_thesis/del13/chromosom_dist_", group, ".svg"), width=6, height=6)
  pie(chrom$total, col=colorRampPalette(brewer.pal(11,'Spectral'))(25), border="white", labels=chrom$label, main = paste0("Chromosomal gene distribution in del13q14: ", group))
  #dev.off()
})

```


