---
title: "Gene interactions in trisomy12"
author: "Almut Luetge"
date: "September 21, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

21.09.2017
####Trisomy12: Gene interaction pattern in the trisomy12 variant 
## Aim: Investigate tumor epistasis  in trisomy12 samples
In the gene interaction network we find trisomy12 being highly interacting with other mutations on gene expression level - especially high interactions are found with IGHV and del13q14. This was found by Junyan on drug response level before (see GB-unit retreat slides).

Drug data led to the following hypothesis:
Trisomy12 samples are sensitiv towards MEK/ERK inhibitors. This sensitivity is due to its activation of NFKB/AP1 through the TLR pathway, which includes the MAPK pathway (MEK/ERK) (see Enrichment test between trisomy12 and WT). 
IGHV mutation can rescue cells from MEK/ERK inhibitors, as it activates NFKB/AP1 etc. on another way? Or just inactivates BCR anyway?

Now we want to narrow down this interaction and see whether we find a hint on gene level as on enriched pathways and gene sets.
    
# Load packages and dataset

Load packages
```{r, message=FALSE, warning=FALSE}
library(Biobase)
library(ggplot2)
library(genefilter)
library(limma)
library(qgraph)
library(gridExtra)
library(reshape2)
library(dplyr)
library(ComplexHeatmap)
library(RColorBrewer)
library(piano)
library(DESeq2)

#global theme for ggplot2
theme_set(theme_bw()+theme(axis.title = element_text(size=20), 
                axis.text = element_text(size =15),
                plot.title = element_text(size=25, color="red",
                                          hjust = 0.5)))

#load gene set collection
#Hallmark
gsc <- loadGSC("/home/almut/Documents/CLL/data/h.all.v6.0.symbols.gmt", type="gmt")
#Kegg
gsc_Kegg <- loadGSC("/home/almut/Documents/CLL/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")


c5 <- loadGSC("/home/almut/Documents/CLL/data/c5.all.v6.0.symbols.gmt", type="gmt")
c6 <- loadGSC("/home/almut/Documents/CLL/data/c6.all.v6.0.symbols.gmt", type="gmt")
c7 <- loadGSC("/home/almut/Documents/CLL/data/c7.all.v6.0.symbols.gmt", type="gmt")

```

Load datasets
```{r, message=FALSE, warning=FALSE}

#dds dataset
load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_variants.RData")    #dds dataset with all relevant variants, filtered for CLL patients 

#filter for parients without NA in trisomy12 
ddsCLL <- ddsCLL[, !is.na(colData(ddsCLL)$trisomy12)]

#normalize dds dataset
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
colnames(RNAnorm) <- colData(RNAnorm)$PatID 



#Interaction object generated by Interaction_network.Rmd. Contains pairs of variants and the resp. gene expression plus p.values for sign differences when interacting 
load("/home/almut/Documents/CLL/data/interaction_allIGHVmeth_250917.RData")
sig_vecTab <- filter(vecTab, p <= 0.1)
sig_vecTab2 <- filter(vecTab, p <= 0.01)

```
 
 
 
##Select for variants: trisomy12 and IGHV/ trisomy12 and del13q14

```{r filter by variant}
ighv <- filter(sig_vecTab, varA %in% "IGHVmeth" & varB %in% "trisomy12")
del13q14 <- filter(sig_vecTab2, varA %in% "del13q14" & varB %in% "trisomy12")

ighv_all <- filter(vecTab, varA %in% "IGHVmeth" & varB %in% "trisomy12" & !geneName %in% "")
del13q14_all <- filter(vecTab, varA %in% "del13q14" & varB %in% "trisomy12" & !geneName %in% "") 



```


##IGHV and Trisomy12

Heatmap of interacting genes: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction ighv, fig.width=30, fig.height=60}
#gene expression data
geneExpression = assay(RNAnorm)[ighv$geneID,]
rownames(geneExpression) <- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% ighv$geneID)]

#scale and censor
geneExpression_new <- log2(geneExpression)
geneExpression_new<- t(scale(t(geneExpression_new)))
geneExpression_new[geneExpression_new > 4] <- 4
geneExpression_new[geneExpression_new < -4] <- -4

mutnames <- c("none", "trisomy12", "ighv", "both")
mutStatus <- data.frame(colData(RNAnorm)) %>% select(PatID, IGHVmeth, trisomy12) %>% mutate(IGHV = ifelse(IGHVmeth %in% "M", 1, 0)) %>% mutate(tri12 = as.numeric(as.character(trisomy12))) %>% mutate(mut = factor(mutnames[1 + tri12 + 2 * IGHV], levels = mutnames)) %>% arrange(mut)


geneExpression_new <- geneExpression_new[,mutStatus$PatID]
#notchTib <- data.frame(colData(RNAnorm)) %>% select(PatID, NOTCH1) %>% mutate(not = as.numeric(as.character(NOTCH1)))
#notch <- as.data.frame(notchTib)
#rownames(notch) <- notch$PatID
#notch <- notch[mutStatus$PatID, ]


#colors
colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
annocol <- colorRampPalette( rev(brewer.pal(4,"Set1")) )(4)
annocolor <- list(Variant = c("none" = annocol[1], "trisomy12" = annocol[2], "ighv" = annocol[3], "both" = annocol[4]))
mutationStatus <- data.frame(mutStatus$mut)
rownames(mutationStatus) <- mutStatus$PatID
colnames(mutationStatus) <- "Variant"

#Column annotation
ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 70), labels_gp = gpar(fontsize = 50),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm")))


#rowcluster
geneExpression_dist <- dist(geneExpression_new)
rowcluster = hclust(geneExpression_dist, method = "ward.D2")

#heatmap
h1 <- Heatmap(geneExpression_new, col = colors ,column_title = "Gene interactions trisomy12 - ighv", column_title_gp = gpar(fontsize = 80, fontface = "bold"), heatmap_legend_param = list(title = "gene expression", title_gp = gpar(fontsize = 50), grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm"), labels_gp = gpar(fontsize = 40)) , row_dend_width = unit(3.5, "cm"), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = TRUE, show_column_dend = FALSE, cluster_columns = FALSE, column_order = mutStatus$PatID, cluster_rows = rowcluster, split = 7)


draw(h1)

```
Even when not clustering by columns - a clear pattern in gene expression can be seen. The group of both also shows a distinct pattern. Tumor epistasis!

Supervised k-means row clustering splits genes in 4 sets:
    - low expression in ighv-M, but only without trisomy12
    - high expression only when both mutations are present. One amplifies the other
    - high expression in tri12 samples, but not if both mutations(ighv-M and tri12) are present. These genes we are interested in to explain drug sensitivities of MEK/ERK inhibitors
    - lower expression when both mutations are present. IGHV-M only samples show highest exp of these genes, but only without tri12 mutation.
    
    
    

##Enrichment analysis

The piano package can use Fisher's method in a rank-based way using significance scores (p-values) of all genes to calculate a score for each gene set based on the significance scores inside compared to the significance scores of all genes not part of the gene set.

This can be used to find pathways that are general associated with a trisomy12 and IGHV interaction, but we can not conclude about directionalities or way of interaction. To implement this we need the above defined cluster, containing genes with consistent changes between our conditions (trisomy12, ighv, both, none). 

As we don't have fold changes (what to compare against what?) or gene set statistics with directions of changes, we have to use list-based GSA, which only includes a subset of genes, that are defined as significant by a cutoff and then determines whether there is an overrepresentation of genes from a gene set among genes that are significant by the cut off (see https://books.google.de/books?isbn=3527696164 p.15-17). 

#Rank-based enrichment - piano package

```{r enrichment tests, fig.height=40, fig.width= 30}
#
#function for enrichment analysis
runGSEA <- function(inputTab, gmtFile, GSAmethod){
    rankTab <- inputTab[order(inputTab[,1],decreasing = TRUE),,drop=FALSE] #re-rank by score 
        res <- runGSA(geneLevelStats = rankTab,
                      geneSetStat = GSAmethod,
                      adjMethod = "fdr", gsc=gmtFile,
                      signifMethod = 'nullDist',
                      gsSizeLim=c(8, Inf))
        networkPlot(res, class = "non", adjusted = T, significance = 0.01, lay = 4, main = "Enriched patways for trisomy12 and IGHV-mut interactions")
        Res <- arrange(GSAsummaryTable(res),desc(`Stat (non-dir.)`))
}

gsaTab <- data.frame(row.names = ighv_all$geneName, stat = ighv_all$p)

resGSEA <- runGSEA(gsaTab, gmtFile = gsc_Kegg, GSAmethod="fisher")

```

Function to plot gsea results
```{r gsea del13q14, fig.width = 15, fig.height=25}
#rename results to plot
resPlot <- resGSEA[, c(1:5)]
colnames(resPlot) <- c("pathway", "gene_number", "stat", "p", "p.adj")

#barplot
ggplot(resPlot %>% mutate(log10Padj = -log10(p.adj)), aes(x = reorder(pathway, log10Padj), y = log10Padj, fill = gene_number)) + geom_bar(stat = "identity") + coord_flip()

```
Enrichment in MAPK-Pathway and GRNH-Pathway - both are related to MEK/ERK. MTOR signaling is among the latter ones. This is not enough to really support drug response data (can serve as a hint).




#List-based enrichment - Piano package Fisher's exact on genes from one cluster only.
```{r list-based GSA, fig.width = 20, fig.height= 30}
#cluster description
#cluster <- c( "masked ighv (high) effect",  "enhanced mut effect", "masked tri12 (high) effect", "masked ighv (low) effect" )
#cluster <- c( "masked ighv (low) effect", "enhanced mut effect",  "masked tri12 (high) effect", "masked ighv (high) effect", "masked tri12 (low) effect")

cluster <- c("masked tri12(low)", "masked ighv(high)", "masked ighv (high)", "masked ighv(low)", "enhanced both", "masked tri12 (high)", "tri12 and both high")

#cutoff is set above in to pBH < 0.1. 

#GSA on cluster defined by kmeans clustering. See cluster in heatmap...
setWiseGSA <- function(clusternr, gmtFile){
  geneNam <- rownames(geneExpression_new[row_order(h1)[[clusternr]],]) 
  pVal <- ighv[which(ighv$geneName %in% geneNam), "p"]
  gsaTab <- data.frame(row.names = geneNam, stat = pVal)
  res <- runGSA(geneLevelStats = gsaTab,
                      geneSetStat = "fisher",
                     adjMethod = "fdr", gsc=gmtFile,
                     signifMethod = 'nullDist',
                      gsSizeLim=c(12, Inf))
  Res <- arrange(GSAsummaryTable(res),desc(`Stat (non-dir.)`))
  #rename results to plot
  resPlot <- Res[, c(1:5)]
  colnames(resPlot) <- c("pathway", "gene_number", "stat", "p", "p.adj")

  #barplot
  ggplot(resPlot %>% mutate(log10Padj = -log10(p.adj)), aes(x = reorder(pathway, log10Padj), y = log10Padj, fill = gene_number)) + geom_bar(stat = "identity") + coord_flip() + ggtitle(cluster[clusternr])
  }

clusternr = c(1:7)

#resListGSA <- 
lapply(as.numeric(clusternr), FUN = "setWiseGSA", gmtFile = c5)

#grid.arrange(grobs = resListGSA, ncol = 2)



```
Gene set enrichment analysis on so low numbers of genes do not work!


##check genomic location of genes in different cluster
```{r chromosome locations}
## Caution do not load earlier as it mask function select
library(biomaRt)
library(genefilter)
library(geneplotter)
library(pheatmap)
library(RColorBrewer)
library(stats)

ensembl <- useMart("ensembl",dataset = "hsapiens_gene_ensembl")
anno <- getBM(values = ighv$geneName, mart = ensembl, attributes = c("hgnc_symbol","chromosome_name","entrezgene","gene_biotype"),filters = "hgnc_symbol")
anno <- anno[!duplicated(anno$hgnc_symbol),]
rownames(anno) <- anno[,1]
geneAnno <- anno[ighv$geneName,]
geneAnno$hgnc_symbol <- NULL


ighv$chromosome <- geneAnno$chromosome_name


```

#Plot chromosome distributions
Is there any connection to genes on chromosome 12 (more or less expressed?)
```{r plot chromosome dist, fig.width= 8, fig.height=8}

chromDist <- function(clusternr){ 
  geneNam <- rownames(geneExpression_new[row_order(h1)[[clusternr]],])
  chrom <- ighv %>% filter(geneName %in% geneNam) %>% dplyr::select(chromosome) %>% group_by(chromosome) %>% summarise(total = n())

  chrom$label <- paste0(chrom$chromosome, "(", chrom$total, ")")

  pie(chrom$total, col=colorRampPalette(brewer.pal(11,'Spectral'))(25), border="white", labels=chrom$label, main = paste0("chrom distribution in cluster:", cluster[clusternr]))
}

lapply(c(1:7), chromDist)

```







##Trisomy12 and del13q14

Heatmap of interacting genes: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction del13q14, fig.width=30, fig.height=60}
#gene expression data
geneExpression = assay(RNAnorm)[del13q14$geneID,]
rownames(geneExpression) <- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% del13q14$geneID)]

#scale and censor
geneExpression_new <- log2(geneExpression)
geneExpression_new<- t(scale(t(geneExpression_new)))
geneExpression_new[geneExpression_new > 4] <- 4
geneExpression_new[geneExpression_new < -4] <- -4

mutnames <- c("none", "trisomy12", "del13q14", "both")
mutStatus <- data.frame(colData(RNAnorm)) %>% dplyr::select(PatID, del13q14, trisomy12) %>% mutate(del13q14 =as.numeric(as.character(del13q14))) %>% mutate(tri12 = as.numeric(as.character(trisomy12))) %>% mutate(mut = factor(mutnames[1 + tri12 + 2 * del13q14], levels = mutnames)) %>% arrange(mut)

geneExpression_new <- geneExpression_new[,mutStatus$PatID]


#colors
colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
annocol <- colorRampPalette( rev(brewer.pal(4,"Set1")) )(4)
annocolor <- list(Variant = c("none" = annocol[1], "trisomy12" = annocol[2], "del13q14" = annocol[3], "both" = annocol[4]))
mutationStatus <- data.frame(mutStatus$mut)
rownames(mutationStatus) <- mutStatus$PatID
colnames(mutationStatus) <- "Variant"

#Column annotation
ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 70), labels_gp = gpar(fontsize = 50),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm")))

#heatmap
h1 <- Heatmap(geneExpression_new, col = colors ,column_title = "Gene interactions trisomy12 - del13q14", column_title_gp = gpar(fontsize = 80, fontface = "bold"), heatmap_legend_param = list(title = "gene expression", title_gp = gpar(fontsize = 50), grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm"), labels_gp = gpar(fontsize = 40)),column_dend_height = unit(3.5, "cm"), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = TRUE, show_column_dend = FALSE, cluster_columns = FALSE, column_order = mutStatus$PatID, split = 3)

draw(h1)


```
Even when not clustering by columns - a clear pattern in gene expression can be seen. The group of both also shows a distinct pattern. Tumor epistasis!


##Enrichment analysis
```{r enrichment tests del13q14, fig.height=30, fig.width= 30}


runGSEA <- function(inputTab, gmtFile, GSAmethod){
    inGMT <- gmtFile
    rankTab <- inputTab[order(inputTab[,1],decreasing = TRUE),,drop=FALSE] #re-rank by score 
    
        res <- runGSA(geneLevelStats = rankTab,
                      geneSetStat = GSAmethod,
                      adjMethod = "fdr", gsc=inGMT,
                      signifMethod = 'nullDist',
                      gsSizeLim=c(2, Inf))
        networkPlot(res, class = "non", adjusted = T, significance = 0.1, lay = 4, main = "Enriched patways for trisomy12 and del13q14 interactions")
        Res <- arrange(GSAsummaryTable(res),desc(`Stat (non-dir.)`))
}

gsaTab <- data.frame(row.names = del13q14_all$geneName, stat = del13q14_all$p)

resGSEA <- runGSEA(gsaTab, gmtFile = gsc, GSAmethod="fisher")

```

Plot gsea results as bars
```{r barplot del13q14, fig.width = 12, fig.height=12}
#rename results to plot
resPlot <- resGSEA[, c(1:5)]
colnames(resPlot) <- c("pathway", "gene_number", "stat", "p", "p.adj")

#barplot
ggplot(resPlot %>% mutate(log10Padj = -log10(p.adj)), aes(x = reorder(pathway, log10Padj), y = log10Padj, fill = gene_number)) + geom_bar(stat = "identity") + coord_flip()

```





























