---
title: "Epistasis_Deseq2"
author: "almut"
date: "15 Februar 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Epistasis using Deseq2
15.02.18
Aim: generate an epistasis model using DEseq2

load packages
```{r libraray, message=FALSE, warning=FALSE}
library(Biobase)
library(ggplot2)
library(genefilter)
library(DESeq2)
library(gridExtra)
library(reshape2)
library(dplyr)
library(geneplotter)
library(RColorBrewer)
library(ComplexHeatmap)
library(topGO)
library(org.Hs.eg.db)
library(circlize)
library(piano)
library(ggpubr)
```

load datasets
```{r data}
#count data
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

```



#Epistasis model
Use deseq2 to determine genes, which can be described by epistatisc interactions
(focus on trisomy12 and IGHV)
```{r deseq}
#filter gene set by differentially expressed genes
#ddsCLL <- ddsCLL[sigID,]


###Deseq
ddsCLL <- estimateSizeFactors(ddsCLL)

#exclude NAs
ddsCLL <- ddsCLL[,!is.na(colData(ddsCLL)[,"IGHV"])]
ddsCLL <- ddsCLL[,!is.na(colData(ddsCLL)[,"trisomy12"])]

RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
colnames(RNAnorm) <- colData(RNAnorm)$PatID 

#design matrix with interaction term
design(ddsCLL) <- as.formula(paste("~ IGHV + trisomy12 + IGHV:trisomy12"))
#rnaRaw <- DESeq(ddsCLL, betaPrior = FALSE)
#resultsNames(rnaRaw)
#is the effect of IGHV different between different trisomy12 groups?
#res <- results(rnaRaw, name = "IGHVU.trisomy121")

#load data set.. Choose one down in the list:
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHTri12.RData")


resOrdered <- res[order(res$pvalue),]
resSig <- subset(resOrdered, padj < 0.1)# & abs(log2FoldChange) > 2)
  #rownames(resSig)
resTab <- as.data.frame(resSig)
resTab$symbol <- rowData(RNAnorm[rownames(resSig),])$symbol
#write.table(resTab, file = "/home/almut/Dokumente/masterarbeit/workinprogress/epistasis_IGHV_tri12_cutoff005.csv")

#save(res, file="/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHV_times_Tri12_only.RData")

#deseqResults
#1.) "~ trisomy12 + IGHV + IGHV*trisomy12"
#load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHTri12.RData")

#2.)"~ Tsig + IGHV*trisomy12"
#load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHV_times_Tri12.RData")

#3.)"IGHV*trisomy12"
#load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHV_times_Tri12_only.RData")


```


#Gene expression of significant epistatic genes
Heatmap of interacting genes: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction ighv, fig.width=35, fig.height=41}

  #filter by variant
  sig_Genes <- rownames(resSig)
  
  #gene expression data
  #filter for parients without NA in trisomy12 
  RNAnorm <- RNAnorm[, !is.na(colData(RNAnorm)["IGHV"]) & !is.na(colData(RNAnorm)["trisomy12"])]
  
  
  geneExpression = assay(RNAnorm)[sig_Genes,]
  rownames(geneExpression) <- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% sig_Genes)]

  #scale and censor
  geneExpression_new <- log2(geneExpression)
  geneExpression_new<- t(scale(t(geneExpression_new)))
  geneExpression_new[geneExpression_new > 6] <- 6
  geneExpression_new[geneExpression_new < -6] <- -6

  mutnames <- c("none", "IGHV", "trisomy12", "both")
  
  mutStatus <- data.frame(colData(RNAnorm)) %>% mutate(IGHVnew = ifelse(IGHV %in% "M", 1, 0)) %>% dplyr::select(-IGHV) %>% mutate(IGHV = IGHVnew) %>% dplyr::select_("PatID", "IGHV", "trisomy12") %>% mutate_("namA" = "IGHV", "namB" = "trisomy12") %>% mutate(naA = as.numeric(as.character(namA))) %>% mutate(naB = as.numeric(as.character(namB))) %>% mutate(mut = factor(mutnames[1 + naA + 2 * naB], levels = mutnames)) %>% arrange(mut)

  geneExpression_new <- geneExpression_new[,mutStatus$PatID]


  #colors
  #colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
  colors = colorRamp2(c(-6,-2,0,2,6), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
  annocol <- get_palette("jco", 10)
  chromcol <- list(chromosome = c("12" = annocol[6], "other" = annocol[5]))
  
  annocolor <- list(Variant = c("none" = annocol[1], annocol[2], annocol[3], "both" = annocol[4]))
  names(annocolor$Variant) <- c("none", "IGHV", "trisomy12", "both")
  mutationStatus <- data.frame(mutStatus$mut)
  rownames(mutationStatus) <- mutStatus$PatID
  colnames(mutationStatus) <- "Variant"

  #Column annotation
  ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 60), labels_gp = gpar(fontsize = 50),  grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm"), gap = unit(15, "mm")))

  #rowcluster
  geneExpression_dist <- dist(geneExpression_new)
  rowcluster = hclust(geneExpression_dist, method = "ward.D2")

  #heatmap
  h1 <- Heatmap(geneExpression_new, col = colors ,column_title = paste0("Gene interactions:", "IGHV", "-", "trisomy12"), column_title_gp = gpar(fontsize = 60, fontface = "bold"), heatmap_legend_param = list(title = "Expr", title_gp = gpar(fontsize = 60), grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm"), gap = unit(15, "mm"), labels_gp = gpar(fontsize = 50), labels = c(-6,-3, 0,3, 6)) , row_dend_width = unit(3.5, "cm"), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = FALSE, show_column_dend = FALSE, cluster_columns = FALSE, cluster_rows = rowcluster, split = 5, gap = unit(0.6,"cm"), column_order = mutStatus$PatID)

  #chromosome annotation
  #chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[sig_Genes,])$chromosome)
rownames(chrom) <- sig_Genes
colnames(chrom ) <- "chromosome"
#select for chromosome12
chrom$chromosome <- ifelse(chrom$chromosome == 12, 12, "other")

  #chromcol <- list(chromosome = c("1" ="#00d3b4", "2" ="#f443b9", "3" = "#6cde56", "4" = "#932db4", "5" ="#dcc608", "6"= "#0a54d9", "7" = "#fea114", "8"= "#0098ec","12" ="#fd4b43","10" = "#01e09f", "11" = "#ff6aab", "9" = "#a8d37e", "13" = "#a59eff","14" = "#be6f00", "15" = "#03d1f5", "16" = "#a02008", "17" = "#89c8ff", "18" = "#7a7600", "19" ="#005b8f", "20" = "#ffad61","21" = "#80386c", "22" = "#e3c27d", "X" ="#843e29", "MT" = "#ff938b"))
  
  ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(2.3, "cm"), annotation_legend_param = list(ncol = 2, title_gp = gpar(fontsize = 60), labels_gp = gpar(fontsize = 50),  grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm")))
  
   
  #Annotate known genes from litertaure and/or most significant genes
  top50 <-  rownames(subset(resOrdered, padj < 0.001))
  int_genes <- rowData(RNAnorm[top50,])$symbol

subset <- as.data.frame(rowData(RNAnorm[sig_Genes,]))
subset <- subset[-which(subset$symbol %in% ""),]
subset <- subset[-which(subset$symbol %in% NA),]

subset <- subset[which(subset$symbol %in% int_genes),]
#subset2 <- subset[which(subset$symbol %in% c("MAP3K8", "CDCP1", "CD38")),]
#subset <- rbind(subset1, subset2)
rownames(subset) <- subset$symbol
geneIDs <- which(rownames(geneExpression_new) %in% rownames(subset))
labels <- rownames(geneExpression_new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 45)), width = unit(9, "cm"))
  

  
  #svg(filename="~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg", width=30, height=50)
  #pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistasis_Deseq.pdf", width=35, height=45)
  draw(h1 + ha_genes + ha_chrom) 
  #dev.off()
  #draw(h1 + ha_chrom + ha_genes)
 IGHVTri12 <- list("sig_Genes" = sig_Genes, "geneExp" = geneExpression_new, "h1"= h1)

```

#Draw count distribution according to different genotype for specific genes
```{r single gene counts, fig.width=7, fig.height=5}
#function to create stripchart plots for specific genes
gene_count <- function(gene_nam){
  geneEnsID <- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% gene_nam)]
  geneNum <- counts(ddsCLL)[geneEnsID,]
  mutPat <- as.data.frame(mutationStatus[colData(ddsCLL)$PatID,])
  colnames(mutPat) <- c("genotype")
  geneDat <- cbind(mutPat, geneNum)
  colnames(geneDat) <- c("genotype", "counts")
  
  p <- ggstripchart(geneDat, x = "genotype", y = "counts",
          color = "genotype",
          palette = "jco",
          add = "mean_sd",
          title = paste(gene_nam),
          ylab = "gene counts",
          font.x = 25, font.y = 25, font.legend = 20) + font("xy.text", size = 20) + font("title", size = 30, face = "bold")
 #ggsave(file=paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epi_genes/genetic_interaction_", gene_nam, ".svg"), plot=p, width=7, height=5)
  p
}

#geneList <- c("TCTN1", "ENPP3", "ABLIM2", "CHAD", "E2F2", "GEN1", "TRAK2", "EML6","SYBU", "SLC30A4", "RAB27B", "MAP2K6", "MAPK13", "MAP3K8", "PLIN4", "TRAK2", "BCRP3", "PAX9", "EBF4", "IRAK4", "TWF1", "TMEM117", "TRAF6", "TLR4", "TLR9", "JAG1", "MYBL2", "CCL22", "MEFV", "DHDH", "EBI3", "ABL1", "BCL6", "BCL10", "CCDC121", "DPY19L2", "MAF", "RRS1", "CDH4", "BCRP4", "BCRP3")#, "BRG1")

#leukocyte activating genes
#geneList <- c( "BCL2", "BCL6","BMI1","BST1","CARD11","CD38","CD40","CD74","CD81","CD320","CDKN1A","CHRNB2","CLCF1","GPR183","IGHD","IGHM","IL2","IL4","IL5","IL7","IL13","IRS2","MEF2C","MIF","NCKAP1L","NFATC2","PELI1","PTPRC","SASH3","SLC39A10","TCF3","TFRC","TICAM1","TIRAP","TLR4","TLR9","TNFRSF4","TNFRSF13C","TNFSF13B","VAV3","WNT3A")

#paper
geneList <- c("LEF1", "TIMELESS", "CHAD", "BCL2A1", "EML6", "PPP1R14A", "EPHB6", "GEN1")

#geneList<- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% IGHVTri12$sig_Genes[row_order(IGHVTri12$h1)[[5]]])]
#geneList <- geneList[which(geneList != "")]

lapply(geneList, gene_count)


```


#show distribution of genes within different cluster
```{r number of genes within cluster}

cluster <- c("buffering_dn","sup_tri", "buffering_up" ,"buffering_up_strong", "inversion_up")

cluster_size <- lapply(c(1:5), function(clusternr){
  size <- length(row_order(IGHVTri12$h1)[[clusternr]])
  name <- cluster[clusternr]
  clust <- c(name, as.numeric(size))
}) %>% do.call(rbind,.) 

cluster_size <- as.data.frame(cluster_size)
colnames(cluster_size) <- c("name", "size")
cluster_size$size <- as.numeric(as.character(cluster_size$size))
distr <- ggbarplot(cluster_size, "name", "size",
   fill = "name", color = "name",
   palette = "jco",
   ylab = "Number of genes",
   xlab = "cluster")
 
 #ggsave(file="/home/almut/Dokumente/masterarbeit/workinprogress/distrib_ofEpiTypes.svg", plot=distr, width=8, height=5)
distr

```


#Enrichment analysis
#List-based enrichment - Piano package Fisher's exact on genes from one cluster only.
```{r list-based GSA, fig.width = 20, fig.height= 30}

#load gene set collection
#Hallmark
gsc <- loadGSC("/home/almut/Dokumente/masterarbeit/data/h.all.v6.0.symbols.gmt", type="gmt")
#Kegg
gsc_Kegg <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
tft <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c3.tft.v6.0.symbols.gmt", type="gmt")
c5 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c5.all.v6.0.symbols.gmt", type="gmt")
c6 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c6.all.v6.0.symbols.gmt", type="gmt")
c7 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c7.all.v6.0.symbols.gmt", type="gmt")
cgp <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cgp.v6.0.symbols.gmt", type="gmt")


```

```{r GSEA piano, fig.height=7, fig.weight=16}
#GSA on cluster defined by kmeans clustering. See cluster in heatmap...
setWiseGSA <- function(clusternr, gmtFile){
  #get sig genes and pvalues for each cluster
  geneNam <- IGHVTri12$sig_Genes[row_order(IGHVTri12$h1)[[clusternr]]] 
  pVal <- resSig[geneNam, "padj"]
  #needs to be symbol/remove those without symbol, but ENSID only
  genName <- rowData(RNAnorm[geneNam,])$symbol
  gsTab <- data.frame(gene = genName, stat = pVal)
  gsTab <- gsTab[-which(gsTab$gene %in% c("", NA)),]
  gsaTab <- data.frame(row.names = gsTab$gene, stat = gsTab$stat)
  res <- runGSA(geneLevelStats = gsaTab,
                      geneSetStat = "fisher",
                     adjMethod = "fdr", gsc=gmtFile,
                     signifMethod = 'nullDist',
                     #nPerm = 50000,
                      gsSizeLim=c(1, Inf))
  Res <- arrange(GSAsummaryTable(res),desc(`Stat (non-dir.)`))
  #rename results to plot
  resPlot <- Res[, c(1:5)]
  colnames(resPlot) <- c("pathway", "gene_number", "stat", "p", "p.adj")

  #barplot
  #ggplot(resPlot %>% mutate(log10Padj = -log10(p.adj)), aes(x = reorder(pathway, log10Padj), y = log10Padj, fill = gene_number)) + geom_bar(stat = "identity") + coord_flip() + ggtitle(cluster[clusternr])
  
  enrichPlot <- resPlot[c(1:10),] %>% mutate(log10Padj = -log10(p.adj)) %>% mutate(genes = ifelse(gene_number > 5, ">5", "<=5"))
  
  p <- ggbarplot(enrichPlot, x = "pathway", y = "log10Padj",
          fill = "genes",           # change fill color by mpg_level
          color = "white",            # Set bar border colors to white
          palette = "jco",            # jco journal color palett. see ?ggpar
          sort.val = "asc",          # Sort the value in ascending order
          sort.by.groups = FALSE,     # Don't sort inside each group
          x.text.angle = 90,          # Rotate vertically x axis texts
          ylab = "-log10(padj)",
          legend.title = "Pathway",
          rotate = TRUE,
          font.x = 20, font.y = 20, font.legend = 20, legend = "right", 
          title = paste0("GSEA in ", cluster[clusternr]),
          ggtheme = theme_pubr()
          ) + font("xy.text", size = 22) + font("title", size = 20, face = "bold")
  
   ggsave(file=paste0("/home/almut/Dokumente/masterarbeit/results/enrichment_epistasis/GSEA_in_", clusternr, ".pdf"), plot=p, width=16, height=7)
#pdf(file= paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/GSEA_in_", clusternr, ".pdf"), width=40, height=35)
  p
 # dev.off()
  
  
  }

clusternr = c(1:5)

#resListGSA <- 
lapply(as.numeric(clusternr), FUN = "setWiseGSA", gmtFile = gsc)

#grid.arrange(grobs = resListGSA, ncol = 2)


```

#Try enrichment analyis using a background gene set with similar expression pattern as described by Bernd: https://support.bioconductor.org/p/84290/
#Using TopGO
```{r enrichment background, fig.width = 16, fig.height = 7}

setWiseTopGO <- function(clusternr){
  #get sig genes and pvalues for each cluster
  sig_Genes <- IGHVTri12$sig_Genes[row_order(IGHVTri12$h1)[[clusternr]]]
  exprMat <- assay(RNAnorm)
  de_idx <- which(rownames(exprMat) %in% sig_Genes)
  backG <- genefinder(exprMat, de_idx, 100, method = "manhattan")
  backG <- rownames(exprMat)[as.vector(sapply(backG, function(x)x$indices))]
  backG <- setdiff(backG, sig_Genes)

  multidensity( list( 
      all= log2(exprMat),
      foreground =log2(exprMat[sig_Genes,]), 
      background = log2(exprMat[backG,])), 
      xlab="log2 normalized counts", main = "Matching for enrichment analysis")

  geneIDs = rownames(ddsCLL)
  inUniverse = geneIDs %in% c(sig_Genes, backG) 
  inSelection =  geneIDs %in% sig_Genes
  alg <- factor(as.integer(inSelection[inUniverse]))
  names(alg) <- geneIDs[inUniverse]

  onts = c("MF", "BP", "CC")

  tab = as.list(onts)
  names(tab) = onts
    for(i in 1:3){

  ## prepare data
    tgd <- new( "topGOdata", ontology=onts[i], allGenes = alg, nodeSize=5,
                 annot=annFUN.org, mapping="org.Hs.eg.db", ID = "ensembl" )

    ## run tests
      resultTopGO.elim <- runTest(tgd, algorithm = "elim", statistic = "Fisher" )
      resultTopGO.classic <- runTest(tgd, algorithm = "classic", statistic = "Fisher" )

  ## look at results
      tab[[i]] <- GenTable( tgd, Fisher.elim = resultTopGO.elim, 
          Fisher.classic = resultTopGO.classic,
          orderBy = "Fisher.classic" , topNodes = 200)
      }
 
topGOResults <- plyr::rbind.fill(tab)
#write.csv(topGOResults, file = paste0("/home/almut/Dokumente/git/Transcriptome_CLL/variants/epistasis/pathways/topGOResults", cluster[clusternr],".csv"))
allRes <- topGOResults[order(as.numeric(topGOResults$Fisher.classic)),]
enrichPlot <- allRes[c(1:10),] %>% mutate(log10Padj = -log10(as.numeric(Fisher.classic))) %>% mutate(genes = ifelse(Significant > 10, ">10", "<=10")) %>% mutate(goterm= paste0(GO.ID, "_", Term))
  
p <- ggbarplot(enrichPlot, x = "goterm", y = "log10Padj",
          fill = "genes",           # change fill color by mpg_level
          color = "white",            # Set bar border colors to white
          palette = "jco",            # jco journal color palett. see ?ggpar
          sort.val = "asc",          # Sort the value in ascending order
          sort.by.groups = FALSE,     # Don't sort inside each group
          x.text.angle = 90,          # Rotate vertically x axis texts
          ylab = "-log10(padj)",
          xlab = "Gene Ontology",
          legend.title = "Num. sig. diff. genes",
          rotate = TRUE,
          font.x = 20, font.y = 20, font.legend = 20, legend = "right", 
          title = paste0("GSEA in ", cluster[clusternr]),
          ggtheme = theme_pubr()
          ) + font("xy.text", size = 22) + font("title", size = 20, face = "bold")

#pdf(file= paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/enrichment/GSEA_in_", clusternr, ".pdf"), width=12, height=7)
#dev.off()
 ggsave(file=paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/enrichment/GSEA_in_", clusternr, ".pdf"), plot=p, width=16, height=7)
p
}

lapply(c(1:5), FUN = "setWiseTopGO")

```

#Scheme to show different ways of mixed epistasis
```{r epistasis scheme, fig.width=7, fig.height=5}
#generate a dataframe
mix <- t(data.frame(buffering_up = c(-0.1, -0.5, 0.5, 5), buffering_dn = c(0.5, 0.2, -0.2, -5), suppression_1 = c(0.5, -5, -6, 0.2), suppression_2 = c(-0.2, 5, 4,-0.5), suppression_3 = c(0.5, 0.1, -6, 1), suppression_4 = c(-0.2, 0.5, 4,-0.5), inversion_up = c(-0.2, -6, -4, 5), inversion_dn = c(0.1, 5, 4, -5)))
colnames(mix) <-  c("none", "IGHV", "trisomy12", "both")

annocol <- get_palette("jco", 10)
annocolor <- list(Variant = c("none" = annocol[1], annocol[2], annocol[3], "both" = annocol[4]))
names(annocolor$Variant) <- c("none", "IGHV", "trisomy12", "both")
variants <- as.data.frame(colnames(mix))
colnames(variants) <- "Variant"
rownames(variants) <- variants$Variant

  #Column annotation
  ha_col = HeatmapAnnotation(df = variants, col = annocolor, annotation_height = unit(0.9, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15),  grid_height = unit(0.9, "cm"), grid_width = unit(0.9, "cm"), gap = unit(15, "mm")))

#heatmap
h1 <- Heatmap(mix, col = colors ,column_title = paste0("Types of mixed epistasis"), column_title_gp = gpar(fontsize = 20, fontface = "bold"), heatmap_legend_param = list(title = "Expr.", title_gp = gpar(fontsize = 20), grid_height = unit(1, "cm"), grid_width = unit(0.5, "cm"), gap = unit(10, "mm"), labels_gp = gpar(fontsize = 20), labels = c(-6,-3, 0,3, 6)) , show_row_dend = F, show_column_names =FALSE , top_annotation = ha_col, show_row_names = FALSE, show_column_dend = FALSE, cluster_columns = FALSE, cluster_rows = FALSE, split = c( rep("Buffering",2), rep("Suppression", 4), rep("Inversion", 2)), gap = unit(0.6,"cm"), row_title_gp = gpar(fontsize=19))

#pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/mixed_epistasis_model.pdf", width=7, height=5)
draw(h1)
#dev.off()

```


