---
title: "Gene-interaction network"
author: "Almut Luetge"
date: "September 18, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

18.09.2017
# Aim: Generate a gene interaction network on phenotypes (in our case variants) as described by Fischer, Sandmann et al..
Junyan established a similar approach for drug response data - this can be used to implement interaction networks based on transcriptome data. 

Gene interaction networks can be used to:
 - assign genes to functional modules
 - unravel mechanisms behind tumor epistasis as found in network e.g
    - TP53 and del17p13 patients
    - IGHV-U and del11p23
    - IGHV-U and del13q14
    - trisomy12 and IGH-V unmutated/del13q14
    
    
Load packages and dataset

```{r, message=FALSE, warning=FALSE}
library(Biobase)
library(ggplot2)
library(genefilter)
library(limma)
#library(qgraph)
library(gridExtra)
library(reshape2)
library(dplyr)
#library(metap)
library(RColorBrewer)
library(ComplexHeatmap)

#global theme for ggplot2
theme_set(theme_bw()+theme(axis.title = element_text(size=20), 
                axis.text = element_text(size =15),
                plot.title = element_text(size=25, color="red",
                                          hjust = 0.5)))
```

Load datasets
```{r, message=FALSE, warning=FALSE}

#dds dataset
#dds data set. gene expression data + patmetadata
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_trimmed_tsig_modIGHV.RData")    #dds dataset with all relevant variants, filtered for CLL patients 

#latest patmeta object with updated patient annotations
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/patmeta_180122.RData")
rownames(patMeta) <- patMeta$Patient.ID
patMeta_new <- patMeta[colData(ddsCLL)$PatID,]

#latest annotation (renew interested annotations only --> change ddsCll object as total by the next chance..)
colData(ddsCLL)$IGHV <- patMeta_new$IGHV.status
colData(ddsCLL)$trisomy12 <- patMeta_new$trisomy12

#save(ddsCLL, file="/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

#DeseqRes: Objects are generated using Deseq and the gene_signatures script. First object contains diff_genes for all variants with IGHVmeth, trisomy12, Tsig and c1c2 as covariable for deseq. Second object contains diff genes for variants with IGHVmeth, Tsig and c1c2 as covariable. Last objects'only considers c1c2 and Tsig as covariable

load("/home/almut/Dokumente/masterarbeit/data/desRes_Tsig_IGHV_tri12_25102017.RData")
diffAll <- res_list

load("/home/almut/Dokumente/masterarbeit/data/desResTrisomy12_Tsig_IGHV_25102017.RData")
diffTri12 <- res_list


load("/home/almut/Dokumente/masterarbeit/data/desResIGHV_Tsig_Tri12_25102017.RData")
diffIGHVmeth <- res_list


#Combine diff genes to one object. Use diffgenes with all cofactors for all variants except trisomy12 and IGHV. For them use the respective object:
diffGenes <- c(diffAll, IGHV = diffIGHVmeth$IGHV, trisomy12 = diffTri12$trisomy12)



```
 
 
Filter for diff expressed genes
```{r diierentail expressed genes}
pCut <- 0.01

difftab <- function(condition){
  dataTab <- data.frame(diffGenes[[condition]])
  dataTab$ID <- rownames(dataTab)
  #filter using pvalues
  dataTab <- filter(dataTab, padj <= pCut) %>%
    arrange(padj) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol) #%>%
    #filter(log2FoldChange > 0)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  #dataTab <- data.frame(row.names = dataTab$Symbol, stat = dataTab$stat)
  rownames(dataTab) <- dataTab$ID
  dataTab
}

cond <- c("del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12", "trisomy12", "IGHV")

sigRes <- lapply(cond, difftab)
names(sigRes) <- cond
#save(sigRes, file = "/home/almut/Documents/CLL/data/sig_DiffGenes_250917.RData")
#sigRes <- load("/home/almut/Documents/CLL/data/sig_DiffGenes_250917.RData")


```


 
Define testgroups 
```{r subgroups}
#table with variants 
varTab <- colData(ddsCLL)[, c("IGHV", "del13q14", "trisomy12", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")]
rownames(varTab) <- colData(ddsCLL)$PatID
varTab[,c(2:12)] <- lapply(varTab[,c(2:12)], function(x) as.numeric(as.character(x)))
varTab <- t(as.matrix(varTab))

#set factor level to 1 and 0 (important for the names in lmfit)
varTab[varTab > 1] <- 0



#normalize dds dataset
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
colnames(RNAnorm) <- colData(RNAnorm)$PatID 


```


Enumerate var pairs
```{r}
varTest <- list()
n=1
for (i in seq(nrow(varTab)-1)) {
  for (j in seq(i+1,nrow(varTab))) {
    varA <- varTab[i,]
    varB <- varTab[j,]
    mutnames <- c("none", "A", "B", "both")
    mut <- factor(mutnames[1 + varA + 2 * varB], levels = mutnames)
    varTest <- c(varTest, list(c(rownames(varTab)[i], rownames(varTab)[j])))
  }
}

#filter for one variant you are interested in:
#varTest <- lapply(varTest, function(pair){ 
#  if(length(which(pair %in% "trisomy12" > 0)))
#    return(pair)
#})

#varTest <- varTest[lapply(varTest,length)>0]

```


Identify significant genes and variants
```{r}
scr <- lapply(varTest, function(varPair) {
  x <- tibble(varA = factor(varTab[varPair[1], ]),
              varB  = factor(varTab[varPair[2], ]))
  noNA <- !is.na(x$varA) & !is.na(x$varB)
  x <- x[noNA,]
  intGenes1 <-rownames(sigRes[[varPair[1]]])
  intGenes2 <- rownames(sigRes[[varPair[2]]])
  intGenes <- c(intGenes1,intGenes2)
  intGenes <- unique(intGenes)
  y <- assay(RNAnorm)[intGenes, noNA]
  fit <- eBayes(lmFit(y, design = model.matrix( ~ varA * varB, data = x))) 
  tibble(geneID = rownames(fit$p.value), 
         p      = fit$p.value[, "varA1:varB1"],
         varA = varPair[1],
         varB = varPair[2])
}) %>% bind_rows %>% mutate(pBH = p.adjust(p, method = "BH")) %>% arrange(p)


scr$geneName <- rowData(RNAnorm[scr$geneID,])$symbol

```

Schweder and Spjotvoll plot
```{r schweder and spjotvoll}
pVal <- scr %>% filter(!is.na(p)) %>% dplyr::select(p) 
plot(1-pVal$p,
(length(pVal$p)-1):0, pch=".", xaxs="i", yaxs="i",
xlab=expression(1-p[i]), ylab=expression(N(p[i])))
Npi <- (length(pVal$p[30000:100000])-1):0
pi <- 1-pVal$p[30000:100000]
LinMod = lm(Npi ~ pi)
slope <- LinMod$coefficients["pi"]
abline(a= 0, slope, col="red3", lwd=2)
abline(h=slope)
text(x=0, y=slope, labels=paste(round(slope)), adj=c(-0.1, 1.3))

#number of true Null hypothesis
Npos <- nrow(pVal) - slope

```




Plot the gene expression in combined variants
```{r, eval=FALSE, fig.height=100, fig.width=15, include=FALSE}
fscr <- filter(scr, pBH <= 0.05)
plotList <- lapply(seq(nrow(fscr)), function(i) {
  
  mutnames <- c("none", fscr[i,]$varA, fscr[i,]$varB, "both")
    
  plotTab <- tibble(geneExpression = assay(RNAnorm)[fscr[i,]$geneID,],
                    varA = varTab[fscr[i,]$varA,],
                    varB = varTab[fscr[i,]$varB,],
                    IGHV = factor(colData(RNAnorm)$IGHV)) %>%
    filter(!is.na(varA) & !is.na(varB)) %>%
    mutate(mut = factor(mutnames[1 + varA + 2 * varB], levels = mutnames))
  
  ggplot(plotTab, aes(x=mut, y=geneExpression)) + geom_boxplot(outlier.shape = NA, fill = NA, color = "blue", width =0.3) + 
    theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
    geom_point(alpha=0.4, aes(col = IGHV)) + ggtitle(fscr[i,]$geneName) + ylab("norm genecounts") + xlab("Variant")
})


g <- arrangeGrob(grobs= plotList, ncol = 3)
#ggsave(file = "/home/almut/Documents/CLL/results/interaction/Interactiongenes_all.svg", plot = g, width = 15, height = 250, limitsize = F)

grid.arrange(grobs = plotList, ncol =3)
```



#### Plot interaction network

Filtering interactions using raw-p value
```{r}
fscr <- filter(scr, p <= 0.05)
nrow(fscr)
```
A p value of 0.05 is in line with the results from the Schweder- Spjotvoll plot (almost exactly the number of expecting true null-hypothesis)



Draw network using qgraph package
Normalized by the total number of diff genes (number of genes used to test)
```{r, fig.width=12, fig.height=12}
tabAll <- tibble(varA = fscr$varA, varB = fscr$varB, p = -log10(fscr$p)) %>% 
  group_by(varA, varB) %>% summarise(n=length(varA)) %>% arrange(desc(n))

nGenes <- lapply(seq(nrow(tabAll)) ,function(i){
  variantA <- tabAll[i,]$varA
  variantB <- tabAll[i,]$varB
  nGen = nrow(sigRes[[variantA]]) + nrow(sigRes[[variantB]]) 
})

tabAll$nGenes <- nGenes
tabAll <- tabAll %>% mutate(nRel = n/as.numeric(nGenes))

tabNew <- tabAll %>% dplyr::select(varA, varB, nRel) %>% arrange(desc(nRel))

qgraph(tabNew, directed = FALSE, vsize = 7, layout = "spring", label.cex =1, shape="circle", label.scale = TRUE, label.color = "red")
```
Strong interaction between del17p13 and del13q14, del17p13 and del11q22.3 and TP53 and BRAF

##Select for variants: del17p13 and del13q14/ del11q22.3

```{r filter by variant}
del11q22 <- filter(fscr, varA %in% "del11q22.3" & varB %in% "del17p13")
del13q14 <- filter(fscr, varA %in% "del13q14" & varB %in% "del17p13")
TP53 <- filter(fscr, varA %in% "del17p13" & varB %in% "TP53")

del11q22_all <- filter(scr,varA %in% "del11q22.3" & varB %in% "del17p13" & !geneName %in% "")
del13q14_all <- filter(scr, varA %in% "del13q14" & varB %in% "del17p13" & !geneName %in% "") 


```


## del17p13 and del11q22

Heatmap of interacting genes: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction ighv, fig.width=30, fig.height=60}
#gene expression data
geneExpression = assay(RNAnorm)[TP53$geneID,]
rownames(geneExpression) <- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% TP53$geneID)]

#scale and censor
geneExpression_new <- log2(geneExpression)
geneExpression_new<- t(scale(t(geneExpression_new)))
geneExpression_new[geneExpression_new > 4] <- 4
geneExpression_new[geneExpression_new < -4] <- -4

mutnames <- c("none", "TP53", "del17p13", "both")
mutStatus <- data.frame(colData(RNAnorm)) %>% dplyr::select(PatID, TP53, del17p13) %>% mutate(TP53 = as.numeric(as.character(TP53))) %>% mutate(del17 = as.numeric(as.character(del17p13))) %>% mutate(mut = factor(mutnames[1 + TP53 + 2 * del17], levels = mutnames)) %>% arrange(mut)

geneExpression_new <- geneExpression_new[,mutStatus$PatID]


#colors
colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
annocol <- colorRampPalette( rev(brewer.pal(4,"Set1")) )(4)
annocolor <- list(Variant = c("none" = annocol[1], "TP53" = annocol[2], "del17p13" = annocol[3], "both" = annocol[4]))
mutationStatus <- data.frame(mutStatus$mut)
rownames(mutationStatus) <- mutStatus$PatID
colnames(mutationStatus) <- "Variant"

#Column annotation
ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 70), labels_gp = gpar(fontsize = 50),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm")))

#rowcluster
geneExpression_dist <- dist(geneExpression_new)
rowcluster = hclust(geneExpression_dist, method = "ward.D2")

#heatmap
h1 <- Heatmap(geneExpression_new, col = colors ,column_title = "Gene interactions TP53 - del17p13", column_title_gp = gpar(fontsize = 80, fontface = "bold"), heatmap_legend_param = list(title = "gene expression", title_gp = gpar(fontsize = 50), grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(5, "mm"), labels_gp = gpar(fontsize = 40)) , row_dend_width = unit(3.5, "cm"), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = TRUE, show_column_dend = FALSE, cluster_columns = FALSE, column_order = mutStatus$PatID, cluster_rows = rowcluster, split = 5)

draw(h1)

```




#### Identify direction of interaction

Calculate vector for each variant pair
```{r}
#function to calcualte the vector for a gene-gene pair
calcVec <- function(varPair, geneName, exprMat, varTab) {
  varA <- varPair[1]
  varB <- varPair[2]
  exprTab <- tibble(patID = colnames(exprMat),
                  expr = exprMat[geneName,],
                  varA = varTab[varA,],
                  varB = varTab[varB,]) %>%
  filter(!is.na(varA) & !is.na(varB)) %>%
  mutate(mut = factor(mutnames[1 + (as.integer(varA)) + 2 * (as.integer(varB)) ], levels = mutnames))
  avgNone <- mean(exprTab[exprTab$mut == "none",]$expr)
  avgA <- mean(exprTab[exprTab$mut == "A",]$expr)
  avgB <- mean(exprTab[exprTab$mut == "B",]$expr)
  avgBoth <- mean(exprTab[exprTab$mut == "both",]$expr)
  return(c(avgA - avgNone, avgB-avgNone, avgBoth-avgNone))
}

exprMat <- assay(RNAnorm)


vecTab <- lapply(seq(nrow(scr)), function(i) {
  varA <- scr[i,]$varA
  varB <- scr[i,]$varB
  geneName <- scr[i,]$geneID
  calcVec(c(varA, varB), geneName, exprMat, varTab)
}) %>% do.call(rbind,.)
colnames(vecTab) <- c("onlyA","onlyB","both")
vecTab <- cbind(scr, vecTab)

save(vecTab, file="/home/almut/Dokumente/git/Transcriptome_CLL/dataset/interaction_all_13022017.RData")

```

Calculate pi-score
```{r}
vecPi <- group_by(vecTab, varA, varB) %>% do(data.frame(onlyA = .$onlyA, onlyB = .$onlyB, geneID =.$geneID,  pi = .$both - (.$onlyA + .$onlyB), p= .$p))

#vecPi <- vecTab %>% filter( varA %in% "IGHV" & varB %in% "trisomy12") %>% do(data.frame(onlyA = .$onlyA, onlyB = .$onlyB, geneID =.$geneID,  pi = .$both - (.$onlyA + .$onlyB), p= .$p))
```

```{r}
#function to calculate fractional variance explained for each pair
varExp <- function(geneID) {
  varA <- "IGHV"
  varB <- "trisomy12"
  exprTab <- tibble(patID = colnames(exprMat),
                  expr = exprMat[geneID,],
                  varA = varTab[varA,],
                  varB = varTab[varB,]) %>%
  filter(!is.na(varA) & !is.na(varB)) %>%
  mutate(mut = factor(mutnames[1 + (as.integer(varA)) + 2 * (as.integer(varB)) ], levels = mutnames))
  None <- mean(exprTab[exprTab$mut == "none",]$expr)
  expA <- exprTab[exprTab$mut == "A",]$expr - None
  expB <- exprTab[exprTab$mut == "B",]$expr - None
  expBoth <- exprTab[exprTab$mut == "both",]$expr -None
  
  
  
  #fit a linear model
  model <- lm(expr ~ varA + varB, data = exprTab)
  
  #anova
  aovRes <- anova(model)
  
  #calculate fractional variance explained
 return(c(aovRes[1,2]/sum(aovRes[,2]),
             aovRes[2,2]/sum(aovRes[,2]),
              model$coefficients[2],
              model$coefficients[3], 
              geneID))
} 
#varVec <- group_by(vecPi, varA, varB) %>% filter(complete.cases(onlyA, onlyB, pi)) %>% do(varExp(.$onlyA, .$onlyB, .$pi))
varVec <- vecPi %>% filter(varA %in% "IGHV"& varB %in% "trisomy12") %>% filter(complete.cases(onlyA, onlyB, pi))# %>% do(varExp(.$onlyA, .$onlyB, .$pi, .$geneID))

varTest <- lapply(varVec$geneID, varExp)%>% do.call(rbind,.)
colnames(varTest) <- c("frA", "frB", "coefA", "coefB", "geneID")
varVec <- as_tibble(varTest)

```




Calculate coefficient and variance explained
```{r}
#function to calculate fractional variance explained for each pair
varExp <- function(onlyA, onlyB, pi) {
  #fit a linear model
  model <- lm(pi ~ onlyA + onlyB )
  
  #anova
  aovRes <- anova(model)
  
  #calculate fractional variance explained
  data.frame(frA = aovRes[1,2]/sum(aovRes[,2]),
             frB = aovRes[2,2]/sum(aovRes[,2]),
             coefA = model$coefficients[2],
             coefB = model$coefficients[3], 
             geneID = geneID)
}

#varVec <- vecPi %>% filter(varA %in% "IGHV"& varB %in% "trisomy12") %>% filter(complete.cases(onlyA, onlyB, pi)) %>% do(varExp(.$onlyA, .$onlyB, .$pi))

```

Significant correlations
```{r}
fscr <- filter(scr, p <=0.05)

TriIGHV <- fscr %>% dplyr::filter(varA %in% "IGHV" & varB %in% "trisomy12")
#comTab <- left_join(fscr, varVec, by = c("varA","varB"))
comTab <- left_join(TriIGHV, varVec, by = c("geneID"))

#quantile
fractionsA <- as.numeric(c(comTab$frA))#, comTab$frB))
thrA <- quantile(fractionsA, probs=c(0.1,0.80))
fractionsB <- as.numeric(c(comTab$frB))#, comTab$frB))
thrB <- quantile(fractionsB, probs=c(0.1,0.80))

#comTab <- comTab %>%  mutate(direction = ifelse(frA > thr[[2]] , "Trisomy12 -> IGHV", ifelse(frA < thr[[1]], "IGHV -> Trisomy12", ifelse(frB > thr[[2]], "IGHV -> Trisomy12", ifelse(frB < thr[[1]], "Trisomy12 -> IGHV", "no direction")))))

#comTab <- comTab[,-11]
comTab <- comTab %>%  mutate(direction = ifelse(frA > thrA[[2]] & frB < thrB[[1]], "Trisomy12 -> IGHV", ifelse(frA < thrA[[1]] & frB > thrB[[2]], "IGHV -> Trisomy12", "no direction")))

#save comTab
save(comTab, file="/home/almut/Documents/CLL/data/direction_IGHVTri12_02122017.RData")
```


Plot correlations between pi-score and gene pairs
```{r, fig.width=26, fig.height=7}
pairTab <- filter(fscr, !duplicated(paste0(varA,varB))) %>% arrange(p)
pList <- lapply(12, function(i) {
  nameA <- pairTab[i,]$varA
  nameB <- pairTab[i,]$varB
  dataTab <- filter(vecPi, varA == nameA, varB == nameB)
  dataTab$p <- NULL
  dataTab$varA <- NULL
  dataTab$varB <- NULL
  colnames(dataTab) <- c(nameA,nameB,"pi")
  plotTab <-  melt(dataTab, id.vars = "pi")
  ggplot(plotTab, aes(x= value, y=pi)) + geom_point(aes(color = variable)) + geom_smooth(method = "lm") + 
           facet_wrap( ~ variable, scale = "free") + ggtitle(sprintf("%s - %s", nameA, nameB)) + theme(legend.position = "none") + 
    geom_abline(slope = 1, intercept = 0, linetype = "dotted") + geom_abline(slope = -1, intercept = 0, linetype = "dotted") +
    ylab("Pi score") + xlab("Effect")
})

gg <- arrangeGrob(grobs= pList, ncol = 2)
#ggsave(file = "/home/almut/Documents/CLL/results/interaction/piScoreIGHV-U.svg",plot = gg, width = 16, height = 108, limitsize = F)

#seq(nrow(pairTab))

grid.arrange(grobs=pList,ncol=2)

```



 
 
 
 
 
 
 
 
 
 
 