---
title: "pseudotime_CLL"
author: "almut"
date: "26 Februar 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Pseudotime
26.02.2018
Aim: Order sample according to a pseudotime by adapting methods from sigle cell RNAseq analysis like diffusion maps. Does this order support our epistasis model?

# 1. data

load libraries
```{r libraries}
library(simpleSingleCell)
```

load data
```{r data}
#count data
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

#generate single cell experiment object from it to use single cell experiment package
all_counts <- counts(ddsCLL)
colnames(all_counts) <- colData(ddsCLL)$PatID

sce <- SingleCellExperiment(list(counts=all_counts))
dim(sce)
colData(sce) <- colData(ddsCLL)


```

#2. QC-plots
```{r qc plots}

#highest expressed genes

plotReducedDim(sce, use_dimred="PCA", ncomponents=2, colour_by="IGHV") + fontsize


```






