---
title: "Trisomy12andIGHV-M"
author: "Almut Luetge"
date: "September 29, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

29.09.2017
##IGHV and trisomy12 Interaction
# Aim: Investigate the interaction of IGHV mutation and trisomy12
Our tumor epistatsis model suggests an interaction between IGHV mutation and trisomy12. In drug response data Junyan found trisomy12 cells sensitive towards MEK/ERK inhibitors. Trisomy12 cells were only sensitive towards MEK/ERK inhibitors when the IGHV-gene was unmutated (but drug response in IGHV-M sample always show a lower response). 

- Is there a specific mechanism how trisomy12 and IGHV- mutation interact?
- How do transcriptional data support drug response findings?
- What is the temporal order of mutations? Does one supress the other? Why?

Hypothesis: In IGHV unmutated cells the trisomy12 mutation regulates tumor growth via the BCR and MEK/ERK pathways. By IGHV mutation the BCR signalling is downregulated and therewith the trisomy12 activated tumor growth. Do cells use other patrhways? Why do we find a mutation that causes a less agressive form of the tumor? Is this a kind of defense mechanism? 
Does this mean that the trisomy12 mutation happens before the IGHV-mutation? Or why should cells mutate if the pathway of action is downregulated by another mutation anyways?


#Analysis
1.) Epistatsis model (details in "Interaction_networks.Rmd")
2.) Gene expression of significant genes in interaction model (details in "epistasis_in_trisomy12.Rmd")
3.) GSEA/Pathway analysis
4.) Pathviewer

    
Load packages and dataset

```{r, message=FALSE, warning=FALSE}
library(Biobase)
library(ggplot2)
library(genefilter)
library(limma)
library(qgraph)
library(gridExtra)
library(reshape2)
library(dplyr)
library(metap)
library(RColorBrewer)
library(ComplexHeatmap)
library(piano)
#library(pathview)
library(DESeq2)
library(ggpubr)

#global theme for ggplot2
theme_set(theme_bw()+theme(axis.title = element_text(size=20), 
                axis.text = element_text(size =15),
                plot.title = element_text(size=25, color="red",
                                          hjust = 0.5)))
```


load and prepare data
```{r data}
#dds dataset
#dds data set. gene expression data + patmetadata
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_trimmed_tsig_modIGHV.RData")    #dds dataset with all relevant variants, filtered for CLL patients 

#latest patmeta object with updated patient annotations
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/patmeta_180122.RData")
rownames(patMeta) <- patMeta$Patient.ID
patMeta_new <- patMeta[colData(ddsCLL)$PatID,]

#latest annotation (renew interested annotations only --> change ddsCll object as total by the next chance..)
colData(ddsCLL)$IGHV <- patMeta_new$IGHV.status
colData(ddsCLL)$trisomy12 <- patMeta_new$trisomy12

#filter for parients without NA in trisomy12 
ddsCLL <- ddsCLL[, !is.na(colData(ddsCLL)$trisomy12)]

#normalize dds dataset
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind=T)
colnames(RNAnorm) <- colData(RNAnorm)$PatID 

#Interaction object generated by Interaction_network.Rmd. Contains pairs of variants and the resp. gene expression plus p.values for sign differences when interacting. Interaction is based on the expression of diff. expressed genes (p.value cut-off 0.01)
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/interaction_all_13022017.RData")
sig_vecTab <- filter(vecTab, p <= 0.01)


#list with sig deseq results for all variants (p.value cut-off 0.01)
load("/home/almut/Dokumente/masterarbeit/data/sig_DiffGenes_250917.RData")

#directions
load("/home/almut/Dokumente/masterarbeit/data/direction_IGHVTri12_02122017.RData")
comTab <- as.data.frame(comTab)
rownames(comTab) <- comTab$geneID

```



Draw network using qgraph package
The network is based on number of sign. interacting genes. As this is based on the differentially expressed genes within the variants, the second network is normalized for the number of overall diff expressed genes for both variants. 
Is this the correct way to normalize? The number of diff. genes depends on the occurrence of the variant in our patient data... 

Normalized by the total number of diff genes (number of genes used to test)
```{r, fig.width=12, fig.height=12}
fscr <- sig_vecTab
tabAll <- tibble(varA = fscr$varA, varB = fscr$varB, p = -log10(fscr$p)) %>% 
  group_by(varA, varB) %>% summarise(n=length(varA)) %>% arrange(desc(n))

nGenes <- lapply(seq(nrow(tabAll)) ,function(i){
  variantA <- tabAll[i,]$varA
  variantB <- tabAll[i,]$varB
  nGen = nrow(sigRes[[variantA]]) + nrow(sigRes[[variantB]]) 
})

tabAll_norm <- tabAll
tabAll_norm$nGenes <- nGenes
tabNew_norm <- tabAll_norm %>% mutate(nRel = n/as.numeric(nGenes))

tabNew_norm <- tabNew_norm %>% dplyr::select(varA, varB, nRel) %>% arrange(desc(nRel))

qgraph(tabAll, directed = FALSE, vsize = 7, layout = "spring", label.cex =1, shape="circle", label.scale = TRUE, label.color = "red")

qgraph(tabNew_norm, directed = FALSE, vsize = 7, layout = "spring", label.cex =1, shape="circle", label.scale = TRUE, label.color = "red")
```
Non nomalized network shows IGHV and trisomy12 as predominant interacting variants. Known interactions as TP53 an del17p13 are not reflected. They can be found in the normalized network (even though other are more dominant). IGHV and trisomy12 are still major interaction "partners", but weaker connected than e.g. del11q22.3 and del17p13. Nevertheless when having a detailed look into the gene expression of the significant interacting genes of those variants (del13q14 - del17p13 resp. del17p13 - del11q22)
the analysis seem to suffer from the low number of overall diff expressed genes: 


#Gene expression of significant epistatic genes
Heatmap of interacting genes: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction ighv, fig.width=35, fig.height=45}

my_heatmap <- function(nameA, nameB, k){
  
  #filter by variant
  sig_Genes <- filter(fscr, varA %in% nameA & varB %in% nameB)
  
  #gene expression data
  #filter for parients without NA in trisomy12 
  RNAnorm <- RNAnorm[, !is.na(colData(RNAnorm)[nameA]) & !is.na(colData(RNAnorm)[nameB])]
  
  
  geneExpression = assay(RNAnorm)[sig_Genes$geneID,]
  rownames(geneExpression) <- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% sig_Genes$geneID)]

  #scale and censor
  geneExpression_new <- log2(geneExpression)
  geneExpression_new<- t(scale(t(geneExpression_new)))
  geneExpression_new[geneExpression_new > 4] <- 4
  geneExpression_new[geneExpression_new < -4] <- -4

  mutnames <- c("none", nameA, nameB, "both")
  
  mutStatus <- data.frame(colData(RNAnorm)) %>% mutate(IGHVnew = ifelse(IGHV %in% "M", 1, 0)) %>% dplyr::select(-IGHV) %>% mutate(IGHV = IGHVnew) %>% dplyr::select_("PatID", nameA, nameB) %>% mutate_("namA" = nameA, "namB" = nameB) %>% mutate(naA = as.numeric(as.character(namA))) %>% mutate(naB = as.numeric(as.character(namB))) %>% mutate(mut = factor(mutnames[1 + naA + 2 * naB], levels = mutnames)) %>% arrange(mut)

  geneExpression_new <- geneExpression_new[,mutStatus$PatID]


  #colors
  colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
  annocol <- colorRampPalette( rev(brewer.pal(4,"Set1")) )(4)
  
  annocolor <- list(Variant = c("none" = annocol[1], annocol[2], annocol[3], "both" = annocol[4]))
  names(annocolor$Variant) <- c("none", nameA, nameB, "both")
  mutationStatus <- data.frame(mutStatus$mut)
  rownames(mutationStatus) <- mutStatus$PatID
  colnames(mutationStatus) <- "Variant"

  #Column annotation
  ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 35),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm"), gap = unit(5, "mm")))

  #rowcluster
  geneExpression_dist <- dist(geneExpression_new)
  rowcluster = hclust(geneExpression_dist, method = "ward.D2")

  #heatmap
  h1 <- Heatmap(geneExpression_new, col = colors ,column_title = paste0("Gene interactions:", nameA, "-", nameB), column_title_gp = gpar(fontsize = 50, fontface = "bold"), heatmap_legend_param = list(title = "expr", title_gp = gpar(fontsize = 40), grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm"), gap = unit(5, "mm"), labels_gp = gpar(fontsize = 35)) , row_dend_width = unit(3.5, "cm"), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = FALSE, show_column_dend = FALSE, cluster_columns = FALSE, cluster_rows = rowcluster, split = k, column_order = mutStatus$PatID)

  #chromosome annotation
  #chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[sig_Genes$geneID,])$chromosome)
rownames(chrom) <- sig_Genes$geneID
colnames(chrom ) <- "chromosome"
  
  chromcol <- list(chromosome = c("1" ="#00d3b4", "2" ="#f443b9", "3" = "#6cde56", "4" = "#932db4", "5" ="#dcc608", "6"= "#0a54d9", "7" = "#fea114", "8"= "#0098ec","12" ="#fd4b43","10" = "#01e09f", "11" = "#ff6aab", "9" = "#a8d37e", "13" = "#a59eff","14" = "#be6f00", "15" = "#03d1f5", "16" = "#a02008", "17" = "#89c8ff", "18" = "#7a7600", "19" ="#005b8f", "20" = "#ffad61","21" = "#80386c", "22" = "#e3c27d", "X" ="#843e29", "MT" = "#ff938b"))
  
  ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(1.3, "cm"), annotation_legend_param = list(ncol = 2, title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 35),  grid_height = unit(1.5, "cm"), grid_width = unit(1.5, "cm"), gap = unit(2.1,"cm")))
  
  
  #direction
  dirColors <-colorRampPalette(brewer.pal(5, "Set1"))(5)
  annoDircolor <- list(direction = c("IGHV -> Trisomy12" = dirColors[1], "Trisomy12 -> IGHV" = dirColors[2], "no direction" = "white")) 
  

  dir <- comTab[sig_Genes$geneID,]
  directionDat <- as.data.frame(dir$direction)
  colnames(directionDat) <- "direction"
  #rownames(directionDat) <- dir$geneID
  ha_dir = rowAnnotation(df = directionDat, col = annoDircolor, annotation_width = unit(c(rep(1.4, 1)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 30),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm"), gap = unit(3, "cm")))
  
   
  #Annotate known genes from litertaure  
  top50 <-   sig_Genes %>% filter(pBH < 0.005) %>% select(geneID)
  int_genes <- rowData(RNAnorm[top50$geneID,])$symbol

subset <- as.data.frame(rowData(RNAnorm[sig_Genes$geneID,]))
subset1 <- subset[which(subset$symbol %in% int_genes),]
subset2 <- subset[which(subset$symbol %in% c("MAP3K8", "CDCP1", "CD38")),]
subset <- rbind(subset1, subset2)
rownames(subset) <- subset$symbol
geneIDs <- which(rownames(geneExpression_new) %in% rownames(subset))
labels <- rownames(geneExpression_new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 35)), width = unit(9, "cm"))
  

  
  #svg(filename="~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg", width=30, height=50)
  pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistatsisTri12IGHV_140218.pdf", width=35, height=45)
  draw(h1 + ha_chrom + ha_genes)
  dev.off()
  draw(h1 + ha_chrom + ha_genes )#+ ha_dir)
return(list("sig_Genes" = sig_Genes, "geneExp" = geneExpression_new, "h1"= h1))
}


#del17TP53 <- my_heatmap("del17p13", "TP53", 3)
#del11Del17 <- my_heatmap("del11q22.3" , "del17p13", 5)
#del13Del17 <- my_heatmap("del13q14", "del17p13", 3)
IGHVTri12 <- my_heatmap("IGHV", "trisomy12", 5)

```
Trisomy12 and IGHV show groups of genes with distinct pattern. The other vraiants also form cluster, but despite some genes diff. expressed when both mutations occur together, epistasis effects seem to be rare and less dominant. This could be due to the lower number of diff. expressed genes (power issue?) or/and less epistasis. 

#IGHV and Trisomy12
Supervised k-means row clustering splits genes in 5 sets:
    - IGHV-U downregulated (none_dn)
    - both down regulated (both_dn)
    - upregulated gene expression in tri12 samples, but not if IGHV is mutated as well (only_tri12_up)
    - upregulated gene expression in IGHV unmutated samples, but not if trismoy12 is mutated (none_up)
    - strong (!) upregulated gene gexpression in both (both_up)
   

Genes in the 5 sets are more or less equally distributed over chromosomes (chrom 12 is among the most represented, but not dominating) (dtails see Rmd epistasis in trisomy12)

#PCA with all labels
```{r PCA, fig.width=7, fig.height=6}
#Calculate PCA

my_PCA <- function(nameA, nameB) {
  #Plot PCA
  
  #filter for parients without NA in trisomy12 
  RNAnorm <- RNAnorm[, !is.na(colData(RNAnorm)[nameA]) & !is.na(colData(RNAnorm)[nameB])]
  exprMat <- assay(RNAnorm)

  #top 5000 most variant genes
  sds <- rowSds(exprMat)
  exprMat <- exprMat[order(sds, decreasing = T)[1:1000],]
  
  
  pcaRes <- prcomp(t(exprMat), scale =T)
  varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
  pcaTab <- data.frame(pcaRes$x[,c(1:10)])
  names(varExp) <- colnames(pcaRes$x)

  #add background information
  pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))
   mutnames <- c("none", nameA, nameB, "both")
  
  mutStatus <- data.frame(colData(RNAnorm)) %>% mutate(IGHVnew = ifelse(IGHV %in% "M", 1, 0)) %>% dplyr::select(-IGHV) %>%   mutate(IGHV = IGHVnew) %>% dplyr::select_("PatID", nameA, nameB) %>% mutate_("namA" = nameA, "namB" = nameB) %>% mutate(naA = as.numeric(as.character(namA))) %>% mutate(naB = as.numeric(as.character(namB))) %>% mutate(mut = factor(mutnames[1 + naA + 2 * naB], levels = mutnames)) %>% arrange(mut)

  rownames(mutStatus) <- mutStatus$PatID
  mutStatus <- mutStatus[rownames(pcaTab),]
  pcaTab <- cbind(pcaTab, data.frame(mutStatus))

  #plot PCA and color samples based on annotations

  
  p <- ggscatter(pcaTab, x = "PC1", y = "PC2", color = "mut", palette = c("#00AFBB", "#E7B800", "#FC4E07", '#2E9FDF'),
  ylab = sprintf("PC2 (%2.1f%%)",varExp[2]), xlab = sprintf("PC1 (%2.1f%%)",varExp[1]), legend = "right", main = "PCA trisomy12 and IGHV") + coord_fixed()
  
  #ggsave(file="~/git/figures_thesis/PCA_epistasis.svg", plot=p, width=7, height=6)
  p
  
}

my_PCA("IGHV", "trisomy12")


```


#Enrichment analysis
#List-based enrichment - Piano package Fisher's exact on genes from one cluster only.
```{r list-based GSA, fig.width = 20, fig.height= 30}

#load gene set collection
#Hallmark
gsc <- loadGSC("/home/almut/Dokumente/masterarbeit/data/h.all.v6.0.symbols.gmt", type="gmt")
#Kegg
gsc_Kegg <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
tft <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c3.tft.v6.0.symbols.gmt", type="gmt")
c5 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c5.all.v6.0.symbols.gmt", type="gmt")
c6 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c6.all.v6.0.symbols.gmt", type="gmt")
c7 <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c7.all.v6.0.symbols.gmt", type="gmt")
cgp <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cgp.v6.0.symbols.gmt", type="gmt")


#cluster description
#cluster <- c("none_down", "both_dn", "only_tri12_up", "none_up", "both_up")

cluster <- c("both_up", "both_dn", "none_up", "none_dn", "tri_up")
#cutoff is set above in to p < 0.1 

```

```{r GSEA, fig.width = 40, fig.height= 35}
#GSA on cluster defined by kmeans clustering. See cluster in heatmap...
setWiseGSA <- function(clusternr, gmtFile){
  geneNam <- rownames(IGHVTri12$geneExp[row_order(IGHVTri12$h1)[[clusternr]],]) 
  pVal <- IGHVTri12$sig_Genes[which(IGHVTri12$sig_Genes$geneName %in% geneNam), "p"]
  gsaTab <- data.frame(row.names = geneNam, stat = pVal)
  res <- runGSA(geneLevelStats = gsaTab,
                      geneSetStat = "fisher",
                     adjMethod = "fdr", gsc=gmtFile,
                     signifMethod = 'nullDist',
                      gsSizeLim=c(1, Inf))
  Res <- arrange(GSAsummaryTable(res),desc(`Stat (non-dir.)`))
  #rename results to plot
  resPlot <- Res[, c(1:5)]
  colnames(resPlot) <- c("pathway", "gene_number", "stat", "p", "p.adj")

  #barplot
  #ggplot(resPlot %>% mutate(log10Padj = -log10(p.adj)), aes(x = reorder(pathway, log10Padj), y = log10Padj, fill = gene_number)) + geom_bar(stat = "identity") + coord_flip() + ggtitle(cluster[clusternr])
  
  enrichPlot <- resPlot %>% mutate(log10Padj = -log10(p.adj)) %>% mutate(genes = ifelse(gene_number > 3, ">3", "<=3"))
  
  p <- ggbarplot(enrichPlot, x = "pathway", y = "log10Padj",
          fill = "genes",           # change fill color by mpg_level
          color = "white",            # Set bar border colors to white
          palette = "jco",            # jco journal color palett. see ?ggpar
          sort.val = "asc",          # Sort the value in ascending order
          sort.by.groups = FALSE,     # Don't sort inside each group
          x.text.angle = 90,          # Rotate vertically x axis texts
          ylab = "Enrichment stat",
          legend.title = "Pathway",
          rotate = TRUE,
          font.x = 65, font.y = 65, font.legend = 65, legend = "right", 
          title = paste0("GSEA in ", cluster[clusternr]),
          ggtheme = theme_pubr()
          ) + font("xy.text", size = 50) + font("title", size = 85, face = "bold")
  
#pdf(file= paste0("/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/GSEA_in_", clusternr, ".pdf"), width=40, height=35)
  p
 # dev.off()
  
  
  }

clusternr = c(1:5)

#resListGSA <- 
lapply(as.numeric(clusternr), FUN = "setWiseGSA", gmtFile = gsc)

#grid.arrange(grobs = resListGSA, ncol = 2)


```
#genesets
Enrichment analysis show many transport and cell localization as well as inflammatory pathways involved in our epistasis model: 
- c7- gene set:
  Cytoskeleton, cell projection, membrane transport processes, response to stimuli, mitchondria (up in tri12)
  
- Kegg:
  endocytosis (all)
  cytokine-cytokine receptor (dn in IGHV)
  MAPK pathway, focal adhesion (up in tri12)
  
- Hallmark:
  inflammatory pathways: Inf, Tnfa, Il (down in IGHV only samples and up in trisomy12)
  Kras, mTor, myc (down in IGHV only samples and up in trisomy12)

- Transcription factor
  AP1 Cytokine receptor
  NFAT (cell motility)
  CEBPB (inflammatory pathways/cytokines)
  HFH4 (cell motility, and immune supression via nfkB)

- Chemical compounds
  Alzheimers disease up 
  TP53 targets up
  apoptosis by doxorubicin up


# Detailed look into single gene expression
How can we biologically explain our findings? And how do they support phenotypical data as the MEK/ERK inhibitor sensitivity of tri12/IGHV-U cells compared to tri12/IGHV-M cells?  Furthermore in cell culture tri12/IGHV-M cells seem to be more vulnerable. Is this connected to a cytokine dependency?

#trisomy12 and VLA4
Trisomy12 has been reported to enhance VLA-4 directed adhesion and motility via upregulate integrin signalling. This is an important part of the leukocyte adhesion cascade, which regulates the migration of CLL-cells to the prosurvival enviroment of the lymphnode. Next to integrins like CD11b, CD18, CD29, and ITGB7, VLA4 is activated by Ras-related protein ligands like CALDAG-GEFI and RAP1B

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4073326/

VLA-4 interacts with the BCR, which is activated in IGHV-U , but not IGHV-M cases. So is there an mechanism connected to cell motility and inflammatory response as cytokine regulation that is activated in trisomy12 cells, but somehow deprived in IGHV-M cells, as they do not use the BCR pathway for tumor progression?

ITGA4 (VLA4 subunit), as well as some regulators of the Ras pathway like  (RAB3,RAB8,..) are upregulated in tri12 only (but not together with IGHV-M) samples
RAP1B-GAP a regulator of RABP1 is upregulated in IGHV-M samples



#pathview
```{r pathview function}

#table with variants 
varTab <- colData(ddsCLL)[, c("IGHVmeth", "del13q14", "trisomy12", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12")]
rownames(varTab) <- colData(ddsCLL)$PatID
varTab[,c(2:12)] <- lapply(varTab[,c(2:12)], function(x) as.numeric(as.character(x)))
varTab <- t(as.matrix(varTab))

#set factor level to 1 and 0 (important for the names in lmfit)
varTab[varTab > 1] <- 0
x <- tibble(IGHV = factor(varTab["IGHVmeth",]),
            tri12  = factor(varTab["trisomy12", ]),
            PatID = colnames(varTab))
  noNA <- !is.na(x$IGHV) & !is.na(x$tri12)
  x <- x[noNA,]

  #patient annotation
IGHVonly <- x %>% filter(IGHV == 1 & tri12 == 0) %>% dplyr::select(PatID)
tri12only <- x %>% filter(IGHV == 0 & tri12 == 1) %>% dplyr::select(PatID)
none <- x %>% filter(IGHV == 0 & tri12 == 0) %>% dplyr::select(PatID)
both <- x %>% filter(IGHV == 1 & tri12 == 1) %>% dplyr::select(PatID)
patgroups <- list("IGHVonly" = IGHVonly, "tri12only" = tri12only, "none" = none, "both" = both)


#gene expression data
#scale and censor
exprMat <- assay(RNAnorm)
exprMat.new <- t(scale(t(exprMat)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

data(paths.hsa)
data(gene.idtype.bods)

my_pathview <- function(pathwayID){
  pathwayDir <- paste0("~/git/Transcriptome_CLL/pathviewPlots/", pathwayID)
  dir.create(pathwayDir)
  setwd(pathwayDir)
  
  for(patGroup in c("IGHVonly", "tri12only", "none", "both")){
    patNam <- patgroups[[patGroup]]$PatID
    expr <- exprMat.new[, patNam]
    expr_Mean <- as_tibble(expr) %>% mutate(mean = dplyr::select(., starts_with(patNam[1])) %>%  rowMeans()) %>% dplyr::select(mean)
    exprMatrix <- as.matrix(expr_Mean)
    rownames(exprMatrix) <- rownames(expr)
    pathview_out <- pathview(gene.data = exprMatrix, pathway.id = pathwayID, species = "hsa", gene.idtype = "ENSEMBL",   kegg.native = T, both.dirs = list(gene = T, cpd = T), bins = list(gene = 10, cpd = 10), limit = list(gene = 4, cpd = 1), node.sum = "mean", out.suffix = patGroup)
   }
}

```

```{r pathview diff pathways, warning= F, message= F}
#B-cell receptor pathway
my_pathview("hsa04662") 

#T-cell receptor pathway
my_pathview("hsa04660") 

#MAPK signaling pathway
my_pathview("hsa04010")
                                                                 
#Ras signaling pathway
my_pathview("hsa04014")

#Rap1 signaling pathway
my_pathview("hsa04015")

#mTor signaling pathway
my_pathview("hsa04150")

#MAPK signaling pathway
my_pathview("hsa04330")

#Focal adhesion pathway
my_pathview("hsa04510")

#TGF-Beta signaling pathway
my_pathview("hsa04350")

#GnrH signaling pathway
my_pathview("hsa04912")
                       
#Actin regulation pathway
my_pathview("hsa04810")
                       
#Leukocyte transendothelial migration pathway
my_pathview("hsa04670")

#Hematopoietic cell lineage pathway
my_pathview("hsa04640")
                       
#TLR signaling pathway
my_pathview("hsa04620")

#Endocytosis
my_pathview("hsa04144")

#Cell adhesion molecules pathway (CAMs)
my_pathview("hsa04514")

#Cytokine-cytokine receptor interaction
my_pathview("hsa04060")

```
#Pathview plots
Results see ../Transcriptome_CLL/pathviewPlots/"pathwayID":
 - B-cell receptor pathway
   clear upregulation in Tri12only (effect is supressed when IGHV is mutated!)  
   
 - T-cell receptor pathway
   clear difference between IGHV M and U, MHC receptor signalling is upregulated in IGHV-U and downregulated in IGHV-M

 - MAPK signaling pathway
   upregulated in Tri12 samples (all IGHV M and U)
                                                                 
 - Ras signaling pathway
   differences between groups (no obvious tendencies)

 - Rap1 signaling pathway
   rap1 is higher expressed when both mutations are present, IGHV and NMDAR/E-cad?

 - mTor signaling pathway
   upregulated in Tri12 samples (all IGHV M and U)
   
 - Notch pathway
   upregulated by IGHV-M, tri12 and both

 - Focal adhesion pathway
   differences between groups (no obvious tendencies)

 - TGF-Beta signaling pathway
   upregulated in both (interaction through Smad4?)

 - GnrH signaling pathway
   not so high expressed in all groups
                       
 - Actin regulation pathway
   not so high expression (Mena and Tri12?)
                       
 - Leukocyte transendothelial migration pathway
   highest expression in tri12 (IGHV-U) samples, downregulated in both (depended on ITGA4, ITGAM, ITGAL) (VLA-4)

 - Hematopoietic cell lineage pathway
   highest expression in tri12 and IGHV-U samples (supressed by IGHV-M?) (CD19, CD20, Cd11b, Flt3L, CD1)
                       
 - TLR signaling pathway
   upregulated in tri12 (all IGHV M and U), high expression of IRAK4

  -Cytokine-Cytokine receptor interactions
   IGHV-M: TNF &TGF family
   IGHV-U: Hematopoietins(IL2RG, IL3RB), Chemokines (CC-subfamily-receptors)
   trisomy12: TGF-beta family, IL2RG down
   both: TGF-Beta, TNF family up, IL2RG down
   
  -Endocytosis
   upregulated in Tri12 - especially early and late endosome
   
  -Cell adhesion Molecules
  IGHV-M: Leukocyte transepithelial migration & Complement, CAMs in Neural system up
  IGHV-U: T cell receptor signaling & Complement, CAMs in Immune system up!
  Both:  T cell receptor signaling down, CAMs in Immune system down!
  Tri12: Leukocyte transepithelial migration & Complement, CAMs in Neural system down
  

#Summary
Endocytosis, B-cell and T-cell receptor signaling pathways show major differences between groups. 
IGHV-U cells use T cell receptor signaling pathway (in line with literature e.g. Zap70), which is downregulated in IGHV-M cells. 
B cell receptor signalling is upregulated in Tri12 cells, dependend on IGHV status (epistasis)
Endocytosis is distinct upregulated in trisomy12 cells independent of IGHV status.
Groups show differences in cytokine/chemokine receptor expression as well as in CAMs. 

 
What is the role of B-cell receptor signaling in trisomy12 cells and why is this dependent on the IGHV status?
What is the role of endocytosis in trisomy12? What is the role of VLA4? Are cells with both mutations more depended on cytokines of the TGF-Beta and TNF family instead of hematopoietins? 


Chatzouli et al.:
- Differences in ERK phosphoyation upon BCR stimulation in IGHV M/U cells. Prolonged Erk activation in IGHV-U cells
- Collaboration of BCR and TLR. BCR AND TLR stimulation is requiered to activate MAPK pathway in M-CLL, but not U-CLL
- " U-CLL cells may be better equipped to transmit BCR-derived signals and thus more responsive to BCR cross linking compared with CLL cells from M-CLL"
  As Tri12 seems to be depended on MEK/ERK pathway (sensitivity to MEK/ERk inhibitors), this could be the reason for diff. in the "both" group.
  
http://www.jimmunol.org/content/192/10/4518.long#ref-36

