---
title: "IGHV_pathway_plots"
author: "almut"
date: "18 Januar 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Aim: Analyse enriched pathways in IGHV

###paper
- heatmapos with enriched pathways combined/ alone
- Beeswarms
- Pathview plots


libraries
```{r load libraries, warning=FALSE, message=FALSE}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(piano)
library(ggsci)
library(RColorBrewer)
library(circlize)
library(pathview)
```

#data
```{r load data}
#dds data set. gene expression data + patmetadata
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

#filter for parients without NA in trisomy12 
ddsCLL <- ddsCLL[, !is.na(colData(ddsCLL)["IGHV"]) & !is.na(colData(ddsCLL)["trisomy12"])]

#differentially expressed genes between IGHV groups 
ighv <- read.csv(file="/home/almut/Dokumente/masterarbeit/results/noTsig/IGHV_diffGenes.csv")


rownames(ighv) <- ighv$X
ighv <- ighv[which(ighv$padj < 0.1 ),-1]
#ighv<- ighv[which(abs(ighv$log2FoldChange) > 2 & ighv$baseMean > 500),]
#ighv<- ighv[which(ighv$baseMean > 500),]

mutStatus <- data.frame(colData(ddsCLL)) %>% arrange(IGHV)
colnames(ddsCLL) <-colData(ddsCLL)$PatID
ddsCLL <- ddsCLL[, mutStatus$PatID]

#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)

#int_genes <- ighv[which(ighv$padj < 0.001 & abs(ighv$log2FoldChange) > 3),]
sig_pathways <- c("KEGG_RIBOSOME", "KEGG_ACUTE_MYELOID_LEUKEMIA","KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY", "KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY","KEGG_NON_SMALL_CELL_LUNG_CANCER", "KEGG_ENDOMETRIAL_CANCER" ,"KEGG_SMALL_CELL_LUNG_CANCER", "KEGG_CHEMOKINE_SIGNALING_PATHWAY","KEGG_CHRONIC_MYELOID_LEUKEMIA", "KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY", "KEGG_ALLOGRAFT_REJECTION", "KEGG_SPLICEOSOME")

gsc_Kegg <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
pathway_name <- sig_pathways[3]
pathway <- gsc_Kegg[["gsc"]][[pathway_name]]
pathway_ensid <- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% pathway)]
pathway_symbol <- rowData(ddsCLL)$symbol[which(rowData(ddsCLL)$symbol %in% pathway)]
path <- data.frame(ensId = pathway_ensid, Symbol = pathway_symbol)
#path <- path[which(path$ensId %in% rownames(ighv)),]

```


Add all relevant information into one matrix
```{r tri12 gene expression}
#filter for sign. genes in Trisomy12 
exprMat <- assay(RNAnorm)
exprIGHV <- exprMat[path$ensId,]
colnames(exprIGHV) <- colData(ddsCLL)$PatID
exprIGHV.new <- log2(exprIGHV)
exprIGHV.new <- t(scale(t(exprIGHV.new)))
exprIGHV.new[exprIGHV.new > 4] <- 4
exprIGHV.new[exprIGHV.new < -4] <- -4
rownames(exprIGHV.new) <- rowData(RNAnorm[as.character(path$ensId),])$symbol

```


#Plot heatmap using complex heatmaps to show expression signature
```{r complex heatmap, fig.width= 22, fig.height=25}

#colors
#colors <- colorRampPalette(rev( brewer.pal(20,"RdBu")) )(20)
colors = colorRamp2(c(-4,-2,0,2,4), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
annocol <- colorRampPalette( brewer.pal(11,"Paired") )(11)
annocol <- get_palette("jco", 10)
annocolor <- list(IGHV = c("M" = annocol[1], "U" = annocol[2]))
rowcolors <-colorRampPalette(brewer.pal(5, "Set1"))(5)
rowcolors[6] <- "white"


feature <- as.data.frame(colData(ddsCLL)[,c("IGHV")]) #, "trisomy12")])
colnames(feature) <- c("IGHV") #"trisomy12")#, 

ha_col <- HeatmapAnnotation(feature, col = annocolor, annotation_height = unit(c(rep(1.9, 1)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 50), labels_gp = gpar(fontsize = 45),  grid_height = unit(1.9, "cm"), grid_width = unit(1.9, "cm")))


h1 <- Heatmap(exprIGHV.new ,                                                     #Cluster param
              #km = 2,
              gap = unit(0.5, "cm"),
              cluster_columns = F,
              #clustering_distance_columns = "euclidean",
              #clustering_method_columns = "ward.D2",
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = paste0("IGHV: ", pathway_name),                      #Graphic parameter
              col = colors,
              column_title_gp = gpar(fontsize = 60, fontface = "bold"), 
              heatmap_legend_param = list(title = "expr", 
                                          title_gp = gpar(fontsize = 50), 
                                          grid_height = unit(1.9, "cm"), 
                                          grid_width = unit(1.9, "cm"), 
                                          gap = unit(2, "cm"), 
                                          labels_gp = gpar(fontsize = 45)), 
              column_dend_height = unit(2.5, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = TRUE, 
              row_names_gp = gpar(fontsize = 40),
              top_annotation = ha_col)


#svg(filename="~/git/figures_thesis/gene_expr/gene_exprTrisomy12_gsea_Hallmark.svg", width=30, height=45)
#pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/gene_exprIGHV_filtered.pdf", width=22, height=25)

draw(h1) #  + ha_genes) #+ ha_row ha_chrom +
#dev.off()

```



#gene counts for specific genes
```{r single gene counts}
#function to create stripchart plots for specific genes
gene_count <- function(gene_nam){
  geneEnsID <- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% gene_nam)]
  geneNum <- counts(ddsCLL)[geneEnsID,]
  mutPat <- as.data.frame(colData(ddsCLL)[, c("IGHV")])
  colnames(mutPat) <- c("genotype")
  geneDat <- cbind(mutPat, geneNum)
  colnames(geneDat) <- c("genotype", "counts")
  
  p <- ggstripchart(geneDat, x = "genotype", y = "counts",
          color = "genotype",
          palette = "jco",
          add = "mean_sd",
          title = paste(gene_nam),
          ylab = "normalized counts")
 # ggsave(file=paste0("/home/almut/Dokumente/masterarbeit/workinprogress/genetic_interaction_", gene_nam, ".svg"), plot=p, width=6, height=5)
  p
}

geneList <- path$Symbol

lapply(geneList, gene_count)


```