---
title: "IGHV_signature"
author: "Almut Luetge"
date: "October 26, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Aim: Define an IGHV expression signature


libraries
```{r load libraries}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(piano)
library(RColorBrewer)

```

#data
```{r load data}
#dds data set. gene expression data + patmetadata
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")

#differentially expressed genes between IGHV groups with Tsig as confounding factor
ighv <- read.csv(file="/home/almut/Dokumente/masterarbeit/results/noFilter/IGHV_diffGenes.csv")
rownames(ighv) <- ighv$X
#ighv<- ighv[c(1:3000),-1]
ighv<- ighv[which(ighv$padj < 0.01),-1]
#ighv <-as.data.frame(ighv[c("ENSG00000004468"),])#  "ENSG00000115085"),]

int_genes <- ighv[which(ighv$padj < 0.001 & abs(ighv$log2FoldChange) > 4.5),]

#Add diff pathway information
#pathways <- read.csv(file="/home/almut/Documents/CLL/results/deseq/IGHV_Tsig_Tri12/IGHV_diffPathways_GSEAKegg.csv")
#pathways <- pathways[,-1] 
#pathways <- pathways %>% mutate(group = ifelse(stat > 0, "up", "down")) %>% filter(ifelse(group %in% "up", p_up < 0.1, p_dn < 0.1))

#order columns by IGHV status
mutStatus <- data.frame(colData(ddsCLL)) %>% mutate(IGHVnew = ifelse(IGHV %in% "M", 1, 0))  %>% arrange(IGHVnew)
colnames(ddsCLL) <-colData(ddsCLL)$PatID
ddsCLL <- ddsCLL[, mutStatus$PatID]

```


Add all relevant information into one matrix
```{r ighv gene expression}
#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)


#filter for sign. genes in IGHV 
exprMat <- assay(RNAnorm)
#exprIGHV <- t(as.matrix(exprMat[rownames(ighv),]))
exprIGHV <- exprMat[rownames(ighv),]
colnames(exprIGHV) <- colData(ddsCLL)$PatID
exprIGHV.new <- log2(exprIGHV)
exprIGHV.new <- t(scale(t(exprIGHV.new)))
exprIGHV.new[exprIGHV.new > 4] <- 4
exprIGHV.new[exprIGHV.new < -4] <- -4
rownames(exprIGHV.new) <- rowData(RNAnorm[rownames(ighv),])$symbol

```

```{r pathway annotation}

#Rowdata
#chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[rownames(ighv),])$chromosome)
rownames(chrom) <- rownames(exprIGHV.new)  
colnames(chrom) <- "chromosome"



#pathway Annotation
#topGO <- sort.int(abs(pathways$stat), index.return = T, decreasing = T)$ix[1:5]
#topPath <- pathways[topGO,]


#Annotate Genes in this pathways
#gsc <- loadGSC("/home/almut/Documents/CLL/data/h.all.v6.0.symbols.gmt", type="gmt")
#gsc_Kegg <- loadGSC("/home/almut/Documents/CLL/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")
#tft <- loadGSC("/home/almut/Documents/CLL/data/c3.tft.v6.0.symbols.gmt", type="gmt")
#c5 <- loadGSC("/home/almut/Documents/CLL/data/c5.all.v6.0.symbols.gmt", type="gmt")
#c6 <- loadGSC("/home/almut/Documents/CLL/data/c6.all.v6.0.symbols.gmt", type="gmt")
#c7 <- loadGSC("/home/almut/Documents/CLL/data/c7.all.v6.0.symbols.gmt", type="gmt")
#cgp <- loadGSC("/home/almut/Documents/CLL/data/c2.cgp.v6.0.symbols.gmt", type="gmt")
#topGsc <- gsc_Kegg$gsc[as.character(topPath$pathway)]

#topPathGenes <- lapply(topGsc, function(x){
#  genes <- ifelse(ighv$Symbol %in% unlist(x), 1 , 0)
#})  %>% do.call(cbind,.)



#topPathGenes <- apply(topPathGenes, 2, function(x) as.factor(as.character(x)))
#rownames(topPathGenes) <- ighv$Symbol


```





#Plot heatmap using complex heatmaps to show expression signature
```{r complex heatmap, fig.width= 30, fig.height=45}

#colors
#colors <- colorRampPalette( rev(brewer.pal(20,"RdBu"))) (10)
colors = colorRamp2(c(-4,-2,0,2,4), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
#annocol <- colorRampPalette( brewer.pal(11,"Paired") )(11)
annocol <- get_palette("jco", 10)
annocolor <- list(IGHV = c("M" = annocol[1], "U" = annocol[2]), Methylation = c("IP" = annocol[5], "LP" = annocol[6], "HP" = annocol[7]),trisomy12 = c("1" = annocol[8], "0" =annocol[4]))
rowcolors <-colorRampPalette(brewer.pal(5, "Set2"))(5)
rowcolors[6] <- "white"

#annoRowcolor <- list()
#  for(i in c(1:length(topGO)))(
#  annoRowcolor[[i]] = c("0"= rowcolors[6], "1" = rowcolors[i]) 
#  )
  
#names(annoRowcolor) <- topPath$pathway

chromcol <- list(chromosome = c("1" ="#00d3b4", "2" ="#f443b9", "3" = "#6cde56", "4" = "#932db4", "5" ="#dcc608", "6"= "#0a54d9", "7" = "#fea114", "8"= "#0098ec","12" ="#fd4b43","10" = "#01e09f", "11" = "#ff6aab", "9" = "#a8d37e", "13" = "#a59eff","14" = "#be6f00", "15" = "#03d1f5", "16" = "#a02008", "17" = "#89c8ff", "18" = "#7a7600", "19" ="#005b8f", "20" = "#ffad61","21" = "#80386c", "22" = "#e3c27d", "X" ="#843e29", "MT" = "#ff938b"))


ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(1.3, "cm"), annotation_legend_param = list(ncol = 2, title_gp = gpar(fontsize = 30), labels_gp = gpar(fontsize = 25),  grid_height = unit(1, "cm"), grid_width = unit(1, "cm"), gap = unit(1,"cm")))

#ha_row = rowAnnotation(df = topPathGenes, col = annoRowcolor, annotation_width = unit(c(rep(1, 7)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15),  grid_height = unit(0.7, "cm"), grid_width = unit(0.7, "cm"), gap = unit(5, "mm")))

ha_col <- HeatmapAnnotation(colData(ddsCLL)[,c("IGHV", "Methylation", "trisomy12")], col = annocolor, annotation_height = unit(c(rep(1.9, 3)), "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 40), labels_gp = gpar(fontsize = 30),  grid_height = unit(1.3, "cm"), grid_width = unit(1.3, "cm"), gap = unit(5, "mm")))




h1 <- Heatmap(exprIGHV.new,                                                     #Cluster param
              km = 2,
              gap = unit(0.5, "cm"),
              #clustering_distance_columns = "euclidean",
              #clustering_method_columns = "ward.D2",
              cluster_columns = F,
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = "Gene signature: IGHV status",                      #Graphic parameter
              col = colors,
              column_title_gp = gpar(fontsize = 60, fontface = "bold"), 
              heatmap_legend_param = list(title = "Expr.", 
                                          title_gp = gpar(fontsize = 50), 
                                          grid_height = unit(1.5, "cm"), 
                                          grid_width = unit(1.5, "cm"), 
                                          gap = unit(15, "mm"), 
                                          labels_gp = gpar(fontsize = 50)), 
              #column_dend_height = unit(3, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = FALSE, 
              top_annotation = ha_col)


#Annotate known genes from litertaure
subset <- as.data.frame(rowData(RNAnorm[rownames(ighv),]))
subset1 <- subset[which(subset$symbol %in% c("ZAP70", "CD38")),]
subset2 <- subset[which(subset$symbol %in% int_genes$Symbol),]
#subset <- subset1
subset <- rbind(subset1, subset2)
rownames(subset) <- subset$symbol
geneIDs <- which(rownames(exprIGHV.new) %in% rownames(subset))
labels <- rownames(exprIGHV.new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 30)), width = unit(8, "cm"))




#pdf(file="~/git/figures_thesis/gene_expr/gene_exprIGHV_gsea_Hallmark.pdf", width=30, height=45)

draw( h1 + ha_genes) # + ha_chrom + ha_row)
#dev.off()




```

#Plot chromosome distributions
Is there any connection to genes on chromosome 12 (more or less expressed?)
```{r plot chromosome dist, fig.width= 8, fig.height=8}

IGHVGroups <- ighv %>%  mutate(group = ifelse(stat > 0, "up", "down"))
rownames(IGHVGroups) <- rownames(ighv)


chromDist <- lapply(c("up", "down"), function(group){
  geneID <- rownames(IGHVGroups)[which(IGHVGroups$group %in% group)]
  chromo <- as.data.frame(rowData(RNAnorm[geneID,])$chromosome)
  rownames(chromo) <- geneID 
  colnames(chromo ) <- "chromosome"
  
  
  chrom <- chromo %>% dplyr::select(chromosome) %>% group_by(chromosome) %>% summarise(total = n())

  chrom$label <- paste0(chrom$chromosome, "(", chrom$total, ")")
  
 # svg(filename=paste0("~/git/figures_thesis/tri12/chromosom_dist_", group, ".svg"), width=6, height=6)
  pie(chrom$total, col=colorRampPalette(brewer.pal(11,'Spectral'))(25), border="white", labels=chrom$label, main = paste0("Chromosomal gene distribution in IGHV: ", group))
  #dev.off()
})

```




Evaluate the B cell receptor pathway - tumor epistatsis?

```{r B cell receptor pathway}
B_genes <-gsc_Kegg$gsc$KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY

#KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY
#KEGG_MAPK_SIGNALING_PATHWAY
#KEGG_PRIMARY_IMMUNODEFICIENCY
#KEGG_ENDOCYTOSIS
#KEGG_WNT_SIGNALING_PATHWAY
#KEGG_HEMATOPOIEITIC_CELL_LINEAGE
#KEGG_RAS_SIGNALING_PATHWAY
#KEGG_NF-KAPPA_B_SIGNALING_PATHWAY
#KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY
#KEGG_FOCAL_ADHESION
#KEGG_CALCIUM_SIGNALING_PATHWAY
#KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY
#KEGG_TGF_BETA_SIGNALING_PATHWAY

#Hallmark
#B_genes <- gsc$gsc$HALLMARK_MTORC1_SIGNALING
#HALLMARK_INTERFERON_GAMMA_RESPONSE
#HALLMARK_P53_PATHWAY
#HALLMARK_G2M_CHECKPOINT
#HALLMARK_IL2_STAT5_SIGNALING
#HALLMARK_UV_RESPONSE_UP
#HALLMARK_PI3K_AKT_MTOR_SIGNALING
#HALLMARK_TNFA_SIGNALING_VIA_NFKB
#HALLMARK_ANGIOGENESIS
#HALLMARK_NOTCH_SIGNALING
#HALLMARK_MTORC1_SIGNALING

#filter for sign. genes in IGHV 
exprMatSymbol <- assay(RNAnorm)
rownames(exprMatSymbol) <- rowData(RNAnorm)$symbol

B_filtered <- B_genes[which(B_genes %in% rownames(exprMatSymbol))]
exprBcell <- exprMatSymbol[B_filtered,]
colnames(exprBcell) <- colData(ddsCLL)$PatID
exprBcell.new <- log2(exprBcell)
exprBcell.new <- t(scale(t(exprBcell.new)))
exprBcell.new[exprBcell.new > 4] <- 4
exprBcell.new[exprBcell.new < -4] <- -4

```



#Plot heatmap using complex heatmaps to show expression signature
```{r B cell receptor signaling, fig.width= 30, fig.height=45}

h2 <- Heatmap(exprBcell.new,                                                     #Cluster param
              #km = 2,
              clustering_distance_columns = "euclidean",
              clustering_method_columns = "ward.D2",
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = "Gene signature: IGHV status",                      #Graphic parameter
              col = colors,
              column_title_gp = gpar(fontsize = 30, fontface = "bold"), 
              heatmap_legend_param = list(title = "Gene expression", 
                                          title_gp = gpar(fontsize = 20), 
                                          grid_height = unit(0.5, "cm"), 
                                          grid_width = unit(0.5, "cm"), 
                                          gap = unit(5, "mm"), 
                                          labels_gp = gpar(fontsize = 15)), 
              column_dend_height = unit(2, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = TRUE, 
              top_annotation = ha_col)




#svg(filename="~/git/figures_thesis/gene_expr/gene_exprIGHV_gsea_Hallmark.svg", width=30, height=45)

draw( h2)
#dev.off()




```



```{r pathview}
library(pathview)
data(paths.hsa)
data(gene.idtype.bods)

#gene expression data
#scale and censor
exprMat <- assay(RNAnorm)
exprMat.new <- t(scale(t(exprMat)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

colnames(exprMat.new) <- colData(RNAnorm)$PatID

my_pathview <- function(pathwayID){
  pathwayDir <- paste0("~/git/Transcriptome_CLL/pathviewPlots/", pathwayID)
  dir.create(pathwayDir)
  setwd(pathwayDir)
  
  for(patGroup in c("M", "U")){
    patNam <- colData(ddsCLL)$PatID[which(colData(ddsCLL)$IGHV %in% patGroup)]
    expr <- exprMat.new[, patNam]
    expr_Mean <- as_tibble(expr) %>% mutate(mean = dplyr::select(., starts_with(patNam[1])) %>%  rowMeans()) %>% dplyr::select(mean)
    exprMatrix <- as.matrix(expr_Mean)
    rownames(exprMatrix) <- rownames(expr)
    pathview_out <- pathview(gene.data = exprMatrix, pathway.id = pathwayID, species = "hsa", gene.idtype = "ENSEMBL",   kegg.native = T, both.dirs = list(gene = T, cpd = T), bins = list(gene = 10, cpd = 10), limit = list(gene = 4, cpd = 1), node.sum = "mean", out.suffix = patGroup)
   }
}

#B-cell receptor pathway
my_pathview("hsa04662") 

#acute myeloid leukemia
my_pathview("hsa05221") 

#arginine and proline metabolism
my_pathview("hsa00330") 

```

