---
title: "Geneoverlap"
author: "Almut Luetge"
date: "August 21, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

22.08.16

#Aim: Find overlaping genes in modules detected in WGCNA and group of differentially expressed genes in C1C2 groups
Basically another way to find association/correlation. used as the eigengenes in WGCNA package do not give a stable informative conclusion about module - varaint relationships and thus need to be confirmed and further explored.



#load libraries
```{r libraries, warning=FALSE}
library(GeneOverlap)
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(pheatmap)
library(RColorBrewer)
library("IHW")

```


#load data sets 
```{r data}
#RNA dataset
load("/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")

#Diff expressed gene list from Feirrera
fileName <- paste("/home/almut/Documents/CLL/data/FeirreraPaper/Supplemental_File2_diffExpGenesC1C2_FC2.csv")
c1c2_genes <- read.csv(fileName,sep = ",",row.names = 1,header = TRUE)

#Diff expressed gene list from Feirrera
fileName <- paste("/home/almut/Documents/CLL/data/FeirreraPaper/Supplemental_File2_diffExpGenesIGHV_FC2.csv")
ighv_genes <- read.csv(fileName,sep = ",",row.names = 1,header = TRUE)




#patMeta data
load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData") 

#Genemodules
load("/home/almut/Documents/CLL/data/geneModules_WGCNA25082017_ProteinCoding.RData")


```


##Create differentially expressed gene sets to compare the overlap:
a)trisomy12 patients
b)del11q22.3
c)del13q14
d)SF3B1
e)TP53
f)NOTCH1
g)ighv (own dataset)


#Deseq
```{r prepare metadata}
##Prepare dds dataset --> add patmeta data and remove non CLL patients

#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
ddsCLL <- dds[,colData(dds)$diag %in% cll_vector]

#number genes and samples
dim(dds)
dim(ddsCLL)

# Remove rows/genes with too few counts
keep <-apply(counts(ddsCLL), 1, function(x) any(x >= 10))
ddsCLL <- ddsCLL[keep,]

#Add meta data
patMat <- patMeta[which(patMeta$Patient.ID %in% colData(ddsCLL)$PatID),]
patMat <-patMat[, c("Patient.ID", "drugGroup", "del11q22.3", "NOTCH1", "SF3B1", "TP53")]
#patMat <- patMat[,-1]
patMat <-as.data.frame(patMat)
patMat[,c(2:6)] <- lapply(patMat[,c(2:6)], function(x) as.factor(x))
rownames(patMat) <- patMat$Patient.ID

patMat <- patMat[colData(ddsCLL)$PatID,]

colData(ddsCLL) <- cbind(colData(ddsCLL), patMat[,c(2:6)])


```





```{r deseq}
###Deseq
ddsCLL <- estimateSizeFactors(ddsCLL)


#write a function to perform deseq for different genetic conditions 
diff <- function(cond){
  ddsCLL_new <- ddsCLL[,!is.na(colData(ddsCLL)[,cond])]
  design(ddsCLL_new) <- as.formula(paste("~ ", paste(cond)))
  rnaRaw <- DESeq(ddsCLL_new, betaPrior = FALSE)
  res <- results(rnaRaw, filterFun=ihw)
  resOrdered <- res[order(res$pvalue),]
  resSig <- subset(resOrdered, padj < 0.01)
  rownames(resSig)
}

gene_conditions <- c("IGVH", "trisomy12", "del13q14", "drugGroup", "del11q22.3", "NOTCH1", "SF3B1", "TP53")

##Takes a lot of time - load data instead
#res_list <- lapply(gene_conditions, diff)
#names(res_list) <- gene_conditions

#save(res_list, file="/home/almut/Documents/CLL/data/diff_genes_Meta.RData")
load("/home/almut/Documents/CLL/data/diff_genes_Meta.RData")
```




#Create Geneoverlap objects
```{r geneoverlap, fig.width=15, fig.height=15}
#function to create an geneoverlap object from module gene list and c1c2 genes
go.obj <- function(module_nam) {
                  mod <- moduleGene[1,]
                  mod <- unlist(mod[module_nam])
                  go.obj1 <- newGeneOverlap(mod, rownames(c1c2_genes), genome.size=nrow(dds))
                  go.obj2 <- testGeneOverlap(go.obj1)
}



modList <- colnames(moduleGene)

go_list <- lapply(modList, go.obj)

feirrera <-list(rownames(c1c2_genes), rownames(ighv_genes))
names(feirrera) <- c("c1c2", "ighv_feirrera")

all_diffGenes <- append(res_list, feirrera) 
goObj <-newGOM(moduleGene[1,], all_diffGenes, genome.size=nrow(dds))

#number of overlapping genes
goObj_intersectSize <- getMatrix(goObj, name = "intersection")
moduleSize <- getGsetA(goObj)
condSize <-getGsetB(goObj)

drawHeatmap(goObj, what="odds.ratio", ncolused=9, grid.col="Green", note.col="black", adj.p=T)


```
##Geneoverlap within lists
```{r within list overlap, fig.width=15, fig.height=15}
goObj_inside <-newGOM(all_diffGenes)

drawHeatmap(goObj_inside, what="odds.ratio")



```

###Investigate correlating modules
##trisomy12 and "red"
```{r trisomy12}
#load gene set collection
#Hallmark
gsc <- loadGSC("/home/almut/Documents/CLL/data/h.all.v6.0.symbols.gmt", type="gmt")
#Kegg
gsc_Kegg <- loadGSC("/home/almut/Documents/CLL/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")

#function to do GSEA
 gsahyp <- function(moduleNam){
   color <- as.data.frame(moduleGene[,moduleNam])
   genes <- unique(as.vector(color$hgnc_symbol))
   res <- runGSAhyper(genes, gsc = gsc_Kegg, adjMethod = "none")
   summary <- as.data.frame(res$contingencyTable)
   networkPlot(res, class = "non", adjusted = F, significance=0.5, main = moduleNam)
 }

#runGSeA 
networks <- lapply(modList[11], gsahyp)
 


```



