---
title: "coombes_data"
author: "almut"
date: "21 August 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Aim look for epistasis in coombes data set

```{r libraries , warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(data.table)
library(RColorBrewer)
library(ComplexHeatmap)
library(ggpubr)
library(circlize)
#library(lumi)
```

```{r data}
expr <- fread("/home/almut/Dokumente/masterarbeit/data/coombes/expressionData.csv")
meta <- fread("/home/almut/Dokumente/masterarbeit/data/coombes/safeClinical.csv")
dict <- fread("/home/almut/Dokumente/masterarbeit/data/coombes/probeFeatures.csv")

```

```{r prepare data}
#meta data
#meta_new <- as.tibble(meta) %>% mutate(IGHV = ifelse(mutation.status %in% "M", 1, 0)) %>% filter(Purity %in% c("Diploid","Tri12"), Dohner %in% c("Normal", "Tri12")) %>% mutate(trisomy12 = ifelse(Purity %in% "Diploid" | Dohner %in% "Normal", 0, 1)) %>% rename(PatID = V1) %>% select(PatID, trisomy12, IGHV)

#meta data
meta_new <- as.tibble(meta) %>% mutate(IGHV = ifelse(mutation.status %in% "M", 1, 0)) %>% mutate(trisomy12 = ifelse(Purity %in% "Tri12" | Dohner %in% "Tri12", 1, 0)) %>% rename(PatID = V1) %>% select(PatID, trisomy12, IGHV)


#expression data
expr_new <- as.data.frame(expr)
rownames(expr_new) <- expr$V1
expr_new <- expr_new[, meta_new$PatID]
expr <- expr[, meta_new$PatID, with=FALSE]

#dict
dict_new <- as.tibble(dict) %>% rename(geneID = V1) %>% select(geneID, SYMBOL)

```

#Model epistatic interaction using limma
Differentiate into 4 groups
```{r groups}
mutnames <- c("none", "IGHV-M", "trisomy12", "both")
metaGroups <- meta_new %>% mutate(mutStatus = factor(mutnames[1 + IGHV + 2 * trisomy12], levels = mutnames)) 
                                    
```


Identify significant methylation sites
```{r}
x <- tibble(varA = factor(meta_new$trisomy12),
            varB  = factor(meta_new$IGHV))
 
fit <- eBayes(lmFit(expr_new, design = model.matrix( ~ varA * varB, data = x))) 
interaction <- tibble(geneID = rownames(expr_new), p = fit$p.value[, "varA1:varB1"]) %>% mutate(pBH = p.adjust(p, method = "BH")) %>% arrange(p)

#save(interaction,file = "/home/almut/Dokumente/masterarbeit/data/coombes/interaction.RData")
#load("/home/almut/Dokumente/masterarbeit/data/coombes/interaction.RData")

sig_interaction <- interaction %>% filter(p <= 0.01)

sig_named <- left_join(sig_interaction, dict_new)

```

#Compare with our dataset
```{r overlap with epistasis in gene expression}
#load differential data freom epistasis_Deseq_IGHV_tri12. Rmd
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHTri12.RData")
resOrdered <- res[order(res$pvalue),]
resSig <- subset(resOrdered, padj < 0.1)
resTab <- as.data.frame(resSig)
#count data for annotation
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")
resTab$symbol <- rowData(ddsCLL[rownames(resSig),])$symbol
sig_expr <- resTab$symbol


overlap <- as.tibble(sig_expr) %>% filter(!value %in% "")  %>% filter(value %in% sig_named$SYMBOL)
overlap_Il <- dict_new %>% filter(SYMBOL %in% overlap$value) 

#overlap_total 
total <- as.tibble(sig_expr) %>% filter(!value %in% "") %>% filter(value %in% dict_new$SYMBOL)

```



#Plot significant genes
Heatmap significant genes: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction ighv, fig.width=35, fig.height=41}

#expressiom data
sig_genes = expr_new[sig_named$geneID,]
sig_genesID <- rownames(sig_genes)


#scale and censor
sig_genes_new <- log2(sig_genes)
sig_genes_new<- t(scale(t(sig_genes_new)))
sig_genes_new[sig_genes_new > 4] <- 4
sig_genes_new[sig_genes_new < -4] <- -4


#order by mutation groups  
mutStatus <- metaGroups %>% arrange(mutStatus)
sig_genes_new <- sig_genes_new[,mutStatus$PatID]


#colors
#colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
colors = colorRamp2(c(4, 1.5, 0, -1.5, -4), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
annocol <- get_palette("jco", 10)
annocolor <- list(Variant = c(annocol[1], annocol[2], annocol[3], annocol[4]))
names(annocolor$Variant) <- c("none", "IGHV-M", "trisomy12", "both")

#column annotation  
mutationStatus <- data.frame(mutStatus$mutStatus)
rownames(mutationStatus) <- mutStatus$PatID
colnames(mutationStatus) <- "Variant"

ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 60), labels_gp = gpar(fontsize = 50),  grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm"), gap = unit(15, "mm")))

  #rowcluster
  genes_dist <- dist(sig_genes_new)
  rowcluster = hclust(genes_dist, method = "ward.D2")

  #heatmap
  h1 <- Heatmap(sig_genes_new, col = colors ,column_title = paste0("Expression interactions:", "IGHV", "-", "trisomy12"), column_title_gp = gpar(fontsize = 60, fontface = "bold"), heatmap_legend_param = list(title = "expr", title_gp = gpar(fontsize = 60), grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm"), gap = unit(15, "mm"), labels_gp = gpar(fontsize = 50)), row_dend_width = unit(3.5, "cm"), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = FALSE, show_column_dend = FALSE, cluster_columns = FALSE, cluster_rows = rowcluster, split = 4, gap = unit(0.6,"cm"), column_order = mutStatus$PatID)

cluster <- as.tibble(rowcluster$labels) %>% rename(geneID = value)
lab <- left_join(cluster, overlap_Il)
index <- which(!is.na(lab$SYMBOL))
labels <- lab$SYMBOL[index]
  
  
ha_genes <- rowAnnotation(link = row_anno_link(at = index, labels = labels, labels_gp = gpar(fontsize = 25)), width = unit(3, "cm"))


  #svg(filename="~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg", width=30, height=50)
  #pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistasis_coombes.pdf", width=35, height=45)
  draw(h1 + ha_genes) 
  #dev.off()

 IGHVTri12 <- list("sig_genes_new" = rownames(sig_genes_new), "Expr" = sig_genes_new, "h1"= h1)

```





