---
title: "ddsAndFeirrera"
author: "Almut Luetge"
date: "August 30, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Aim: Do we fid epistasis in feirrera data 
```{r libraries, warnings = F, message = F}
library(tidyverse)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(ggdendro)
library(ComplexHeatmap)
library(reshape2)
library(ggpubr)
```

1. load data

```{r data }
#RNA dataset
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")
dds <-ddsCLL


#feirrera dataset (controls are removed and genes with less than 10 counts --> see(feirrera/Feirrera_originaldata.Rmd))
load("/home/almut/Dokumente/masterarbeit/data/FeirreraPaper/feirreraCLL.RData")

#Variance stabilization transformation of the raw data
fCLL <- estimateSizeFactors(feirreraCLL)
RNAnorm <- varianceStabilizingTransformation(fCLL, blind=T)

#load diff expressed genes by IGHV status
ighv <- read.csv("/home/almut/Dokumente/masterarbeit/results/noTsig/IGHV_diffGenes.csv")
ighv <- as.tibble(ighv) %>% filter(padj < 0.01)
diff <- ighv %>% filter(!X %in% "") %>% select(X)
diff_ID <- c(as.character(diff$X))
diff_ID_total <- diff_ID[which(diff_ID %in% rownames(RNAnorm))]


```

2. Add M-status and trisomy12 to feirrera dataset
- define IGHV status by cluster based on diff genes in IGHV in our cohort
- define trisomy12 status by expression of genes on chrom 12
```{r add genotype data by clustering, fig.width=8, fig.height=6}

ighv_expr <-RNAnorm[diff_ID_total,]

exprMat <- assay(ighv_expr)


sampleDists <- dist(t(exprMat))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- colData(ighv_expr)$FeirreraID

colnames(sampleDistMat) <- NULL

sampleDistDat <- t(exprMat)
rownames(sampleDistDat) <-colData(ighv_expr)$FeirreraID


#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"RdBu")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")

#clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists,


#hclust object
ighv_cluster <- hclust(dist(sampleDistDat), method = "ward.D2", members = NULL)

## S3 method for class 'hclust'
plot(ighv_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "Cluster by groups from Feirrera et al.",
     sub = NULL, xlab = "All_patients", ylab = "Height")


dendr    <- dendro_data(ighv_cluster, type="rectangle") # convert for ggplot
clust    <- cutree(ighv_cluster,k=2)                    # find 2 clusters
clust.df <- data.frame(label=names(clust), cluster=factor(clust))

# dendr[["labels"]] has the labels, merge with clust.df based on label column
dendr[["labels"]] <- merge(dendr[["labels"]], clust.df, by="label")

# plot the dendrogram; note use of color=cluster in geom_text(...)
ggplot() + geom_segment(data=segment(dendr), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr), aes(x, y, label=label, hjust=0, color=cluster), 
           size=8) +
        coord_flip() + scale_y_reverse(expand=c(0.9, 0)) + 
        ggtitle("Clustering by ighv-genes") +
        theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_text(size = 20),
        axis.text.x=element_text(size = 20),
        axis.title = element_blank(),
        panel.background=element_rect(fill="white"),
        panel.grid=element_blank(),
        plot.title = element_text(size = 80)) 

##Add group information to patients
colData(feirreraCLL)$ighv <- clust.df$cluster[match(colData(feirreraCLL)$FeirreraID,rownames(clust.df))] 


```

#trisomy12 cases 
```{r trisomy12, fig.width=8, fig.height=6}
tri12 <- as.tibble(rowData(feirreraCLL)) %>% mutate(ensembl = rownames(feirreraCLL)) %>% filter(chromosome %in% "12")

tri12_expr <-RNAnorm[tri12$ensembl,]

exprMat <- assay(tri12_expr)


sampleDists <- dist(t(exprMat))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- colData(ighv_expr)$FeirreraID

colnames(sampleDistMat) <- NULL

sampleDistDat <- t(exprMat)
rownames(sampleDistDat) <-colData(ighv_expr)$FeirreraID


#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"RdBu")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")

#clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists,


#hclust object
tri12_cluster <- hclust(dist(sampleDistDat), method = "ward.D2", members = NULL)

## S3 method for class 'hclust'
plot(tri12_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "Cluster by genes on chrom 12",
     sub = NULL, xlab = "All_patients", ylab = "Height")


dendr    <- dendro_data(tri12_cluster, type="rectangle") # convert for ggplot
clust    <- cutree(tri12_cluster,k=2)                    # find 2 clusters
clust.df <- data.frame(label=names(clust), cluster=factor(clust))

# dendr[["labels"]] has the labels, merge with clust.df based on label column
dendr[["labels"]] <- merge(dendr[["labels"]], clust.df, by="label")

# plot the dendrogram; note use of color=cluster in geom_text(...)
ggplot() + geom_segment(data=segment(dendr), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr), aes(x, y, label=label, hjust=0, color=cluster), 
           size=8) +
        coord_flip() + scale_y_reverse(expand=c(0.9, 0)) + 
        ggtitle("Clustering by genes on chrom 12") +
        theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_text(size = 20),
        axis.text.x=element_text(size = 20),
        axis.title = element_blank(),
        panel.background=element_rect(fill="white"),
        panel.grid=element_blank(),
        plot.title = element_text(size = 80)) 

##Add group information to patients
colData(feirreraCLL)$trisomy12 <- clust.df$cluster[match(colData(feirreraCLL)$FeirreraID,rownames(clust.df))] 

#save(feirreraCLL, file= "/home/almut/Dokumente/masterarbeit/data/FeirreraPaper/feirreraCLL_inferredColData.RData")


```

##Epistasis
```{r epistasis}
RNAnorm <- varianceStabilizingTransformation(feirreraCLL, blind=T)
colnames(RNAnorm) <- colData(RNAnorm)$FeirreraID 

#design matrix with interaction term
design(feirreraCLL) <- as.formula(paste("~ ighv + trisomy12 + ighv:trisomy12"))
rnaRaw <- DESeq(feirreraCLL, betaPrior = FALSE)
resultsNames(rnaRaw)
#is the effect of IGHV different between different trisomy12 groups?
res <- results(rnaRaw, name = "ighv2.trisomy122")

resOrdered <- res[order(res$pvalue),]
resSig <- subset(resOrdered, padj < 0.1)# & abs(log2FoldChange) > 2)
  #rownames(resSig)
resTab <- as.data.frame(resSig)
resTab$symbol <- rowData(RNAnorm[rownames(resSig),])$symbol



```

#Gene expression of significant epistatic genes
Heatmap of interacting genes: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction ighv, fig.width=35, fig.height=41}

  #filter by variant
  sig_Genes <- rownames(resSig)
  
  #gene expression data
  #filter for parients without NA in trisomy12 
  RNAnorm <- RNAnorm[, !is.na(colData(RNAnorm)["ighv"]) & !is.na(colData(RNAnorm)["trisomy12"])]
  
  
  geneExpression = assay(RNAnorm)[sig_Genes,]
  rownames(geneExpression) <- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% sig_Genes)]

  #scale and censor
  geneExpression_new <- log2(geneExpression)
  geneExpression_new<- t(scale(t(geneExpression_new)))
  geneExpression_new[geneExpression_new > 6] <- 6
  geneExpression_new[geneExpression_new < -6] <- -6

  mutnames <- c("none", "IGHV", "trisomy12", "both")
  
  colData(RNAnorm)$trisomy12 = as.numeric(as.character( colData(feirreraCLL)$trisomy12)) - 1
  mutStatus <- data.frame(colData(RNAnorm)) %>% mutate(IGHV = ifelse(ighv %in% "2", 1, 0)) %>% dplyr::select(-ighv) %>% dplyr::select_("FeirreraID", "IGHV", "trisomy12") %>% mutate_("namA" = "IGHV", "namB" = "trisomy12") %>% mutate(naA = as.numeric(as.character(namA))) %>% mutate(naB = as.numeric(as.character(namB))) %>% mutate(mut = factor(mutnames[1 + naA + 2 * naB], levels = mutnames)) %>% arrange(mut)

  geneExpression_new <- geneExpression_new[,as.character(mutStatus$FeirreraID)]


  #colors
  #colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
  colors = colorRamp2(c(-6,-2,0,2,6), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
  annocol <- get_palette("jco", 10)
  chromcol <- list(chromosome = c("12" = annocol[6], "other" = annocol[5]))
  
  annocolor <- list(Variant = c("none" = annocol[1], annocol[2], annocol[3], "both" = annocol[4]))
  names(annocolor$Variant) <- c("none", "IGHV", "trisomy12", "both")
  mutationStatus <- data.frame(mutStatus$mut)
  rownames(mutationStatus) <- mutStatus$PatID
  colnames(mutationStatus) <- "Variant"

  #Column annotation
  ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, "cm"), annotation_legend_param = list(title_gp = gpar(fontsize = 60), labels_gp = gpar(fontsize = 50),  grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm"), gap = unit(15, "mm")))

  #rowcluster
  geneExpression_dist <- dist(geneExpression_new)
  rowcluster = hclust(geneExpression_dist, method = "ward.D2")

  #heatmap
  h1 <- Heatmap(geneExpression_new, col = colors ,column_title = paste0("Gene interactions:", "IGHV", "-", "trisomy12"), column_title_gp = gpar(fontsize = 60, fontface = "bold"), heatmap_legend_param = list(title = "Expr", title_gp = gpar(fontsize = 60), grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm"), gap = unit(15, "mm"), labels_gp = gpar(fontsize = 50), labels = c(-6,-3, 0,3, 6)) , row_dend_width = unit(3.5, "cm"), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = FALSE, show_column_dend = FALSE, cluster_columns = FALSE, cluster_rows = rowcluster, split = 5, gap = unit(0.6,"cm"), column_order = mutStatus$PatID)

  #chromosome annotation
  #chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[sig_Genes,])$chromosome)
rownames(chrom) <- sig_Genes
colnames(chrom ) <- "chromosome"
#select for chromosome12
chrom$chromosome <- ifelse(chrom$chromosome == 12, 12, "other")

  #chromcol <- list(chromosome = c("1" ="#00d3b4", "2" ="#f443b9", "3" = "#6cde56", "4" = "#932db4", "5" ="#dcc608", "6"= "#0a54d9", "7" = "#fea114", "8"= "#0098ec","12" ="#fd4b43","10" = "#01e09f", "11" = "#ff6aab", "9" = "#a8d37e", "13" = "#a59eff","14" = "#be6f00", "15" = "#03d1f5", "16" = "#a02008", "17" = "#89c8ff", "18" = "#7a7600", "19" ="#005b8f", "20" = "#ffad61","21" = "#80386c", "22" = "#e3c27d", "X" ="#843e29", "MT" = "#ff938b"))
  
  ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(2.3, "cm"), annotation_legend_param = list(ncol = 2, title_gp = gpar(fontsize = 60), labels_gp = gpar(fontsize = 50),  grid_height = unit(2.5, "cm"), grid_width = unit(2.5, "cm")))
  
   
  #Annotate known genes from litertaure and/or most significant genes
  top50 <-  rownames(subset(resOrdered, padj < 0.001))
  int_genes <- rowData(RNAnorm[top50,])$symbol

subset <- as.data.frame(rowData(RNAnorm[sig_Genes,]))
subset <- subset[-which(subset$symbol %in% ""),]
subset <- subset[-which(subset$symbol %in% NA),]

subset <- subset[which(subset$symbol %in% int_genes),]
#subset2 <- subset[which(subset$symbol %in% c("MAP3K8", "CDCP1", "CD38")),]
#subset <- rbind(subset1, subset2)
rownames(subset) <- subset$symbol
geneIDs <- which(rownames(geneExpression_new) %in% rownames(subset))
labels <- rownames(geneExpression_new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 45)), width = unit(9, "cm"))
  

  
  #svg(filename="~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg", width=30, height=50)
  #pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistasis_Deseq.pdf", width=35, height=45)
  draw(h1 + ha_genes + ha_chrom) 
  #dev.off()
  #draw(h1 + ha_chrom + ha_genes)
 IGHVTri12 <- list("sig_Genes" = sig_Genes, "geneExp" = geneExpression_new, "h1"= h1)

```

#Compare with our dataset
```{r overlap with epistasis in gene expression}
#load differential data freom epistasis_Deseq_IGHV_tri12. Rmd
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHTri12.RData")
resOrdered_cll <- res[order(res$pvalue),]
resSig_cll <- subset(resOrdered_cll, padj < 0.1)
resTab_cll <- as.data.frame(resSig_cll)
#count data for annotation
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")
resTab_cll$symbol <- rowData(ddsCLL[rownames(resSig_cll),])$symbol
sig_expr_cll <- resTab_cll$symbol

overlap <- as.tibble(rowData(ddsCLL)) %>% filter(symbol %in% rowData(feirreraCLL)$symbol)
overlap_Il <- as.tibble(rowData(ddsCLL)) %>% filter(!symbol %in% "", !is.na(symbol)) %>% filter(symbol %in% resTab$symbol)

#overlap_total 
total <- as.tibble(sig_expr_cll) %>% filter(!value %in% "", !is.na(value)) %>% filter(value %in% resTab$symbol)

```
