---
title: "feirrera_countdata"
author: "Almut Luetge"
date: "August 29, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Reanalyse feirrera count data - Compare with our datasets
Do we find the same cluster in the feirrera dataset? As they stated?
How do their data behave compared to our?

1. Libraries
```{r libraries, warnings = FALSE, message= FALSE}
library(tidyverse)
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(pheatmap)
library(RColorBrewer)
library(ggdendro)
library(corrplot)
library(polycor)
library(readr)
library(VennDiagram)
```



2. load data

```{r data }
#RNA dataset
load("/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")
#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
dds <- dds[,colData(dds)$diag %in% cll_vector]


#Diff expressed gene list from Dvinge
fileName <- paste("/home/almut/Documents/CLL/data/FeirreraPaper/Incubation_signature.csv")
incubat_genes <- read_delim(fileName, delim ="\t")

load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData") 

c1c2_group <- patMeta[which(patMeta$Patient.ID %in% colData(dds)$PatID), c("c1c2", "Patient.ID")]
rownames(c1c2_group) <- c1c2_group$Patient.ID
colData(dds)$c1c2 <- c1c2_group$c1c2[match(colData(dds)$PatID,rownames(c1c2_group))] 


#load count dataset from feirrera
fileName <- paste("/home/almut/Dokumente/masterarbeit/data/FeirreraPaper/Feirrera_geneReadcount.txt")
feirreraTab <- read_table2(fileName)


```

3. Prepare feirrera dataset for dds dataset
```{r feirrera_dds}
##New dataset
FeirreraID <- colnames(feirreraTab[,-c(1,2)])

#get patient informations from dds object
metaTab <- as.data.frame(FeirreraID)

#order metadata and countdata

countTable <- as.data.frame(feirreraTab[,-c(1,2)])
rownames(countTable) <- feirreraTab$geneid


# to later build dds-dataset (function will break otherwise)
colnames(countTable) <- seq(ncol(countTable))




```


Gene annotation

```{r Gene annotation,  message=FALSE, warning=FALSE}
#Get annotations for genes (using biomaRt)
## Caution do not load earlier as it mask function select
library(biomaRt)
library(genefilter)
library(geneplotter)
library(pheatmap)
library(RColorBrewer)
library(stats)

ensembl <- useMart("ensembl",dataset = "hsapiens_gene_ensembl")
anno <- getBM(values = rownames(countTable),mart = ensembl, attributes = c("ensembl_gene_id","hgnc_symbol","description","chromosome_name","entrezgene","gene_biotype"),filters = "ensembl_gene_id")
anno <- anno[!duplicated(anno$ensembl_gene_id),]
rownames(anno) <- anno[,1]
geneAnno <- anno[rownames(countTable),]
geneAnno$ensembl_gene_id <- NULL


```


4.) Create deseq dataset

```{r deseq}
#Generate a new DESeq object
stopifnot(ncol(countTable) == nrow(metaTab))
feirrera <- DESeqDataSetFromMatrix(countData = countTable,colData = metaTab, design = ~1)

mcols(feirrera)$symbol <- geneAnno$hgnc_symbol
mcols(feirrera)$description <- geneAnno$description
mcols(feirrera)$chromosome <- geneAnno$chromosome_name
mcols(feirrera)$entrezgene <- geneAnno$entrezgene
mcols(feirrera)$biotype <- geneAnno$gene_biotype

#save the object
#save(feirrera,file="/home/almut/Documents/CLL/data/FeirreraPaper/feirrera.RData")

```


5.VSN
```{r vsn}
#number genes and samples
dim(feirrera)
ngenes <-nrow(feirrera)
nsamples <-ncol(feirrera)

#Remove non tumor samples
nonTumor <- c("C429","C429b","C429c","C430","C430b","C430c", "C431", "C431b","C431c","C432", "C432b","C432c","C433", "C433b", "C433c","C434", "C434b","C434c", "C435", "C435b","C435c","C436", "C436b","C436c","C437","C437b","C437c")
feirreraCLL <- feirrera[,-c(which(colData(feirrera)$FeirreraID %in% nonTumor))]

# Remove rows/genes with too few counts
keep <-apply(counts(feirreraCLL), 1, function(x) any(x >= 10))
feirreraCLL <- feirreraCLL[keep,]
dim(feirreraCLL)


#remove genes associated to sex
feirreraCLL  <- feirreraCLL[!rowData(feirreraCLL )$chromosome %in% c("Y"),]


#Variance stabilization transformation of the raw data
feirreraCLL <- estimateSizeFactors(feirreraCLL)
RNAnorm <- varianceStabilizingTransformation(feirreraCLL, blind=T)
```



6.Hierachical clustering
```{r s-to-s-distances, fig.width=20, fig.height=20, cache=TRUE}
sampleDists <- dist(t(assay(RNAnorm)))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- colData(RNAnorm)$FeirreraID

colnames(sampleDistMat) <- NULL

sampleDistDat <- t(assay(RNAnorm))
rownames(sampleDistDat) <- colData(RNAnorm)$FeirreraID


#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"Blues")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")


```


```{r hclust, fig.width = 40, fig.heigth = 90}
#hclust object
c1c2_cluster <- hclust(dist(sampleDistDat), method = "ward.D2", members = NULL)

## S3 method for class 'hclust'
plot(c1c2_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "Cluster by groups from Feirrera et al.",
     sub = NULL, xlab = "CLL_patients", ylab = "Height")


dendr    <- dendro_data(c1c2_cluster, type="rectangle") # convert for ggplot
clust    <- cutree(c1c2_cluster,k=2)                    # find 2 clusters
clust.df <- data.frame(label=names(clust), cluster=factor(clust))

# dendr[["labels"]] has the labels, merge with clust.df based on label column
dendr[["labels"]] <- merge(dendr[["labels"]], clust.df, by="label")

# plot the dendrogram; note use of color=cluster in geom_text(...)
ggplot() + geom_segment(data=segment(dendr), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr), aes(x, y, label=label, hjust=0, color=cluster), 
           size=8) +
        coord_flip() + scale_y_reverse(expand=c(0.9, 0)) + 
        ggtitle("Clustering by c1c2-genes") +
        theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_text(size = 20),
        axis.text.x=element_text(size = 20),
        axis.title = element_blank(),
        panel.background=element_rect(fill="white"),
        panel.grid=element_blank(),
        plot.title = element_text(size = 80)) 
        


## S3 method for class 'hclust'
plot(c1c2_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "Cluster by groups from Feirrera et al.",
     sub = NULL, xlab = "CLL_patients", ylab = "Height")


##Add group information to patients
colData(feirreraCLL)$c1c2 <- clust.df$cluster[match(colData(feirreraCLL)$FeirreraID,rownames(clust.df))] 
#save(feirreraCLL,file="/home/almut/Documents/CLL/data/FeirreraPaper/feirreraCLL.RData")


colData(RNAnorm)$c1c2 <- clust.df$cluster[match(colData(RNAnorm)$FeirreraID,rownames(clust.df))] 
```


```{r pca , fig.height=16, fig.width=16}
#Plot PCA based on c1c2genes only
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:10000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))
pcaTab$c1c2 <- as.factor(pcaTab$c1c2)

#plot PCA and color samples based on annotations
featureList <- c("c1c2")

pList <- lapply(featureList, function(feature) {
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 3) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) #+
    #theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) 
 })

grid.arrange(grobs = pList, ncol =1)

```


```{r, hierachical clustering, fig.width=18, fig.height=70}
colAnno <- as.data.frame(pcaTab[,"c1c2"])
colnames(colAnno) <- "c1c2"
rownames(colAnno) <-colnames(RNAnorm)


#scale and censor
exprMat.new <- log2(exprMat[c(1:700),])
exprMat.new <- t(scale(t(exprMat.new)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, 
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol),
         clustering_method = "ward.D2", fontsize = 20, fontsize_row = 15)
```











#7.) Test for incubation signature genes
PNAS paper states a connection of the c1c2 signature to incubation time or a reduced Non sense mediated decay. Test for the genes that are found to be assocated to incubation time therein. 

prepare sample set
```{r incub genes}
incub_sign <- dplyr::select(incubat_genes, geneID, PBMC.none.48h) %>% filter(abs(PBMC.none.48h) >= 5)

randomgenes <- sample(rownames(feirreraCLL), 1000)

dim(feirreraCLL)
#subset with CLL patients only
dds_incub <- feirreraCLL[which(rownames(feirreraCLL) %in% randomgenes),]        #incub_sign$geneID),]   #randomgenes),] 


#number genes and samples
dim(dds_incub)
ngenes <-nrow(dds_incub)
nsamples <-ncol(dds_incub)

# Remove rows/genes with too few counts
keep <-apply(counts(dds_incub), 1, function(x) any(x >= 10))
dds_incub <- dds_incub[keep,]
dim(dds_incub)


#Variance stabilization transformation of the raw data
dds_incub <- estimateSizeFactors(dds_incub)
RNAnorm_incub <- varianceStabilizingTransformation(dds_incub, blind=T)



```

PCA - incuubation-signature genes only

```{r pca , fig.height=10, fig.width=10}
#Plot PCA based on incubat-genes only
exprMat <- assay(RNAnorm_incub)

#top 5000 most variant genes
#sds <- rowSds(exprMat)
#exprMat <- exprMat[order(sds, decreasing = T)[1:5000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm_incub)))
pcaTab$c1c2 <- as.factor(pcaTab$c1c2)

#plot PCA and color samples based on annotations
featureList <- c("c1c2")

pList <- lapply(featureList, function(feature) {
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 3) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) #+
    #theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) 
 })

grid.arrange(grobs = pList, ncol =1)

```
Sample still cluster by c1c2 genes --> does this confirm the incubation signature?


##Why does the clustering always return --> no matter which and how many genes we choose?

```{r, hierachical clustering, fig.width=18, fig.height=70}
colAnno <- as.data.frame(pcaTab[,"c1c2"])
colnames(colAnno) <- "c1c2"
#ownames(colAnno) <-pcaTab$FeirreraID


#scale and censor
exprMat.new <- log2(exprMat)
exprMat.new <- t(scale(t(exprMat.new)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, 
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol),
         clustering_method = "ward.D2", fontsize = 20, fontsize_row = 15)


```







```{r compare gene expression}
c2 <- colData(feirreraCLL)$FeirreraID[which(colData(feirreraCLL)$c1c2 == 2)]
c1 <- colData(feirreraCLL)$FeirreraID[which(colData(feirreraCLL)$c1c2 == 1)]


```


#Try clustering with non-tumor samples.. Do they have the same intrasample signature? In the PNAS paper all blood samples should be affected by this 
Remove rows/genes with too few counts


```{r include non tuimor samples}
# Remove rows/genes with too few counts
keep <-apply(counts(feirrera), 1, function(x) any(x >= 10))
feirrera <- feirrera[keep,]
dim(feirrera)


#Variance stabilization transformation of the raw data
feirrera <- estimateSizeFactors(feirrera)
RNAnorm <- varianceStabilizingTransformation(feirrera, blind=T)
```


.Hierachical clustering
```{r s-to-s-distances, fig.width=20, fig.height=20, cache=TRUE}
sampleDists <- dist(t(assay(RNAnorm)))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- colData(RNAnorm)$FeirreraID

colnames(sampleDistMat) <- NULL

sampleDistDat <- t(assay(RNAnorm))
rownames(sampleDistDat) <- colData(RNAnorm)$FeirreraID


#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"RdBu")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")


```

```{r hclust 3 samples, fig.width = 40, fig.heigth = 90}
#hclust object
c1c2c3_cluster <- hclust(dist(sampleDistDat), method = "ward.D2", members = NULL)

## S3 method for class 'hclust'
plot(c1c2c3_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "Cluster by groups from Feirrera et al.",
     sub = NULL, xlab = "All_patients", ylab = "Height")


dendr    <- dendro_data(c1c2c3_cluster, type="rectangle") # convert for ggplot
clust    <- cutree(c1c2c3_cluster,k=3)                    # find 2 clusters
clust.df <- data.frame(label=names(clust), cluster=factor(clust))

# dendr[["labels"]] has the labels, merge with clust.df based on label column
dendr[["labels"]] <- merge(dendr[["labels"]], clust.df, by="label")

# plot the dendrogram; note use of color=cluster in geom_text(...)
ggplot() + geom_segment(data=segment(dendr), aes(x=x, y=y, xend=xend, yend=yend)) + 
  geom_text(data=label(dendr), aes(x, y, label=label, hjust=0, color=cluster), 
           size=8) +
        coord_flip() + scale_y_reverse(expand=c(0.9, 0)) + 
        ggtitle("Clustering by c1c2c3-genes") +
        theme(axis.line.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.y=element_text(size = 20),
        axis.text.x=element_text(size = 20),
        axis.title = element_blank(),
        panel.background=element_rect(fill="white"),
        panel.grid=element_blank(),
        plot.title = element_text(size = 80)) 
        


## S3 method for class 'hclust'
plot(c1c2c3_cluster, labels = rownames(sampleDistDat), hang = 0.1, check = TRUE,
     axes = TRUE, frame.plot = FALSE, ann = TRUE,
     main = "Cluster by groups from Feirrera et al.",
     sub = NULL, xlab = "CLL_patients", ylab = "Height")


##Add group information to patients
colData(feirrera)$c1c2c3 <- clust.df$cluster[match(colData(feirrera)$FeirreraID,rownames(clust.df))] 
colData(RNAnorm)$c1c2c3 <- clust.df$cluster[match(colData(RNAnorm)$FeirreraID,rownames(clust.df))] 
```

#Do we still find the same c1c2 cluster?
```{r check c1c2 cluster, fig.width=15, fig.heigth = 15}
#intersect between c1 and new c1
c2_new <- colData(feirrera)$FeirreraID[which(colData(feirrera)$c1c2c3 == 2)]
c1_new <- colData(feirrera)$FeirreraID[which(colData(feirrera)$c1c2c3 == 1)]
c3_new <- colData(feirrera)$FeirreraID[which(colData(feirrera)$c1c2c3 == 3)]

#venn diagramm of all groups
ClusterGroups <- list("c1 " = as.character(c1), "c2" = as.character(c2), "c1_new" = as.character(c1_new), "c2_new" = as.character(c2_new), "c3_new" = as.character(c3_new))
#overlapCluster <- calculate.overlap(ClusterGroups)

library(gplots)
venn(ClusterGroups)

#venn_part <-get.venn.partitions(ClusterGroups, force.unique = TRUE, keep.elements = TRUE,
#hierarchical = FALSE)

#vennPlot <-VennDiagram::venn.diagram(ClusterGroups, NULL)
#grid.arrange(grobs = vennPlot)



```



##Only 3 sample were assigned to the other c1/c2 group --> Cluster are consistent
```{r PCA with non-tumor samples}

#Plot PCA based on incubat-genes only
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
#sds <- rowSds(exprMat)
#exprMat <- exprMat[order(sds, decreasing = T)[10000:15000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))
pcaTab$c1c2 <- as.factor(pcaTab$c1c2)

#plot PCA and color samples based on annotations
featureList <- c("c1c2c3")

pList <- lapply(featureList, function(feature) {
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 3) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) #+
    #theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) 
 })

grid.arrange(grobs = pList, ncol =1)

```










