---
title: "Incubation_signature"
author: "Almut Luetge"
date: "August 29, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Look for the incubation signature from Dvinge et al.in our and the feirrera data set
In Dvinge et al. they found the c1c2 being associated to the incubation time and blood collection procedure. They find a reduction in nonsense mediated decay responsible for this.




```{r libraries, warnings = FALSE, message= FALSE}
library(tidyverse)
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(pheatmap)
library(RColorBrewer)
library(ggdendro)
library(corrplot)
library(polycor)
library(readr)
```



1. load data

```{r data }
#RNA dataset
load("/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")
#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
dds <- dds[,colData(dds)$diag %in% cll_vector]
#remove Y-chromosome
dds <- dds[!rowData(dds)$chromosome %in% c("Y"),]

#Diff expressed gene list from Dvinge
fileName <- paste("/home/almut/Documents/CLL/data/FeirreraPaper/Incubation_signature.csv")
incubat_genes <- read_delim(fileName, delim ="\t")

load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData") 

c1c2_group <- patMeta[which(patMeta$Patient.ID %in% colData(dds)$PatID), c("c1c2", "Patient.ID")]
rownames(c1c2_group) <- c1c2_group$Patient.ID
colData(dds)$c1c2 <- c1c2_group$c1c2[match(colData(dds)$PatID,rownames(c1c2_group))] 

drug_group <- patMeta[which(patMeta$Patient.ID %in% colData(dds)$PatID), c("drugGroup", "Patient.ID")]
rownames(drug_group) <- drug_group$Patient.ID
colData(dds)$drugGroup <- drug_group$drugGroup[match(colData(dds)$PatID,rownames(drug_group))] 

#annotate methylation cluster
methCluster <- patMeta[which(patMeta$Patient.ID %in% colData(dds)$PatID), c("Methylation_Cluster", "Patient.ID")]
rownames(methCluster) <- methCluster$Patient.ID
colData(dds)$methClust <- methCluster$Methylation_Cluster[match(colData(dds)$PatID,rownames(methCluster))] 



```


#Select genes which are at least 2 fold changed after 48h in none cryo-preserved samples 
```{r signature genes}
incub_sign <- dplyr::select(incubat_genes, geneID, PBMC.none.48h) %>% filter(abs(PBMC.none.48h) >= 2)

```



2. Hierachical clustering of dds dataset using the incubation signature 

```{r Preprocessing}
#subset with CLL patients only
#dds_incub <- dds[which(rownames(dds) %in% incub_sign$geneID),]
dds_incub <-dds[,colData(dds)$c1c2 == 2]

#number genes and samples
dim(dds_incub)
ngenes <-nrow(dds_incub)
nsamples <-ncol(dds_incub)

# Remove rows/genes with too few counts
keep <-apply(counts(dds_incub), 1, function(x) any(x >= 10))
dds_incub <- dds_incub[keep,]
dim(dds_incub)


#Variance stabilization transformation of the raw data
dds_incub <- estimateSizeFactors(dds_incub)
RNAnorm <- varianceStabilizingTransformation(dds_incub, blind=T)


```


```{r s-to-s-distances, fig.width=40, fig.height=40, cache=TRUE}
sampleDists <- dist(t(assay(RNAnorm)))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- paste0(colData(RNAnorm)$PatID, "_", colData(RNAnorm)$IGVH, "_", colData(RNAnorm)$drugGroup)

sampleDistDat <- t(assay(RNAnorm))
rownames(sampleDistDat) <- colData(RNAnorm)$PatID


colnames(sampleDistMat) <- NULL

#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(20,"RdBu")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2", fontsize = 25)


```


##PCA on incubation genes only
```{r pca , fig.height=22, fig.width=16}
#Plot PCA based on c1c2genes only
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[5000:10000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))
pcaTab$batch <- as.factor(pcaTab$batch)

#plot PCA and color samples based on annotations
featureList <- c("batch","IGVH","methClust","c1c2","RNAprep", "del13q14", "drugGroup", "trisomy12")

pList <- lapply(featureList, function(feature) {
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 3) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) #+
    #theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) 
 })

grid.arrange(grobs = pList, ncol =2)

```


##Gene expression of c1c2 genes
```{r, hierachical clustering, fig.width=18, fig.height=70}
colAnno <- pcaTab[,c("batch","IGVH","trisomy12","Gender","RNAprep", "drugGroup")]

#scale and censor
exprMat.new <- log2(exprMat[c(1:500),])
exprMat.new <- t(scale(t(exprMat.new)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

pheatmap(exprMat.new, treeheight_row = 0, scale = "row", annotation_col = colAnno, 
         labels_row = paste0(rowData(RNAnorm[rownames(exprMat.new),])$symbol),
         clustering_method = "ward.D2", fontsize = 20, fontsize_row = 15)
```





