---
title: "Cluster_sampleprepMetadata"
author: "Almut Luetge"
date: "August 30, 2017"
output: html_document
---

#Aim: Check wether cluster can be related to sampleprep parameter
In Dvine et al. they find the c1c2 groups associated to sample preparation methods.  Do we find any association in our data? The cluster seem to be in deed sample specific...

```{r libraries, warnings = F}
library(tidyverse)
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(pheatmap)
library(RColorBrewer)
library(ggdendro)
library(corrplot)
library(polycor)
```

1. load data

```{r data }
#RNA dataset
load("/home/almut/Documents/CLL/data/2017-08-16_ddsrna_trimmed.RData")
#subset with CLL patients only
cll_vector <- c("CLL", "atyp CLL", "Atyp. CLL","CLL/ET")
dds <- dds[,colData(dds)$diag %in% cll_vector]



#patmeta data (including c1c2 status)
load("/home/almut/Documents/CLL/data/patmetaGroups_170324.RData") 

#sample preparation data
fileName <- paste("/home/almut/Documents/CLL/data/qyr_patient_sample.txt")
samplePrep <- read_delim(fileName, delim =",")

#reduce to important Infos
samplePrep <- dplyr::select(samplePrep, patPatientID, patHIPOID,patComment, smpusrIDref, smpLeukocytes, smpPBLymphocytes,	smpLymphocyteMissing,	smpMixedWithErythrocytes,	smpErythrocyteLysis,	smpProcessedNextDay,	smpNoCells,	smpUnreliableCellCount, smpRNAseq) %>% dplyr::filter(!is.na(smpRNAseq))


#annotate PatIDs uniform 
Hipos <- patMeta[which(patMeta$HIPO.ID %in% samplePrep$patHIPOID), c("HIPO.ID", "Patient.ID")]
rownames(Hipos) <- Hipos$HIPO.ID
samplePrep$patPatientID <- Hipos$Patient.ID[match(samplePrep$patHIPOID,rownames(Hipos))] 

samplePrep <- dplyr::filter(samplePrep, !is.na(patPatientID))

#remove duplicated sample
samplePrep <- dplyr::filter(samplePrep, patPatientID != "P0508")


```


2.)Add sample data to colData of dds object

```{r prepare data set}

dds <- dds[,colData(dds)$PatID %in% samplePrep$patPatientID]

samplePrepMat <- dplyr::filter(samplePrep, patPatientID %in% colData(dds)$PatID)
samplePrepMat[,c(3,4,7:12)] <- lapply(samplePrepMat[,c(3,4,7:12)], function(x) as.factor(x))
samplePrepMat <- as.data.frame(samplePrepMat)
rownames(samplePrepMat) <- samplePrepMat$patPatientID
samplePrepMat <- samplePrepMat[colData(dds)$PatID,]
colData(dds) <- cbind(colData(dds), samplePrepMat)



#generate Matrix with interesting features
patMat <- patMeta[which(patMeta$Patient.ID %in% colData(dds)$PatID),]
genomDat <- apply(patMat[,c(12:98)], 2, function(x) as.numeric(as.character(x)))
intGenom <- genomDat[, which(colSums(genomDat, na.rm=T) > 10)]

patMat <-patMat[, c("Patient.ID", "gender","diagnosis", "treatment", "IGHV.status","Methylation_Cluster", colnames(intGenom), "c1c2", "drugGroup")]
patMat <- lapply(patMat, function(x) as.factor(x))
patMat <- as.data.frame(patMat)
rownames(patMat) <- patMat$Patient.ID
patMat <- patMat[colData(dds)$PatID,]
colData(dds) <- cbind(colData(dds), patMat)
patMat <- cbind(patMat, colData(dds)[,c("RNAprep", "batch")])

```

###Perform sample normalization and clustering
```{r vst normalization}
# Remove rows/genes with too few counts
keep <-apply(counts(dds), 1, function(x) any(x >= 10))
dds <- dds[keep,]

#Variance stabilization transformation of the raw data
dds <- estimateSizeFactors(dds)
RNAnorm <- varianceStabilizingTransformation(dds, blind=T)




```





Hierachical Clustering

Exploratory analysis of the RNAseq data
Sample distances


```{r s-to-s-distances, fig.width=20, fig.height=20, cache=TRUE}
sampleDists <- dist(t(assay(RNAnorm)))
sampleDistMat <- as.matrix(sampleDists)
rownames(sampleDistMat) <- paste0(colData(RNAnorm)$PatID, "_", colData(RNAnorm)$IGVH, "_", colData(RNAnorm)$c1c2)

colnames(sampleDistMat) <- NULL

#Plot haetmap
colors <- colorRampPalette( rev(brewer.pal(9,"Blues")) )(255)
pheatmap(sampleDistMat, clustering_distance_rows = sampleDists, clustering_distance_cols = sampleDists, col = colors, clustering_method = "ward.D2")


```


Principal Component Analysis
Try different number of genes - most variant can be related to gender (remove y/x-chromosomes first?!)



```{r pca, fig.height=22, fig.width=16}
#Plot PCA
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:20000],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))
pcaTab$batch <- as.factor(pcaTab$batch)

#plot PCA and color samples based on annotations
#featureList <- c("batch","IGVH","trisomy12","Gender","RNAprep", "del13q14")
featureList <- c("smpusrIDref", "smpLeukocytes", "smpPBLymphocytes",	"batch",	"smpMixedWithErythrocytes",	"smpProcessedNextDay"	,"c1c2", "IGVH")


pList <- lapply(featureList, function(feature) {
  ggplot(pcaTab, aes_string(x="PC1",y="PC2", color = feature)) + geom_point(size = 5) + theme_bw() + 
    xlab(sprintf("PC1 (%2.1f%%)",varExp[1])) + ylab(sprintf("PC2 (%2.1f%%)",varExp[2])) + 
    ggtitle(feature) + theme(plot.title = element_text(hjust = 0.5, color="red", size = 20)) +
    theme(axis.title.x = element_text(face="bold", size=20), axis.text.x=element_text(size=16)) + theme(axis.title.y = element_text(face="bold", size=20), axis.text.y= element_text(size=16)) +
    theme(legend.box.margin = margin(6, 6, 6, 6), legend.text = element_text(size = 15), legend.title = element_text(face = "bold", size = 10)) 
 })

grid.arrange(grobs = pList, ncol =2)

```

No association in PCA plot.
What about corrplots? 