---
title: "Survival plots"
author: "Almut Luetge"
date: "August 31, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Aim: Check survival data and time to treatment for the trisomy12 and IGHV groups
We find gene expression in IGHV signatures dependent on the trisomy12 status. Trisomy12 sample form their own cluster close to the IGHV mutated samples. The epistatsis model supports interactive gene expression. Do we find differences in survival data of IGHV samples dependent on trisomy12 status as well? 

```{r libraries, warnings = F, message=FALSE}
#library(tidyverse)
library(DESeq2)
library(VennDiagram)
library(dplyr)
library(magrittr)
library(gridExtra)
library(pheatmap)
library(matrixStats)
library(piano)
library(limma)
library(ggvis)
library(pheatmap)
library(RColorBrewer)
library(ggdendro)
library(corrplot)
library(survminer)
library(survival)
```

Load data

```{r data sets}
#RNA dataset
load("/home/almut/Documents/CLL/data/ddsrnaCLL_trimmed_tsig_modIGHV.RData")


#survival data
load("/home/almut/Documents/CLL/data/surv_160201.RData")
survTab <- as.data.frame(survT) 
dds <- ddsCLL[,colData(ddsCLL)$PatID %in% rownames(survT)]
dim(dds)
survTab <- survTab[colData(dds)$PatID,]
stopifnot(rownames(survTab) == colData(dds)$PatID)
colData(dds) <- cbind(colData(dds), survTab[,c(1:10)])

```


Survival plot using survminer
```{r survival plot, fig.width= 8, fig.height=8}
survData <- colData(dds)
#survIGHVU <- colData(dds)[which(colData(dds)$IGHV %in% "U"),]

my_surv <- function(variant){
#survival
survival <- Surv(survData$T5, survData$treatedAfter)
fit_surv<- survfit(as.formula(paste("survival ~", paste(variant))), data = survData)
ggsurvplot(fit_surv, main = "Survival within c1/c2 groups after Feirrera et al.", legend.title = "Group", conf.int = F, pval = T, risk.table = TRUE, fun = "event", risk.table.col = "strata",  ylab ="Percent of alive patients", font.x = 20, font.y = 20) #, ggtheme = theme_bw(), legend.labs = c("no_Tri12", "Tri12"))

}
#Including IGHV
fit_surv<- survfit(Surv(T6, died) ~ MED12 + IGVH, data = survData)
ggsurvplot(fit_surv, main = "Survival within c1/c2 groups after Feirrera et al.", legend.title = "Group", conf.int = F, pval = T, risk.table = TRUE, risk.table.col = "strata",  ylab ="Percent of alive patients", font.x = 20, font.y = 20) #, ggtheme = theme_bw(), legend.labs = c("IGHV-M", "IGHV-U", "Tri12 + IGHV-M", "Tri12 +IGHV-U"))

#Time to Treatment
fit_treatment<- survfit(Surv(T5, treatedAfter) ~ MED12 + IGVH, data = survData)
ggsurvplot(fit_treatment, main = "Time to treatment within c1/c2 groups after Feirrera et al.", legend.title = "Group", conf.int = F, pval = T, risk.table = TRUE, fun = "event", risk.table.col = "strata", ylab ="Percent of treated patients", font.x = 20, font.y = 20, font.labs = 20) #, ggtheme = theme_bw(), legend.labs = c("IGHV-M", "IGHV-U", "Tri12 + IGHV-M", "Tri12 +IGHV-U"))

#Including IGHV
fit_treatment<- survfit(Surv(T5, treatedAfter) ~ MED12, data = survData)
ggsurvplot(fit_treatment, main = "Time to treatment within c1/c2 groups after Feirrera et al.", legend.title = "Group", conf.int = F, pval = T, risk.table = TRUE, fun = "event", risk.table.col = "strata", ylab ="Percent of treated patients", font.x = 20, font.y = 20) #, legend.labs = c("no_Tri12", "Tri12"))


gene_conditions <- c("del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12", "trisomy12")
lapply(gene_conditions, my_surv)


```

Is there a relation between time to treatment and trisomy12 mutation? They are still worse than the IGHV-mutated patients, but better than IGHV-unmutated group.. Is this caused by the comparatively low number of IGHV-U + tri12 patients? 











