<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="almut" />

<meta name="date" content="2022-04-24" />

<title>Time from diagnosis to sampling</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">transcriptome_cll</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/almutlue/transcriptome_cll">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Time from diagnosis to sampling</h1>
<h4 class="author">almut</h4>
<h4 class="date">24 April 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>transcriptome_cll/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190511code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190511)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190511code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190511)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomalmutluetranscriptomeclltree55417a13eb7e0152351df785e6a9bf666dfefe81targetblank55417a1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/almutlue/transcriptome_cll/tree/55417a13eb7e0152351df785e6a9bf666dfefe81" target="_blank">55417a1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomalmutluetranscriptomeclltree55417a13eb7e0152351df785e6a9bf666dfefe81targetblank55417a1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/almutlue/transcriptome_cll/tree/55417a13eb7e0152351df785e6a9bf666dfefe81" target="_blank">55417a1</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/epistasis_abruzzo.Rmd
    Untracked:  data/abruzzo/

Unstaged changes:
    Modified:   analysis/de_diagnosis_time_sampling.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/de_diagnosis_time_sampling.Rmd</code>) and HTML (<code>docs/de_diagnosis_time_sampling.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/55417a13eb7e0152351df785e6a9bf666dfefe81/analysis/de_diagnosis_time_sampling.Rmd" target="_blank">55417a1</a>
</td>
<td>
aluetge
</td>
<td>
2022-04-27
</td>
<td>
rerun with pretreatment
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="association-between-gene-expression-and-time-from-diagnosis-to-sampling" class="section level2">
<h2>Association between gene expression and time from diagnosis to sampling</h2>
<p>Aim: Find gene signatures for mutational status of CLL patients</p>
<p>load packages</p>
<pre class="r"><code>library(DESeq2)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(ComplexHeatmap)
library(matrixStats)
library(here)
library(variancePartition)
library(ggtern)</code></pre>
<p>load data</p>
<pre class="r"><code>data_dir &lt;- here(&quot;data&quot;)
output_dir &lt;- here(&quot;output&quot;)
figure_dir &lt;- here(&quot;output/figures&quot;)

#dds data set. gene expression data + patmetadata
load(paste0(data_dir, &quot;/ddsrnaCLL_150218.RData&quot;))

#load meta data including genotyping info
load(paste0(data_dir, &quot;/patmeta_170324.RData&quot;))
excluded_columns &lt;- c(&quot;HIPO.ID&quot;,&quot;PID&quot;, &quot;gender&quot;,&quot;project&quot;, &quot;diagnosis&quot;, &quot;date.of.diagnosis&quot;, &quot;date.of.first.treatment&quot;, &quot;IGHV.status&quot;,&quot;Methylation_Cluster&quot;)

#get pretreatment status
pretreat = read.csv(paste0(data_dir, &quot;/diagnosis_information.csv&quot;), sep = &quot;;&quot;)
rownames(pretreat) &lt;- pretreat$Patient.ID

patMeta &lt;- as.tibble(patMeta) %&gt;% filter(Patient.ID %in% ddsCLL$PatID) %&gt;% dplyr::select(-one_of(excluded_columns))</code></pre>
<pre><code>Warning: `as.tibble()` was deprecated in tibble 2.0.0.
Please use `as_tibble()` instead.
The signature and semantics have changed, see `?as_tibble`.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</code></pre>
<pre class="r"><code>variants &lt;- patMeta %&gt;% dplyr::select(-Patient.ID) %&gt;%  dplyr::select(-treatment) %&gt;%
  mutate_if(is.factor, as.character) %&gt;% 
  mutate_if(is.character, as.numeric) %&gt;% 
  dplyr::select(colnames(.)[colSums(.,na.rm = TRUE) &gt; 4]) %&gt;% 
  colnames()

var_add &lt;- variants[!variants %in% names(colData(ddsCLL))] 
rownames(patMeta) &lt;-  patMeta$Patient.ID</code></pre>
<pre><code>Warning: Setting row names on a tibble is deprecated.</code></pre>
<pre class="r"><code>patMeta &lt;- patMeta[colData(ddsCLL)$PatID,]
pretreat &lt;- pretreat[colData(ddsCLL)$PatID,]
cd &lt;- cbind(colData(ddsCLL), patMeta[,var_add])
colData(ddsCLL) &lt;- cd
pretreat$norm_time_diag &lt;- scale(pretreat$days.from.diagnosis.to.sampling)
colData(ddsCLL)$time_diag &lt;- pretreat$norm_time_diag </code></pre>
<pre class="r"><code>###Deseq
ddsCLL &lt;- estimateSizeFactors(ddsCLL)

ddsCLL &lt;- ddsCLL[,!is.na(ddsCLL$time_diag)]</code></pre>
<pre><code>Warning in NSBS(i, x, exact = exact, strict.upper.bound = !allow.append, :
subscript is an array, passing it thru as.vector() first</code></pre>
<pre class="r"><code>design(ddsCLL) &lt;- as.formula(paste(&quot;~ time_diag&quot;))
rnaRaw &lt;- DESeq(ddsCLL, betaPrior = FALSE)</code></pre>
<pre><code>using pre-existing size factors</code></pre>
<pre><code>estimating dispersions</code></pre>
<pre><code>gene-wise dispersion estimates</code></pre>
<pre><code>mean-dispersion relationship</code></pre>
<pre><code>final dispersion estimates</code></pre>
<pre><code>fitting model and testing</code></pre>
<pre class="r"><code>res &lt;- results(rnaRaw)
resOrdered &lt;- res[order(res$pvalue),]</code></pre>
<p>Diff genes</p>
<pre class="r"><code>pCut &lt;- 0.05

dataTab &lt;- data.frame(resOrdered)
dataTab$ID &lt;- rownames(dataTab)
#filter using pvalues
dataTab &lt;- filter(dataTab, padj &lt;= pCut) %&gt;% 
  arrange(padj) %&gt;%
  mutate(Symbol = rowData(ddsCLL[ID,])$symbol) 

dataTab &lt;- dataTab[!duplicated(dataTab$Symbol),]
dataTab &lt;- dataTab[!is.na(dataTab$Symbol),]
rownames(dataTab) &lt;- dataTab$ID
write.csv(dataTab, file=paste0(output_dir,&quot;/diff_genes/diagnosis_time_to_sampling_diffGenes.csv&quot;))</code></pre>
</div>
<div id="ma-plot" class="section level2">
<h2>MA-plot</h2>
<pre class="r"><code>DESeq2::plotMA(res, ylim=c(-0.5,0.5), main= &quot;time_diagnosis_to_sampling&quot;)</code></pre>
<p><img src="figure/de_diagnosis_time_sampling.Rmd/ma-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(res$pvalue, breaks=100, col=&quot;skyblue&quot;, border=&quot;slateblue&quot;,
                     main=paste0(&quot;Histogramm of pvalues: Time from diagnosis to sampling&quot;),plot = TRUE)</code></pre>
<p><img src="figure/de_diagnosis_time_sampling.Rmd/ma-plot-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="variance-explained" class="section level2">
<h2>Variance explained</h2>
<pre class="r"><code>expr &lt;- as.matrix(assays(ddsCLL)$counts)
meta_sub &lt;- as.data.frame(colData(ddsCLL)[, c(&quot;time_diag&quot;, &quot;IGHV&quot;)])
form &lt;- as.formula(paste0(&quot;~ time_diag + (1|IGHV)&quot;))
varPart &lt;- fitExtractVarPartModel(expr, form, meta_sub, BPPARAM=MulticoreParam(4))</code></pre>
<pre><code>Warning in .fitExtractVarPartModel(exprObj, formula, data, REML = REML, : Variables contain NA&#39;s: IGHV 
Samples with missing data will be dropped.</code></pre>
<pre><code>Dividing work into 100 chunks...</code></pre>
<pre><code>
Total:184 s</code></pre>
<p>Plott variance partitioning</p>
<pre class="r"><code>t1 &lt;- ggtern(data=varPart, aes(time_diag, IGHV, Residuals)) +
  geom_point(aes(colour = &quot;blue&quot;), size= 3, alpha = 0.6, show.legend = FALSE)  +
  #scale_colour_manual(aes(), values = &quot;blue&quot;) +
  #stat_density_tern(aes(fill=..level.., alpha=..level..),geom=&#39;polygon&#39;,bdl = 0.01,bdl.val = 1, base = &quot;identity&quot;, n = 20) +
  Llab(&quot;time\n \t diagnosis&quot;) +
  Tlab(&quot;IGHV&quot;) +
  Rlab(&quot;Residuals&quot;) +
  ggtitle(&quot;Percent variance explained by ...:&quot;) +
  theme(plot.title = element_text(size=14, hjust = 0.3),
        plot.subtitle = element_text(size=20),    
        panel.grid.major = element_blank(),    
        axis.text.x = element_blank(),    
        axis.text.y = element_blank(),    
        panel.spacing = unit(0.2, &quot;lines&quot;),    
        legend.position = &quot;none&quot;,    
        strip.text = element_text(face=&quot;bold&quot;, size=22))
t1</code></pre>
<p><img src="figure/de_diagnosis_time_sampling.Rmd/var%20part-1.png" width="384" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_CH.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=de_CH.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=de_CH.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_CH.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggtern_3.3.5                variancePartition_1.24.0   
 [3] BiocParallel_1.28.3         limma_3.50.1               
 [5] here_1.0.1                  ComplexHeatmap_2.10.0      
 [7] gridExtra_2.3               forcats_0.5.1              
 [9] stringr_1.4.0               purrr_0.3.4                
[11] readr_2.1.2                 tidyr_1.2.0                
[13] tibble_3.1.6                ggplot2_3.3.5              
[15] tidyverse_1.3.1             magrittr_2.0.2             
[17] dplyr_1.0.8                 DESeq2_1.34.0              
[19] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[21] MatrixGenerics_1.6.0        matrixStats_0.61.0         
[23] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[25] IRanges_2.28.0              S4Vectors_0.32.3           
[27] BiocGenerics_0.40.0        

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.4.1        circlize_0.4.14       
  [4] workflowr_1.7.0        plyr_1.8.7             splines_4.1.3         
  [7] digest_0.6.29          foreach_1.5.2          htmltools_0.5.2       
 [10] fansi_1.0.3            memoise_2.0.1          cluster_2.1.3         
 [13] doParallel_1.0.17      tzdb_0.2.0             Biostrings_2.62.0     
 [16] annotate_1.72.0        modelr_0.1.8           bayesm_3.1-4          
 [19] prettyunits_1.1.1      colorspace_2.0-3       blob_1.2.2            
 [22] rvest_1.0.2            haven_2.4.3            xfun_0.30             
 [25] crayon_1.5.0           RCurl_1.98-1.6         jsonlite_1.8.0        
 [28] genefilter_1.76.0      lme4_1.1-28            survival_3.2-13       
 [31] iterators_1.0.14       glue_1.6.2             gtable_0.3.0          
 [34] zlibbioc_1.40.0        XVector_0.34.0         compositions_2.0-4    
 [37] GetoptLong_1.0.5       DelayedArray_0.20.0    shape_1.4.6           
 [40] DEoptimR_1.0-11        scales_1.1.1           DBI_1.1.2             
 [43] Rcpp_1.0.8.3           xtable_1.8-4           progress_1.2.2        
 [46] clue_0.3-60            latex2exp_0.9.4        bit_4.0.4             
 [49] httr_1.4.2             gplots_3.1.1           RColorBrewer_1.1-2    
 [52] ellipsis_0.3.2         farver_2.1.0           pkgconfig_2.0.3       
 [55] XML_3.99-0.9           sass_0.4.1             dbplyr_2.1.1          
 [58] locfit_1.5-9.5         utf8_1.2.2             labeling_0.4.2        
 [61] tidyselect_1.1.2       rlang_1.0.2            reshape2_1.4.4        
 [64] later_1.3.0            AnnotationDbi_1.56.2   munsell_0.5.0         
 [67] cellranger_1.1.0       tools_4.1.3            cachem_1.0.6          
 [70] cli_3.2.0              generics_0.1.2         RSQLite_2.2.11        
 [73] broom_0.7.12           evaluate_0.15          fastmap_1.1.0         
 [76] yaml_2.3.5             knitr_1.37             bit64_4.0.5           
 [79] fs_1.5.2               robustbase_0.95-0      caTools_1.18.2        
 [82] KEGGREST_1.34.0        nlme_3.1-157           whisker_0.4           
 [85] xml2_1.3.3             compiler_4.1.3         pbkrtest_0.5.1        
 [88] rstudioapi_0.13        png_0.1-7              reprex_2.0.1          
 [91] geneplotter_1.72.0     bslib_0.3.1            stringi_1.7.6         
 [94] highr_0.9              lattice_0.20-45        Matrix_1.4-1          
 [97] nloptr_2.0.0           tensorA_0.36.2         vctrs_0.3.8           
[100] pillar_1.7.0           lifecycle_1.0.1        jquerylib_0.1.4       
[103] GlobalOptions_0.1.2    bitops_1.0-7           httpuv_1.6.5          
[106] R6_2.5.1               promises_1.2.0.1       KernSmooth_2.23-20    
[109] codetools_0.2-18       boot_1.3-28            MASS_7.3-56           
[112] gtools_3.9.2           assertthat_0.2.1       proto_1.0.0           
[115] rprojroot_2.0.2        rjson_0.2.21           withr_2.5.0           
[118] GenomeInfoDbData_1.2.7 parallel_4.1.3         hms_1.1.1             
[121] minqa_1.2.4            rmarkdown_2.13         git2r_0.30.1          
[124] lubridate_1.8.0       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
