<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="aluetge" />

<meta name="date" content="2019-07-26" />

<title>epistasis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">transcriptome_cll</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/almutlue/transcriptome_cll">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">epistasis</h1>
<h4 class="author">aluetge</h4>
<h4 class="date">2019-07-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>transcriptome_cll/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190511code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190511)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190511code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190511)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomalmutluetranscriptomeclltreecbac5c20fb1cfe83dd9f4f73132861dd916a2bbftargetblankcbac5c2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/almutlue/transcriptome_cll/tree/cbac5c20fb1cfe83dd9f4f73132861dd916a2bbf" target="_blank">cbac5c2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomalmutluetranscriptomeclltreecbac5c20fb1cfe83dd9f4f73132861dd916a2bbftargetblankcbac5c2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/c2.cp.kegg.v6.0.symbols.gmt
    Untracked:  data/ddsrnaCLL_150218.RData
    Untracked:  docs/figure/ATM.Rmd/
    Untracked:  docs/figure/Del17p13.Rmd/
    Untracked:  docs/figure/Del8p12.Rmd/
    Untracked:  docs/figure/Gain8q24.Rmd/
    Untracked:  docs/figure/Med12.Rmd/
    Untracked:  docs/figure/Notch1.Rmd/
    Untracked:  docs/figure/TP53.Rmd/
    Untracked:  output/diff_genes/
    Untracked:  output/diff_meth.rds
    Untracked:  output/diff_meth_IGHV.rds
    Untracked:  output/figures/
    Untracked:  output/res_epistatsis_ighv_tri12.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="epistasis-in-cll" class="section level1">
<h1>Epistasis in CLL</h1>
<p>We find correlation and coexpression of several muatations and genetic variants in CLL. How is this reflected on the transcriptome level? Is there a connection?</p>
<div id="ighv-and-trisomy12" class="section level2">
<h2>IGHV and Trisomy12</h2>
<p>IGHV and trisomy12 are the most severe genomic modification on transcriptome level. How do they affect each other?</p>
<p>load packages</p>
<pre class="r"><code>library(Biobase)
library(ggplot2)
library(genefilter)
library(DESeq2)
library(gridExtra)
library(reshape2)
library(dplyr)
library(geneplotter)
library(RColorBrewer)
library(ComplexHeatmap)
library(circlize)
library(piano)
library(ggpubr)
library(here)</code></pre>
<p>load datasets</p>
<pre class="r"><code>data_dir &lt;- here(&quot;data&quot;)
output_dir &lt;- here(&quot;output&quot;)
figure_dir &lt;- here(&quot;output/figures&quot;)

#Countdata
load(paste0(data_dir, &quot;/ddsrnaCLL_150218.RData&quot;))</code></pre>
</div>
</div>
<div id="epistasis-model" class="section level1">
<h1>Epistasis model</h1>
<p>Use deseq2 to determine genes, which can be described by epistatisc interactions (focus on trisomy12 and IGHV)</p>
<pre class="r"><code>###Deseq
ddsCLL &lt;- estimateSizeFactors(ddsCLL)

#exclude NAs
ddsCLL &lt;- ddsCLL[,!is.na(colData(ddsCLL)[,&quot;IGHV&quot;])]
ddsCLL &lt;- ddsCLL[,!is.na(colData(ddsCLL)[,&quot;trisomy12&quot;])]

RNAnorm &lt;- varianceStabilizingTransformation(ddsCLL, blind=T)
colnames(RNAnorm) &lt;- colData(RNAnorm)$PatID 

#design matrix with interaction term
design(ddsCLL) &lt;- as.formula(paste(&quot;~ IGHV + trisomy12 + IGHV:trisomy12&quot;))
#rnaRaw &lt;- DESeq(ddsCLL, betaPrior = FALSE)
#resultsNames(rnaRaw)
#res &lt;- results(rnaRaw, name = &quot;IGHVU.trisomy121&quot;)

#saveRDS(res, paste0(output_dir, &quot;/res_epistatsis_ighv_tri12.rds&quot;))
res &lt;- readRDS(paste0(output_dir, &quot;/res_epistatsis_ighv_tri12.rds&quot;))

resOrdered &lt;- res[order(res$pvalue),]
resOrderedTab &lt;- as.data.frame(resOrdered)
resOrderedTab$symbol &lt;- rowData(RNAnorm[rownames(resOrdered),])$symbol

resSig &lt;- subset(resOrdered, padj &lt; 0.1)# &amp; abs(log2FoldChange) &gt; 2)
resTab &lt;- as.data.frame(resSig)
resTab$symbol &lt;- rowData(RNAnorm[rownames(resSig),])$symbol</code></pre>
</div>
<div id="gene-expression-of-epistatic-genes" class="section level1">
<h1>Gene expression of epistatic genes</h1>
<div id="heatmap-of-interacting-genes" class="section level2">
<h2>Heatmap of interacting genes</h2>
<pre class="r"><code>  #filter by variant
  sig_Genes &lt;- rownames(resSig)
  
  #gene expression data
  geneExpression = assay(RNAnorm)[sig_Genes,]
  rownames(geneExpression) &lt;- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% sig_Genes)]

  #scale and censor
  geneExpression_new &lt;- log2(geneExpression)
  geneExpression_new&lt;- t(scale(t(geneExpression_new)))
  geneExpression_new[geneExpression_new &gt; 6] &lt;- 6
  geneExpression_new[geneExpression_new &lt; -6] &lt;- -6

  mutnames &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
  
  mutStatus &lt;- data.frame(colData(RNAnorm)) %&gt;% mutate(IGHVnew = ifelse(IGHV %in% &quot;M&quot;, 1, 0)) %&gt;% dplyr::select(-IGHV) %&gt;% mutate(IGHV = IGHVnew) %&gt;% dplyr::select_(&quot;PatID&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;) %&gt;% mutate_(&quot;namA&quot; = &quot;IGHV&quot;, &quot;namB&quot; = &quot;trisomy12&quot;) %&gt;% mutate(naA = as.numeric(as.character(namA))) %&gt;% mutate(naB = as.numeric(as.character(namB))) %&gt;% mutate(mut = factor(mutnames[1 + naA + 2 * naB], levels = mutnames)) %&gt;% arrange(mut)</code></pre>
<pre><code>Warning: select_() is deprecated. 
Please use select() instead

The &#39;programming&#39; vignette or the tidyeval book can help you
to program with select() : https://tidyeval.tidyverse.org
This warning is displayed once per session.</code></pre>
<pre><code>Warning: mutate_() is deprecated. 
Please use mutate() instead

The &#39;programming&#39; vignette or the tidyeval book can help you
to program with mutate() : https://tidyeval.tidyverse.org
This warning is displayed once per session.</code></pre>
<pre class="r"><code>  geneExpression_new &lt;- geneExpression_new[,mutStatus$PatID]


  #colors
  #colors &lt;- colorRampPalette( rev(brewer.pal(11,&quot;RdBu&quot;)) )(255)
  colors = colorRamp2(c(-6,-2,0,2,6), c(&quot;#2166ac&quot;,&quot;#4393c3&quot;, &quot;#f7f7f7&quot;, &quot;#d6604d&quot;,&quot;#b2182b&quot;))
  annocol &lt;- get_palette(&quot;jco&quot;, 10)
  chromcol &lt;- list(chromosome = c(&quot;12&quot; = annocol[6], &quot;other&quot; = annocol[5]))
  
  annocolor &lt;- list(Variant = c(&quot;none&quot; = annocol[1], annocol[2], annocol[3], &quot;both&quot; = annocol[4]))
  names(annocolor$Variant) &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
  mutationStatus &lt;- data.frame(mutStatus$mut)
  rownames(mutationStatus) &lt;- mutStatus$PatID
  colnames(mutationStatus) &lt;- &quot;Variant&quot;

  #Column annotation
  ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, &quot;cm&quot;), annotation_legend_param = list(title_gp = gpar(fontsize = 60), labels_gp = gpar(fontsize = 50),  grid_height = unit(2.5, &quot;cm&quot;), grid_width = unit(2.5, &quot;cm&quot;), gap = unit(15, &quot;mm&quot;)))

  #rowcluster
  geneExpression_dist &lt;- dist(geneExpression_new)
  rowcluster = hclust(geneExpression_dist, method = &quot;ward.D2&quot;)

  #heatmap
  h1 &lt;- Heatmap(geneExpression_new, col = colors ,column_title = paste0(&quot;Gene interactions:&quot;, &quot;IGHV&quot;, &quot;-&quot;, &quot;trisomy12&quot;), column_title_gp = gpar(fontsize = 60, fontface = &quot;bold&quot;), heatmap_legend_param = list(title = &quot;Expr&quot;, title_gp = gpar(fontsize = 60), grid_height = unit(2.5, &quot;cm&quot;), grid_width = unit(2.5, &quot;cm&quot;), gap = unit(15, &quot;mm&quot;), labels_gp = gpar(fontsize = 50), labels = c(-6,-3, 0,3, 6)) , row_dend_width = unit(3.5, &quot;cm&quot;), show_row_dend = T, show_column_names =FALSE , top_annotation = ha_col, show_row_names = FALSE, show_column_dend = FALSE, cluster_columns = FALSE, cluster_rows = rowcluster, split = 5, gap = unit(0.6,&quot;cm&quot;), column_order = mutStatus$PatID)

  #chromosome annotation
  #chromosome distribution
chrom &lt;- as.data.frame(rowData(RNAnorm[sig_Genes,])$chromosome)
rownames(chrom) &lt;- sig_Genes
colnames(chrom ) &lt;- &quot;chromosome&quot;
#select for chromosome12
chrom$chromosome &lt;- ifelse(chrom$chromosome == 12, 12, &quot;other&quot;)
  
  ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(2.3, &quot;cm&quot;), annotation_legend_param = list(ncol = 2, title_gp = gpar(fontsize = 60), labels_gp = gpar(fontsize = 50),  grid_height = unit(2.5, &quot;cm&quot;), grid_width = unit(2.5, &quot;cm&quot;)))
  
   
  #Annotate most significant genes
  top50 &lt;-  rownames(subset(resOrdered, padj &lt; 0.001))
  int_genes &lt;- rowData(RNAnorm[top50,])$symbol

subset &lt;- as.data.frame(rowData(RNAnorm[sig_Genes,]))
subset &lt;- subset[-which(subset$symbol %in% &quot;&quot;),]
subset &lt;- subset[-which(subset$symbol %in% NA),]

subset &lt;- subset[which(subset$symbol %in% int_genes),]
rownames(subset) &lt;- subset$symbol
geneIDs &lt;- which(rownames(geneExpression_new) %in% rownames(subset))
labels &lt;- rownames(geneExpression_new)[geneIDs]
ha_genes &lt;- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 45)), width = unit(9, &quot;cm&quot;))</code></pre>
<pre><code>Warning: anno_link() is deprecated, please use anno_mark() instead.</code></pre>
<pre class="r"><code>  #svg(filename=&quot;~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg&quot;, width=30, height=50)
  #pdf(file=&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistasis_Deseq.pdf&quot;, width=35, height=45)
  draw(h1 + ha_genes + ha_chrom) </code></pre>
<p><img src="figure/epistasis.Rmd/heatmap%20of%20interaction%20ighv-1.png" width="3360" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  #dev.off()
  #draw(h1 + ha_chrom + ha_genes)
 IGHVTri12 &lt;- list(&quot;sig_Genes&quot; = sig_Genes, &quot;geneExp&quot; = geneExpression_new, &quot;h1&quot;= h1)</code></pre>
</div>
<div id="genewise-count-distribution" class="section level2">
<h2>Genewise count distribution</h2>
<pre class="r"><code>#function to create stripchart plots for specific genes
gene_count &lt;- function(gene_nam){
  geneEnsID &lt;- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% gene_nam)]
  geneNum &lt;- counts(ddsCLL)[geneEnsID,]
  mutPat &lt;- as.data.frame(mutationStatus[colData(ddsCLL)$PatID,])
  colnames(mutPat) &lt;- c(&quot;genotype&quot;)
  geneDat &lt;- cbind(mutPat, geneNum)
  colnames(geneDat) &lt;- c(&quot;genotype&quot;, &quot;counts&quot;)
  
  p &lt;- ggstripchart(geneDat, x = &quot;genotype&quot;, y = &quot;counts&quot;,
          color = &quot;genotype&quot;,
          palette = &quot;jco&quot;,
          add = &quot;mean_sd&quot;,
          title = paste(gene_nam),
          ylab = &quot;gene counts&quot;,
          font.x = 25, font.y = 25, font.legend = 20) + font(&quot;xy.text&quot;, size = 20) + font(&quot;title&quot;, size = 30, face = &quot;bold&quot;)
 #ggsave(file=paste0(&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epi_genes/genetic_interaction_&quot;, gene_nam, &quot;.svg&quot;), plot=p, width=7, height=5)
  p
}


#interesting genes
geneList &lt;- c(&quot;LEF1&quot;, &quot;TIMELESS&quot;, &quot;CHAD&quot;, &quot;BCL2A1&quot;, &quot;EML6&quot;, &quot;PPP1R14A&quot;, &quot;EPHB6&quot;, &quot;GEN1&quot;, &quot;EBF1&quot;, &quot;EBF4&quot;)


lapply(geneList, gene_count)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[7]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[8]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[9]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[10]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="clusterwise-count-distribution" class="section level2">
<h2>Clusterwise count distribution</h2>
<pre class="r"><code>cluster &lt;- c(&quot;buffering_dn&quot;,&quot;sup_tri&quot;, &quot;buffering_up&quot; ,&quot;buffering_up_strong&quot;, &quot;inversion_up&quot;)

cluster_size &lt;- lapply(c(1:5), function(clusternr){
  size &lt;- length(row_order(IGHVTri12$h1)[[clusternr]])
  name &lt;- cluster[clusternr]
  clust &lt;- c(name, as.numeric(size))
}) %&gt;% do.call(rbind,.) 

cluster_size &lt;- as.data.frame(cluster_size)
colnames(cluster_size) &lt;- c(&quot;name&quot;, &quot;size&quot;)
cluster_size$size &lt;- as.numeric(as.character(cluster_size$size))
distr &lt;- ggbarplot(cluster_size, &quot;name&quot;, &quot;size&quot;,
   fill = &quot;name&quot;, color = &quot;name&quot;,
   palette = &quot;jco&quot;,
   ylab = &quot;Number of genes&quot;,
   xlab = &quot;cluster&quot;)
 
 #ggsave(file=&quot;/home/almut/Dokumente/masterarbeit/workinprogress/distrib_ofEpiTypes.svg&quot;, plot=distr, width=8, height=5)
distr</code></pre>
<p><img src="figure/epistasis.Rmd/number%20of%20genes%20within%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="enrichment-analysis" class="section level1">
<h1>Enrichment analysis</h1>
<p>List-based enrichment - Piano package Fisher’s exact on genes from one cluster only.</p>
<pre class="r"><code>#load gene set collection
#Hallmark
gsc &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/h.all.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)
#Kegg
gsc_Kegg &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)
tft &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/c3.tft.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)
c5 &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/c5.all.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)
c6 &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/c6.all.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)
c7 &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/c7.all.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)
cgp &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/c2.cgp.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)</code></pre>
<pre class="r"><code>#GSA on cluster defined by kmeans clustering. See cluster in heatmap...
setWiseGSA &lt;- function(clusternr, gmtFile){
  #get sig genes and pvalues for each cluster
  geneNam &lt;- IGHVTri12$sig_Genes[row_order(IGHVTri12$h1)[[clusternr]]] 
  pVal &lt;- resSig[geneNam, &quot;padj&quot;]
  #needs to be symbol/remove those without symbol, but ENSID only
  genName &lt;- rowData(RNAnorm[geneNam,])$symbol
  gsTab &lt;- data.frame(gene = genName, stat = pVal)
  gsTab &lt;- gsTab[-which(gsTab$gene %in% c(&quot;&quot;, NA)),]
  gsaTab &lt;- data.frame(row.names = gsTab$gene, stat = gsTab$stat)
  res &lt;- runGSA(geneLevelStats = gsaTab,
                      geneSetStat = &quot;fisher&quot;,
                     adjMethod = &quot;fdr&quot;, gsc=gmtFile,
                     signifMethod = &#39;nullDist&#39;,
                     #nPerm = 50000,
                      gsSizeLim=c(1, Inf))
  Res &lt;- arrange(GSAsummaryTable(res),desc(`Stat (non-dir.)`))
  #rename results to plot
  resPlot &lt;- Res[, c(1:5)]
  colnames(resPlot) &lt;- c(&quot;pathway&quot;, &quot;gene_number&quot;, &quot;stat&quot;, &quot;p&quot;, &quot;p.adj&quot;)

  #barplot
  #ggplot(resPlot %&gt;% mutate(log10Padj = -log10(p.adj)), aes(x = reorder(pathway, log10Padj), y = log10Padj, fill = gene_number)) + geom_bar(stat = &quot;identity&quot;) + coord_flip() + ggtitle(cluster[clusternr])
  
  enrichPlot &lt;- resPlot[c(1:10),] %&gt;% mutate(log10Padj = -log10(p.adj)) %&gt;% mutate(genes = ifelse(gene_number &gt; 5, &quot;&gt;5&quot;, &quot;&lt;=5&quot;))
  
  p &lt;- ggbarplot(enrichPlot, x = &quot;pathway&quot;, y = &quot;log10Padj&quot;,
          fill = &quot;genes&quot;,           # change fill color by mpg_level
          color = &quot;white&quot;,            # Set bar border colors to white
          palette = &quot;jco&quot;,            # jco journal color palett. see ?ggpar
          sort.val = &quot;asc&quot;,          # Sort the value in ascending order
          sort.by.groups = FALSE,     # Don&#39;t sort inside each group
          x.text.angle = 90,          # Rotate vertically x axis texts
          ylab = &quot;-log10(padj)&quot;,
          legend.title = &quot;Pathway&quot;,
          rotate = TRUE,
          font.x = 20, font.y = 20, font.legend = 20, legend = &quot;right&quot;, 
          title = paste0(&quot;GSEA in &quot;, cluster[clusternr]),
          ggtheme = theme_pubr()
          ) + font(&quot;xy.text&quot;, size = 22) + font(&quot;title&quot;, size = 20, face = &quot;bold&quot;)
  
   ggsave(file=paste0(&quot;/home/almut/Dokumente/masterarbeit/results/enrichment_epistasis/GSEA_in_&quot;, clusternr, &quot;.pdf&quot;), plot=p, width=16, height=7)
#pdf(file= paste0(&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/GSEA_in_&quot;, clusternr, &quot;.pdf&quot;), width=40, height=35)
  p
 # dev.off()
  
  
  }

clusternr = c(1:5)


lapply(as.numeric(clusternr), FUN = &quot;setWiseGSA&quot;, gmtFile = gsc)</code></pre>
<pre><code>Running gene set analysis:</code></pre>
<pre><code>Checking arguments...done!</code></pre>
<pre><code>Final gene/gene-set association: 34 genes and 35 gene sets</code></pre>
<pre><code>  Details:</code></pre>
<pre><code>  Calculating gene set statistics from 34 out of 118 gene-level statistics</code></pre>
<pre><code>  Removed 4352 genes from GSC due to lack of matching gene statistics</code></pre>
<pre><code>  Removed 15 gene sets containing no genes after gene removal</code></pre>
<pre><code>  Removed additionally 0 gene sets not matching the size limits</code></pre>
<pre><code>  Loaded additional information for 35 gene sets</code></pre>
<pre><code>Calculating gene set statistics...done!
Calculating gene set significance...done!
Adjusting for multiple testing...done!</code></pre>
<pre><code>Running gene set analysis:</code></pre>
<pre><code>Checking arguments...done!</code></pre>
<pre><code>Final gene/gene-set association: 37 genes and 33 gene sets</code></pre>
<pre><code>  Details:</code></pre>
<pre><code>  Calculating gene set statistics from 37 out of 220 gene-level statistics</code></pre>
<pre><code>  Removed 4349 genes from GSC due to lack of matching gene statistics</code></pre>
<pre><code>  Removed 17 gene sets containing no genes after gene removal</code></pre>
<pre><code>  Removed additionally 0 gene sets not matching the size limits</code></pre>
<pre><code>  Loaded additional information for 33 gene sets</code></pre>
<pre><code>Calculating gene set statistics...done!
Calculating gene set significance...done!
Adjusting for multiple testing...done!</code></pre>
<pre><code>Running gene set analysis:</code></pre>
<pre><code>Checking arguments...done!</code></pre>
<pre><code>Final gene/gene-set association: 36 genes and 35 gene sets</code></pre>
<pre><code>  Details:</code></pre>
<pre><code>  Calculating gene set statistics from 36 out of 111 gene-level statistics</code></pre>
<pre><code>  Removed 4350 genes from GSC due to lack of matching gene statistics</code></pre>
<pre><code>  Removed 15 gene sets containing no genes after gene removal</code></pre>
<pre><code>  Removed additionally 0 gene sets not matching the size limits</code></pre>
<pre><code>  Loaded additional information for 35 gene sets</code></pre>
<pre><code>Calculating gene set statistics...done!
Calculating gene set significance...done!
Adjusting for multiple testing...done!</code></pre>
<pre><code>Running gene set analysis:</code></pre>
<pre><code>Checking arguments...done!</code></pre>
<pre><code>Final gene/gene-set association: 20 genes and 23 gene sets</code></pre>
<pre><code>  Details:</code></pre>
<pre><code>  Calculating gene set statistics from 20 out of 105 gene-level statistics</code></pre>
<pre><code>  Removed 4366 genes from GSC due to lack of matching gene statistics</code></pre>
<pre><code>  Removed 27 gene sets containing no genes after gene removal</code></pre>
<pre><code>  Removed additionally 0 gene sets not matching the size limits</code></pre>
<pre><code>  Loaded additional information for 23 gene sets</code></pre>
<pre><code>Calculating gene set statistics...done!
Calculating gene set significance...done!
Adjusting for multiple testing...done!</code></pre>
<pre><code>Running gene set analysis:</code></pre>
<pre><code>Checking arguments...done!</code></pre>
<pre><code>Final gene/gene-set association: 39 genes and 29 gene sets</code></pre>
<pre><code>  Details:</code></pre>
<pre><code>  Calculating gene set statistics from 39 out of 164 gene-level statistics</code></pre>
<pre><code>  Removed 4347 genes from GSC due to lack of matching gene statistics</code></pre>
<pre><code>  Removed 21 gene sets containing no genes after gene removal</code></pre>
<pre><code>  Removed additionally 0 gene sets not matching the size limits</code></pre>
<pre><code>  Loaded additional information for 29 gene sets</code></pre>
<pre><code>Calculating gene set statistics...done!
Calculating gene set significance...done!
Adjusting for multiple testing...done!</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/epistasis.Rmd/GSEA%20piano-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/epistasis.Rmd/GSEA%20piano-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/epistasis.Rmd/GSEA%20piano-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/epistasis.Rmd/GSEA%20piano-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/epistasis.Rmd/GSEA%20piano-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mixed-epistatsis-scheme" class="section level1">
<h1>Mixed epistatsis scheme</h1>
<p>Scheme to show different ways of mixed epistasis</p>
<pre class="r"><code>#generate a dataframe
mix &lt;- t(data.frame(buffering_up = c(-0.1, -0.5, 0.5, 5), buffering_dn = c(0.5, 0.2, -0.2, -5), suppression_1 = c(0.5, -5, -6, 0.2), suppression_2 = c(-0.2, 5, 4,-0.5), suppression_3 = c(0.5, 0.1, -6, 1), suppression_4 = c(-0.2, 0.5, 4,-0.5), inversion_up = c(-0.2, -6, -4, 5), inversion_dn = c(0.1, 5, 4, -5)))
colnames(mix) &lt;-  c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)

annocol &lt;- get_palette(&quot;jco&quot;, 10)
annocolor &lt;- list(Variant = c(&quot;none&quot; = annocol[1], annocol[2], annocol[3], &quot;both&quot; = annocol[4]))
names(annocolor$Variant) &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
variants &lt;- as.data.frame(colnames(mix))
colnames(variants) &lt;- &quot;Variant&quot;
rownames(variants) &lt;- variants$Variant

  #Column annotation
  ha_col = HeatmapAnnotation(df = variants, col = annocolor, annotation_height = unit(0.9, &quot;cm&quot;), annotation_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15),  grid_height = unit(0.9, &quot;cm&quot;), grid_width = unit(0.9, &quot;cm&quot;), gap = unit(15, &quot;mm&quot;)))

#heatmap
h1 &lt;- Heatmap(mix, col = colors ,column_title = paste0(&quot;Types of mixed epistasis&quot;), column_title_gp = gpar(fontsize = 20, fontface = &quot;bold&quot;), heatmap_legend_param = list(title = &quot;Expr.&quot;, title_gp = gpar(fontsize = 20), grid_height = unit(1, &quot;cm&quot;), grid_width = unit(0.5, &quot;cm&quot;), gap = unit(10, &quot;mm&quot;), labels_gp = gpar(fontsize = 20), labels = c(-6,-3, 0,3, 6)) , show_row_dend = F, show_column_names =FALSE , top_annotation = ha_col, show_row_names = FALSE, show_column_dend = FALSE, cluster_columns = FALSE, cluster_rows = FALSE, split = c( rep(&quot;Buffering&quot;,2), rep(&quot;Suppression&quot;, 4), rep(&quot;Inversion&quot;, 2)), gap = unit(0.6,&quot;cm&quot;), row_title_gp = gpar(fontsize=19))

#pdf(file=&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/mixed_epistasis_model.pdf&quot;, width=7, height=5)
draw(h1)</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20scheme-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=de_DE.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=de_DE.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] here_0.1                    ggpubr_0.2                 
 [3] magrittr_1.5                piano_2.0.2                
 [5] circlize_0.4.6              ComplexHeatmap_2.0.0       
 [7] RColorBrewer_1.1-2          geneplotter_1.62.0         
 [9] annotate_1.62.0             XML_3.98-1.20              
[11] AnnotationDbi_1.46.0        lattice_0.20-38            
[13] dplyr_0.8.1                 reshape2_1.4.3             
[15] gridExtra_2.3               DESeq2_1.24.0              
[17] SummarizedExperiment_1.14.0 DelayedArray_0.10.0        
[19] BiocParallel_1.18.0         matrixStats_0.54.0         
[21] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[23] IRanges_2.18.1              S4Vectors_0.22.0           
[25] genefilter_1.66.0           ggplot2_3.1.1              
[27] Biobase_2.44.0              BiocGenerics_0.30.0        

loaded via a namespace (and not attached):
 [1] fgsea_1.10.0           colorspace_1.4-1       rjson_0.2.20          
 [4] rprojroot_1.3-2        htmlTable_1.13.1       XVector_0.24.0        
 [7] GlobalOptions_0.1.0    base64enc_0.1-3        fs_1.3.1              
[10] clue_0.3-57            rstudioapi_0.10        DT_0.7                
[13] bit64_0.9-7            splines_3.6.0          knitr_1.23            
[16] jsonlite_1.6           Formula_1.2-3          workflowr_1.4.0       
[19] cluster_2.1.0          png_0.1-7              shinydashboard_0.7.1  
[22] shiny_1.3.2            compiler_3.6.0         backports_1.1.4       
[25] assertthat_0.2.1       Matrix_1.2-17          lazyeval_0.2.2        
[28] limma_3.40.2           later_0.8.0            visNetwork_2.0.7      
[31] acepack_1.4.1          htmltools_0.3.6        tools_3.6.0           
[34] igraph_1.2.4.1         gtable_0.3.0           glue_1.3.1            
[37] GenomeInfoDbData_1.2.1 fastmatch_1.1-0        Rcpp_1.0.1            
[40] slam_0.1-45            gdata_2.18.0           xfun_0.7              
[43] stringr_1.4.0          mime_0.7               gtools_3.8.1          
[46] zlibbioc_1.30.0        scales_1.0.0           promises_1.0.1        
[49] relations_0.6-8        sets_1.0-18            yaml_2.2.0            
[52] memoise_1.1.0          rpart_4.1-15           latticeExtra_0.6-28   
[55] stringi_1.4.3          RSQLite_2.1.1          checkmate_1.9.3       
[58] caTools_1.17.1.2       shape_1.4.4            rlang_0.3.4           
[61] pkgconfig_2.0.2        bitops_1.0-6           evaluate_0.14         
[64] purrr_0.3.2            labeling_0.3           htmlwidgets_1.3       
[67] bit_1.1-14             tidyselect_0.2.5       ggsci_2.9             
[70] plyr_1.8.4             R6_2.4.0               gplots_3.0.1.1        
[73] Hmisc_4.2-0            DBI_1.0.0              pillar_1.4.1          
[76] foreign_0.8-71         withr_2.1.2            survival_2.44-1.1     
[79] RCurl_1.95-4.12        nnet_7.3-12            tibble_2.1.3          
[82] crayon_1.3.4           KernSmooth_2.23-15     rmarkdown_1.13        
[85] GetoptLong_0.1.7       locfit_1.5-9.1         data.table_1.12.2     
[88] marray_1.62.0          blob_1.1.1             git2r_0.25.2          
[91] digest_0.6.19          xtable_1.8-4           httpuv_1.5.1          
[94] munsell_0.5.0          shinyjs_1.0           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
