<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="aluetge" />

<meta name="date" content="2019-07-26" />

<title>epistasis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">transcriptome_cll</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/almutlue/transcriptome_cll">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">epistasis</h1>
<h4 class="author">aluetge</h4>
<h4 class="date">2019-07-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>transcriptome_cll/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190511code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190511)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190511code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190511)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomalmutluetranscriptomeclltree8abaeff7705b302f61136a6c043ee9c689497431targetblank8abaeffa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/almutlue/transcriptome_cll/tree/8abaeff7705b302f61136a6c043ee9c689497431" target="_blank">8abaeff</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomalmutluetranscriptomeclltree8abaeff7705b302f61136a6c043ee9c689497431targetblank8abaeffa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/figures/r_objects/BRAF/enrichment/

Untracked files:
    Untracked:  analysis/methylation_epistasis.Rmd
    Untracked:  docs/figure/methylation_epistasis.Rmd/
    Untracked:  output/desRes_250720.RData
    Untracked:  output/diff_meth_IP_vs_HP_LP.rds
    Untracked:  output/figures/paper_fig/figure1/
    Untracked:  output/figures/paper_fig/figure_IGHV/
    Untracked:  output/figures/paper_fig/figure_c1c2.pdf
    Untracked:  output/figures/paper_fig/figure_c1c2.svg
    Untracked:  output/figures/paper_fig/figure_c1c2/
    Untracked:  output/figures/paper_fig/figure_epi/
    Untracked:  output/figures/paper_fig/figure_epi_meth.pdf
    Untracked:  output/figures/paper_fig/figure_epi_meth.svg
    Untracked:  output/figures/paper_fig/figure_variants_mut_load.pdf
    Untracked:  output/figures/paper_fig/figure_variants_mut_load.svg
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/AACSP1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ADAMTS7.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ADGRB2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/AKAP12.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ANGPT2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ANKRD62.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ARHGAP22.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ASAP3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/BCL7A.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/BDKRB2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/BTBD16.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/C10orf10.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/C10orf35.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/C14orf132.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/C1orf106.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CACHD1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CACNB2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CAMP.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CECR2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CHL1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CHMP4C.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CHST3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CLEC2B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CLEC9A.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CMPK2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CNPY1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CNR1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/COBLL1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/COL1A2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CORO2B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CPNE8.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CPXM1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CRY1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CSGALNACT1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/CSMD1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/DGKH.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/DPP4.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/DPY19L2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/DPYD-AS1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/DUOXA2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/EBF1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/EFNA2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/EGLN3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FAIM2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FAM20C.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FBXO27.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FGFR1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FIRRE.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FKBP10.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FRMD4B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FRMPD1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/FXYD6.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/GGT2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/GGT3P.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/GLDN.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/GPR34.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/GPSM1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/GTSF1L.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/HAR1B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/HNRNPA1P57.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/HTRA3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IFI44L.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IFNA14.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IFNA21.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IFNB1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IFNW1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IFNWP9.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGHV2-5.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGHV3-15.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGHV3-48.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGHV3-7.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGHV4-34.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGKV1-16.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGKV1-8.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGKV2-24.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGKV3-11.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGKV3D-15.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGKV4-1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGLC7.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGLV2-14.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IGLV3-9.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/INA.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/INSR.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/IQSEC3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/JUP.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KALRN.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KANK2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KCNJ2-AS1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KCNJ2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KCNK9.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KIAA0895.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KLK4.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/KRT19P2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/L3MBTL4.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LDOC1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LINC00930.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LINC00982.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LINC01033.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LINC01866.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LINC02170.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LPL.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LRMDA.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LRP5.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/LYPD6B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MAPK4.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MAST4.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MCRIP2P1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MOCOS.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MRO.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MTMR11.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MYL12BP2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MYL9.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MYLK-AS2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MYLK.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/MYO3A.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/NAP1L2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/NCAPGP1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/NEURL1B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/NGEF.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/NPTX1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/NUGGC.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/OSBPL5.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/OVCH1-AS1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/P4HA2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PCDHGC3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PDLIM3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PLD1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PLEK2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PLEKHG4B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PPP2R2C.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PRAMENP.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PRR18.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PTGFRN.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PTH2R.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PTPRB.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/PXDNL.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/RASAL1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/RASSF6.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/RNA5SP460.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/RPS20P22.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SCAMP5.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SCHIP1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SH3BP4.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SH3RF1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SLC12A1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SLC16A9.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SLC39A12.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SLC4A10.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SLC4A8.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SMIM10.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SNORA70F.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SORT1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/SPTA1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TAS1R1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TERT.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TFAP2A-AS1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TFEC.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TGFB2-AS1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/THAP10.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/THRB.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TMEM200A.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TP63.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TRPM4.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/TSPAN13.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/UNC13B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/VSIG2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/WNK2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/WNK3.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/WNT11.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/WNT5A.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/WNT5B.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/WSCD2.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ZAR1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ZNF112.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ZNF334.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ZNF663P.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ZNF667-AS1.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ZNF667.rds
    Untracked:  output/figures/r_objects/Methylation_IP/de_genes/ZNF804A.rds
    Untracked:  output/figures/r_objects/epi_meth/
    Untracked:  output/figures/r_objects/epistasis/de_genes/ABLIM2.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/CAMK2N1.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/CRIM1.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/E2F2.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/ENPP3.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/EPHB1.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/FAM212A.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/FGD6.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/FGF2.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/GP5.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/JAG1.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/RAI14.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/SLC4A8.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/SYBU.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/TCTN1.rds
    Untracked:  output/figures/r_objects/epistasis/de_genes/TMPRSS11E.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_dot2.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_dot_buffering.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_dot_hm.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_dot_inversion.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_dot_supression.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_dot_synergy.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_net_hm.rds
    Untracked:  output/figures/r_objects/epistasis/enrich_net_kegg.rds
    Untracked:  output/figures/r_objects/epistasis/epi_methylation_heatmap.rds
    Untracked:  output/figures/r_objects/epistasis/epistasis_scheme_lolli.rds
    Untracked:  output/figures/r_objects/ighv/enrich_dot2.rds
    Untracked:  output/figures/r_objects/ighv/enrich_dot_hm.rds
    Untracked:  output/figures/r_objects/ighv/enrich_net_hm.rds
    Untracked:  output/figures/r_objects/ighv/enrich_net_kegg.rds
    Untracked:  output/figures/r_objects/mean_sd_by_c1c2.rds
    Untracked:  output/figures/r_objects/trisomy12/de_genes/HRAS.rds
    Untracked:  output/figures/r_objects/trisomy12/de_genes/NT5E.rds
    Untracked:  output/figures/r_objects/trisomy12/de_genes/PTPN6.rds
    Untracked:  output/figures/r_objects/trisomy12/de_genes/VAV1.rds
    Untracked:  output/figures/r_objects/trisomy12/enrich_dot2.rds
    Untracked:  output/figures/r_objects/trisomy12/enrich_dot_hm.rds
    Untracked:  output/figures/r_objects/trisomy12/enrich_net_hm.rds
    Untracked:  output/figures/r_objects/trisomy12/enrich_net_kegg.rds
    Untracked:  output/res_ICGC_epistatsis_ighv_tri12.rds

Unstaged changes:
    Modified:   analysis/IGHV.Rmd
    Modified:   analysis/de_analysis.Rmd
    Modified:   analysis/epistasis.Rmd
    Modified:   analysis/general_eda.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/methylation_IP_vs_HPLP.Rmd
    Modified:   analysis/summary_de_genes.Rmd
    Modified:   analysis/summary_variants.Rmd
    Modified:   analysis/trisomy12.Rmd
    Modified:   output/cluster500exprgenes.pdf
    Modified:   output/diff_genes/ACTN2_diffGenes.csv
    Modified:   output/diff_genes/ATM_diffGenes.csv
    Modified:   output/diff_genes/BRAF_diffGenes.csv
    Modified:   output/diff_genes/Chromothripsis_diffGenes.csv
    Modified:   output/diff_genes/EGR2_diffGenes.csv
    Modified:   output/diff_genes/IGHV_diffGenes.csv
    Modified:   output/diff_genes/KLHL6_diffGenes.csv
    Modified:   output/diff_genes/KRAS_diffGenes.csv
    Modified:   output/diff_genes/MED12_diffGenes.csv
    Modified:   output/diff_genes/MGA_diffGenes.csv
    Modified:   output/diff_genes/NFKBIE_diffGenes.csv
    Modified:   output/diff_genes/NOTCH1_diffGenes.csv
    Modified:   output/diff_genes/PCLO_diffGenes.csv
    Modified:   output/diff_genes/SF3B1_diffGenes.csv
    Modified:   output/diff_genes/TP53_diffGenes.csv
    Modified:   output/diff_genes/XPO1_diffGenes.csv
    Modified:   output/diff_genes/del11q22.3_diffGenes.csv
    Modified:   output/diff_genes/del13q14_diffGenes.csv
    Modified:   output/diff_genes/del15q15.1_diffGenes.csv
    Modified:   output/diff_genes/del17p13_diffGenes.csv
    Modified:   output/diff_genes/del8p12_diffGenes.csv
    Modified:   output/diff_genes/gain14q32_diffGenes.csv
    Modified:   output/diff_genes/gain2p25.3_diffGenes.csv
    Modified:   output/diff_genes/gain8q24_diffGenes.csv
    Modified:   output/diff_genes/meth_IP_vs_HP_diffGenes.csv
    Modified:   output/diff_genes/trisomy12_diffGenes.csv
    Modified:   output/figures/paper_fig/figure1.pdf
    Modified:   output/figures/paper_fig/figure1.svg
    Modified:   output/figures/paper_fig/figure_IGHV.pdf
    Modified:   output/figures/paper_fig/figure_IGHV.svg
    Modified:   output/figures/paper_fig/figure_IP_group.pdf
    Modified:   output/figures/paper_fig/figure_IP_group.svg
    Modified:   output/figures/paper_fig/figure_epi.pdf
    Modified:   output/figures/paper_fig/figure_epi.svg
    Modified:   output/figures/paper_fig/figure_tri12.pdf
    Modified:   output/figures/paper_fig/figure_tri12.svg
    Modified:   output/figures/paper_fig/generate_figures.Rmd
    Modified:   output/figures/r_objects/BRAF/de_genes/ABCC6P1.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/ARHGEF37.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/BIRC5.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/BNIP3P41.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/CA1.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/DHRS9.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/DSP.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/FAM184A.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/FAM96AP2.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/FLT4.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/HCAR1.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/HJURP.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/IGHV3-11.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/IGHV3-23.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/IQCB2P.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/IQGAP3.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/KIF14.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/KIF18B.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/LINC00658.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/LINC01203.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/LINC01358.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/MELK.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/NRP2.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/OR2B6.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/PKMYT1.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/PLIN5.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/PPP1R14C.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/PPP1R3C.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/PTPRB.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/RGPD2.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/RNF157.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/RPSAP69.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/SAMD12.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/SEPT3.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/SERPINE2.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/SH3RF1.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/SLC38A11.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/SPRY4.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/TAGLN3.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/TFEC.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/TMEM151B.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/TMPRSS3.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/ZFHX4.rds
    Modified:   output/figures/r_objects/BRAF/de_genes/ZNF503-AS1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/ARHGAP32.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/BARX2.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/CDYLP1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/CLDN1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/DAZL.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/ELFN2.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/ENPP3.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/FAM41C.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/FST.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/GRIK3.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/ICOS.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGHA2.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGHV1-69.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGHV3-21.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGKV1-12.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGKV1-39.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGKV1OR-2.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGKV3-15.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGLC6.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGLV2-11.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/IGLV3-21.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/LAMC3.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/LINC01203.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/LINC01250.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/MECOM.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/MIR4538.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/MRGPRX4.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/MYB.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/NETO1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/NREP.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/NRIP1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/PHLPP1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/PON1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/PPP1R9A.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/PRDM16.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/RNASE1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/RSPO1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/SEPT10.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/SGMS2.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/SLC25A27.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/SNCAIP.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/SOWAHC.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/SOX11.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/STARD4-AS1.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/T.rds
    Modified:   output/figures/r_objects/Methylation_IP/de_genes/TGFB2.rds
    Modified:   output/figures/r_objects/Methylation_IP/meth_ip_heatmap.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/BCL2A1.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/CHAD.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/EBF1.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/EBF4.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/EML6.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/EPHB6.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/GEN1.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/LEF1.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/PPP1R14A.rds
    Modified:   output/figures/r_objects/epistasis/de_genes/TIMELESS.rds
    Modified:   output/figures/r_objects/epistasis/epistasis_heatmap.rds
    Modified:   output/figures/r_objects/epistasis/epistasis_scheme.rds
    Modified:   output/figures/r_objects/heatmap_top500genes.rds
    Modified:   output/figures/r_objects/ighv/de_genes/ADAM29.rds
    Modified:   output/figures/r_objects/ighv/de_genes/ADAMTS7.rds
    Modified:   output/figures/r_objects/ighv/de_genes/ATOX1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/BCAT1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/C1orf106.rds
    Modified:   output/figures/r_objects/ighv/de_genes/CD38.rds
    Modified:   output/figures/r_objects/ighv/de_genes/CDHR1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/CLEC9A.rds
    Modified:   output/figures/r_objects/ighv/de_genes/CPNE8.rds
    Modified:   output/figures/r_objects/ighv/de_genes/DGKH.rds
    Modified:   output/figures/r_objects/ighv/de_genes/DNAH14.rds
    Modified:   output/figures/r_objects/ighv/de_genes/DPY19L2.rds
    Modified:   output/figures/r_objects/ighv/de_genes/EGR3.rds
    Modified:   output/figures/r_objects/ighv/de_genes/FARP1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/FGFR1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/FRMD4B.rds
    Modified:   output/figures/r_objects/ighv/de_genes/GFI1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/IGLC6.rds
    Modified:   output/figures/r_objects/ighv/de_genes/IGLV3-19.rds
    Modified:   output/figures/r_objects/ighv/de_genes/KANK2.rds
    Modified:   output/figures/r_objects/ighv/de_genes/KCNK9.rds
    Modified:   output/figures/r_objects/ighv/de_genes/LDOC1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/LPL.rds
    Modified:   output/figures/r_objects/ighv/de_genes/MAPK4.rds
    Modified:   output/figures/r_objects/ighv/de_genes/MSI2.rds
    Modified:   output/figures/r_objects/ighv/de_genes/MYL12BP2.rds
    Modified:   output/figures/r_objects/ighv/de_genes/MYLK.rds
    Modified:   output/figures/r_objects/ighv/de_genes/NETO1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/NFAT5.rds
    Modified:   output/figures/r_objects/ighv/de_genes/NPTX1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/NUGGC.rds
    Modified:   output/figures/r_objects/ighv/de_genes/PEG10.rds
    Modified:   output/figures/r_objects/ighv/de_genes/PLD1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/PLEKHG4B.rds
    Modified:   output/figures/r_objects/ighv/de_genes/PRR18.rds
    Modified:   output/figures/r_objects/ighv/de_genes/SEPT10.rds
    Modified:   output/figures/r_objects/ighv/de_genes/SLC12A1.rds
    Modified:   output/figures/r_objects/ighv/de_genes/SLC16A9.rds
    Modified:   output/figures/r_objects/ighv/de_genes/SOWAHC.rds
    Modified:   output/figures/r_objects/ighv/de_genes/VSIG10.rds
    Modified:   output/figures/r_objects/ighv/de_genes/WSB2.rds
    Modified:   output/figures/r_objects/ighv/de_genes/ZAP70.rds
    Modified:   output/figures/r_objects/ighv/de_genes/ZNF334.rds
    Modified:   output/figures/r_objects/ighv/ighv_enrichment_dn.rds
    Modified:   output/figures/r_objects/ighv/ighv_gene_status.rds
    Modified:   output/figures/r_objects/ighv/ighv_heatmap.rds
    Modified:   output/figures/r_objects/pca_top500genes.rds
    Modified:   output/figures/r_objects/summary_de_genes_all.rds
    Modified:   output/figures/r_objects/summary_var_freq_IGHV.rds
    Modified:   output/figures/r_objects/summary_variant_patient.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/ABLIM2.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/ACAP3.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/ADD2.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/CABLES1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/CACNA1D.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/CHDH.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/CHFR.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/CORO2B.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/CTLA4.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/DIRAS1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/EML6.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/EPHB1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/FAHD2B.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/FAM212A.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/GPR146.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/IGHV2-5.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/IGKV1-17.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/IGLV3-21.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/IGLV6-57.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/IQSEC1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/KCND3.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/KLF3-AS1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/LARGE1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/LIX1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/LRRC66.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/MERTK.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/MIR4489.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/MOXD1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/PCBD1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/PCSK7.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/PODXL.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/PPP1R14A.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/RASGRF1.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/RHBDF2.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/RWDD2A.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/SLC16A10.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/SNX22.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/SOCS3.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/SPPL2B.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/SYN3.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/SYNPO.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/TBC1D16.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/TBC1D29.rds
    Modified:   output/figures/r_objects/trisomy12/de_genes/UACA.rds
    Modified:   output/figures/r_objects/trisomy12/tri12_dosage.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/transcriptome_cll/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/epistasis.html" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/a132f77a820392f3dc7f05ad566a74156974309e/analysis/epistasis.Rmd" target="_blank">a132f77</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
<td>
wflow_publish(c(“analysis/epistasis.Rmd”, “analysis/Gain8q24.Rmd”, “analysis/general_eda.Rmd”, “analysis/Med12.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/transcriptome_cll/2cd07c0ba69936230dbbcd36fb108c16634294c2/docs/epistasis.html" target="_blank">2cd07c0</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/d6075084faf2e7b2a8b88921cc87f18a64f0ebd3/analysis/epistasis.Rmd" target="_blank">d607508</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-17
</td>
<td>
wflow_publish(c(“analysis/epistasis.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/almutlue/transcriptome_cll/cc24f928b35f1d70caf1f312608fad772de22bab/docs/epistasis.html" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
<td>
Build site.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="epistasis-in-cll" class="section level1">
<h1>Epistasis in CLL</h1>
<p>We find correlation and coexpression of several muatations and genetic variants in CLL. How is this reflected on the transcriptome level? Is there a connection?</p>
<div id="ighv-and-trisomy12" class="section level2">
<h2>IGHV and Trisomy12</h2>
<p>IGHV and trisomy12 are the most severe genomic modification on transcriptome level. How do they affect each other?</p>
<p>load packages</p>
<pre class="r"><code>library(Biobase)
library(ggplot2)
library(genefilter)
library(DESeq2)
library(gridExtra)
library(reshape2)
library(dplyr)
library(geneplotter)
library(RColorBrewer)
library(ComplexHeatmap)
library(circlize)
library(piano)
library(ggpubr)
library(here)
library(clusterProfiler)
library(msigdbr)
library(org.Hs.eg.db)
library(enrichplot)
library(purrr)
library(data.table)
set.seed(1000)</code></pre>
<p>load datasets</p>
<pre class="r"><code>data_dir &lt;- here(&quot;data&quot;)
output_dir &lt;- here(&quot;output&quot;)
figure_dir &lt;- here(&quot;output/figures&quot;)

#Countdata
load(paste0(data_dir, &quot;/ddsrnaCLL_150218.RData&quot;))</code></pre>
</div>
</div>
<div id="epistasis-model" class="section level1">
<h1>Epistasis model</h1>
<p>Use deseq2 to determine genes, which can be described by epistatisc interactions (focus on trisomy12 and IGHV)</p>
<pre class="r"><code>###Deseq
ddsCLL &lt;- estimateSizeFactors(ddsCLL)

#exclude NAs
ddsCLL &lt;- ddsCLL[,!is.na(colData(ddsCLL)[,&quot;IGHV&quot;])]
ddsCLL &lt;- ddsCLL[,!is.na(colData(ddsCLL)[,&quot;trisomy12&quot;])]

RNAnorm &lt;- varianceStabilizingTransformation(ddsCLL, blind=T)
colnames(RNAnorm) &lt;- colData(RNAnorm)$PatID 

#design matrix with interaction term
design(ddsCLL) &lt;- as.formula(paste(&quot;~ IGHV + trisomy12 + IGHV:trisomy12&quot;))
#rnaRaw &lt;- DESeq(ddsCLL, betaPrior = FALSE)
#resultsNames(rnaRaw)
#res &lt;- results(rnaRaw, name = &quot;IGHVU.trisomy121&quot;)

#saveRDS(res, paste0(output_dir, &quot;/res_epistatsis_ighv_tri12.rds&quot;))
res &lt;- readRDS(paste0(output_dir, &quot;/res_epistatsis_ighv_tri12.rds&quot;))

resOrdered &lt;- res[order(res$pvalue),]
resOrderedTab &lt;- as.data.frame(resOrdered)
resOrderedTab$symbol &lt;- rowData(RNAnorm[rownames(resOrdered),])$symbol

resSig &lt;- subset(resOrdered, padj &lt; 0.1)
resTab &lt;- as.data.frame(resSig)
resTab$symbol &lt;- rowData(RNAnorm[rownames(resSig),])$symbol</code></pre>
</div>
<div id="gene-expression-of-epistatic-genes" class="section level1">
<h1>Gene expression of epistatic genes</h1>
<div id="heatmap-of-interacting-genes" class="section level2">
<h2>Heatmap of interacting genes</h2>
<pre class="r"><code>  #filter by variant
  sig_Genes &lt;- rownames(resSig)
  
  #gene expression data
  geneExpression = assay(RNAnorm)[sig_Genes,]
  rownames(geneExpression) &lt;- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% sig_Genes)]

  #scale and censor
  geneExpression_new &lt;- log2(geneExpression)
  geneExpression_new&lt;- t(scale(t(geneExpression_new)))
  geneExpression_new[geneExpression_new &gt; 4] &lt;- 4
  geneExpression_new[geneExpression_new &lt; -4] &lt;- -4

  mutnames &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
  
  mutStatus &lt;- data.frame(colData(RNAnorm)) %&gt;% mutate(IGHVnew = ifelse(IGHV %in% &quot;M&quot;, 1, 0)) %&gt;% 
    dplyr::select(-IGHV) %&gt;% mutate(IGHV = IGHVnew) %&gt;% 
    dplyr::select_(&quot;PatID&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;) %&gt;% 
    mutate_(&quot;namA&quot; = &quot;IGHV&quot;, &quot;namB&quot; = &quot;trisomy12&quot;) %&gt;% 
    mutate(naA = as.numeric(as.character(namA))) %&gt;% 
    mutate(naB = as.numeric(as.character(namB))) %&gt;% 
    mutate(mut = factor(mutnames[1 + naA + 2 * naB], levels = mutnames)) %&gt;% arrange(mut)</code></pre>
<pre><code>Warning: select_() is deprecated. 
Please use select() instead

The &#39;programming&#39; vignette or the tidyeval book can help you
to program with select() : https://tidyeval.tidyverse.org
This warning is displayed once per session.</code></pre>
<pre><code>Warning: mutate_() is deprecated. 
Please use mutate() instead

The &#39;programming&#39; vignette or the tidyeval book can help you
to program with mutate() : https://tidyeval.tidyverse.org
This warning is displayed once per session.</code></pre>
<pre class="r"><code>  geneExpression_new &lt;- geneExpression_new[,mutStatus$PatID]


  #colors
  #colors &lt;- colorRampPalette( rev(brewer.pal(11,&quot;RdBu&quot;)) )(255)
  colors = colorRamp2(c(-4, -1, 0, 1, 4), c(&quot;#2166ac&quot;, &quot;#4393c3&quot;, &quot;#f7f7f7&quot;, &quot;#d6604d&quot;, &quot;#b2182b&quot;))
  annocol &lt;- get_palette(&quot;uchicago&quot;, 9)
  chromcol &lt;- list(chromosome = c(&quot;12&quot; = annocol[6], &quot;other&quot; = annocol[5]))
  
  annocolor &lt;- list(Variant = c(&quot;none&quot; = annocol[3], &quot;IGHV&quot; = annocol[5], &quot;trisomy12&quot; = annocol[7], &quot;both&quot; = annocol[9]))
  names(annocolor$Variant) &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
  mutationStatus &lt;- data.frame(mutStatus$mut)
  rownames(mutationStatus) &lt;- mutStatus$PatID
  colnames(mutationStatus) &lt;- &quot;Variant&quot;

  #Column annotation
  ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, annotation_height = unit(1.8, &quot;cm&quot;),
                             annotation_legend_param = list(title_gp = gpar(fontsize = 23), 
                                                            labels_gp = gpar(fontsize = 18),  
                                                            grid_height = unit(0.7, &quot;cm&quot;), 
                                                            grid_width = unit(0.3, &quot;cm&quot;), 
                                                            gap = unit(15, &quot;mm&quot;)))

  #rowcluster
  geneExpression_dist &lt;- dist(geneExpression_new)
  rowcluster = hclust(geneExpression_dist, method = &quot;ward.D2&quot;)

  #heatmap
  h1 &lt;- Heatmap(geneExpression_new, 
                col = colors,
                column_title = paste0(&quot;Gene interactions:&quot;, &quot;IGHV&quot;, &quot;-&quot;, &quot;trisomy12&quot;), 
                column_title_gp = gpar(fontsize = 23, fontface = &quot;bold&quot;), 
                heatmap_legend_param = list(title = &quot;Expr&quot;, 
                                            title_gp = gpar(fontsize = 23), 
                                            grid_height = unit(0.7, &quot;cm&quot;), 
                                            grid_width = unit(0.3, &quot;cm&quot;), 
                                            gap = unit(15, &quot;mm&quot;), 
                                            labels_gp = gpar(fontsize = 18), 
                                            labels = c(-4, -1, 0, 1, 4)), 
                row_dend_width = unit(0.7, &quot;cm&quot;), 
                show_row_dend = T, 
                show_column_names =FALSE ,
                top_annotation = ha_col,
                show_row_names = FALSE, 
                show_column_dend = FALSE, 
                row_title_gp = gpar(fontsize = 0),
                cluster_columns = FALSE, 
                cluster_rows = rowcluster, 
                split = 4, gap = unit(0.2,&quot;cm&quot;), 
                column_order = mutStatus$PatID)

  #chromosome annotation
  #chromosome distribution
chrom &lt;- as.data.frame(rowData(RNAnorm[sig_Genes,])$chromosome)
rownames(chrom) &lt;- sig_Genes
colnames(chrom ) &lt;- &quot;chromosome&quot;
#select for chromosome12
chrom$chromosome &lt;- ifelse(chrom$chromosome == 12, 12, &quot;other&quot;)
  
  ha_chrom = rowAnnotation(df = chrom, 
                           col = chromcol, 
                           annotation_width = unit(0.8, &quot;cm&quot;), 
                           annotation_legend_param = list(ncol = 2, 
                                                          title_gp = gpar(fontsize = 23), 
                                                          labels_gp = gpar(fontsize = 18),  
                                                          grid_height = unit(0.7, &quot;cm&quot;), 
                                                          grid_width = unit(0.3, &quot;cm&quot;)))
  
   
  #Annotate most significant genes
  top50 &lt;-  rownames(resTab[which(abs(resTab$stat) &gt; 5 ),])
  int_genes &lt;- rowData(RNAnorm[top50,])$symbol

subset &lt;- as.data.frame(rowData(RNAnorm[sig_Genes,]))
subset &lt;- subset[-which(subset$symbol %in% &quot;&quot;),]
subset &lt;- subset[-which(subset$symbol %in% NA),]

subset &lt;- subset[which(subset$symbol %in% int_genes),]
rownames(subset) &lt;- subset$symbol
geneIDs &lt;- which(rownames(geneExpression_new) %in% rownames(subset))
labels &lt;- rownames(geneExpression_new)[geneIDs]
ha_genes &lt;- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, 
                                               labels_gp = gpar(fontsize = 15)), 
                          width = unit(2.5, &quot;cm&quot;))</code></pre>
<pre><code>Warning: anno_link() is deprecated, please use anno_mark() instead.</code></pre>
<pre class="r"><code>  #svg(filename=&quot;~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg&quot;, width=30, height=50)
  #pdf(file=&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistasis_Deseq.pdf&quot;, width=35, height=45)
p1 &lt;- draw(h1 + ha_genes ) </code></pre>
<p><img src="figure/epistasis.Rmd/heatmap%20of%20interaction%20ighv-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-of-interaction-ighv-1">
Past versions of “heatmap of interaction ighv-1.png”
</button>
</p>
<div id="fig-no-spaces-heatmap-of-interaction-ighv-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/heatmap of interaction ighv-1.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/heatmap of interaction ighv-1.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  #dev.off()
  #draw(h1 + ha_chrom + ha_genes)
#saveRDS(p1, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/epistasis_heatmap.rds&quot;))
IGHVTri12 &lt;- list(&quot;sig_Genes&quot; = sig_Genes, &quot;geneExp&quot; = geneExpression_new, &quot;h1&quot;= h1)</code></pre>
</div>
<div id="heatmap-filtered-of-interacting-genes" class="section level2">
<h2>Heatmap filtered of interacting genes</h2>
<pre class="r"><code> resSig &lt;- subset(resOrdered, padj &lt; 0.01 &amp; abs(stat) &gt; 4)
  #filter by variant
  sig_Genes &lt;- rownames(resSig)
  
  #gene expression data
  geneExpression = assay(RNAnorm)[sig_Genes,]
  rownames(geneExpression) &lt;- rowData(RNAnorm)$symbol[which(rownames(RNAnorm) %in% sig_Genes)]

  #scale and censor
  geneExpression_new &lt;- log2(geneExpression)
  geneExpression_new&lt;- t(scale(t(geneExpression_new)))
  geneExpression_new[geneExpression_new &gt; 4] &lt;- 4
  geneExpression_new[geneExpression_new &lt; -4] &lt;- -4

  mutnames &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
  
  mutStatus &lt;- data.frame(colData(RNAnorm)) %&gt;% mutate(IGHVnew = ifelse(IGHV %in% &quot;M&quot;, 1, 0)) %&gt;% 
    dplyr::select(-IGHV) %&gt;% mutate(IGHV = IGHVnew) %&gt;% 
    dplyr::select_(&quot;PatID&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;) %&gt;% 
    mutate_(&quot;namA&quot; = &quot;IGHV&quot;, &quot;namB&quot; = &quot;trisomy12&quot;) %&gt;% 
    mutate(naA = as.numeric(as.character(namA))) %&gt;% 
    mutate(naB = as.numeric(as.character(namB))) %&gt;% 
    mutate(mut = factor(mutnames[1 + naA + 2 * naB], levels = mutnames)) %&gt;% arrange(mut)

  geneExpression_new &lt;- geneExpression_new[,mutStatus$PatID]


  #colors
  #colors &lt;- colorRampPalette( rev(brewer.pal(11,&quot;RdBu&quot;)) )(255)
  colors = colorRamp2(c(-4, -1, 0, 1, 4), c(&quot;#2166ac&quot;, &quot;#4393c3&quot;, &quot;#f7f7f7&quot;, &quot;#d6604d&quot;, &quot;#b2182b&quot;))
  annocol &lt;- get_palette(&quot;uchicago&quot;, 9)
  chromcol &lt;- list(chromosome = c(&quot;12&quot; = annocol[6], &quot;other&quot; = annocol[5]))
  
  annocolor &lt;- list(Variant = c(&quot;none&quot; = annocol[3], &quot;IGHV&quot; = annocol[5], &quot;trisomy12&quot; = annocol[7], &quot;both&quot; = annocol[9]))
  names(annocolor$Variant) &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
  mutationStatus &lt;- data.frame(mutStatus$mut)
  rownames(mutationStatus) &lt;- mutStatus$PatID
  colnames(mutationStatus) &lt;- &quot;Variant&quot;

  #Column annotation
  ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, simple_anno_size = unit(0.9, &quot;cm&quot;),
                             annotation_legend_param = list(title_gp = gpar(fontsize = 23), 
                                                            labels_gp = gpar(fontsize = 18),  
                                                            grid_height = unit(1, &quot;cm&quot;), 
                                                            grid_width = unit(0.3, &quot;cm&quot;), 
                                                            gap = unit(15, &quot;mm&quot;)))

  #rowcluster
  geneExpression_dist &lt;- dist(geneExpression_new)
  rowcluster = hclust(geneExpression_dist, method = &quot;ward.D2&quot;)

  #heatmap
  h1 &lt;- Heatmap(geneExpression_new, 
                col = colors,
                column_title = paste0(&quot;Gene interactions:&quot;, &quot;IGHV&quot;, &quot;-&quot;, &quot;trisomy12&quot;), 
                column_title_gp = gpar(fontsize = 23, fontface = &quot;bold&quot;), 
                heatmap_legend_param = list(title = &quot;Expr&quot;, 
                                            title_gp = gpar(fontsize = 23), 
                                            grid_height = unit(0.7, &quot;cm&quot;), 
                                            grid_width = unit(0.3, &quot;cm&quot;), 
                                            gap = unit(15, &quot;mm&quot;), 
                                            labels_gp = gpar(fontsize = 18), 
                                            labels = c(-4, -1, 0, 1, 4)), 
                row_dend_width = unit(0.7, &quot;cm&quot;), 
                show_row_dend = T, 
                show_column_names =FALSE ,
                top_annotation = ha_col,
                show_row_names = FALSE, 
                show_column_dend = FALSE, 
                row_title_gp = gpar(fontsize = 0),
                cluster_columns = FALSE, 
                cluster_rows = rowcluster, 
                split = 4, gap = unit(0.2,&quot;cm&quot;), 
                column_order = mutStatus$PatID)


  
  #svg(filename=&quot;~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg&quot;, width=30, height=50)
  #pdf(file=&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistasis_Deseq.pdf&quot;, width=35, height=45)
p1 &lt;- draw(h1) </code></pre>
<p><img src="figure/epistasis.Rmd/heatmap%20of%20interaction%20ighv%20filtered-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  #dev.off()
  #draw(h1 + ha_chrom + ha_genes)
saveRDS(p1, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/epistasis_heatmap.rds&quot;))</code></pre>
</div>
<div id="genewise-count-distribution" class="section level2">
<h2>Genewise count distribution</h2>
<pre class="r"><code>cluster &lt;- c(&quot;buffering&quot;, &quot;supression&quot;, &quot;synergy&quot;, &quot;inversion&quot;)
gene_by_cat &lt;- lapply(1:4, function(clusternr){
  genes &lt;- IGHVTri12$sig_Genes[row_order(IGHVTri12$h1)[[clusternr]]]
  gene_symbol &lt;- rowData(ddsCLL)[genes, &quot;symbol&quot;]
}) %&gt;% set_names(cluster)

#function to create stripchart plots for specific genes
gene_count &lt;- function(gene_nam){
  gene_cat &lt;- names(gene_by_cat) %&gt;% map(function(cat){
    cat_gene &lt;- &quot;none&quot;
    cat_gene &lt;- ifelse(any(gene_nam %in% gene_by_cat[[cat]]), cat, cat_gene) 
    }) %&gt;% unlist()
  gene_cat &lt;- gene_cat[!gene_cat %in% &quot;none&quot;]
  gene_cat &lt;- ifelse(gene_nam %in% &quot;EBF1&quot;, &quot;synergy&quot;, gene_cat)
  gene_cat &lt;- ifelse(gene_nam %in% &quot;FGF2&quot;, &quot;suppression&quot;, gene_cat)
  ddsCLL$IGHV_tri12 &lt;- mutationStatus[colData(ddsCLL)$PatID,]
  geneEnsID &lt;- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% gene_nam)]
  gc &lt;- plotCounts(ddsCLL, gene = geneEnsID, intgroup = &quot;IGHV_tri12&quot;, returnData=TRUE)
  p &lt;- ggboxplot(gc, x = &quot;IGHV_tri12&quot;, y = &quot;count&quot;,
          color = &quot;IGHV_tri12&quot;,
          size = 1.2,
          palette = c(annocol[3], annocol[5], annocol[7], annocol[9]),
          add = &quot;jitter&quot;,
          outlier.shape = NA,
          add.params = list(size = 2.5),
          yscale = &quot;log10&quot;,
          title = paste0(gene_nam, &quot;: &quot;, gene_cat),
          font.x = 20, font.y = 20, font.legend = 20, 
          ylab = &quot;normalized counts&quot;) + font(&quot;xy.text&quot;, size = 20) + font(&quot;title&quot;, size = 20, face = &quot;bold&quot;)
   #ggsave(file=paste0(&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epi_genes/genetic_interaction_&quot;, gene_nam, &quot;.svg&quot;),      plot=p, width=7, height=5)
  saveRDS(p, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/de_genes/&quot;, gene_nam, &quot;.rds&quot;))
  p
}


#interesting genes
diff &lt;- resTab[which(abs(resTab$stat) &gt; 6 ),]
geneList &lt;- diff$symbol
geneList &lt;- geneList[-which(geneList %in% &quot;&quot;)]
geneList &lt;- c(geneList, &quot;LEF1&quot;, &quot;TIMELESS&quot;, &quot;CHAD&quot;, &quot;BCL2A1&quot;, &quot;EML6&quot;, &quot;PPP1R14A&quot;, &quot;EPHB6&quot;, &quot;GEN1&quot;, &quot;EBF1&quot;, 
              &quot;EBF4&quot;, &quot;SLC4A8&quot;, &quot;CAMK2N1&quot;, &quot;FGF2&quot;)


lapply(geneList, gene_count)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-1">
Past versions of “single gene counts-1.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-1.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-1.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-2">
Past versions of “single gene counts-2.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-2.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-2.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-3">
Past versions of “single gene counts-3.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-3.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-3.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-4">
Past versions of “single gene counts-4.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-4.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-4.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-5">
Past versions of “single gene counts-5.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-5.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-5.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-6">
Past versions of “single gene counts-6.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-6.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-6.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[7]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-7">
Past versions of “single gene counts-7.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-7.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-7.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[8]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-8">
Past versions of “single gene counts-8.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-8.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-8.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[9]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-9">
Past versions of “single gene counts-9.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-9.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-9.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[10]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-single-gene-counts-10">
Past versions of “single gene counts-10.png”
</button>
</p>
<div id="fig-no-spaces-single-gene-counts-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/single gene counts-10.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/single gene counts-10.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[11]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[12]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[13]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[14]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[15]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[16]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[17]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[18]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[19]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[20]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[21]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-21.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[22]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-22.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[23]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-23.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[24]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-24.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[25]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-25.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[26]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-26.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[27]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-27.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[28]]</code></pre>
<p><img src="figure/epistasis.Rmd/single%20gene%20counts-28.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="clusterwise-count-distribution" class="section level2">
<h2>Clusterwise count distribution</h2>
<pre class="r"><code>cluster_size &lt;- lapply(c(1:4), function(clusternr){
  size &lt;- length(row_order(IGHVTri12$h1)[[clusternr]])
  name &lt;- cluster[clusternr]
  clust &lt;- c(name, as.numeric(size))
}) %&gt;% do.call(rbind,.) 

cluster_size &lt;- as.data.frame(cluster_size)
colnames(cluster_size) &lt;- c(&quot;name&quot;, &quot;size&quot;)
cluster_size$size &lt;- as.numeric(as.character(cluster_size$size))
distr &lt;- ggbarplot(cluster_size, &quot;name&quot;, &quot;size&quot;,
   fill = &quot;name&quot;, color = &quot;name&quot;,
   palette = &quot;jco&quot;,
   ylab = &quot;Number of genes&quot;,
   xlab = &quot;cluster&quot;)
 
 #ggsave(file=&quot;/home/almut/Dokumente/masterarbeit/workinprogress/distrib_ofEpiTypes.svg&quot;, plot=distr, width=8, height=5)
distr</code></pre>
<p><img src="figure/epistasis.Rmd/number%20of%20genes%20within%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-number-of-genes-within-cluster-1">
Past versions of “number of genes within cluster-1.png”
</button>
</p>
<div id="fig-no-spaces-number-of-genes-within-cluster-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/number of genes within cluster-1.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/number of genes within cluster-1.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis" class="section level2">
<h2>Gene set enrichment analysis</h2>
<p>Gene sets</p>
<pre class="r"><code>#load gene set collection
#Hallmark
gsc &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/h.all.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)
#Kegg
gsc_Kegg &lt;- loadGSC(&quot;/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt&quot;, type=&quot;gmt&quot;)


diff_res &lt;- resOrderedTab
diff_res$ID &lt;- rownames(diff_res)

#clusterProfiler
diff_res &lt;- diff_res[-which(diff_res$symbol %in% c(&quot;&quot;, NA)),]
gene_list &lt;- diff_res$stat %&gt;% set_names(diff_res$symbol)
dup &lt;- names(gene_list)[duplicated(names(gene_list))]
gene_list &lt;- gene_list[-which(names(gene_list) %in% dup)]
gene_list &lt;- sort(gene_list, decreasing = TRUE)
gene_lfc &lt;- diff_res$log2FoldChange %&gt;% set_names(diff_res$symbol)
gene_lfc &lt;- sort(gene_lfc, decreasing = TRUE)
de_gene &lt;- diff_res %&gt;% filter(padj &lt; 0.01) 
de_gene &lt;- de_gene$symbol

de_ens &lt;- diff_res %&gt;% filter(padj &lt; 0.01)
de_ens &lt;- de_ens$ID
#Get Gene IDs
gene_id &lt;- bitr(de_ens, fromType = &quot;ENSEMBL&quot;,
        toType = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;),
        OrgDb = org.Hs.eg.db)</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(de_ens, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;,
&quot;SYMBOL&quot;), : 9.63% of input gene IDs are fail to map...</code></pre>
<pre class="r"><code>gene_list_id &lt;- bitr(diff_res$ID, fromType = &quot;ENSEMBL&quot;,
        toType = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;),
        OrgDb = org.Hs.eg.db)</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(diff_res$ID, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;, :
18.07% of input gene IDs are fail to map...</code></pre>
<pre class="r"><code>names(gene_list_id) &lt;- c(&quot;ID&quot;, &quot;ENTREZID&quot;, &quot;symbol&quot;)
diff_id &lt;- left_join(gene_list_id, diff_res)</code></pre>
<pre><code>Joining, by = c(&quot;ID&quot;, &quot;symbol&quot;)</code></pre>
<pre class="r"><code>gene_list_id &lt;- diff_id$stat %&gt;% set_names(diff_id$ENTREZID)
gene_list_id &lt;- sort(gene_list_id, decreasing = TRUE)
gene_lfc_id &lt;- diff_id$log2FoldChange %&gt;% set_names(diff_id$ENTREZID)
gene_lfc_id &lt;- sort(gene_lfc_id, decreasing = TRUE)

#convert gsc
m_t2g &lt;- msigdbr(species = &quot;Homo sapiens&quot;, category = &quot;H&quot;) %&gt;% 
  dplyr::select(gs_name, human_gene_symbol)

#Hallmark
em2 &lt;- GSEA(gene_list, TERM2GENE = m_t2g, pvalueCutoff = 0.1)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre class="r"><code>em &lt;- enricher(de_gene, TERM2GENE = m_t2g, pvalueCutoff = 0.2)

#Kegg
kk &lt;- enrichKEGG(gene_id$ENTREZID,
                 organism     = &#39;hsa&#39;,
                 pvalueCutoff = 0.2)

kk2 &lt;- gseKEGG(geneList     = gene_list_id,
               organism     = &#39;hsa&#39;,
               nPerm        = 1000,
               minGSSize    = 50,
               pvalueCutoff = 0.2,
               verbose      = FALSE)

kk2x &lt;- setReadable(kk2, &#39;org.Hs.eg.db&#39;, &#39;ENTREZID&#39;)</code></pre>
<p>Visualize ClusterProfiler results</p>
<pre class="r"><code>barplot(kk, showCategory=5)</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(em, showCategory=5)</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dot1 &lt;- clusterProfiler::dotplot(em2, showCategory=10) + ggtitle(&quot;GSEA epistatsis trisomy12 and IGHV&quot;) +
  theme_pubr() +
  theme(legend.position=&quot;right&quot;) + 
  theme(plot.title = element_text(face = &quot;bold&quot;)) </code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<pre class="r"><code>dot1</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clusterProfiler::dotplot(em, showCategory=10) + ggtitle(&quot;Enrichment for epistatsis trisomy12 and IGHV&quot;)</code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clusterProfiler::dotplot(kk2, showCategory=10) + ggtitle(&quot;GSEA for epistatsis trisomy12 and IGHV&quot;)</code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dot2 &lt;- clusterProfiler::dotplot(kk, showCategory=10) + ggtitle(&quot;Enrichment for epistatsis trisomy12 and IGHV&quot;) +
  theme_pubr() +
  theme(legend.position=&quot;right&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;))</code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<pre class="r"><code>dot2</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ridgeplot(em2)</code></pre>
<pre><code>Picking joint bandwidth of 0.282</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ridgeplot(kk2)</code></pre>
<pre><code>Picking joint bandwidth of 0.284</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gseaplot2(em2, geneSetID = 3, title = em2$Description[3])</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gseaplot2(kk2, geneSetID = 2, title = kk2$Description[2])</code></pre>
<p><img src="figure/epistasis.Rmd/clusterProfiler-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(dot1, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/enrich_dot_hm.rds&quot;))
saveRDS(dot2, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/enrich_dot2.rds&quot;))</code></pre>
<p>network plot</p>
<pre class="r"><code># Networks Hallmark
 em2_sub &lt;- em2
 em2_sub@result &lt;- em2@result[which(em2@result$Description %in% c(&quot;HALLMARK_TNFA_SIGNALING_VIA_NFKB&quot;,
                                                                  &quot;HALLMARK_IL2_STAT5_SIGNALING&quot;,
                                                                    &quot;HALLMARK_MYC_TARGETS_V2&quot;)),]
p_net &lt;- cnetplot(em2_sub, categorySize=&quot;pvalue&quot;, foldChange=gene_lfc) + 
  scale_colour_gradientn(colors = c(&quot;#581845&quot;, &quot;#900C3F&quot;, &quot;#C70039&quot;, &quot;#FF5733&quot;, &quot;#FFC300&quot;, &quot;#DAF7A6&quot;)) + 
  guides(size = FALSE) + 
  labs(color = &quot;logFC&quot;)</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for
&#39;colour&#39;, which will replace the existing scale.</code></pre>
<pre class="r"><code>p_net  </code></pre>
<p><img src="figure/epistasis.Rmd/network-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Networks KEGG
 kk2_sub &lt;- kk2x
 kk2_sub@result &lt;- kk2x@result[which(kk2x@result$Description %in% c(&quot;Cytokine-cytokine receptor interaction&quot;,
                                                                    &quot;Hematopoietic cell lineage&quot;,
                                                                    &quot;Antigen processing and presentation&quot;,
                                                                    &quot;Apoptosis&quot;)),]

pnet_kegg &lt;- cnetplot(kk2_sub, categorySize=&quot;pvalue&quot;, foldChange=gene_lfc) + 
  scale_colour_gradientn(colors = c(&quot;#581845&quot;, &quot;#900C3F&quot;, &quot;#C70039&quot;, &quot;#FF5733&quot;, &quot;#FFC300&quot;, &quot;#DAF7A6&quot;)) + 
  guides(size = FALSE) + 
  labs(color = &quot;logFC&quot;)</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for
&#39;colour&#39;, which will replace the existing scale.</code></pre>
<pre class="r"><code>pnet_kegg</code></pre>
<p><img src="figure/epistasis.Rmd/network-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(pnet_kegg, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/enrich_net_kegg.rds&quot;))
saveRDS(p_net, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/enrich_net_hm.rds&quot;))</code></pre>
<div id="enrichment-analysis-per-epistatsis-type" class="section level3">
<h3>Enrichment analysis per epistatsis type</h3>
<p>List-based enrichment - Piano package Fisher’s exact on genes from one cluster only.</p>
<pre class="r"><code>cluster_gsea &lt;- function(clusternr){
  gene_in &lt;- IGHVTri12$sig_Genes[row_order(IGHVTri12$h1)[[clusternr]]] 
  gene_id &lt;- bitr(gene_in, fromType = &quot;ENSEMBL&quot;,
        toType = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;),
        OrgDb = org.Hs.eg.db)

  names(gene_id) &lt;- c(&quot;ID&quot;, &quot;ENTREZID&quot;, &quot;symbol&quot;)

  #Hallmark
  em &lt;- enricher(gene_id$symbol, TERM2GENE = m_t2g, pvalueCutoff = 0.5, qvalueCutoff = 0.5)

  #Kegg
  kk &lt;- enrichKEGG(gene_id$ENTREZID,
                 organism     = &#39;hsa&#39;,
                 pvalueCutoff = 0.5,
                 qvalueCutoff = 0.5)
  plot(barplot(em, showCategory=5))
  plot(barplot(kk, showCategory=5))
}

lapply(1:4, cluster_gsea)</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(gene_in, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;,
&quot;SYMBOL&quot;), : 20.51% of input gene IDs are fail to map...</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(gene_in, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;,
&quot;SYMBOL&quot;), : 24.37% of input gene IDs are fail to map...</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/list-based%20GSA-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(gene_in, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;,
&quot;SYMBOL&quot;), : 17.69% of input gene IDs are fail to map...</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/list-based%20GSA-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(gene_in, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;,
&quot;SYMBOL&quot;), : 27.35% of input gene IDs are fail to map...</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/list-based%20GSA-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/list-based%20GSA-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/epistasis.Rmd/list-based%20GSA-12.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea-per-epistasis-type" class="section level3">
<h3>GSEA per epistasis type</h3>
<p>Exclude genes from other epistasis types</p>
<pre class="r"><code>epitype_gsea &lt;- function(clusternr){
   gene_in &lt;- IGHVTri12$sig_Genes[row_order(IGHVTri12$h1)[[clusternr]]] 
   gene_out &lt;- IGHVTri12$sig_Genes[!IGHVTri12$sig_Genes %in% gene_in]
   diff_res &lt;- resOrderedTab
   diff_res$ID &lt;- rownames(diff_res)

   #clusterProfiler
   #filter genes
   diff_res &lt;- diff_res[-which(diff_res$symbol %in% c(&quot;&quot;, NA)),]
   diff_res &lt;- diff_res[!duplicated(diff_res$symbol),]
   diff_res &lt;- diff_res[!rownames(diff_res) %in% gene_out,]
   gene_list &lt;- diff_res$stat %&gt;% set_names(diff_res$symbol)
   gene_list &lt;- sort(gene_list, decreasing = TRUE)
   gene_list_id &lt;- bitr(diff_res$ID, fromType = &quot;ENSEMBL&quot;,
        toType = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;),
        OrgDb = org.Hs.eg.db)
    names(gene_list_id) &lt;- c(&quot;ID&quot;, &quot;ENTREZID&quot;, &quot;symbol&quot;)
    diff_id &lt;- left_join(gene_list_id, diff_res)
    gene_list_id &lt;- diff_id$stat %&gt;% set_names(diff_id$ENTREZID)
    gene_list_id &lt;- sort(gene_list_id, decreasing = TRUE)
    gene_lfc_id &lt;- diff_id$log2FoldChange %&gt;% set_names(diff_id$ENTREZID)
    gene_lfc_id &lt;- sort(gene_lfc_id, decreasing = TRUE)

    #Hallmark
    em2 &lt;- GSEA(gene_list, TERM2GENE = m_t2g, pvalueCutoff = 0.1)
    #Kegg
    kk2 &lt;- gseKEGG(geneList     = gene_list_id,
               organism     = &#39;hsa&#39;,
               nPerm        = 1000,
               minGSSize    = 50,
               pvalueCutoff = 0.2,
               verbose      = FALSE)

    kk2x &lt;- setReadable(kk2, &#39;org.Hs.eg.db&#39;, &#39;ENTREZID&#39;)
    dot1 &lt;- clusterProfiler::dotplot(em2, showCategory=12) + ggtitle(paste0(&quot;GSEA epistasis &quot;, cluster[clusternr])) +
      theme_pubr() +
      theme(legend.position=&quot;right&quot;) + 
      theme(plot.title = element_text(face = &quot;bold&quot;)) 
    plot(dot1)
    saveRDS(dot1, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/enrich_dot_&quot;, cluster[clusternr],&quot;.rds&quot;))
    dot2 &lt;- clusterProfiler::dotplot(kk2, showCategory=10) + ggtitle(paste0(&quot;GSEA epistasis &quot;, cluster[clusternr])) +
    theme_pubr() +
    theme(legend.position=&quot;right&quot;) +
    theme(plot.title = element_text(face = &quot;bold&quot;))
    plot(dot2)

    plot(ridgeplot(em2))
    plot(ridgeplot(kk2))
    em2
}

em_list &lt;- lapply(1:4, epitype_gsea) %&gt;% set_names(cluster)</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(diff_res$ID, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;, :
18.27% of input gene IDs are fail to map...</code></pre>
<pre><code>Joining, by = c(&quot;ID&quot;, &quot;symbol&quot;)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`
wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/epistasis%20gsea-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.216</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.22</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(diff_res$ID, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;, :
18.29% of input gene IDs are fail to map...</code></pre>
<pre><code>Joining, by = c(&quot;ID&quot;, &quot;symbol&quot;)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/epistasis%20gsea-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.218</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.22</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(diff_res$ID, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;, :
18.27% of input gene IDs are fail to map...</code></pre>
<pre><code>Joining, by = c(&quot;ID&quot;, &quot;symbol&quot;)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/epistasis%20gsea-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.223</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.243</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning in bitr(diff_res$ID, fromType = &quot;ENSEMBL&quot;, toType = c(&quot;ENTREZID&quot;, :
18.22% of input gene IDs are fail to map...</code></pre>
<pre><code>Joining, by = c(&quot;ID&quot;, &quot;symbol&quot;)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/epistasis.Rmd/epistasis%20gsea-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.26</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Picking joint bandwidth of 0.276</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20gsea-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#get unique pathways per type
sig_pw &lt;- lapply(1:4, function(cluster_nr){
  pw_unique &lt;- em_list[[cluster[cluster_nr]]]@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% dplyr::select(ID)
  pw_unique$ID
}) %&gt;% set_names(cluster)

dup_pw &lt;- unlist(sig_pw)[duplicated(unlist(sig_pw))]
all_pw &lt;- Reduce(intersect, sig_pw)
unique_pw &lt;- sig_pw %&gt;% map(function(pw){pw &lt;- pw[!pw %in% all_pw]})
unique_pw </code></pre>
<pre><code>$buffering
character(0)

$supression
[1] &quot;HALLMARK_HYPOXIA&quot;             &quot;HALLMARK_PANCREAS_BETA_CELLS&quot;
[3] &quot;HALLMARK_ALLOGRAFT_REJECTION&quot; &quot;HALLMARK_APOPTOSIS&quot;          
[5] &quot;HALLMARK_E2F_TARGETS&quot;        

$synergy
[1] &quot;HALLMARK_PANCREAS_BETA_CELLS&quot;     &quot;HALLMARK_INFLAMMATORY_RESPONSE&quot;  
[3] &quot;HALLMARK_ALLOGRAFT_REJECTION&quot;     &quot;HALLMARK_G2M_CHECKPOINT&quot;         
[5] &quot;HALLMARK_HYPOXIA&quot;                 &quot;HALLMARK_E2F_TARGETS&quot;            
[7] &quot;HALLMARK_IL6_JAK_STAT3_SIGNALING&quot;

$inversion
[1] &quot;HALLMARK_ALLOGRAFT_REJECTION&quot;      
[2] &quot;HALLMARK_HYPOXIA&quot;                  
[3] &quot;HALLMARK_IL6_JAK_STAT3_SIGNALING&quot;  
[4] &quot;HALLMARK_INFLAMMATORY_RESPONSE&quot;    
[5] &quot;HALLMARK_INTERFERON_GAMMA_RESPONSE&quot;
[6] &quot;HALLMARK_PANCREAS_BETA_CELLS&quot;      
[7] &quot;HALLMARK_G2M_CHECKPOINT&quot;           
[8] &quot;HALLMARK_APOPTOSIS&quot;                
[9] &quot;HALLMARK_E2F_TARGETS&quot;              </code></pre>
</div>
</div>
</div>
<div id="mixed-epistatsis-scheme" class="section level1">
<h1>Mixed epistatsis scheme</h1>
<p>Scheme to show different ways of mixed epistasis</p>
<pre class="r"><code>#generate a dataframe
mix &lt;- t(data.frame(synergy = c(-0.1, -0.5, 0.5, 5), buffering_dn = c(0.5, 0.2, -0.2, -5), suppression_1 = c(0.5, -5, -6, 0.2), suppression_2 = c(-0.2, 5, 4,-0.5), suppression_3 = c(0.5, 0.1, -6, 1), suppression_4 = c(-0.2, 0.5, 4,-0.5), inversion_up = c(-0.2, -6, -4, 5), inversion_dn = c(0.1, 5, 4, -5)))
colnames(mix) &lt;-  c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)

annocol &lt;- get_palette(&quot;uchicago&quot;, 9)
annocolor &lt;- list(Variant = c(&quot;none&quot; = annocol[3], &quot;IGHV&quot; = annocol[5], &quot;trisomy12&quot; = annocol[7], &quot;both&quot; = annocol[9]))
names(annocolor$Variant) &lt;- c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
variants &lt;- as.data.frame(colnames(mix))
colnames(variants) &lt;- &quot;Variant&quot;
rownames(variants) &lt;- variants$Variant

  #Column annotation
  ha_col = HeatmapAnnotation(df = variants, 
                             col = annocolor, 
                             simple_anno_size = unit(0.9, &quot;cm&quot;), 
                             annotation_legend_param = list(title_gp = gpar(fontsize = 20), 
                                                            labels_gp = gpar(fontsize = 15),  
                                                            grid_height = unit(0.9, &quot;cm&quot;), 
                                                            grid_width = unit(0.9, &quot;cm&quot;), 
                                                            gap = unit(15, &quot;mm&quot;)))

#heatmap
h1 &lt;- Heatmap(mix, 
              col = colors,
              column_title = paste0(&quot;Types of mixed epistasis&quot;), 
              column_title_gp = gpar(fontsize = 20, fontface = &quot;bold&quot;), 
              heatmap_legend_param = list(title = &quot;Expr.&quot;, 
                                          title_gp = gpar(fontsize = 20), 
                                          grid_height = unit(1, &quot;cm&quot;), 
                                          grid_width = unit(0.5, &quot;cm&quot;), 
                                          gap = unit(10, &quot;mm&quot;), 
                                          labels_gp = gpar(fontsize = 20), 
                                          labels = c(-6,-3, 0,3, 6)) , 
              show_row_dend = F, 
              show_column_names =FALSE , 
              top_annotation = ha_col, 
              show_row_names = FALSE, 
              show_column_dend = FALSE, 
              cluster_columns = FALSE, 
              cluster_rows = FALSE, 
              split = c( rep(&quot;Synergy&quot;,1), rep(&quot;Buffering&quot;,1), rep(&quot;Suppression&quot;, 4), rep(&quot;Inversion&quot;, 2)), 
              gap = unit(0.8,&quot;cm&quot;), 
              row_title_gp = gpar(fontsize=19))

#pdf(file=&quot;/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/mixed_epistasis_model.pdf&quot;, width=7, height=5)
draw(h1)</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20scheme-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-epistasis-scheme-1">
Past versions of “epistasis scheme-1.png”
</button>
</p>
<div id="fig-no-spaces-epistasis-scheme-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/712d2e92f0600e7f359092f190d279dbbf8b1a55/docs/figure/epistasis.Rmd/epistasis scheme-1.png" target="_blank">712d2e9</a>
</td>
<td>
aluetge
</td>
<td>
2019-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/almutlue/transcriptome_cll/blob/cc24f928b35f1d70caf1f312608fad772de22bab/docs/figure/epistasis.Rmd/epistasis scheme-1.png" target="_blank">cc24f92</a>
</td>
<td>
aluetge
</td>
<td>
2019-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()

saveRDS(h1, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/epistasis_scheme.rds&quot;))</code></pre>
<pre class="r"><code>annocol &lt;- get_palette(&quot;uchicago&quot;, 9)
annocolor &lt;- c(&quot;none&quot; = annocol[3], &quot;IGHV&quot; = annocol[5], &quot;trisomy12&quot; = annocol[7], &quot;both&quot; = annocol[9])

#generate a dataframe
mix &lt;- t(data.frame(synergy = c(10, 30, 25, 450), buffering = c(400, 450, 425, 50), suppression = c(20, 30, 450, 40), inversion = c(20, 420, 450, 35)))

colnames(mix) &lt;-  c(&quot;none&quot;, &quot;IGHV&quot;, &quot;trisomy12&quot;, &quot;both&quot;)
mix &lt;- data.frame(mix)
mix$epistasis_type &lt;- rownames(mix)
mix_long &lt;- melt(setDT(mix), id.vars = c(&quot;epistasis_type&quot;), variable.name = &quot;genotype&quot;)
mix_long$epistasis_type &lt;- factor(mix_long$epistasis_type, levels = mix$epistasis_type)

colnames(mix_long) &lt;- c(&quot;epistasis_type&quot;, &quot;genotype&quot;, &quot;gene_count&quot;)


p &lt;- ggplot(mix_long, aes(x = genotype, y = gene_count)) +
  geom_segment( aes(x=genotype, xend=genotype, y=0, yend=gene_count), color=&quot;grey&quot;) +
  geom_point( aes(x=genotype, y=gene_count, color=genotype), size=7 ) +
  scale_y_continuous(breaks=c(0,240,480),
        labels=c(&quot;low&quot;, &quot;medium&quot;, &quot;high&quot;), limits = c(0,500)) +
  facet_wrap(~epistasis_type, ncol=2) +
  theme_pubr() +
    theme(legend.position=&quot;right&quot;) +
    theme(plot.title = element_text(face = &quot;bold&quot;, size = 24),
          axis.title = element_text(face = &quot;bold&quot;, size = 18),
          legend.position = &quot;none&quot;,
          axis.text = element_text(size = 14),
          strip.text.x = element_text(face = &quot;bold&quot;, size = 18),
          panel.border = element_rect(fill = NA, colour = &quot;black&quot;)
      ) +
  ggtitle(&quot;scheme types of epistasis&quot;) +
  xlab(&quot;genotype&quot;) +
  ylab(&quot;mean gene counts&quot;) +
  scale_colour_manual(values = annocolor)
p</code></pre>
<p><img src="figure/epistasis.Rmd/epistasis%20scheme%20lollipop-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(p, file = paste0(output_dir, &quot;/figures/r_objects/epistasis/epistasis_scheme_lolli.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=de_DE.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=de_DE.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] data.table_1.12.2           purrr_0.3.2                
 [3] enrichplot_1.4.0            org.Hs.eg.db_3.8.2         
 [5] msigdbr_7.0.1               clusterProfiler_3.12.0     
 [7] here_0.1                    ggpubr_0.2                 
 [9] magrittr_1.5                piano_2.0.2                
[11] circlize_0.4.6              ComplexHeatmap_2.0.0       
[13] RColorBrewer_1.1-2          geneplotter_1.62.0         
[15] annotate_1.62.0             XML_3.98-1.20              
[17] AnnotationDbi_1.46.0        lattice_0.20-38            
[19] dplyr_0.8.1                 reshape2_1.4.3             
[21] gridExtra_2.3               DESeq2_1.24.0              
[23] SummarizedExperiment_1.14.0 DelayedArray_0.10.0        
[25] BiocParallel_1.18.0         matrixStats_0.54.0         
[27] GenomicRanges_1.36.0        GenomeInfoDb_1.20.0        
[29] IRanges_2.18.1              S4Vectors_0.22.0           
[31] genefilter_1.66.0           ggplot2_3.1.1              
[33] Biobase_2.44.0              BiocGenerics_0.30.0        

loaded via a namespace (and not attached):
  [1] backports_1.1.4        Hmisc_4.2-0            fastmatch_1.1-0       
  [4] workflowr_1.4.0        plyr_1.8.4             igraph_1.2.4.1        
  [7] lazyeval_0.2.2         shinydashboard_0.7.1   splines_3.6.0         
 [10] urltools_1.7.3         digest_0.6.19          htmltools_0.3.6       
 [13] GOSemSim_2.10.0        viridis_0.5.1          GO.db_3.8.2           
 [16] gdata_2.18.0           checkmate_1.9.3        memoise_1.1.0         
 [19] cluster_2.1.0          limma_3.40.2           graphlayouts_0.6.0    
 [22] prettyunits_1.0.2      colorspace_1.4-1       blob_1.1.1            
 [25] ggrepel_0.8.1          xfun_0.7               crayon_1.3.4          
 [28] RCurl_1.95-4.12        jsonlite_1.6           survival_2.44-1.1     
 [31] glue_1.3.1             polyclip_1.10-0        gtable_0.3.0          
 [34] zlibbioc_1.30.0        XVector_0.24.0         UpSetR_1.4.0          
 [37] GetoptLong_0.1.7       shape_1.4.4            scales_1.0.0          
 [40] DOSE_3.10.2            DBI_1.0.0              relations_0.6-8       
 [43] Rcpp_1.0.1             progress_1.2.2         viridisLite_0.3.0     
 [46] xtable_1.8-4           htmlTable_1.13.1       clue_0.3-57           
 [49] gridGraphics_0.5-0     europepmc_0.3          foreign_0.8-71        
 [52] bit_1.1-14             Formula_1.2-3          DT_0.7                
 [55] httr_1.4.0             htmlwidgets_1.3        fgsea_1.10.0          
 [58] gplots_3.0.1.1         acepack_1.4.1          pkgconfig_2.0.2       
 [61] farver_2.0.3           nnet_7.3-12            locfit_1.5-9.1        
 [64] labeling_0.3           ggplotify_0.0.5        tidyselect_0.2.5      
 [67] rlang_0.3.4            later_0.8.0            munsell_0.5.0         
 [70] tools_3.6.0            visNetwork_2.0.7       RSQLite_2.1.1         
 [73] ggridges_0.5.2         evaluate_0.14          stringr_1.4.0         
 [76] yaml_2.2.0             knitr_1.23             bit64_0.9-7           
 [79] fs_1.3.1               tidygraph_1.1.2        caTools_1.17.1.2      
 [82] ggraph_2.0.2           whisker_0.3-2          mime_0.7              
 [85] slam_0.1-45            xml2_1.2.0             DO.db_2.9             
 [88] compiler_3.6.0         rstudioapi_0.10        png_0.1-7             
 [91] marray_1.62.0          tibble_2.1.3           tweenr_1.0.1          
 [94] stringi_1.4.3          Matrix_1.2-17          ggsci_2.9             
 [97] shinyjs_1.0            pillar_1.4.1           BiocManager_1.30.4    
[100] triebeard_0.3.0        GlobalOptions_0.1.0    cowplot_0.9.4         
[103] bitops_1.0-6           httpuv_1.5.1           qvalue_2.16.0         
[106] R6_2.4.0               latticeExtra_0.6-28    promises_1.0.1        
[109] KernSmooth_2.23-15     MASS_7.3-51.4          gtools_3.8.1          
[112] assertthat_0.2.1       rprojroot_1.3-2        rjson_0.2.20          
[115] withr_2.1.2            GenomeInfoDbData_1.2.1 hms_0.4.2             
[118] rpart_4.1-15           tidyr_0.8.3            rvcheck_0.1.8         
[121] rmarkdown_1.13         git2r_0.25.2           sets_1.0-18           
[124] ggforce_0.3.1          shiny_1.3.2            base64enc_0.1-3       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
