---
title: "Differentially expressed genes"
author: "almut"
date: "15 November 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Differentially expressed genes in CLL

Aim: Find gene signatures for mutational status of CLL patients

load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(ComplexHeatmap)
library(matrixStats)
library(here)
library(reshape2)
```


load data
```{r datasets}
data_dir <- here("data")
output_dir <- here("output")
figure_dir <- here("output/figures")

#dds data set. gene expression data + patmetadata
load(paste0(data_dir, "/ddsrnaCLL_150218.RData"))

#load meta data including genotyping info
load(paste0(data_dir, "/patmeta_170324.RData"))
excluded_columns <- c("HIPO.ID","PID", "gender","project", "diagnosis", "date.of.diagnosis", "treatment", "date.of.first.treatment", "IGHV.status","Methylation_Cluster")

patMeta <- as.tibble(patMeta) %>% filter(Patient.ID %in% ddsCLL$PatID) %>% dplyr::select(-one_of(excluded_columns))

variants <- patMeta %>% dplyr::select(-Patient.ID) %>% 
  mutate_if(is.factor, as.character) %>% 
  mutate_if(is.character, as.numeric) %>% 
  dplyr::select(colnames(.)[colSums(.,na.rm = TRUE) > 4]) %>% 
  colnames()

var_add <- variants[!variants %in% names(colData(ddsCLL))] 
rownames(patMeta) <-  patMeta$Patient.ID
patMeta <- patMeta[colData(ddsCLL)$PatID,]
cd <- cbind(colData(ddsCLL), patMeta[,var_add])
colData(ddsCLL) <- cd

```


```{r deseq}
###Deseq
ddsCLL <- estimateSizeFactors(ddsCLL)

#write a function to perform deseq for different genetic conditions 
diff <- function(cond){
  gc()
  ddsCLL_new <- ddsCLL[,!is.na(colData(ddsCLL)[,cond])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"IGHV"])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"trisomy12"])]
  colData(ddsCLL_new)[,"IGHV"] <-droplevels(colData(ddsCLL_new)[,"IGHV"])
  design(ddsCLL_new) <- as.formula(paste("~ IGHV + trisomy12 + ", paste(cond)))
  rnaRaw <- DESeq(ddsCLL_new, betaPrior = FALSE)
  res <- results(rnaRaw)
  resOrdered <- res[order(res$pvalue),]
}

#gene_conditions <- c("del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1", "TP53", "ATM", "MED12", "trisomy12")
variants_not12 <- variants[!variants %in% "trisomy12"]
#res_list <- lapply(variants_not12, diff)
#names(res_list) <- variants_not12

diff_notri12 <- function(cond){
  ddsCLL_new <- ddsCLL[,!is.na(colData(ddsCLL)[,cond])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"trisomy12"])]
  design(ddsCLL_new) <- as.formula(paste("~ trisomy12 + ", paste(cond)))
  rnaRaw <- DESeq(ddsCLL_new, betaPrior = FALSE)
  res <- results(rnaRaw)
  resOrdered <- res[order(res$pvalue),]
}

#res_list[["IGHV"]] <- diff_notri12("IGHV")

diff_noighv <- function(cond){
  ddsCLL_new <- ddsCLL[,!is.na(colData(ddsCLL)[,cond])]
  ddsCLL_new <- ddsCLL_new[,!is.na(colData(ddsCLL_new)[,"IGHV"])]
  design(ddsCLL_new) <- as.formula(paste("~ IGHV + ", paste(cond)))
  rnaRaw <- DESeq(ddsCLL_new, betaPrior = FALSE)
  res <- results(rnaRaw)
  resOrdered <- res[order(res$pvalue),]
}

#res_list[["trisomy12"]] <- diff_noighv("trisomy12")

#save(res_list, file=paste0(output_dir,"/desRes_250720.RData"))

load(paste0(output_dir,"/desRes_250720.RData"))
```

Diff genes
```{r create csv of diff genes}

pCut <- 0.1

difftab <- function(condition){
  dataTab <- data.frame(res_list[[condition]])
  dataTab$ID <- rownames(dataTab)
  #filter using pvalues
  dataTab <- filter(dataTab, padj <= pCut) %>%
    arrange(padj) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol)# %>%
    #filter(abs(log2FoldChange) > 2)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  rownames(dataTab) <- dataTab$ID
  write.csv(dataTab, file=paste0(output_dir,"/diff_genes/", condition, "_diffGenes.csv"))
  dataTab
}

cond <- c(variants, "IGHV")

#Only run when you want to write result tables! Change path according to test! 
sigRes <- lapply(cond, difftab)
names(sigRes) <- cond


```


### MA-plots
```{r ma-plots}
#Check Ma plots

myMaPlot <-function(condition){
  DESeq2::plotMA(res_list[[condition]], ylim=c(-5,5), main= paste(condition))
}

lapply(cond, myMaPlot)

```


### Histogramm of pvalues

```{r hist}

myHist <- function(condition){
  res <- res_list[[condition]]
  hist(res$pvalue, breaks=100, col="skyblue", border="slateblue", 
                    main=paste0("Histogramm of pvalues:", condition),plot = TRUE)
}

hist_cond <- lapply(cond, myHist)

```

