---
title: "Time from diagnosis to sampling"
author: "almut"
date: "24 April 2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Association between gene expression and time from diagnosis to sampling

Aim: Find gene signatures for mutational status of CLL patients

load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(dplyr)
library(magrittr)
library(tidyverse)
library(gridExtra)
library(ComplexHeatmap)
library(matrixStats)
library(here)
library(variancePartition)
library(ggtern)
```


load data
```{r datasets}
data_dir <- here("data")
output_dir <- here("output")
figure_dir <- here("output/figures")

#dds data set. gene expression data + patmetadata
load(paste0(data_dir, "/ddsrnaCLL_150218.RData"))

#load meta data including genotyping info
load(paste0(data_dir, "/patmeta_170324.RData"))
excluded_columns <- c("HIPO.ID","PID", "gender","project", "diagnosis", "date.of.diagnosis", "date.of.first.treatment", "IGHV.status","Methylation_Cluster")

#get pretreatment status
pretreat = read.csv(paste0(data_dir, "/diagnosis_information.csv"), sep = ";")
rownames(pretreat) <- pretreat$Patient.ID

patMeta <- as.tibble(patMeta) %>% filter(Patient.ID %in% ddsCLL$PatID) %>% dplyr::select(-one_of(excluded_columns))

variants <- patMeta %>% dplyr::select(-Patient.ID) %>%  dplyr::select(-treatment) %>%
  mutate_if(is.factor, as.character) %>% 
  mutate_if(is.character, as.numeric) %>% 
  dplyr::select(colnames(.)[colSums(.,na.rm = TRUE) > 4]) %>% 
  colnames()

var_add <- variants[!variants %in% names(colData(ddsCLL))] 
rownames(patMeta) <-  patMeta$Patient.ID
patMeta <- patMeta[colData(ddsCLL)$PatID,]
pretreat <- pretreat[colData(ddsCLL)$PatID,]
cd <- cbind(colData(ddsCLL), patMeta[,var_add])
colData(ddsCLL) <- cd
pretreat$norm_time_diag <- scale(pretreat$days.from.diagnosis.to.sampling)
colData(ddsCLL)$time_diag <- pretreat$norm_time_diag 


```


```{r deseq}
###Deseq
ddsCLL <- estimateSizeFactors(ddsCLL)

ddsCLL <- ddsCLL[,!is.na(ddsCLL$time_diag)]
design(ddsCLL) <- as.formula(paste("~ time_diag"))
rnaRaw <- DESeq(ddsCLL, betaPrior = FALSE)
res <- results(rnaRaw)
resOrdered <- res[order(res$pvalue),]

```

Diff genes
```{r create csv of diff genes}

pCut <- 0.05

dataTab <- data.frame(resOrdered)
dataTab$ID <- rownames(dataTab)
#filter using pvalues
dataTab <- filter(dataTab, padj <= pCut) %>% 
  arrange(padj) %>%
  mutate(Symbol = rowData(ddsCLL[ID,])$symbol) 

dataTab <- dataTab[!duplicated(dataTab$Symbol),]
dataTab <- dataTab[!is.na(dataTab$Symbol),]
rownames(dataTab) <- dataTab$ID
write.csv(dataTab, file=paste0(output_dir,"/diff_genes/diagnosis_time_to_sampling_diffGenes.csv"))

```


## MA-plot
```{r ma-plot}

DESeq2::plotMA(res, ylim=c(-0.5,0.5), main= "time_diagnosis_to_sampling")

hist(res$pvalue, breaks=100, col="skyblue", border="slateblue",
                     main=paste0("Histogramm of pvalues: Time from diagnosis to sampling"),plot = TRUE)
```



## Variance explained
```{r var exp}
expr <- as.matrix(assays(ddsCLL)$counts)
meta_sub <- as.data.frame(colData(ddsCLL)[, c("time_diag", "IGHV")])
form <- as.formula(paste0("~ time_diag + (1|IGHV)"))
varPart <- fitExtractVarPartModel(expr, form, meta_sub, BPPARAM=MulticoreParam(4))

```

Plott variance partitioning
```{r var part}
type_data
t1 <- ggtern(data=type_data, aes(vp_time_dias, vp_IGHV, vp_res)) +
  guides(color = guide_legend(override.aes = list(size = 3, alpha = 1))) +
  geom_point(aes(colour = dataset),size= 1.6, alpha = 0.9)  +
  #scale_colour_manual(values = cols_data) +
  #stat_density_tern(aes(fill=..level.., alpha=..level..),geom='polygon',bdl = 0.01,bdl.val = 1, base = "identity", n = 20) +
  geom_density_tern(
        bins=10,
        bdl = 0.02,
        color='black',
        base = "identity", n = 20,
        bdl.val = 0.1) +
  Llab("cell\n \t type") +
  Tlab("batch") +
  Rlab("int.") +
  ggtitle("Percent variance explained by ...:") +
t1
```

