---
title: "epistasis_methylation"
author: "almut"
date: "11 Juli 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Aim: Do we find epistatis pattern between trisomy12 and IGHV in methylation data?

load packages
```{r libraray, message=FALSE, warning=FALSE}
library(tidyverse)
library(ggpubr)
library(DESeq2)
library(limma)
library(RColorBrewer)
library(ComplexHeatmap)
library(ggpubr)
library(circlize)
library(here)
#library("BloodCancerMultiOmics2017")
```

load datasets
```{r data}

output_dir <- here("output")
figure_dir <- here("output/figures")

load("/home/almut/Dokumente/masterarbeit/data/patmeta.RData")
load("/home/almut/Dokumente/masterarbeit/data/mutCOM.RData")
load("/home/almut/Dokumente/masterarbeit/data/methData.RData")

```

filter CLL patients with IGHV and trisomy12 info
```{r prepare data sets}
patmeta$PatID <- rownames(patmeta)
patCLL <- patmeta %>% filter(Diagnosis %in% "CLL") %>% filter(!is.na(IGHV))
mutations <-data.frame(assayData(mutCOM)$binary)
mutations$PatID <- rownames(mutations)
mutations_sel <- mutations %>% filter(!is.na(trisomy12)) %>% select(trisomy12,PatID)

metaData <- inner_join(patCLL, mutations_sel)
meth <- assay(methData)
metaData <- metaData %>% filter(PatID %in% colnames(meth)) %>% mutate(IGHV_status=ifelse(IGHV %in% "M", 1, 0))

methCLL <- meth[, metaData$PatID]

```

#Model epistatic interaction using limma
Differentiate into 4 groups
```{r groups}
mutnames <- c("none", "IGHV-M", "trisomy12", "both")
metaGroups <- metaData %>% mutate(mutStatus = factor(mutnames[1 + IGHV_status + 2 * trisomy12], levels = mutnames)) %>% select(mutStatus, trisomy12, IGHV_status, PatID)
                                    
```


Identify significant methylation sites
```{r}
x <- tibble(varA = factor(metaData$trisomy12),
            varB  = factor(metaData$IGHV_status))
 
fit <- eBayes(lmFit(methCLL, design = model.matrix( ~ varA + varB + varA * varB, data = x))) 
interaction <- tibble(methID = rownames(fit$p.value), 
                      p = fit$p.value[, "varA1:varB1"]) %>% mutate(pBH = p.adjust(p, method = "BH")) %>% arrange(p)

sig_interaction <- interaction %>% filter(p <= 0.01)
```


#Plot significant methylation sites
Heatmap significant meth sites: order columns by variant status (both, varA, varB, none)

```{r heatmap of interaction ighv, fig.width=15, fig.height=15}

#methylation data
 sig_meth = methCLL[sig_interaction$methID,]

#annotate genes
sig_genes <- rowData(methData[sig_interaction$methID,])$UCSC_RefGene_Name
sig_genes <- gsub('[A-z0-9]*\\;', '', sig_genes)

sig_gen <- data.frame(genNam =sig_genes,
                     genID = sig_interaction$methID)
sig_genes <- sig_genes[-which(sig_genes %in% "")]

sig_gen <- as.tibble(sig_gen) %>% filter(genNam %in% sig_genes)

#order by mutation groups  
mutStatus <- metaGroups %>% arrange(mutStatus)
sig_meth <- sig_meth[,mutStatus$PatID]


#colors
#colors <- colorRampPalette( rev(brewer.pal(11,"RdBu")) )(255)
colors = colorRamp2(c(0,0.1,0.5,0.9,1), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
annocol <- get_palette("uchicago", 9)
annocolor <- list(Variant = c(annocol[3], annocol[5], annocol[7], annocol[9]))
names(annocolor$Variant) <- c("none", "IGHV-M", "trisomy12", "both")

#column annotation  
mutationStatus <- data.frame(mutStatus$mut)
rownames(mutationStatus) <- mutStatus$PatID
colnames(mutationStatus) <- "Variant"

ha_col = HeatmapAnnotation(df = mutationStatus, col = annocolor, 
                           simple_anno_size = unit(0.8, "cm"),
                           annotation_name_gp = gpar(fontsize = 25),
                            annotation_legend_param = list(title_gp = gpar(fontsize = 23), 
                                                           labels_gp = gpar(fontsize = 18),  
                                                           grid_height = unit(1, "cm"), 
                                                           grid_width = unit(0.5, "cm"),
                                                           gap = unit(0.5, "cm")))

  #rowcluster
  meth_dist <- dist(sig_meth)
  rowcluster = hclust(meth_dist, method = "ward.D2")

  #heatmap
  h1 <- Heatmap(sig_meth, col = colors ,
                column_title = paste0("Methylation interactions:", "IGHV", "-", "trisomy12"), 
                column_title_gp = gpar(fontsize = 25, fontface = "bold"), 
                heatmap_legend_param = list(title = "expr", 
                                          title_gp = gpar(fontsize = 23), 
                                          grid_height = unit(1, "cm"), 
                                          grid_width = unit(0.5, "cm"), 
                                          gap = unit(2, "cm"), 
                                          labels_gp = gpar(fontsize = 18)), 
                row_dend_width = unit(0.5, "cm"), 
                show_row_dend = F, 
                show_column_names =FALSE,
                show_row_names =FALSE, 
                top_annotation = ha_col,
                show_column_dend = FALSE, 
                cluster_columns = FALSE, 
                cluster_rows = rowcluster,
                split = 3, 
                gap = unit(0.2,"cm"), 
                column_order = mutStatus$PatID)


geneIDs <- which(rownames(sig_meth) %in% sig_gen$genID)
labels <- sig_gen$genNam
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 45)), width = unit(9, "cm"))


  #svg(filename="~/git/figures_thesis/gene_expr/epistatsisTri12IGHV.svg", width=30, height=50)
  #pdf(file="/home/almut/Dokumente/git/Transcriptome_CLL/paper/figures/epistasis_methylation.pdf", width=35, height=45)
  draw(h1) 
  #dev.off()

 IGHVTri12 <- list("sig_meth" = rownames(sig_meth), "MethExp" = sig_meth, "h1"= h1)
saveRDS(h1, file = paste0(output_dir, "/figures/r_objects/epistasis/epi_methylation_heatmap.rds"))
```


```{r overlap with epistasis in gene expression}
#load differential data freom epistasis_Deseq_IGHV_tri12. Rmd
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/epistaticGenes_Deseq2_IGHTri12.RData")
resOrdered <- res[order(res$pvalue),]
resSig <- subset(resOrdered, padj < 0.1)
resTab <- as.data.frame(resSig)
#count data for annotation
load("/home/almut/Dokumente/git/Transcriptome_CLL/dataset/ddsrnaCLL_150218.RData")
resTab$symbol <- rowData(ddsCLL[rownames(resSig),])$symbol
sig_expr <- resTab$symbol

#sig_meth data
sig_genes <- gsub('[A-z0-9]*\\;', '', sig_genes)

overlap <- as.tibble(sig_expr) %>% filter(!value %in% "")  %>% filter(value %in% sig_genes)



#function to create stripchart plots for specific genes
meth_count <- function(gene_nam){
  meth_loc <- sig_gen$genID[which(sig_gen$genNam %in% gene_nam)]
  gc <- data.frame("intensity" = sig_meth[meth_loc,], 
                   "variant" = mutationStatus)
  p <- ggboxplot(gc, x = "Variant", y = "intensity",
          color = "Variant",
          size = 1.2,
          palette = c(annocol[3], annocol[5], annocol[7], annocol[9]),
          outlier.shape = NA, 
          add = "jitter",
          add.params = list(size = 2.5),
          title = paste(gene_nam),
          font.x = 20, font.y = 20, font.legend = 20, 
          ylab = "normalized counts") + font("xy.text", size = 20) + font("title", size = 20, face = "bold")
  saveRDS(p, file = paste0(output_dir, "/figures/r_objects/epi_meth/de_genes/", gene_nam, ".rds"))
  p
}


lapply(overlap[[1]], meth_count)

```

