---
title: "Methylation IP vs rest"
author: "almut"
date: "07 October 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Methylation groups
Analyse genes associated with methylation groups

libraries
```{r load libraries, warning=F, message=FALSE}
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(ggpubr)
library(ComplexHeatmap)
library(RColorBrewer)
library(circlize)
library(here)
library(piano)
library(clusterProfiler)
library(msigdbr)
library(org.Hs.eg.db)
library(enrichplot)
```

# Data preprocessing

data
```{r load data}
data_dir <- here("data")
output_dir <- here("output")
figure_dir <- here("output/figures")

#dds data set. gene expression data + patmetadata
load(paste0(data_dir, "/ddsrnaCLL_150218.RData"))

#arrange columns
mutStatus <- data.frame(colData(ddsCLL)) %>% arrange(Methylation) %>% filter(!is.na(Methylation)) %>% mutate("IP" = ifelse(Methylation %in% c("LP", "HP"), 0, 1))
colnames(ddsCLL) <-colData(ddsCLL)$PatID
ddsCLL <- ddsCLL[, mutStatus$PatID]
colData(ddsCLL)$IP <- mutStatus$IP
table(colData(ddsCLL)$IP)
```

Normalize
```{r RNAnorm}
#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)

```

# Exploratory data analysis

Clustering on the most variable genes already split methylation groups. How do methylation groups affect highly variable genes? 
Can we distinguish all 3 groups?

PCA
```{r pca, fig.height=6, fig.width=6}
#Plot PCA
exprMat <- assay(RNAnorm)

#top 5000 most variant genes
sds <- rowSds(exprMat)
exprMat <- exprMat[order(sds, decreasing = T)[1:150],]


#Calculate PCA
pcaRes <- prcomp(t(exprMat), scale =T)
varExp <- (pcaRes$sdev^2 / sum(pcaRes$sdev^2)) * 100
pcaTab <- data.frame(pcaRes$x[,c(1:10)])
names(varExp) <- colnames(pcaRes$x)

#add background information
pcaTab <- cbind(pcaTab, data.frame(colData(RNAnorm)))


#plot PCA and color samples based on annotations
annocol <- get_palette("jco", 10)
  
p <- ggscatter(pcaTab, x = "PC1", y = "PC2", color = "Methylation", palette = c( annocol[7], annocol[5], annocol[6]),
  ylab = sprintf("PC2 (%2.1f%%)",varExp[2]), xlab = sprintf("PC1 (%2.1f%%)",varExp[1]), legend = "right", main = "PCA Methylation groups") + coord_fixed()

p

#ggsave(file=paste0(figure_dir, "/pca_Meth_IP_all_top150.svg"), plot=p, width=5, height=5)
```

# Gene expression

##Differential expression analysis

deseq2
```{r deseq2, eval = FALSE}

###Deseq
ddsCLL <- estimateSizeFactors(ddsCLL)

# deseq2 function
ddsCLL <- ddsCLL[,!is.na(ddsCLL$Methylation)]
ddsCLL$Methylation <- relevel(ddsCLL$Methylation, ref = "IP")
design(ddsCLL) <- as.formula(paste("~ Methylation"))
rnaRaw <- DESeq(ddsCLL, betaPrior = FALSE)
saveRDS(rnaRaw, file= paste0(output_dir,"/diff_meth_IP_vs_HP_LP.rds"))
rnaRaw <- readRDS(paste0(output_dir,"/diff_meth_IP_vs_HP_LP.rds"))

res_hp <- results(rnaRaw, name="Methylation_HP_vs_IP")
res_hp <- res_hp[order(res_hp$pvalue),]
res_lp <- results(rnaRaw, name="Methylation_LP_vs_IP")
res_lp <- res_lp[order(res_lp$pvalue),]

res_lp_hp <- results(rnaRaw, c("Methylation", "LP", "HP"))

```


Filter differentially expressed genes
```{r diff genes}
rearrange_res <- function(res_mat){
  dataTab <- data.frame(res_mat)
  dataTab$ID <- rownames(dataTab)

  #filter using pvalues
  dataTab <- dataTab %>% arrange(padj) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  rownames(dataTab) <- dataTab$ID
  dataTab
}

tab_hp <- rearrange_res(res_hp)
tab_lp <- rearrange_res(res_lp)
tab_lp_hp <- rearrange_res(res_lp_hp)

#write.csv(dataTab, file=paste0(output_dir, "/diff_genes/meth_IP_vs_all_diffGenes.csv"))
write.csv(tab_hp, file=paste0(output_dir, "/diff_genes/meth_IP_vs_HP_diffGenes.csv"))
write.csv(tab_lp, file=paste0(output_dir, "/diff_genes/meth_IP_vs_LP_diffGenes.csv"))


#Get differentially expressed genes
genes_hp <- tab_hp %>% filter(padj <= 0.01, abs(stat) > 5)
genes_hp_sig <- tab_hp %>% filter(padj <= 0.05)
genes_lp <- tab_lp %>% filter(padj <= 0.01, abs(stat) > 5)
genes_lp_sig <- tab_lp %>% filter(padj <= 0.05)
genes_lp_hp <- tab_lp_hp %>% filter(padj <= 0.01) 
dim(genes_hp_sig)
dim(genes_lp_sig)


#Exclude genes differentially expressed between hp and lp
both <- c(genes_hp_sig$ID, genes_lp_sig$ID) 
genes_diff_sig <- both[!both %in% genes_lp_hp$ID]
length(genes_diff_sig)

both <- c(genes_hp$ID, genes_lp$ID) 
genes_diff <- both[!both %in% genes_lp_hp$ID]
length(genes_diff)

genes_hp <- genes_hp[genes_hp$ID %in% genes_diff, ]
genes_lp <- genes_lp[genes_lp$ID %in% genes_diff, ]

genes <- rbind(genes_hp, genes_lp)
genes <- genes[order(genes$padj),]
```


## Expression heatmap

```{r complex heatmap, fig.width= 15, fig.height=14}
#Differentially expressed genes
exprMat <- assay(RNAnorm)
exprMat<- exprMat[unique(genes$ID),]

#scale gene expression
colnames(exprMat) <- colData(ddsCLL)$PatID
exprMat.new <- log2(exprMat)
exprMat.new <- t(scale(t(exprMat.new)))
exprMat.new[exprMat.new > 4] <- 4
exprMat.new[exprMat.new < -4] <- -4

#colors
colors = colorRamp2(c(-4,-1,0,1,4), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
annocol <- get_palette("jco", 10)
annocolor <- list(Methylation = c("IP" = annocol[5], "LP" = annocol[6], "HP" =  annocol[7]), IGHV = c("M" = annocol[1], "U" = annocol[2])) 

#Annotation
feature <- as.data.frame(colData(ddsCLL)[,c("Methylation", "IGHV")]) 
colnames(feature) <- c("Methylation", "IGHV") 

#gene symbol as rownames
rownames(exprMat.new) <- rowData(RNAnorm[rownames(exprMat),])$symbol

ha_col <- HeatmapAnnotation(df=feature, col = annocolor, annotation_height = unit(c(rep(1.9, 1)), "cm"), 
                           simple_anno_size = unit(0.7, "cm"),
                           annotation_name_gp = gpar(fontsize = 30),
                            annotation_legend_param = list(title_gp = gpar(fontsize = 23), 
                                                           labels_gp = gpar(fontsize = 18),  
                                                           grid_height = unit(0.7, "cm"), 
                                                           grid_width = unit(0.3, "cm"),
                                                           gap = unit(1, "cm")))

#Annotate top 40 genes
sub_names <- unique(genes[,"Symbol"])[1:40]
sub_names <- sub_names[-which(sub_names %in% "")]
geneIDs <- which(rownames(exprMat.new) %in% sub_names)
rownames(exprMat.new)[-geneIDs] <- ""


h1 <- Heatmap(exprMat.new ,                                                     
              km = 2,
              gap = unit(0.5, "cm"),
              cluster_columns = F,
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = paste0("Gene signature: IP group"),                      
              col = colors,
              column_title_gp = gpar(fontsize = 25, fontface = "bold"), 
              heatmap_legend_param = list(title = "expr", 
                                          title_gp = gpar(fontsize = 23), 
                                          grid_height = unit(0.7, "cm"), 
                                          grid_width = unit(0.3, "cm"), 
                                          gap = unit(1, "cm"), 
                                          labels_gp = gpar(fontsize = 18)), 
              column_dend_height = unit(1, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = TRUE, 
              row_names_gp = gpar(fontsize = 17),
              top_annotation = ha_col)



#labels <- rownames(exprMat.new)[geneIDs]
#ha_genes <- rowAnnotation(link = anno_mark(at = geneIDs, which = "row", labels = labels, labels_gp = gpar(fontsize = 15)))

#svg(filename=paste0(figure_dir, "/gene_expr_Methylation_IP_all.svg"), width=30, height=35)
#pdf(file=paste0(figure_dir,"/gene_expr_Methylation_IP_all.pdf"), width=30, height=45)
p1 <- draw( h1) # + ha_genes)
#dev.off()

saveRDS(p1, file = paste0(output_dir, "/figures/r_objects/Methylation_IP/meth_ip_heatmap.rds"))


```


# Gene and Sample specific expression - top genes
```{r single gene counts, message=FALSE, fig.width=8, fig.height=6}
#function to create stripchart plots for specific genes
gene_count <- function(gene_nam){
  geneEnsID <- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% gene_nam)]
  geneNum <- exprMat[geneEnsID,]
  mutPat <- as.data.frame(colData(ddsCLL)[, c("Methylation")])
  colnames(mutPat) <- c("genotype")
  geneDat <- cbind(mutPat, geneNum)
  colnames(geneDat) <- c("genotype", "counts")
  
  p <- ggstripchart(geneDat, x = "genotype", y = "counts",
          color = "genotype",
          size = 3,
          palette = c(annocol[5], annocol[7], annocol[6]),
          add = "mean_sd",
          add.params = list(size = 1.2),
          title = paste(gene_nam),
          font.x = 20, font.y = 20, font.legend = 20, 
          ylab = "normalized counts") + font("xy.text", size = 20) + font("title", size = 20, face = "bold")
 # ggsave(file=paste0(figure_dir, "/methylation_IP_all/genetic_interaction_", gene_nam, ".svg"), plot=p, width=6, height=5)
  saveRDS(p, file = paste0(output_dir, "/figures/r_objects/Methylation_IP/de_genes/", gene_nam, ".rds"))
  p
}

geneList <- genes %>% filter(abs(log2FoldChange) > 2) %>% select(Symbol) %>% unique() 
geneList <- geneList$Symbol[-which(geneList$Symbol %in% "")]

lapply(geneList, gene_count)

```


## Gene set enrichment analysis

Gene sets
```{r gene sets}
#load gene set collection
#Hallmark
gsc <- loadGSC("/home/almut/Dokumente/masterarbeit/data/h.all.v6.0.symbols.gmt", type="gmt")
#Kegg
gsc_Kegg <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")


#clusterProfiler

#Get Gene IDs
gene_id <- bitr(genes_diff_sig, fromType = "ENSEMBL",
        toType = c("ENTREZID", "SYMBOL"),
        OrgDb = org.Hs.eg.db)

names(gene_id) <- c("ID", "ENTREZID", "Symbol")


#convert gsc
m_t2g <- msigdbr(species = "Homo sapiens", category = "H") %>% 
  dplyr::select(gs_name, human_gene_symbol)

#Hallmark
em <- enricher(gene_id$Symbol, TERM2GENE = m_t2g)

#Kegg
kk <- enrichKEGG(gene_id$ENTREZID,
                 organism     = 'hsa',
                 pvalueCutoff = 0.1)


barplot(kk, showCategory=5)
barplot(em, showCategory=5)

dotplot(em, showCategory=10) + ggtitle("Enrichment for IP group")

dot2 <- dotplot(kk, showCategory=10) + ggtitle("Enrichment for trisomy12") +
  theme_pubr() +
  theme(legend.position="right") +
  theme(plot.title = element_text(face = "bold"))
dot2



```


```{r gsea, eval=FALSE, fig.height=7, fig.width=16, include=FALSE}

variant <- "IP"
gmtFile <- loadGSC(paste0(data_dir,"/c2.cp.kegg.v6.0.symbols.gmt"), type="gmt")

diff_res <- dataTab
diff_res  <- diff_res[-which(diff_res$Symbol %in% c("", NA)),]
#get genes and pvalues 
geneNam <- diff_res$Symbol 
pVal <- diff_res$padj
logFold <- diff_res$log2FoldChange
stat <- diff_res$stat
gsTab <- data.frame(gene = geneNam, stat = stat)
gsaTab <- data.frame(row.names = gsTab$gene, stat = gsTab$stat)
res <- runGSA(geneLevelStats = gsaTab,
                      geneSetStat = "gsea",
                     adjMethod = "fdr", gsc=gmtFile,
                     signifMethod = "geneSampling",
                      nPerm = 50000,
                      gsSizeLim=c(1, Inf))
Res_up <- arrange(GSAsummaryTable(res), `p adj (dist.dir.up)`)                 
Res_dn <- arrange(GSAsummaryTable(res), `p adj (dist.dir.dn)`)
  

#Plot
resPlot <- Res_dn[, c(1:3,7,8,9)]
colnames(resPlot) <- c("pathway", "gene_number", "stat", "p.adj","genes_up" , "genes_dn")


enrichPlot <- resPlot %>% filter(p.adj < 0.1) %>% mutate(log10Padj = -log10(p.adj)) #%>% mutate(genes = ifelse(gene_number > 5, ">5", "<=5"))
enrichPlot$log10Padj[which(enrichPlot$log10Padj == Inf)] <- 5

p <- ggbarplot(enrichPlot, x = "pathway", y = "log10Padj",
          fill = "gene_number",         
          color = "white",           
          palette =  "gsea",            
          sort.val = "asc",          
          sort.by.groups = FALSE,     
          ylab = "-log10(padj)",
          legend.title = "#diff.genes",
          rotate = TRUE,
          font.x = 20, font.y = 20, font.legend = 20, legend = "right", 
          title = "Methylation groups - Kegg",
          ggtheme = theme_pubr()) +
  font("xy.text", size = 16) + 
  font("title", size = 20, face = "bold")
  
#ggsave(file=paste0(figure_dir, "/GSEA_Meth_IP_vs_all_Kegg.svg"), plot=p, width=14, height=7)

p


saveRDS(p, file = paste0(output_dir, "/figures/r_objects/Methylation_IP/meth_ip_enrichment.rds"))
```

