---
title: "Summary DE genes"
author: "almut"
date: "14 November 2019"
output: html_document
---
# Summary DE genes by variants

Aim: Generate a summary plot giving an overview on the number of differntially expressed genes / number upreagulated / number downregulated genes and chromosomal distribution for CNVs.


libraries
```{r load libraries , warning=F, message=FALSE}
suppressPackageStartupMessages({
  library(tidyverse)
  library(ggplot2)
  library(here)
  library(ggpubr)
  library(ggsci)
  library(RColorBrewer)
  library(circlize)
  library(reshape2)
  library(ggsci)
  library(DESeq2)
})
```

data
```{r data set}
data_dir <- here("data")
output_dir <- here("output")
figure_dir <- here("output/figures")

#dds data set. gene expression data + patmetadata
load(paste0(data_dir, "/ddsrnaCLL_150218.RData"))

#load meta data including genotyping info
load(paste0(data_dir, "/patmeta_170324.RData"))
excluded_columns <- c("HIPO.ID","PID", "gender","project", "diagnosis", "date.of.diagnosis", "treatment", "date.of.first.treatment", "IGHV.status","Methylation_Cluster")

patMeta <- as.tibble(patMeta) %>% filter(Patient.ID %in% ddsCLL$PatID) %>% dplyr::select(-one_of(excluded_columns))

variants <- patMeta %>% dplyr::select(-Patient.ID) %>% 
  mutate_if(is.factor, as.character) %>% 
  mutate_if(is.character, as.numeric) %>% 
  dplyr::select(colnames(.)[colSums(.,na.rm = TRUE) > 4]) %>% 
  colnames()

var_add <- variants[!variants %in% names(colData(ddsCLL))] 
rownames(patMeta) <-  patMeta$Patient.ID
patMeta <- patMeta[colData(ddsCLL)$PatID,]
cd <- cbind(colData(ddsCLL), patMeta[,var_add])
colData(ddsCLL) <- cd

#variants <- c( "trisomy12", "del13q14", "del8p12", "gain8q24", "del11q22.3", "del17p13", "BRAF", "NOTCH1", "SF3B1","TP53", "ATM", "MED12","IGHV")

variants <- c(variants, "IGHV")
```

Summarize de genes in a table
```{r summary table}
#Function to summarize de genes
variant_stats <- function(variant, chr){
   diff_genes <-read.csv(file=paste0(output_dir, "/diff_genes/", variant, "_diffGenes.csv"))
  sig_genes <- as.tibble(diff_genes) %>% filter(padj < 0.01 & abs(log2FoldChange) > 0) 
  up_genes <- sig_genes %>% filter(log2FoldChange > 0) %>% mutate(chrom = rowData(ddsCLL)$chromosome[which(rownames(ddsCLL) %in% X)])
  dn_genes <- sig_genes %>% filter(log2FoldChange < 0) %>% mutate(chrom = rowData(ddsCLL)$chromosome[which(rownames(ddsCLL) %in% X)])
  chr_up <- ifelse(!is.na(chr), length(which(up_genes$chrom %in% chr)), 0)
  chr_dn <- ifelse(!is.na(chr), length(which(dn_genes$chrom %in% chr)), 0)
  
  summary <- tibble(
    variant = variant,
    n_genes = nrow(sig_genes),
    up_genes = nrow(up_genes),
    dn_genes = nrow(dn_genes),
    nr_up = chr_up,
    nr_dn = chr_dn,
    nr_chrom = chr_dn + chr_up,
    nr_other = n_genes - chr_dn - chr_up
  )
  }

# Variant chromosomal location
chromo <- c( "2", "8", "8", "11", "12", "13", "15", "17", NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA)
input <- cbind(variants, chromo)

summary <- mapply(variant_stats, variants, chromo)
sum_dat <- data.frame(matrix(unlist(summary), nrow=length(variants), byrow=T),stringsAsFactors=FALSE)
colnames(sum_dat) <- c("variant", "n_genes", "up_genes", "dn_genes", "nr_up", "nr_dn", "nr_chrom", "nr_other")
sum_dat <- sum_dat %>% filter(as.numeric(n_genes) > 100)
sum_order <- as.tibble(sum_dat) %>% arrange(desc(-1*as.numeric(n_genes))) %>% dplyr::select(variant)

#sum_dat_long <- melt(sum_dat[,c("variant","up_genes", "dn_genes")], id.vars = c("variant")) #, variable_name = c("up_genes", "dn_genes"))

sum_dat_long <- melt(sum_dat[,c("variant","nr_other", "nr_chrom")], id.vars = c("variant")) #, variable_name = c("up_genes", "dn_genes"))

colnames(sum_dat_long) <- c("variant", "location", "genes")
sum_dat_long$dir_color <-paste0(sum_dat_long$variant, "_", sum_dat_long$location)

sum_dat_long$genes <- as.numeric(as.character(sum_dat_long$genes))
sum_dat_long <- as.tibble(sum_dat_long) %>% arrange(desc(genes)) %>% mutate(group=ifelse(variant %in% c("trisomy12", "IGHV"), 1, 2))

```

## Plots

### Summary all variants
```{r barplot, fig.height= 5, fig.width=8}
col_pal1 <- pal_igv(palette = c("default"), alpha = 1)(nrow(sum_dat))
col_pal2 <- pal_igv(palette = c("default"), alpha = 0.4)(nrow(sum_dat))

color <- c(col_pal1, col_pal2)
color <- sort(color)

#pdf(file=paste0(figure_dir, "/sum_diffGenes_0.05_2.pdf"), width=8, height=5)
ggbarplot(sum_dat_long[order(match(sum_dat_long$variant, sum_order$variant)),], "variant", "genes",
  fill = "dir_color", palette = color,
  label = FALSE,  font.x =23 ,font.y = 23, font.tickslab = 18, x.text.angle = 90, legend = "none") 
#dev.off()

```


### IGHV and trisomy12
```{r barplot ighv, fig.height= 2.3, fig.width=6}

sum_dat_long1 <- as.tibble(sum_dat_long) %>% filter(variant %in% c("trisomy12", "IGHV"))

#pdf(file=paste0(figure_dir, "/sum_diffGenes_noTsig_IGHVTri12.pdf"), width=5, height=2.3)
p1 <- ggbarplot(sum_dat_long1[order(match(sum_dat_long1$variant, sum_order$variant)),], "variant", "genes",
  fill = "dir_color", palette = color[c(1:4)],
  label = FALSE,  font.x =23 ,font.y = 23, font.tickslab = 18, x.text.angle = 0, legend = "none", rotate = T) 
#dev.off()

p1

saveRDS(p1, file = paste0(output_dir, "/figures/r_objects/summary_de_genes_IGHV_tri12.rds"))
```


### All other
```{r barplot others, fig.height= 5.5, fig.width=6}

sum_dat_long2 <- as.tibble(sum_dat_long) %>% filter(!variant %in% c("trisomy12", "IGHV")) %>% filter

#pdf(file=paste0(figure_dir, "/sum_diffGenes_noTsig.pdf"), width=6, height=7.5)
p2 <- ggbarplot(sum_dat_long2[order(match(sum_dat_long2$variant, sum_order$variant)),], "variant", "genes",
  fill = "dir_color", palette = color[c(5:50)],
  label = FALSE,  font.x =23 ,font.y = 23, font.tickslab = 18, x.text.angle = 0, legend = "none", rotate = T) 
#dev.off()


p2

saveRDS(p2, file = paste0(output_dir, "/figures/r_objects/summary_de_genes_all.rds"))
```

## summary table
```{r table}
#pdf(file=paste0(output_dir,"/sum_diffGenes_table.pdf"), width=5, height=7.5)
sum_dat
#dev.off()

```
