---
title: "Trisomy12"
author: "Almut Luetge"
date: "September 6, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Trisomy12 signature

# Differentially expressed genes

### 1. Differential expression analysis
load packages
```{r packages, warning=FALSE, message=FALSE}
library(DESeq2)
library(tidyverse)
library(ggsci)
library(matrixStats)
library(piano)
library(reshape2)
library(genefilter)
library(Biobase)
library(ComplexHeatmap)
library(ggplot2)
library(gtable)
library(grid)
library(circlize)
library(gridExtra)
library(ggpubr)
library(RColorBrewer)
library(here)
library(clusterProfiler)
library(msigdbr)
library(org.Hs.eg.db)
library(enrichplot)
```


load data
```{r datasets}
data_dir <- here("data")
output_dir <- here("output")
figure_dir <- here("output/figures")

#dds data set. gene expression data + patmetadata
load(paste0(data_dir, "/ddsrnaCLL_150218.RData"))

variant <- "trisomy12"
#filter for patients without NA in variant
ddsCLL <- ddsCLL[, !is.na(colData(ddsCLL)[,variant])]

#differentially expressed genes between Trisomy12 groups (see differential expression.html)
diff_all <- read.csv(file=paste0(output_dir, "/diff_genes/", variant, "_diffGenes.csv"))


rownames(diff_all) <- diff_all$X
diff_all <- diff_all[which(diff_all$padj < 0.01 ),-1]
diff <- diff_all[which(abs(diff_all$stat) > 8),]


mutStatus <- data.frame(colData(ddsCLL)) %>% arrange(trisomy12)
colnames(ddsCLL) <-colData(ddsCLL)$PatID
ddsCLL <- ddsCLL[, mutStatus$PatID]

#expression data
ddsCLL <- estimateSizeFactors(ddsCLL)
RNAnorm <- varianceStabilizingTransformation(ddsCLL, blind = T)

```


Expression matrix
```{r var gene expression}

#filter for sign. genes in variant
exprMat <- assay(RNAnorm)
exprVariant <- exprMat[rownames(diff),]
colnames(exprVariant) <- colData(ddsCLL)$PatID
exprVariant.new <- log2(exprVariant)
exprVariant.new <- t(scale(t(exprVariant.new)))
exprVariant.new[exprVariant.new > 4] <- 4
exprVariant.new[exprVariant.new < -4] <- -4
rownames(exprVariant.new) <- rowData(RNAnorm[rownames(diff),])$symbol

```


## Expression signature
```{r complex heatmap, fig.width= 15, fig.height=15}

#colors
colors = colorRamp2(c(-4,-2,0,2,4), c("#2166ac","#4393c3", "#f7f7f7", "#d6604d","#b2182b"))
annocol <- get_palette("jco", 10)
annocolor <- list(trisomy12 = c("1" = annocol[8], "0" = annocol[4]))
rowcolors <-colorRampPalette(brewer.pal(5, "Set1"))(5)
rowcolors[6] <- "white"


feature <- as.data.frame(colData(ddsCLL)[,c(variant)])
colnames(feature) <- c(variant)  

ha_col <- HeatmapAnnotation(df=feature, col = annocolor, annotation_height = unit(c(rep(1.9, 1)), "cm"), 
                           simple_anno_size = unit(0.7, "cm"),
                           annotation_name_gp = gpar(fontsize = 25),
                            annotation_legend_param = list(title_gp = gpar(fontsize = 23), 
                                                           labels_gp = gpar(fontsize = 18),  
                                                           grid_height = unit(0.7, "cm"), 
                                                           grid_width = unit(0.3, "cm"),
                                                           gap = unit(0.5, "cm")))

#Rowannotation
#chromosome distribution
chrom <- as.data.frame(rowData(RNAnorm[rownames(diff),])$chromosome)
rownames(chrom) <- rownames(diff) 
colnames(chrom ) <- "chr"

#select for chromosome12
chrom$chr <- ifelse(chrom$chr == 12, 12, "other")
chromcolor <- pal_jama("default")(7)
chromcol <- list(chr = c("12" = chromcolor[1], "other" = chromcolor[5]))

ha_chrom = rowAnnotation(df = chrom, col = chromcol, annotation_width = unit(0.7, "cm"), 
                         show_annotation_name = FALSE,
                         simple_anno_size = unit(0.7, "cm"),
                         annotation_legend_param = list(ncol = 1, title_gp = gpar(fontsize = 23), 
                                                        labels_gp = gpar(fontsize = 18),  
                                                        grid_height = unit(0.7, "cm"), 
                                                        grid_width = unit(0.3, "cm"), 
                                                        gap = unit(0.5,"cm")))

h1 <- Heatmap(exprVariant.new ,                                                     
              km = 2,
              gap = unit(0.5, "cm"),
              cluster_columns = F,
              clustering_distance_rows = "pearson",
              clustering_method_rows = "ward.D2",
              column_title = paste0("Gene signature: ", variant),                      
              col = colors,
              column_title_gp = gpar(fontsize = 25, fontface = "bold"), 
              heatmap_legend_param = list(title = "expr", 
                                          title_gp = gpar(fontsize = 23), 
                                          grid_height = unit(0.7, "cm"), 
                                          grid_width = unit(0.3, "cm"), 
                                          gap = unit(2, "cm"), 
                                          labels_gp = gpar(fontsize = 18)), 
              column_dend_height = unit(1, "cm"),
              show_row_dend = FALSE, 
              show_column_names = FALSE , 
              show_row_names = FALSE, 
              row_names_gp = gpar(fontsize = 0),
              top_annotation = ha_col)

#Annotate top 50 genes
sub_names <- diff[1:50,"Symbol"]
sub_names <- sub_names[-which(sub_names %in% "")]
geneIDs <- which(rownames(exprVariant.new) %in% sub_names)
labels <- rownames(exprVariant.new)[geneIDs]
ha_genes <- rowAnnotation(link = row_anno_link(at = geneIDs, labels = labels, labels_gp = gpar(fontsize = 15)), width = unit(3, "cm"))

#svg(filename=paste0(figure_dir, "/", variant, "_gene_expr.svg"), width=30, height=45)
#pdf(file=paste0(figure_dir, "/", variant, "_gene_expr.pdf"), width=22, height=25)

p1 <- draw(h1 + ha_chrom + ha_genes) 
#dev.off()
saveRDS(p1, file = paste0(output_dir, "/figures/r_objects/trisomy12/tri12_heatmap.rds"))
```


# Sample and gene specific expression -  top genes

```{r single gene counts, fig.width=6, fig.height=5}
#function to create stripchart plots for specific genes
gene_count <- function(gene_nam){
  geneEnsID <- rownames(ddsCLL)[which(rowData(ddsCLL)$symbol %in% gene_nam)]
  gc <- plotCounts(ddsCLL, gene = geneEnsID, intgroup = variant, returnData=TRUE)
  p <- ggboxplot(gc, x = variant, y = "count",
          color = variant,
          size = 1.2,
          palette = c(annocol[4],annocol[8]),
          outlier.shape = NA, 
          add = "jitter",
          add.params = list(size = 2.5),
          yscale = "log10",
          title = paste(gene_nam),
          font.x = 20, font.y = 20, font.legend = 20, 
          ylab = "normalized counts") + font("xy.text", size = 20) + font("title", size = 20, face = "bold")
  saveRDS(p, file = paste0(output_dir, "/figures/r_objects/trisomy12/de_genes/", gene_nam, ".rds"))
  p
}

diff <- diff_all[which(abs(diff_all$stat) > 9 & !rowData(RNAnorm[rownames(diff_all),])$chromosome %in% "12"),]
geneList <- c(as.character(diff$Symbol), "CTLA4", "CHFR", "SOCS3", "MERTK", "HRAS", "VAV1", "PTPN6", "NT5E")

lapply(geneList, gene_count)


```


## Gene set enrichment analysis

## Gene set enrichment analysis

Gene sets
```{r gene sets}
#load gene set collection
#Hallmark
gsc <- loadGSC("/home/almut/Dokumente/masterarbeit/data/h.all.v6.0.symbols.gmt", type="gmt")
#Kegg
gsc_Kegg <- loadGSC("/home/almut/Dokumente/masterarbeit/data/c2.cp.kegg.v6.0.symbols.gmt", type="gmt")


#get all de outputs
load(paste0(output_dir,"/desRes_250720.RData"))
difftab <- function(condition){
  dataTab <- data.frame(res_list[[condition]])
  dataTab$ID <- rownames(dataTab)
  #filter using pvalues
  dataTab <- dataTab %>%
    arrange(padj) %>%
    mutate(Symbol = rowData(ddsCLL[ID,])$symbol)# %>%
    #filter(abs(log2FoldChange) > 2)
  dataTab <- dataTab[!duplicated(dataTab$Symbol),]
  dataTab <- dataTab[!is.na(dataTab$Symbol),]
  rownames(dataTab) <- dataTab$ID
   dataTab
}

diff_res <- difftab(variant)

#clusterProfiler
diff_res <- diff_res[-which(diff_res$Symbol %in% c("", NA)),]
gene_list <- diff_res$stat %>% set_names(diff_res$Symbol)
gene_list <- sort(gene_list, decreasing = TRUE)
gene_lfc <- diff_res$log2FoldChange %>% set_names(diff_res$Symbol)
gene_lfc <- sort(gene_lfc, decreasing = TRUE)
de_gene <- diff_res %>% filter(padj < 0.01) 
de_gene <- de_gene$Symbol

de_ens <- diff_res %>% filter(padj < 0.01)
de_ens <- de_ens$ID
#Get Gene IDs
gene_id <- bitr(de_ens, fromType = "ENSEMBL",
        toType = c("ENTREZID", "SYMBOL"),
        OrgDb = org.Hs.eg.db)

gene_list_id <- bitr(diff_res$ID, fromType = "ENSEMBL",
        toType = c("ENTREZID", "SYMBOL"),
        OrgDb = org.Hs.eg.db)
names(gene_list_id) <- c("ID", "ENTREZID", "Symbol")
diff_id <- left_join(gene_list_id, diff_res)
gene_list_id <- diff_id$stat %>% set_names(diff_id$ENTREZID)
gene_list_id <- sort(gene_list_id, decreasing = TRUE)
gene_lfc_id <- diff_id$log2FoldChange %>% set_names(diff_id$ENTREZID)
gene_lfc_id <- sort(gene_lfc_id, decreasing = TRUE)

#convert gsc
m_t2g <- msigdbr(species = "Homo sapiens", category = "H") %>% 
  dplyr::select(gs_name, human_gene_symbol)

#Hallmark
em2 <- GSEA(gene_list, TERM2GENE = m_t2g, pvalueCutoff = 0.1)
em <- enricher(de_gene, TERM2GENE = m_t2g)

#Kegg
kk <- enrichKEGG(gene_id$ENTREZID,
                 organism     = 'hsa',
                 pvalueCutoff = 0.1)

kk2 <- gseKEGG(geneList     = gene_list_id,
               organism     = 'hsa',
               nPerm        = 1000,
               minGSSize    = 50,
               pvalueCutoff = 0.1,
               verbose      = FALSE)

kk2x <- setReadable(kk2, 'org.Hs.eg.db', 'ENTREZID')
```
Visualize ClusterProfiler results
```{r clusterProfiler}
barplot(kk, showCategory=5)
barplot(em, showCategory=5)

dot1 <- dotplot(em2, showCategory=10) + ggtitle("GSEA for trisomy12") +
  theme_pubr() +
  theme(legend.position="right") + 
  theme(plot.title = element_text(face = "bold")) 
dot1
dotplot(em, showCategory=10) + ggtitle("Enrichment for trisomy12")

dotplot(kk2, showCategory=10) + ggtitle("GSEA for trisomy12")
dot2 <- dotplot(kk, showCategory=10) + ggtitle("Enrichment for trisomy12") +
  theme_pubr() +
  theme(legend.position="right") +
  theme(plot.title = element_text(face = "bold"))
dot2

ridgeplot(em2)
ridgeplot(kk2)

gseaplot2(em2, geneSetID = 3, title = em2$Description[1])
gseaplot2(kk2, geneSetID = 2, title = kk2$Description[1])

saveRDS(dot1, file = paste0(output_dir, "/figures/r_objects/trisomy12/enrich_dot_hm.rds"))
saveRDS(dot2, file = paste0(output_dir, "/figures/r_objects/trisomy12/enrich_dot2.rds"))
```

```{r network, fig.width=10, fig.height=10}
# Networks Hallmark
 em2_sub <- em2
 em2_sub@result <- em2@result[which(em2@result$Description %in% c("HALLMARK_INTERFERON_GAMMA_RESPONSE",
                                                                    "HALLMARK_TNFA_SIGNALING_VIA_NFKB",
                                                                    "HALLMARK_PI3K_AKT_MTOR_SIGNALING")),]
p_net <- cnetplot(em2_sub, categorySize="pvalue", foldChange=gene_lfc) + 
  scale_colour_gradientn(colors = c("#581845", "#900C3F", "#C70039", "#FF5733", "#FFC300", "#DAF7A6")) + 
  guides(size = FALSE) + 
  labs(color = "logFC")

p_net  

# Networks KEGG
 kk2_sub <- kk2x
 kk2_sub@result <- kk2x@result[which(kk2x@result$Description %in% c("MAPK signaling pathway",
                                                                    "mTOR signaling pathway",
                                                                    "Chemokine signaling pathway")),]

pnet_kegg <- cnetplot(kk2_sub, categorySize="pvalue", foldChange=gene_lfc) + 
  scale_color_gradient(low="blue", high="red") +
  guides(size = FALSE) + 
  labs(color = "logFC")

pnet_kegg

saveRDS(pnet_kegg, file = paste0(output_dir, "/figures/r_objects/trisomy12/enrich_net_kegg.rds"))
saveRDS(p_net, file = paste0(output_dir, "/figures/r_objects/trisomy12/enrich_net_hm.rds"))

```




# Dosage effect

```{r stripchart, fig.width=4, fig.height=6}

tri12Groups <- diff_all  %>%  mutate(group = ifelse(stat > 0, "up", "down"))
rownames(tri12Groups) <- rownames(diff_all )

#specify colors/change order to get a better color for chromosome 12
palette <- pal_igv(palette = c("default"), alpha = 1)(24)
palette <- c(palette[c(5:24)], palette[c(1:4)])

#chrom dist
tri12_chrom <- as.tibble(tri12Groups) %>% mutate(Chromosome=rowData(RNAnorm[as.character(ID),])$chromosome) %>% 
  group_by(Chromosome, group) %>% summarise(DE_genes = n()) %>% 
  mutate(Chromosome_bin= ifelse(Chromosome == 12, "12", "other")) %>% 
  filter(!Chromosome %in% c("MT", "X")) 

tri12_order <- as.tibble(tri12Groups) %>% 
  mutate(Chromosome=rowData(RNAnorm[as.character(ID),])$chromosome) %>% 
  group_by(Chromosome) %>% 
  summarise(DE_genes = n()) %>% 
  filter(!Chromosome %in% c("MT", "X")) %>% 
  arrange(as.numeric(Chromosome))

tri12_chrom$DE_genes[tri12_chrom$group %in% "down"] <- tri12_chrom$DE_genes[tri12_chrom$group %in% "down"] * -1

#plot
p <- ggbarplot(tri12_chrom[order(match(tri12_chrom$Chromosome, tri12_order$Chromosome)),], "Chromosome", "DE_genes",
  fill = "group", 
  palette = c("#5B8FA8FF", "#FFB547FF"),
  font.x =23 ,
  font.y = 23, 
  font.tickslab = 18,
  rotate = T) 

p
#sum_dat_long2[order(match(sum_dat_long2$variant, sum_order$variant)),]
#save plot 
#ggsave(file=paste0(figure_dir, "/chromosom_dist_tri12.svg", plot=p, width=8, height=5)  
  
saveRDS(p, file = paste0(output_dir, "/figures/r_objects/trisomy12/tri12_dosage.rds"))

```
